<html><head><title>GeneValidator - validation explanations</title><meta content="text/html; charset=UTF-8" http-equiv="content-type"><style type="text/css">.lst-kix_euwmeg8cbpt8-4>li:before{content:"\0025cb  "}.lst-kix_g3lepnw8n0x6-4>li:before{content:"\0025cb  "}.lst-kix_9drbe3jtkkuh-4>li:before{content:"" counter(lst-ctn-kix_9drbe3jtkkuh-4,lower-latin) ". "}ul.lst-kix_hzkhz4hhlr8l-6{list-style-type:none}ul.lst-kix_hzkhz4hhlr8l-7{list-style-type:none}ul.lst-kix_hzkhz4hhlr8l-8{list-style-type:none}.lst-kix_z0ttp4mliy4e-6>li:before{content:"\0025cf  "}.lst-kix_tjc8wwk9cseo-2>li:before{content:"\0025a0  "}ol.lst-kix_9drbe3jtkkuh-5.start{counter-reset:lst-ctn-kix_9drbe3jtkkuh-5 0}ul.lst-kix_pizqq4oicz3j-5{list-style-type:none}ul.lst-kix_pizqq4oicz3j-4{list-style-type:none}ul.lst-kix_hzkhz4hhlr8l-0{list-style-type:none}ul.lst-kix_pizqq4oicz3j-3{list-style-type:none}ul.lst-kix_hzkhz4hhlr8l-1{list-style-type:none}ul.lst-kix_pizqq4oicz3j-2{list-style-type:none}ul.lst-kix_hzkhz4hhlr8l-2{list-style-type:none}.lst-kix_tjc8wwk9cseo-3>li:before{content:"\0025cf  "}ul.lst-kix_pizqq4oicz3j-1{list-style-type:none}ul.lst-kix_hzkhz4hhlr8l-3{list-style-type:none}ul.lst-kix_pizqq4oicz3j-0{list-style-type:none}ul.lst-kix_hzkhz4hhlr8l-4{list-style-type:none}ul.lst-kix_hzkhz4hhlr8l-5{list-style-type:none}.lst-kix_dpynbghv8b9j-3>li:before{content:"\0025cf  "}ul.lst-kix_pizqq4oicz3j-6{list-style-type:none}ul.lst-kix_pizqq4oicz3j-7{list-style-type:none}ul.lst-kix_pizqq4oicz3j-8{list-style-type:none}.lst-kix_hzkhz4hhlr8l-6>li:before{content:"\0025cf  "}.lst-kix_7rljtqi4tpf8-2>li:before{content:"\0025a0  "}.lst-kix_yghjnj8s9xuc-2>li:before{content:"\0025a0  "}.lst-kix_cbmcswhr1ftl-6>li:before{content:"\0025cf  "}.lst-kix_dpynbghv8b9j-0>li:before{content:"\0025cf  "}.lst-kix_lf42ux7u6753-5>li:before{content:"\0025a0  "}.lst-kix_sjinkw26u0wp-3>li:before{content:"\0025cf  "}.lst-kix_pdbqztwjr65g-8>li:before{content:"\0025a0  "}.lst-kix_qzhxzy2sbaso-7>li:before{content:"\0025cb  "}.lst-kix_3jnvhqhv71nk-4>li:before{content:"\0025cb  "}.lst-kix_fscfg01f3awm-4>li:before{content:"\0025cb  "}.lst-kix_olh7ea290trr-4>li:before{content:"\0025cb  "}.lst-kix_xq12dxu3f08q-6>li:before{content:"\0025cf  "}.lst-kix_hzkhz4hhlr8l-4>li:before{content:"\0025cb  "}.lst-kix_jamjx369pdr8-2>li:before{content:"\0025a0  "}.lst-kix_3jnvhqhv71nk-5>li:before{content:"\0025a0  "}.lst-kix_aps1ppntqbvm-5>li:before{content:"\0025a0  "}.lst-kix_x7iifx7mthae-8>li:before{content:"\0025a0  "}.lst-kix_vqtjz6k7exn7-1>li:before{content:"\0025cb  "}.lst-kix_g96zcqu3kh41-8>li:before{content:"\0025a0  "}.lst-kix_gr4a4jnxityy-3>li:before{content:"\0025cf  "}ul.lst-kix_lf42ux7u6753-0{list-style-type:none}.lst-kix_whzqgipsg1bt-2>li:before{content:"\0025a0  "}.lst-kix_g96zcqu3kh41-0>li:before{content:"\0025cf  "}ul.lst-kix_lf42ux7u6753-2{list-style-type:none}.lst-kix_puzs83wo6j0o-3>li:before{content:"\0025cf  "}ul.lst-kix_lf42ux7u6753-1{list-style-type:none}ul.lst-kix_lf42ux7u6753-4{list-style-type:none}ul.lst-kix_lf42ux7u6753-3{list-style-type:none}ul.lst-kix_lf42ux7u6753-6{list-style-type:none}.lst-kix_e5f9ugvogyec-7>li:before{content:"\0025cb  "}ul.lst-kix_lf42ux7u6753-5{list-style-type:none}ul.lst-kix_lf42ux7u6753-8{list-style-type:none}ul.lst-kix_lf42ux7u6753-7{list-style-type:none}.lst-kix_48dltnwvsqfp-1>li:before{content:"\0025cb  "}.lst-kix_4ed4y1s6lj5w-0>li:before{content:"\0025cf  "}.lst-kix_z0ttp4mliy4e-3>li:before{content:"\0025cf  "}.lst-kix_fscfg01f3awm-8>li:before{content:"\0025a0  "}.lst-kix_yghjnj8s9xuc-7>li:before{content:"\0025cb  "}.lst-kix_10avetia2xt4-3>li:before{content:"\0025cf  "}.lst-kix_508olx789l51-3>li:before{content:"\0025cf  "}.lst-kix_10avetia2xt4-8>li:before{content:"\0025a0  "}.lst-kix_9a6zvrglxdu8-1>li:before{content:"\0025cb  "}.lst-kix_z0ttp4mliy4e-7>li:before{content:"\0025cb  "}.lst-kix_vz631vhbp5pc-0>li:before{content:"\0025cf  "}.lst-kix_kmsnp6mmp404-3>li:before{content:"\0025cf  "}.lst-kix_q40ki98g1i63-6>li:before{content:"\0025cf  "}.lst-kix_aps1ppntqbvm-1>li:before{content:"\0025cb  "}ol.lst-kix_9drbe3jtkkuh-2.start{counter-reset:lst-ctn-kix_9drbe3jtkkuh-2 0}.lst-kix_9drbe3jtkkuh-0>li{counter-increment:lst-ctn-kix_9drbe3jtkkuh-0}.lst-kix_7kppodwmns3b-2>li:before{content:"\0025a0  "}.lst-kix_vqtjz6k7exn7-3>li:before{content:"\0025cf  "}ol.lst-kix_9drbe3jtkkuh-0.start{counter-reset:lst-ctn-kix_9drbe3jtkkuh-0 0}.lst-kix_ao4z72tcnso6-3>li:before{content:"\0025cf  "}.lst-kix_pizqq4oicz3j-4>li:before{content:"\0025cb  "}.lst-kix_owpom8h3nkwo-7>li:before{content:"\0025cb  "}.lst-kix_7kppodwmns3b-5>li:before{content:"\0025a0  "}.lst-kix_puzs83wo6j0o-2>li:before{content:"\0025a0  "}.lst-kix_e5f9ugvogyec-0>li:before{content:"\0025cf  "}.lst-kix_gr4a4jnxityy-0>li:before{content:"\0025cf  "}ul.lst-kix_dpynbghv8b9j-8{list-style-type:none}.lst-kix_euwmeg8cbpt8-6>li:before{content:"\0025cf  "}ul.lst-kix_dpynbghv8b9j-6{list-style-type:none}ul.lst-kix_dpynbghv8b9j-7{list-style-type:none}ul.lst-kix_dpynbghv8b9j-4{list-style-type:none}ul.lst-kix_dpynbghv8b9j-5{list-style-type:none}ul.lst-kix_dpynbghv8b9j-2{list-style-type:none}ul.lst-kix_dpynbghv8b9j-3{list-style-type:none}ul.lst-kix_puzs83wo6j0o-6{list-style-type:none}ul.lst-kix_dpynbghv8b9j-1{list-style-type:none}ul.lst-kix_puzs83wo6j0o-5{list-style-type:none}ul.lst-kix_dpynbghv8b9j-0{list-style-type:none}ul.lst-kix_puzs83wo6j0o-8{list-style-type:none}ul.lst-kix_puzs83wo6j0o-7{list-style-type:none}ul.lst-kix_puzs83wo6j0o-0{list-style-type:none}.lst-kix_pdbqztwjr65g-5>li:before{content:"\0025a0  "}ul.lst-kix_puzs83wo6j0o-2{list-style-type:none}.lst-kix_myo05blafbk-5>li:before{content:"\0025a0  "}ul.lst-kix_puzs83wo6j0o-1{list-style-type:none}ul.lst-kix_puzs83wo6j0o-4{list-style-type:none}ul.lst-kix_puzs83wo6j0o-3{list-style-type:none}.lst-kix_48dltnwvsqfp-5>li:before{content:"\0025a0  "}.lst-kix_e5f9ugvogyec-8>li:before{content:"\0025a0  "}.lst-kix_fscfg01f3awm-0>li:before{content:"\0025cf  "}.lst-kix_riib1nssx348-5>li:before{content:"\0025a0  "}.lst-kix_aps1ppntqbvm-2>li:before{content:"\0025a0  "}.lst-kix_508olx789l51-4>li:before{content:"\0025cb  "}.lst-kix_48dltnwvsqfp-7>li:before{content:"\0025cb  "}.lst-kix_10avetia2xt4-4>li:before{content:"\0025cb  "}.lst-kix_7kppodwmns3b-6>li:before{content:"\0025cf  "}.lst-kix_48dltnwvsqfp-3>li:before{content:"\0025cf  "}.lst-kix_4tq1geabcxa-2>li:before{content:"\0025a0  "}ol.lst-kix_9drbe3jtkkuh-0{list-style-type:none}.lst-kix_qzhxzy2sbaso-1>li:before{content:"\0025cb  "}.lst-kix_7rljtqi4tpf8-3>li:before{content:"\0025cf  "}.lst-kix_4tq1geabcxa-7>li:before{content:"\0025cb  "}.lst-kix_puzs83wo6j0o-5>li:before{content:"\0025a0  "}.lst-kix_b24v03uyl4k9-4>li:before{content:"\0025cb  "}.lst-kix_velyrizglpvf-6>li:before{content:"\0025cf  "}.lst-kix_sjinkw26u0wp-5>li:before{content:"\0025a0  "}.lst-kix_xc244bhq3ube-8>li:before{content:"\0025a0  "}.lst-kix_6064le485ne7-6>li:before{content:"\0025cf  "}.lst-kix_hzkhz4hhlr8l-1>li:before{content:"\0025cb  "}.lst-kix_l7t8lma3v5dd-7>li:before{content:"\0025cb  "}.lst-kix_jamjx369pdr8-5>li:before{content:"\0025a0  "}.lst-kix_vqtjz6k7exn7-0>li:before{content:"\0025cf  "}.lst-kix_6064le485ne7-0>li:before{content:"\0025cf  "}.lst-kix_e5f9ugvogyec-1>li:before{content:"\0025cb  "}.lst-kix_4ed4y1s6lj5w-7>li:before{content:"\0025cb  "}.lst-kix_olh7ea290trr-7>li:before{content:"\0025cb  "}.lst-kix_48dltnwvsqfp-6>li:before{content:"\0025cf  "}.lst-kix_velyrizglpvf-0>li:before{content:"\0025cf  "}.lst-kix_mk8t5sy0k589-2>li:before{content:"\0025a0  "}.lst-kix_yghjnj8s9xuc-6>li:before{content:"\0025cf  "}.lst-kix_b24v03uyl4k9-0>li:before{content:"\0025cf  "}.lst-kix_mk8t5sy0k589-4>li:before{content:"\0025cb  "}.lst-kix_g96zcqu3kh41-6>li:before{content:"\0025cf  "}.lst-kix_whzqgipsg1bt-3>li:before{content:"\0025cf  "}.lst-kix_pizqq4oicz3j-8>li:before{content:"\0025a0  "}.lst-kix_euwmeg8cbpt8-0>li:before{content:"\0025cf  "}.lst-kix_olh7ea290trr-6>li:before{content:"\0025cf  "}ol.lst-kix_9drbe3jtkkuh-3{list-style-type:none}ol.lst-kix_9drbe3jtkkuh-4{list-style-type:none}ol.lst-kix_9drbe3jtkkuh-1{list-style-type:none}.lst-kix_g3lepnw8n0x6-5>li:before{content:"\0025a0  "}ol.lst-kix_9drbe3jtkkuh-2{list-style-type:none}ol.lst-kix_9drbe3jtkkuh-7{list-style-type:none}ol.lst-kix_9drbe3jtkkuh-8{list-style-type:none}ol.lst-kix_9drbe3jtkkuh-5{list-style-type:none}ol.lst-kix_9drbe3jtkkuh-6{list-style-type:none}.lst-kix_owpom8h3nkwo-3>li:before{content:"\0025cf  "}.lst-kix_gr4a4jnxityy-8>li:before{content:"\0025a0  "}ul.lst-kix_g96zcqu3kh41-1{list-style-type:none}.lst-kix_dpynbghv8b9j-5>li:before{content:"\0025a0  "}ul.lst-kix_g96zcqu3kh41-2{list-style-type:none}ul.lst-kix_g96zcqu3kh41-3{list-style-type:none}ul.lst-kix_g96zcqu3kh41-4{list-style-type:none}ul.lst-kix_jamjx369pdr8-8{list-style-type:none}ul.lst-kix_jamjx369pdr8-7{list-style-type:none}.lst-kix_7rljtqi4tpf8-8>li:before{content:"\0025a0  "}ul.lst-kix_jamjx369pdr8-6{list-style-type:none}ul.lst-kix_jamjx369pdr8-5{list-style-type:none}ul.lst-kix_g96zcqu3kh41-0{list-style-type:none}.lst-kix_ao4z72tcnso6-1>li:before{content:"\0025cb  "}.lst-kix_g3lepnw8n0x6-0>li:before{content:"\0025cf  "}.lst-kix_x7iifx7mthae-0>li:before{content:"\0025cf  "}.lst-kix_mk8t5sy0k589-5>li:before{content:"\0025a0  "}.lst-kix_q40ki98g1i63-7>li:before{content:"\0025cb  "}.lst-kix_10avetia2xt4-6>li:before{content:"\0025cf  "}.lst-kix_6064le485ne7-1>li:before{content:"\0025cb  "}.lst-kix_9drbe3jtkkuh-6>li:before{content:"" counter(lst-ctn-kix_9drbe3jtkkuh-6,decimal) ". "}.lst-kix_cmqbbecr9wqr-2>li:before{content:"\0025a0  "}.lst-kix_6064le485ne7-2>li:before{content:"\0025a0  "}.lst-kix_l7t8lma3v5dd-3>li:before{content:"\0025cf  "}ul.lst-kix_jamjx369pdr8-3{list-style-type:none}ul.lst-kix_jamjx369pdr8-4{list-style-type:none}ul.lst-kix_jamjx369pdr8-1{list-style-type:none}ul.lst-kix_jamjx369pdr8-2{list-style-type:none}.lst-kix_9drbe3jtkkuh-3>li:before{content:"" counter(lst-ctn-kix_9drbe3jtkkuh-3,decimal) ". "}ul.lst-kix_jamjx369pdr8-0{list-style-type:none}.lst-kix_7kppodwmns3b-3>li:before{content:"\0025cf  "}.lst-kix_sjinkw26u0wp-1>li:before{content:"\0025cb  "}.lst-kix_tjc8wwk9cseo-4>li:before{content:"\0025cb  "}ul.lst-kix_sjinkw26u0wp-0{list-style-type:none}.lst-kix_7rljtqi4tpf8-6>li:before{content:"\0025cf  "}.lst-kix_l7t8lma3v5dd-5>li:before{content:"\0025a0  "}.lst-kix_4ed4y1s6lj5w-8>li:before{content:"\0025a0  "}ul.lst-kix_g3lepnw8n0x6-0{list-style-type:none}ul.lst-kix_g3lepnw8n0x6-1{list-style-type:none}ul.lst-kix_g3lepnw8n0x6-2{list-style-type:none}ul.lst-kix_g3lepnw8n0x6-3{list-style-type:none}ul.lst-kix_g3lepnw8n0x6-4{list-style-type:none}ul.lst-kix_g3lepnw8n0x6-5{list-style-type:none}ul.lst-kix_g3lepnw8n0x6-6{list-style-type:none}ul.lst-kix_g3lepnw8n0x6-7{list-style-type:none}.lst-kix_mk8t5sy0k589-3>li:before{content:"\0025cf  "}ul.lst-kix_g3lepnw8n0x6-8{list-style-type:none}.lst-kix_jamjx369pdr8-4>li:before{content:"\0025cb  "}.lst-kix_xc244bhq3ube-7>li:before{content:"\0025cb  "}.lst-kix_48dltnwvsqfp-8>li:before{content:"\0025a0  "}.lst-kix_puzs83wo6j0o-4>li:before{content:"\0025cb  "}.lst-kix_9drbe3jtkkuh-5>li:before{content:"" counter(lst-ctn-kix_9drbe3jtkkuh-5,lower-roman) ". "}.lst-kix_vz631vhbp5pc-8>li:before{content:"\0025a0  "}.lst-kix_lf42ux7u6753-6>li:before{content:"\0025cf  "}.lst-kix_xq12dxu3f08q-0>li:before{content:"\0025cf  "}.lst-kix_mk8t5sy0k589-6>li:before{content:"\0025cf  "}.lst-kix_508olx789l51-1>li:before{content:"\0025cb  "}ul.lst-kix_sjinkw26u0wp-8{list-style-type:none}.lst-kix_sjinkw26u0wp-6>li:before{content:"\0025cf  "}ul.lst-kix_sjinkw26u0wp-7{list-style-type:none}ul.lst-kix_sjinkw26u0wp-6{list-style-type:none}.lst-kix_owpom8h3nkwo-4>li:before{content:"\0025cb  "}ul.lst-kix_sjinkw26u0wp-5{list-style-type:none}.lst-kix_ao4z72tcnso6-6>li:before{content:"\0025cf  "}ul.lst-kix_sjinkw26u0wp-4{list-style-type:none}ul.lst-kix_sjinkw26u0wp-3{list-style-type:none}ul.lst-kix_sjinkw26u0wp-2{list-style-type:none}.lst-kix_gr4a4jnxityy-1>li:before{content:"\0025cb  "}ul.lst-kix_sjinkw26u0wp-1{list-style-type:none}.lst-kix_hzkhz4hhlr8l-3>li:before{content:"\0025cf  "}.lst-kix_cmqbbecr9wqr-6>li:before{content:"\0025cf  "}.lst-kix_whzqgipsg1bt-6>li:before{content:"\0025cf  "}.lst-kix_x7t5b7jbczs4-7>li:before{content:"\0025cb  "}.lst-kix_pizqq4oicz3j-6>li:before{content:"\0025cf  "}.lst-kix_pdbqztwjr65g-7>li:before{content:"\0025cb  "}.lst-kix_x7t5b7jbczs4-2>li:before{content:"\0025a0  "}.lst-kix_7kppodwmns3b-1>li:before{content:"\0025cb  "}.lst-kix_velyrizglpvf-1>li:before{content:"\0025cb  "}.lst-kix_x7iifx7mthae-6>li:before{content:"\0025cf  "}ul.lst-kix_l7t8lma3v5dd-7{list-style-type:none}ul.lst-kix_l7t8lma3v5dd-6{list-style-type:none}ul.lst-kix_l7t8lma3v5dd-5{list-style-type:none}ul.lst-kix_l7t8lma3v5dd-4{list-style-type:none}.lst-kix_dpynbghv8b9j-7>li:before{content:"\0025cb  "}ul.lst-kix_l7t8lma3v5dd-3{list-style-type:none}.lst-kix_pizqq4oicz3j-2>li:before{content:"\0025a0  "}ul.lst-kix_l7t8lma3v5dd-2{list-style-type:none}ul.lst-kix_l7t8lma3v5dd-1{list-style-type:none}ul.lst-kix_l7t8lma3v5dd-0{list-style-type:none}.lst-kix_l7t8lma3v5dd-2>li:before{content:"\0025a0  "}.lst-kix_whzqgipsg1bt-5>li:before{content:"\0025a0  "}ul.lst-kix_l7t8lma3v5dd-8{list-style-type:none}.lst-kix_riib1nssx348-8>li:before{content:"\0025a0  "}.lst-kix_myo05blafbk-7>li:before{content:"\0025cb  "}.lst-kix_vz631vhbp5pc-2>li:before{content:"\0025a0  "}.lst-kix_cbmcswhr1ftl-0>li:before{content:"\0025cf  "}.lst-kix_qzhxzy2sbaso-4>li:before{content:"\0025cb  "}.lst-kix_3jnvhqhv71nk-7>li:before{content:"\0025cb  "}.lst-kix_ao4z72tcnso6-0>li:before{content:"\0025cf  "}.lst-kix_3jnvhqhv71nk-6>li:before{content:"\0025cf  "}.lst-kix_x7t5b7jbczs4-5>li:before{content:"\0025a0  "}.lst-kix_mk8t5sy0k589-8>li:before{content:"\0025a0  "}.lst-kix_lf42ux7u6753-0>li:before{content:"\0025cf  "}.lst-kix_owpom8h3nkwo-8>li:before{content:"\0025a0  "}.lst-kix_mk8t5sy0k589-1>li:before{content:"\0025cb  "}.lst-kix_3kgcjgrfg4nh-2>li:before{content:"\0025a0  "}.lst-kix_7kppodwmns3b-4>li:before{content:"\0025cb  "}ul.lst-kix_aps1ppntqbvm-2{list-style-type:none}.lst-kix_b24v03uyl4k9-8>li:before{content:"\0025a0  "}ul.lst-kix_aps1ppntqbvm-3{list-style-type:none}ul.lst-kix_aps1ppntqbvm-4{list-style-type:none}.lst-kix_whzqgipsg1bt-8>li:before{content:"\0025a0  "}ul.lst-kix_aps1ppntqbvm-5{list-style-type:none}ul.lst-kix_aps1ppntqbvm-0{list-style-type:none}ul.lst-kix_aps1ppntqbvm-1{list-style-type:none}ul.lst-kix_4tq1geabcxa-8{list-style-type:none}ul.lst-kix_4tq1geabcxa-6{list-style-type:none}ul.lst-kix_4tq1geabcxa-7{list-style-type:none}ul.lst-kix_4tq1geabcxa-4{list-style-type:none}ul.lst-kix_4tq1geabcxa-5{list-style-type:none}ul.lst-kix_4tq1geabcxa-2{list-style-type:none}ul.lst-kix_4tq1geabcxa-3{list-style-type:none}.lst-kix_myo05blafbk-3>li:before{content:"\0025cf  "}ul.lst-kix_4tq1geabcxa-0{list-style-type:none}ul.lst-kix_4tq1geabcxa-1{list-style-type:none}ul.lst-kix_mk8t5sy0k589-3{list-style-type:none}ul.lst-kix_mk8t5sy0k589-4{list-style-type:none}.lst-kix_myo05blafbk-0>li:before{content:"\0025cf  "}ul.lst-kix_mk8t5sy0k589-5{list-style-type:none}ul.lst-kix_z0ttp4mliy4e-8{list-style-type:none}.lst-kix_myo05blafbk-6>li:before{content:"\0025cf  "}ul.lst-kix_mk8t5sy0k589-6{list-style-type:none}ul.lst-kix_z0ttp4mliy4e-7{list-style-type:none}ul.lst-kix_mk8t5sy0k589-0{list-style-type:none}.lst-kix_48dltnwvsqfp-2>li:before{content:"\0025a0  "}ul.lst-kix_mk8t5sy0k589-1{list-style-type:none}ul.lst-kix_mk8t5sy0k589-2{list-style-type:none}.lst-kix_3kgcjgrfg4nh-7>li:before{content:"\0025cb  "}.lst-kix_xq12dxu3f08q-2>li:before{content:"\0025a0  "}ul.lst-kix_yghjnj8s9xuc-7{list-style-type:none}ul.lst-kix_fscfg01f3awm-2{list-style-type:none}ul.lst-kix_yghjnj8s9xuc-8{list-style-type:none}ul.lst-kix_fscfg01f3awm-1{list-style-type:none}ul.lst-kix_yghjnj8s9xuc-5{list-style-type:none}ul.lst-kix_fscfg01f3awm-0{list-style-type:none}ul.lst-kix_yghjnj8s9xuc-6{list-style-type:none}.lst-kix_q40ki98g1i63-3>li:before{content:"\0025cf  "}ul.lst-kix_fscfg01f3awm-6{list-style-type:none}ul.lst-kix_fscfg01f3awm-5{list-style-type:none}ul.lst-kix_fscfg01f3awm-4{list-style-type:none}.lst-kix_vqtjz6k7exn7-5>li:before{content:"\0025a0  "}.lst-kix_9a6zvrglxdu8-2>li:before{content:"\0025a0  "}ul.lst-kix_fscfg01f3awm-3{list-style-type:none}.lst-kix_b24v03uyl4k9-1>li:before{content:"\0025cb  "}.lst-kix_508olx789l51-0>li:before{content:"\0025cf  "}ul.lst-kix_z0ttp4mliy4e-0{list-style-type:none}ul.lst-kix_yghjnj8s9xuc-0{list-style-type:none}ul.lst-kix_z0ttp4mliy4e-1{list-style-type:none}ul.lst-kix_fscfg01f3awm-8{list-style-type:none}ul.lst-kix_z0ttp4mliy4e-2{list-style-type:none}ul.lst-kix_fscfg01f3awm-7{list-style-type:none}ul.lst-kix_z0ttp4mliy4e-3{list-style-type:none}ul.lst-kix_yghjnj8s9xuc-3{list-style-type:none}ul.lst-kix_z0ttp4mliy4e-4{list-style-type:none}ul.lst-kix_yghjnj8s9xuc-4{list-style-type:none}ul.lst-kix_mk8t5sy0k589-8{list-style-type:none}ul.lst-kix_z0ttp4mliy4e-5{list-style-type:none}ul.lst-kix_yghjnj8s9xuc-1{list-style-type:none}ul.lst-kix_mk8t5sy0k589-7{list-style-type:none}.lst-kix_lf42ux7u6753-7>li:before{content:"\0025cb  "}ul.lst-kix_z0ttp4mliy4e-6{list-style-type:none}ul.lst-kix_yghjnj8s9xuc-2{list-style-type:none}.lst-kix_9a6zvrglxdu8-3>li:before{content:"\0025cf  "}.lst-kix_x7iifx7mthae-5>li:before{content:"\0025a0  "}.lst-kix_3jnvhqhv71nk-3>li:before{content:"\0025cf  "}ul.lst-kix_7rljtqi4tpf8-0{list-style-type:none}.lst-kix_yghjnj8s9xuc-1>li:before{content:"\0025cb  "}ul.lst-kix_7rljtqi4tpf8-1{list-style-type:none}ul.lst-kix_7rljtqi4tpf8-3{list-style-type:none}.lst-kix_sjinkw26u0wp-4>li:before{content:"\0025cb  "}ul.lst-kix_q40ki98g1i63-6{list-style-type:none}ul.lst-kix_7rljtqi4tpf8-2{list-style-type:none}ul.lst-kix_q40ki98g1i63-5{list-style-type:none}ul.lst-kix_7rljtqi4tpf8-5{list-style-type:none}ul.lst-kix_q40ki98g1i63-4{list-style-type:none}ul.lst-kix_7rljtqi4tpf8-4{list-style-type:none}ul.lst-kix_q40ki98g1i63-3{list-style-type:none}ul.lst-kix_7rljtqi4tpf8-7{list-style-type:none}ul.lst-kix_q40ki98g1i63-2{list-style-type:none}ul.lst-kix_7rljtqi4tpf8-6{list-style-type:none}ul.lst-kix_q40ki98g1i63-1{list-style-type:none}ul.lst-kix_q40ki98g1i63-0{list-style-type:none}ul.lst-kix_7rljtqi4tpf8-8{list-style-type:none}.lst-kix_qzhxzy2sbaso-0>li:before{content:"\0025cf  "}.lst-kix_pizqq4oicz3j-3>li:before{content:"\0025cf  "}ul.lst-kix_q40ki98g1i63-8{list-style-type:none}.lst-kix_g96zcqu3kh41-7>li:before{content:"\0025cb  "}ul.lst-kix_q40ki98g1i63-7{list-style-type:none}ul.lst-kix_e5f9ugvogyec-8{list-style-type:none}ul.lst-kix_cmqbbecr9wqr-3{list-style-type:none}ul.lst-kix_e5f9ugvogyec-7{list-style-type:none}ul.lst-kix_cmqbbecr9wqr-4{list-style-type:none}ul.lst-kix_cmqbbecr9wqr-1{list-style-type:none}.lst-kix_mk8t5sy0k589-7>li:before{content:"\0025cb  "}ul.lst-kix_cmqbbecr9wqr-2{list-style-type:none}ul.lst-kix_e5f9ugvogyec-4{list-style-type:none}.lst-kix_10avetia2xt4-2>li:before{content:"\0025a0  "}ul.lst-kix_e5f9ugvogyec-3{list-style-type:none}ul.lst-kix_cmqbbecr9wqr-0{list-style-type:none}ul.lst-kix_e5f9ugvogyec-6{list-style-type:none}ul.lst-kix_e5f9ugvogyec-5{list-style-type:none}.lst-kix_vz631vhbp5pc-6>li:before{content:"\0025cf  "}.lst-kix_xc244bhq3ube-4>li:before{content:"\0025cb  "}.lst-kix_kmsnp6mmp404-4>li:before{content:"\0025cb  "}.lst-kix_b24v03uyl4k9-7>li:before{content:"\0025cb  "}.lst-kix_b24v03uyl4k9-6>li:before{content:"\0025cf  "}ol.lst-kix_9drbe3jtkkuh-4.start{counter-reset:lst-ctn-kix_9drbe3jtkkuh-4 0}.lst-kix_b24v03uyl4k9-5>li:before{content:"\0025a0  "}.lst-kix_4ed4y1s6lj5w-6>li:before{content:"\0025cf  "}.lst-kix_e5f9ugvogyec-6>li:before{content:"\0025cf  "}ul.lst-kix_cmqbbecr9wqr-8{list-style-type:none}ul.lst-kix_cmqbbecr9wqr-7{list-style-type:none}.lst-kix_508olx789l51-6>li:before{content:"\0025cf  "}ul.lst-kix_cmqbbecr9wqr-6{list-style-type:none}ul.lst-kix_cmqbbecr9wqr-5{list-style-type:none}.lst-kix_4tq1geabcxa-5>li:before{content:"\0025a0  "}ol.lst-kix_9drbe3jtkkuh-6.start{counter-reset:lst-ctn-kix_9drbe3jtkkuh-6 0}.lst-kix_3jnvhqhv71nk-2>li:before{content:"\0025a0  "}.lst-kix_velyrizglpvf-7>li:before{content:"\0025cb  "}ul.lst-kix_qzhxzy2sbaso-7{list-style-type:none}ul.lst-kix_qzhxzy2sbaso-8{list-style-type:none}.lst-kix_9drbe3jtkkuh-8>li:before{content:"" counter(lst-ctn-kix_9drbe3jtkkuh-8,lower-roman) ". "}ul.lst-kix_qzhxzy2sbaso-5{list-style-type:none}ul.lst-kix_qzhxzy2sbaso-6{list-style-type:none}.lst-kix_b24v03uyl4k9-2>li:before{content:"\0025a0  "}ul.lst-kix_aps1ppntqbvm-6{list-style-type:none}ul.lst-kix_aps1ppntqbvm-7{list-style-type:none}ul.lst-kix_qzhxzy2sbaso-0{list-style-type:none}ul.lst-kix_aps1ppntqbvm-8{list-style-type:none}.lst-kix_4tq1geabcxa-1>li:before{content:"\0025cb  "}ul.lst-kix_e5f9ugvogyec-1{list-style-type:none}ul.lst-kix_qzhxzy2sbaso-3{list-style-type:none}ul.lst-kix_e5f9ugvogyec-2{list-style-type:none}ul.lst-kix_qzhxzy2sbaso-4{list-style-type:none}ul.lst-kix_qzhxzy2sbaso-1{list-style-type:none}ul.lst-kix_e5f9ugvogyec-0{list-style-type:none}ul.lst-kix_qzhxzy2sbaso-2{list-style-type:none}.lst-kix_cbmcswhr1ftl-5>li:before{content:"\0025a0  "}.lst-kix_q40ki98g1i63-5>li:before{content:"\0025a0  "}.lst-kix_kmsnp6mmp404-1>li:before{content:"\0025cb  "}.lst-kix_z0ttp4mliy4e-2>li:before{content:"\0025a0  "}.lst-kix_olh7ea290trr-0>li:before{content:"\0025cf  "}.lst-kix_x7t5b7jbczs4-4>li:before{content:"\0025cb  "}.lst-kix_ao4z72tcnso6-8>li:before{content:"\0025a0  "}.lst-kix_9drbe3jtkkuh-1>li:before{content:"" counter(lst-ctn-kix_9drbe3jtkkuh-1,lower-latin) ". "}.lst-kix_gr4a4jnxityy-7>li:before{content:"\0025cb  "}.lst-kix_x7t5b7jbczs4-1>li:before{content:"\0025cb  "}.lst-kix_whzqgipsg1bt-7>li:before{content:"\0025cb  "}.lst-kix_9drbe3jtkkuh-1>li{counter-increment:lst-ctn-kix_9drbe3jtkkuh-1}.lst-kix_jamjx369pdr8-7>li:before{content:"\0025cb  "}.lst-kix_mk8t5sy0k589-0>li:before{content:"\0025cf  "}.lst-kix_ao4z72tcnso6-2>li:before{content:"\0025a0  "}.lst-kix_48dltnwvsqfp-4>li:before{content:"\0025cb  "}.lst-kix_g96zcqu3kh41-3>li:before{content:"\0025cf  "}ul.lst-kix_xq12dxu3f08q-8{list-style-type:none}ul.lst-kix_xq12dxu3f08q-7{list-style-type:none}ul.lst-kix_xq12dxu3f08q-6{list-style-type:none}.lst-kix_9drbe3jtkkuh-4>li{counter-increment:lst-ctn-kix_9drbe3jtkkuh-4}ul.lst-kix_xq12dxu3f08q-5{list-style-type:none}ul.lst-kix_xq12dxu3f08q-4{list-style-type:none}ul.lst-kix_xq12dxu3f08q-3{list-style-type:none}ul.lst-kix_xq12dxu3f08q-2{list-style-type:none}.lst-kix_fscfg01f3awm-6>li:before{content:"\0025cf  "}.lst-kix_4ed4y1s6lj5w-2>li:before{content:"\0025a0  "}.lst-kix_kmsnp6mmp404-5>li:before{content:"\0025a0  "}.lst-kix_vz631vhbp5pc-5>li:before{content:"\0025a0  "}ul.lst-kix_xq12dxu3f08q-0{list-style-type:none}.lst-kix_qzhxzy2sbaso-2>li:before{content:"\0025a0  "}ul.lst-kix_xq12dxu3f08q-1{list-style-type:none}.lst-kix_xq12dxu3f08q-8>li:before{content:"\0025a0  "}.lst-kix_x7t5b7jbczs4-6>li:before{content:"\0025cf  "}.lst-kix_9drbe3jtkkuh-6>li{counter-increment:lst-ctn-kix_9drbe3jtkkuh-6}ul.lst-kix_g96zcqu3kh41-5{list-style-type:none}ul.lst-kix_g96zcqu3kh41-6{list-style-type:none}ul.lst-kix_g96zcqu3kh41-7{list-style-type:none}.lst-kix_euwmeg8cbpt8-2>li:before{content:"\0025a0  "}ul.lst-kix_g96zcqu3kh41-8{list-style-type:none}.lst-kix_10avetia2xt4-1>li:before{content:"\0025cb  "}.lst-kix_6064le485ne7-4>li:before{content:"\0025cb  "}.lst-kix_myo05blafbk-4>li:before{content:"\0025cb  "}.lst-kix_tjc8wwk9cseo-8>li:before{content:"\0025a0  "}.lst-kix_9a6zvrglxdu8-8>li:before{content:"\0025a0  "}.lst-kix_9a6zvrglxdu8-4>li:before{content:"\0025cb  "}.lst-kix_yghjnj8s9xuc-0>li:before{content:"\0025cf  "}.lst-kix_x7iifx7mthae-7>li:before{content:"\0025cb  "}.lst-kix_vz631vhbp5pc-3>li:before{content:"\0025cf  "}.lst-kix_g96zcqu3kh41-1>li:before{content:"\0025cb  "}.lst-kix_7rljtqi4tpf8-5>li:before{content:"\0025a0  "}.lst-kix_tjc8wwk9cseo-5>li:before{content:"\0025a0  "}.lst-kix_sjinkw26u0wp-2>li:before{content:"\0025a0  "}.lst-kix_4tq1geabcxa-4>li:before{content:"\0025cb  "}.lst-kix_q40ki98g1i63-8>li:before{content:"\0025a0  "}ul.lst-kix_gr4a4jnxityy-2{list-style-type:none}ul.lst-kix_gr4a4jnxityy-3{list-style-type:none}ul.lst-kix_gr4a4jnxityy-4{list-style-type:none}ul.lst-kix_gr4a4jnxityy-5{list-style-type:none}ul.lst-kix_gr4a4jnxityy-0{list-style-type:none}ul.lst-kix_gr4a4jnxityy-1{list-style-type:none}.lst-kix_owpom8h3nkwo-1>li:before{content:"\0025cb  "}ul.lst-kix_gr4a4jnxityy-6{list-style-type:none}ul.lst-kix_gr4a4jnxityy-7{list-style-type:none}.lst-kix_9drbe3jtkkuh-7>li{counter-increment:lst-ctn-kix_9drbe3jtkkuh-7}ul.lst-kix_gr4a4jnxityy-8{list-style-type:none}.lst-kix_48dltnwvsqfp-0>li:before{content:"\0025cf  "}.lst-kix_kmsnp6mmp404-7>li:before{content:"\0025cb  "}ul.lst-kix_vz631vhbp5pc-2{list-style-type:none}ul.lst-kix_vz631vhbp5pc-1{list-style-type:none}ul.lst-kix_vz631vhbp5pc-4{list-style-type:none}ul.lst-kix_vz631vhbp5pc-3{list-style-type:none}ul.lst-kix_vz631vhbp5pc-6{list-style-type:none}ul.lst-kix_vz631vhbp5pc-5{list-style-type:none}ul.lst-kix_vz631vhbp5pc-8{list-style-type:none}ul.lst-kix_vz631vhbp5pc-7{list-style-type:none}ul.lst-kix_vz631vhbp5pc-0{list-style-type:none}.lst-kix_jamjx369pdr8-3>li:before{content:"\0025cf  "}.lst-kix_xq12dxu3f08q-5>li:before{content:"\0025a0  "}.lst-kix_xc244bhq3ube-1>li:before{content:"\0025cb  "}ul.lst-kix_ao4z72tcnso6-5{list-style-type:none}ul.lst-kix_ao4z72tcnso6-6{list-style-type:none}ul.lst-kix_ao4z72tcnso6-3{list-style-type:none}ul.lst-kix_ao4z72tcnso6-4{list-style-type:none}.lst-kix_owpom8h3nkwo-6>li:before{content:"\0025cf  "}ul.lst-kix_ao4z72tcnso6-1{list-style-type:none}ul.lst-kix_ao4z72tcnso6-2{list-style-type:none}.lst-kix_aps1ppntqbvm-3>li:before{content:"\0025cf  "}ul.lst-kix_ao4z72tcnso6-0{list-style-type:none}.lst-kix_yghjnj8s9xuc-4>li:before{content:"\0025cb  "}.lst-kix_fscfg01f3awm-3>li:before{content:"\0025cf  "}ul.lst-kix_ao4z72tcnso6-8{list-style-type:none}ul.lst-kix_ao4z72tcnso6-7{list-style-type:none}.lst-kix_6064le485ne7-5>li:before{content:"\0025a0  "}.lst-kix_fscfg01f3awm-1>li:before{content:"\0025cb  "}.lst-kix_cbmcswhr1ftl-1>li:before{content:"\0025cb  "}.lst-kix_vz631vhbp5pc-1>li:before{content:"\0025cb  "}.lst-kix_owpom8h3nkwo-2>li:before{content:"\0025a0  "}.lst-kix_vqtjz6k7exn7-7>li:before{content:"\0025cb  "}ul.lst-kix_myo05blafbk-1{list-style-type:none}ul.lst-kix_myo05blafbk-2{list-style-type:none}ul.lst-kix_myo05blafbk-3{list-style-type:none}ul.lst-kix_myo05blafbk-4{list-style-type:none}ul.lst-kix_myo05blafbk-5{list-style-type:none}ul.lst-kix_myo05blafbk-6{list-style-type:none}.lst-kix_dpynbghv8b9j-6>li:before{content:"\0025cf  "}ul.lst-kix_myo05blafbk-7{list-style-type:none}ul.lst-kix_myo05blafbk-8{list-style-type:none}.lst-kix_gr4a4jnxityy-2>li:before{content:"\0025a0  "}ul.lst-kix_myo05blafbk-0{list-style-type:none}.lst-kix_aps1ppntqbvm-8>li:before{content:"\0025a0  "}.lst-kix_tjc8wwk9cseo-0>li:before{content:"\0025cf  "}.lst-kix_puzs83wo6j0o-7>li:before{content:"\0025cb  "}.lst-kix_xq12dxu3f08q-3>li:before{content:"\0025cf  "}.lst-kix_fscfg01f3awm-2>li:before{content:"\0025a0  "}.lst-kix_z0ttp4mliy4e-8>li:before{content:"\0025a0  "}.lst-kix_euwmeg8cbpt8-3>li:before{content:"\0025cf  "}.lst-kix_sjinkw26u0wp-7>li:before{content:"\0025cb  "}.lst-kix_gr4a4jnxityy-6>li:before{content:"\0025cf  "}.lst-kix_l7t8lma3v5dd-1>li:before{content:"\0025cb  "}.lst-kix_xc244bhq3ube-3>li:before{content:"\0025cf  "}.lst-kix_xc244bhq3ube-2>li:before{content:"\0025a0  "}.lst-kix_x7t5b7jbczs4-8>li:before{content:"\0025a0  "}.lst-kix_qzhxzy2sbaso-6>li:before{content:"\0025cf  "}.lst-kix_hzkhz4hhlr8l-0>li:before{content:"\0025cf  "}ul.lst-kix_4ed4y1s6lj5w-8{list-style-type:none}ul.lst-kix_4ed4y1s6lj5w-7{list-style-type:none}.lst-kix_z0ttp4mliy4e-0>li:before{content:"\0025cf  "}ul.lst-kix_4ed4y1s6lj5w-6{list-style-type:none}ul.lst-kix_4ed4y1s6lj5w-5{list-style-type:none}.lst-kix_q40ki98g1i63-4>li:before{content:"\0025cb  "}.lst-kix_e5f9ugvogyec-3>li:before{content:"\0025cf  "}ul.lst-kix_whzqgipsg1bt-3{list-style-type:none}ul.lst-kix_whzqgipsg1bt-2{list-style-type:none}ul.lst-kix_whzqgipsg1bt-1{list-style-type:none}ul.lst-kix_whzqgipsg1bt-0{list-style-type:none}ul.lst-kix_whzqgipsg1bt-7{list-style-type:none}ul.lst-kix_whzqgipsg1bt-6{list-style-type:none}ul.lst-kix_whzqgipsg1bt-5{list-style-type:none}ul.lst-kix_whzqgipsg1bt-4{list-style-type:none}.lst-kix_puzs83wo6j0o-0>li:before{content:"\0025cf  "}.lst-kix_x7iifx7mthae-1>li:before{content:"\0025cb  "}ul.lst-kix_whzqgipsg1bt-8{list-style-type:none}.lst-kix_9a6zvrglxdu8-6>li:before{content:"\0025cf  "}.lst-kix_x7t5b7jbczs4-0>li:before{content:"\0025cf  "}.lst-kix_pdbqztwjr65g-4>li:before{content:"\0025cb  "}.lst-kix_7kppodwmns3b-0>li:before{content:"\0025cf  "}.lst-kix_508olx789l51-5>li:before{content:"\0025a0  "}.lst-kix_vz631vhbp5pc-4>li:before{content:"\0025cb  "}.lst-kix_9a6zvrglxdu8-0>li:before{content:"\0025cf  "}.lst-kix_10avetia2xt4-7>li:before{content:"\0025cb  "}.lst-kix_q40ki98g1i63-0>li:before{content:"\0025cf  "}.lst-kix_3kgcjgrfg4nh-3>li:before{content:"\0025cf  "}ul.lst-kix_riib1nssx348-3{list-style-type:none}.lst-kix_6064le485ne7-7>li:before{content:"\0025cb  "}ul.lst-kix_riib1nssx348-4{list-style-type:none}ul.lst-kix_riib1nssx348-1{list-style-type:none}ul.lst-kix_riib1nssx348-2{list-style-type:none}ul.lst-kix_riib1nssx348-0{list-style-type:none}.lst-kix_tjc8wwk9cseo-1>li:before{content:"\0025cb  "}.lst-kix_ao4z72tcnso6-4>li:before{content:"\0025cb  "}.lst-kix_velyrizglpvf-5>li:before{content:"\0025a0  "}ul.lst-kix_olh7ea290trr-8{list-style-type:none}ul.lst-kix_olh7ea290trr-7{list-style-type:none}ul.lst-kix_olh7ea290trr-6{list-style-type:none}ul.lst-kix_olh7ea290trr-5{list-style-type:none}ul.lst-kix_olh7ea290trr-4{list-style-type:none}.lst-kix_9drbe3jtkkuh-8>li{counter-increment:lst-ctn-kix_9drbe3jtkkuh-8}ul.lst-kix_olh7ea290trr-3{list-style-type:none}ul.lst-kix_olh7ea290trr-2{list-style-type:none}ul.lst-kix_olh7ea290trr-1{list-style-type:none}ul.lst-kix_riib1nssx348-8{list-style-type:none}ul.lst-kix_olh7ea290trr-0{list-style-type:none}ul.lst-kix_riib1nssx348-7{list-style-type:none}ul.lst-kix_riib1nssx348-6{list-style-type:none}ul.lst-kix_riib1nssx348-5{list-style-type:none}.lst-kix_puzs83wo6j0o-8>li:before{content:"\0025a0  "}.lst-kix_q40ki98g1i63-1>li:before{content:"\0025cb  "}.lst-kix_4ed4y1s6lj5w-1>li:before{content:"\0025cb  "}.lst-kix_lf42ux7u6753-1>li:before{content:"\0025cb  "}.lst-kix_pdbqztwjr65g-6>li:before{content:"\0025cf  "}.lst-kix_euwmeg8cbpt8-8>li:before{content:"\0025a0  "}.lst-kix_cbmcswhr1ftl-2>li:before{content:"\0025a0  "}.lst-kix_aps1ppntqbvm-4>li:before{content:"\0025cb  "}.lst-kix_7kppodwmns3b-8>li:before{content:"\0025a0  "}.lst-kix_euwmeg8cbpt8-7>li:before{content:"\0025cb  "}ol.lst-kix_9drbe3jtkkuh-3.start{counter-reset:lst-ctn-kix_9drbe3jtkkuh-3 0}ul.lst-kix_x7t5b7jbczs4-8{list-style-type:none}.lst-kix_3kgcjgrfg4nh-1>li:before{content:"\0025cb  "}.lst-kix_cmqbbecr9wqr-0>li:before{content:"\0025cf  "}ul.lst-kix_b24v03uyl4k9-1{list-style-type:none}ul.lst-kix_x7t5b7jbczs4-4{list-style-type:none}ul.lst-kix_3kgcjgrfg4nh-6{list-style-type:none}ul.lst-kix_b24v03uyl4k9-0{list-style-type:none}ul.lst-kix_x7t5b7jbczs4-5{list-style-type:none}ul.lst-kix_3kgcjgrfg4nh-5{list-style-type:none}ul.lst-kix_x7t5b7jbczs4-6{list-style-type:none}ul.lst-kix_3kgcjgrfg4nh-8{list-style-type:none}.lst-kix_ao4z72tcnso6-7>li:before{content:"\0025cb  "}ul.lst-kix_x7t5b7jbczs4-7{list-style-type:none}ul.lst-kix_3kgcjgrfg4nh-7{list-style-type:none}ul.lst-kix_9a6zvrglxdu8-0{list-style-type:none}ul.lst-kix_x7t5b7jbczs4-0{list-style-type:none}ul.lst-kix_x7t5b7jbczs4-1{list-style-type:none}ul.lst-kix_x7t5b7jbczs4-2{list-style-type:none}ul.lst-kix_x7t5b7jbczs4-3{list-style-type:none}ul.lst-kix_9a6zvrglxdu8-4{list-style-type:none}ul.lst-kix_48dltnwvsqfp-2{list-style-type:none}ul.lst-kix_9a6zvrglxdu8-3{list-style-type:none}ul.lst-kix_48dltnwvsqfp-3{list-style-type:none}ul.lst-kix_9a6zvrglxdu8-2{list-style-type:none}ul.lst-kix_48dltnwvsqfp-0{list-style-type:none}.lst-kix_lf42ux7u6753-8>li:before{content:"\0025a0  "}ul.lst-kix_9a6zvrglxdu8-1{list-style-type:none}ul.lst-kix_48dltnwvsqfp-1{list-style-type:none}ul.lst-kix_48dltnwvsqfp-7{list-style-type:none}.lst-kix_euwmeg8cbpt8-1>li:before{content:"\0025cb  "}ul.lst-kix_48dltnwvsqfp-6{list-style-type:none}ul.lst-kix_48dltnwvsqfp-5{list-style-type:none}ul.lst-kix_48dltnwvsqfp-4{list-style-type:none}ul.lst-kix_48dltnwvsqfp-8{list-style-type:none}ul.lst-kix_b24v03uyl4k9-4{list-style-type:none}ul.lst-kix_3kgcjgrfg4nh-1{list-style-type:none}ul.lst-kix_b24v03uyl4k9-5{list-style-type:none}ul.lst-kix_3kgcjgrfg4nh-2{list-style-type:none}ul.lst-kix_b24v03uyl4k9-2{list-style-type:none}.lst-kix_velyrizglpvf-4>li:before{content:"\0025cb  "}.lst-kix_kmsnp6mmp404-6>li:before{content:"\0025cf  "}ul.lst-kix_3kgcjgrfg4nh-3{list-style-type:none}ul.lst-kix_b24v03uyl4k9-3{list-style-type:none}.lst-kix_riib1nssx348-3>li:before{content:"\0025cf  "}ul.lst-kix_3kgcjgrfg4nh-4{list-style-type:none}ul.lst-kix_b24v03uyl4k9-8{list-style-type:none}ul.lst-kix_b24v03uyl4k9-6{list-style-type:none}.lst-kix_9drbe3jtkkuh-2>li{counter-increment:lst-ctn-kix_9drbe3jtkkuh-2}ul.lst-kix_b24v03uyl4k9-7{list-style-type:none}.lst-kix_kmsnp6mmp404-8>li:before{content:"\0025a0  "}ul.lst-kix_3kgcjgrfg4nh-0{list-style-type:none}.lst-kix_l7t8lma3v5dd-4>li:before{content:"\0025cb  "}.lst-kix_4ed4y1s6lj5w-5>li:before{content:"\0025a0  "}ol.lst-kix_9drbe3jtkkuh-1.start{counter-reset:lst-ctn-kix_9drbe3jtkkuh-1 0}.lst-kix_xc244bhq3ube-5>li:before{content:"\0025a0  "}.lst-kix_tjc8wwk9cseo-6>li:before{content:"\0025cf  "}.lst-kix_sjinkw26u0wp-0>li:before{content:"\0025cf  "}.lst-kix_dpynbghv8b9j-4>li:before{content:"\0025cb  "}.lst-kix_riib1nssx348-0>li:before{content:"\0025cf  "}.lst-kix_pdbqztwjr65g-1>li:before{content:"\0025cb  "}.lst-kix_jamjx369pdr8-1>li:before{content:"\0025cb  "}.lst-kix_xc244bhq3ube-0>li:before{content:"\0025cf  "}ul.lst-kix_9a6zvrglxdu8-5{list-style-type:none}.lst-kix_3kgcjgrfg4nh-5>li:before{content:"\0025a0  "}ul.lst-kix_9a6zvrglxdu8-6{list-style-type:none}ul.lst-kix_9a6zvrglxdu8-7{list-style-type:none}ul.lst-kix_9a6zvrglxdu8-8{list-style-type:none}.lst-kix_jamjx369pdr8-6>li:before{content:"\0025cf  "}.lst-kix_hzkhz4hhlr8l-7>li:before{content:"\0025cb  "}.lst-kix_whzqgipsg1bt-0>li:before{content:"\0025cf  "}ol.lst-kix_9drbe3jtkkuh-8.start{counter-reset:lst-ctn-kix_9drbe3jtkkuh-8 0}.lst-kix_cbmcswhr1ftl-7>li:before{content:"\0025cb  "}.lst-kix_q40ki98g1i63-2>li:before{content:"\0025a0  "}.lst-kix_puzs83wo6j0o-6>li:before{content:"\0025cf  "}.lst-kix_myo05blafbk-1>li:before{content:"\0025cb  "}.lst-kix_jamjx369pdr8-0>li:before{content:"\0025cf  "}.lst-kix_lf42ux7u6753-2>li:before{content:"\0025a0  "}.lst-kix_yghjnj8s9xuc-3>li:before{content:"\0025cf  "}.lst-kix_riib1nssx348-6>li:before{content:"\0025cf  "}.lst-kix_yghjnj8s9xuc-5>li:before{content:"\0025a0  "}.lst-kix_7rljtqi4tpf8-4>li:before{content:"\0025cb  "}.lst-kix_cmqbbecr9wqr-1>li:before{content:"\0025cb  "}.lst-kix_puzs83wo6j0o-1>li:before{content:"\0025cb  "}.lst-kix_3jnvhqhv71nk-0>li:before{content:"\0025cf  "}ul.lst-kix_kmsnp6mmp404-0{list-style-type:none}ul.lst-kix_kmsnp6mmp404-1{list-style-type:none}.lst-kix_7rljtqi4tpf8-0>li:before{content:"\0025cf  "}.lst-kix_3kgcjgrfg4nh-6>li:before{content:"\0025cf  "}.lst-kix_vqtjz6k7exn7-6>li:before{content:"\0025cf  "}.lst-kix_whzqgipsg1bt-1>li:before{content:"\0025cb  "}ul.lst-kix_kmsnp6mmp404-4{list-style-type:none}.lst-kix_pizqq4oicz3j-7>li:before{content:"\0025cb  "}ul.lst-kix_kmsnp6mmp404-5{list-style-type:none}ul.lst-kix_kmsnp6mmp404-2{list-style-type:none}ul.lst-kix_kmsnp6mmp404-3{list-style-type:none}.lst-kix_g3lepnw8n0x6-2>li:before{content:"\0025a0  "}.lst-kix_9a6zvrglxdu8-5>li:before{content:"\0025a0  "}ul.lst-kix_kmsnp6mmp404-8{list-style-type:none}.lst-kix_6064le485ne7-3>li:before{content:"\0025cf  "}ul.lst-kix_kmsnp6mmp404-7{list-style-type:none}.lst-kix_4ed4y1s6lj5w-3>li:before{content:"\0025cf  "}ul.lst-kix_kmsnp6mmp404-6{list-style-type:none}.lst-kix_10avetia2xt4-0>li:before{content:"\0025cf  "}.lst-kix_hzkhz4hhlr8l-2>li:before{content:"\0025a0  "}.lst-kix_g3lepnw8n0x6-1>li:before{content:"\0025cb  "}.lst-kix_kmsnp6mmp404-0>li:before{content:"\0025cf  "}.lst-kix_7rljtqi4tpf8-1>li:before{content:"\0025cb  "}.lst-kix_lf42ux7u6753-3>li:before{content:"\0025cf  "}ul.lst-kix_vqtjz6k7exn7-2{list-style-type:none}ul.lst-kix_10avetia2xt4-6{list-style-type:none}ul.lst-kix_vqtjz6k7exn7-3{list-style-type:none}ul.lst-kix_10avetia2xt4-7{list-style-type:none}ul.lst-kix_vqtjz6k7exn7-0{list-style-type:none}ul.lst-kix_10avetia2xt4-4{list-style-type:none}.lst-kix_cmqbbecr9wqr-7>li:before{content:"\0025cb  "}ul.lst-kix_vqtjz6k7exn7-1{list-style-type:none}ul.lst-kix_10avetia2xt4-5{list-style-type:none}ul.lst-kix_10avetia2xt4-2{list-style-type:none}ul.lst-kix_10avetia2xt4-3{list-style-type:none}.lst-kix_fscfg01f3awm-5>li:before{content:"\0025a0  "}.lst-kix_vqtjz6k7exn7-2>li:before{content:"\0025a0  "}ul.lst-kix_10avetia2xt4-0{list-style-type:none}.lst-kix_pdbqztwjr65g-2>li:before{content:"\0025a0  "}ul.lst-kix_10avetia2xt4-1{list-style-type:none}ul.lst-kix_vqtjz6k7exn7-8{list-style-type:none}ul.lst-kix_vqtjz6k7exn7-6{list-style-type:none}ul.lst-kix_vqtjz6k7exn7-7{list-style-type:none}.lst-kix_jamjx369pdr8-8>li:before{content:"\0025a0  "}ul.lst-kix_vqtjz6k7exn7-4{list-style-type:none}ul.lst-kix_10avetia2xt4-8{list-style-type:none}ul.lst-kix_vqtjz6k7exn7-5{list-style-type:none}.lst-kix_4ed4y1s6lj5w-4>li:before{content:"\0025cb  "}.lst-kix_pdbqztwjr65g-3>li:before{content:"\0025cf  "}.lst-kix_508olx789l51-7>li:before{content:"\0025cb  "}.lst-kix_xq12dxu3f08q-1>li:before{content:"\0025cb  "}.lst-kix_gr4a4jnxityy-4>li:before{content:"\0025cb  "}.lst-kix_qzhxzy2sbaso-3>li:before{content:"\0025cf  "}.lst-kix_tjc8wwk9cseo-7>li:before{content:"\0025cb  "}ul.lst-kix_velyrizglpvf-0{list-style-type:none}.lst-kix_9a6zvrglxdu8-7>li:before{content:"\0025cb  "}ul.lst-kix_velyrizglpvf-2{list-style-type:none}ul.lst-kix_velyrizglpvf-1{list-style-type:none}ul.lst-kix_velyrizglpvf-4{list-style-type:none}ul.lst-kix_velyrizglpvf-3{list-style-type:none}ul.lst-kix_velyrizglpvf-6{list-style-type:none}ul.lst-kix_velyrizglpvf-5{list-style-type:none}ul.lst-kix_velyrizglpvf-8{list-style-type:none}ul.lst-kix_velyrizglpvf-7{list-style-type:none}.lst-kix_fscfg01f3awm-7>li:before{content:"\0025cb  "}.lst-kix_pizqq4oicz3j-0>li:before{content:"\0025cf  "}.lst-kix_z0ttp4mliy4e-1>li:before{content:"\0025cb  "}.lst-kix_e5f9ugvogyec-4>li:before{content:"\0025cb  "}.lst-kix_vqtjz6k7exn7-4>li:before{content:"\0025cb  "}.lst-kix_cmqbbecr9wqr-5>li:before{content:"\0025a0  "}.lst-kix_7kppodwmns3b-7>li:before{content:"\0025cb  "}.lst-kix_velyrizglpvf-8>li:before{content:"\0025a0  "}.lst-kix_sjinkw26u0wp-8>li:before{content:"\0025a0  "}.lst-kix_x7t5b7jbczs4-3>li:before{content:"\0025cf  "}ul.lst-kix_6064le485ne7-0{list-style-type:none}.lst-kix_cbmcswhr1ftl-8>li:before{content:"\0025a0  "}.lst-kix_riib1nssx348-1>li:before{content:"\0025cb  "}.lst-kix_myo05blafbk-2>li:before{content:"\0025a0  "}.lst-kix_4tq1geabcxa-0>li:before{content:"\0025cf  "}ul.lst-kix_pdbqztwjr65g-2{list-style-type:none}ul.lst-kix_pdbqztwjr65g-1{list-style-type:none}ul.lst-kix_pdbqztwjr65g-0{list-style-type:none}.lst-kix_aps1ppntqbvm-6>li:before{content:"\0025cf  "}.lst-kix_g96zcqu3kh41-2>li:before{content:"\0025a0  "}ul.lst-kix_pdbqztwjr65g-6{list-style-type:none}ul.lst-kix_pdbqztwjr65g-5{list-style-type:none}ul.lst-kix_pdbqztwjr65g-4{list-style-type:none}ul.lst-kix_pdbqztwjr65g-3{list-style-type:none}ul.lst-kix_pdbqztwjr65g-8{list-style-type:none}ul.lst-kix_pdbqztwjr65g-7{list-style-type:none}ul.lst-kix_6064le485ne7-2{list-style-type:none}ul.lst-kix_6064le485ne7-1{list-style-type:none}.lst-kix_dpynbghv8b9j-1>li:before{content:"\0025cb  "}ul.lst-kix_6064le485ne7-4{list-style-type:none}ul.lst-kix_6064le485ne7-3{list-style-type:none}.lst-kix_3jnvhqhv71nk-1>li:before{content:"\0025cb  "}ul.lst-kix_6064le485ne7-6{list-style-type:none}ul.lst-kix_6064le485ne7-5{list-style-type:none}.lst-kix_3kgcjgrfg4nh-0>li:before{content:"\0025cf  "}ul.lst-kix_6064le485ne7-8{list-style-type:none}.lst-kix_cbmcswhr1ftl-3>li:before{content:"\0025cf  "}ul.lst-kix_6064le485ne7-7{list-style-type:none}.lst-kix_ao4z72tcnso6-5>li:before{content:"\0025a0  "}.lst-kix_cbmcswhr1ftl-4>li:before{content:"\0025cb  "}.lst-kix_pizqq4oicz3j-1>li:before{content:"\0025cb  "}.lst-kix_hzkhz4hhlr8l-5>li:before{content:"\0025a0  "}.lst-kix_9drbe3jtkkuh-0>li:before{content:"" counter(lst-ctn-kix_9drbe3jtkkuh-0,decimal) ". "}ul.lst-kix_508olx789l51-2{list-style-type:none}ul.lst-kix_508olx789l51-1{list-style-type:none}ul.lst-kix_508olx789l51-0{list-style-type:none}.lst-kix_4tq1geabcxa-3>li:before{content:"\0025cf  "}.lst-kix_kmsnp6mmp404-2>li:before{content:"\0025a0  "}.lst-kix_508olx789l51-2>li:before{content:"\0025a0  "}ul.lst-kix_508olx789l51-7{list-style-type:none}ul.lst-kix_508olx789l51-8{list-style-type:none}.lst-kix_aps1ppntqbvm-0>li:before{content:"\0025cf  "}ul.lst-kix_508olx789l51-5{list-style-type:none}.lst-kix_xq12dxu3f08q-7>li:before{content:"\0025cb  "}ul.lst-kix_508olx789l51-6{list-style-type:none}ul.lst-kix_508olx789l51-3{list-style-type:none}ul.lst-kix_508olx789l51-4{list-style-type:none}.lst-kix_g3lepnw8n0x6-7>li:before{content:"\0025cb  "}.lst-kix_euwmeg8cbpt8-5>li:before{content:"\0025a0  "}.lst-kix_508olx789l51-8>li:before{content:"\0025a0  "}.lst-kix_g96zcqu3kh41-4>li:before{content:"\0025cb  "}.lst-kix_9drbe3jtkkuh-7>li:before{content:"" counter(lst-ctn-kix_9drbe3jtkkuh-7,lower-latin) ". "}ul.lst-kix_tjc8wwk9cseo-1{list-style-type:none}ul.lst-kix_tjc8wwk9cseo-0{list-style-type:none}ul.lst-kix_owpom8h3nkwo-8{list-style-type:none}ul.lst-kix_tjc8wwk9cseo-3{list-style-type:none}ul.lst-kix_owpom8h3nkwo-7{list-style-type:none}ul.lst-kix_tjc8wwk9cseo-2{list-style-type:none}ul.lst-kix_tjc8wwk9cseo-5{list-style-type:none}ul.lst-kix_tjc8wwk9cseo-4{list-style-type:none}ul.lst-kix_owpom8h3nkwo-4{list-style-type:none}ul.lst-kix_tjc8wwk9cseo-7{list-style-type:none}ul.lst-kix_owpom8h3nkwo-3{list-style-type:none}ul.lst-kix_tjc8wwk9cseo-6{list-style-type:none}ul.lst-kix_owpom8h3nkwo-6{list-style-type:none}ul.lst-kix_owpom8h3nkwo-5{list-style-type:none}ul.lst-kix_tjc8wwk9cseo-8{list-style-type:none}ul.lst-kix_owpom8h3nkwo-0{list-style-type:none}ul.lst-kix_owpom8h3nkwo-2{list-style-type:none}ul.lst-kix_owpom8h3nkwo-1{list-style-type:none}.lst-kix_riib1nssx348-4>li:before{content:"\0025cb  "}.lst-kix_yghjnj8s9xuc-8>li:before{content:"\0025a0  "}.lst-kix_l7t8lma3v5dd-8>li:before{content:"\0025a0  "}.lst-kix_dpynbghv8b9j-8>li:before{content:"\0025a0  "}.lst-kix_riib1nssx348-7>li:before{content:"\0025cb  "}.lst-kix_3kgcjgrfg4nh-8>li:before{content:"\0025a0  "}.lst-kix_g96zcqu3kh41-5>li:before{content:"\0025a0  "}.lst-kix_qzhxzy2sbaso-5>li:before{content:"\0025a0  "}.lst-kix_4tq1geabcxa-8>li:before{content:"\0025a0  "}.lst-kix_3jnvhqhv71nk-8>li:before{content:"\0025a0  "}.lst-kix_6064le485ne7-8>li:before{content:"\0025a0  "}.lst-kix_9drbe3jtkkuh-3>li{counter-increment:lst-ctn-kix_9drbe3jtkkuh-3}.lst-kix_velyrizglpvf-3>li:before{content:"\0025cf  "}.lst-kix_b24v03uyl4k9-3>li:before{content:"\0025cf  "}.lst-kix_riib1nssx348-2>li:before{content:"\0025a0  "}.lst-kix_dpynbghv8b9j-2>li:before{content:"\0025a0  "}.lst-kix_g3lepnw8n0x6-3>li:before{content:"\0025cf  "}ul.lst-kix_4ed4y1s6lj5w-0{list-style-type:none}ul.lst-kix_4ed4y1s6lj5w-4{list-style-type:none}ul.lst-kix_4ed4y1s6lj5w-3{list-style-type:none}ul.lst-kix_4ed4y1s6lj5w-2{list-style-type:none}ul.lst-kix_4ed4y1s6lj5w-1{list-style-type:none}.lst-kix_aps1ppntqbvm-7>li:before{content:"\0025cb  "}.lst-kix_gr4a4jnxityy-5>li:before{content:"\0025a0  "}ul.lst-kix_3jnvhqhv71nk-2{list-style-type:none}.lst-kix_olh7ea290trr-1>li:before{content:"\0025cb  "}ul.lst-kix_3jnvhqhv71nk-3{list-style-type:none}ul.lst-kix_3jnvhqhv71nk-0{list-style-type:none}ul.lst-kix_3jnvhqhv71nk-1{list-style-type:none}ul.lst-kix_cbmcswhr1ftl-0{list-style-type:none}.lst-kix_x7iifx7mthae-4>li:before{content:"\0025cb  "}ul.lst-kix_cbmcswhr1ftl-6{list-style-type:none}ul.lst-kix_cbmcswhr1ftl-5{list-style-type:none}.lst-kix_lf42ux7u6753-4>li:before{content:"\0025cb  "}ul.lst-kix_cbmcswhr1ftl-8{list-style-type:none}ul.lst-kix_cbmcswhr1ftl-7{list-style-type:none}ul.lst-kix_3jnvhqhv71nk-8{list-style-type:none}ul.lst-kix_cbmcswhr1ftl-2{list-style-type:none}ul.lst-kix_3jnvhqhv71nk-7{list-style-type:none}ul.lst-kix_cbmcswhr1ftl-1{list-style-type:none}ul.lst-kix_3jnvhqhv71nk-6{list-style-type:none}ul.lst-kix_cbmcswhr1ftl-4{list-style-type:none}ul.lst-kix_3jnvhqhv71nk-5{list-style-type:none}.lst-kix_9drbe3jtkkuh-2>li:before{content:"" counter(lst-ctn-kix_9drbe3jtkkuh-2,lower-roman) ". "}ul.lst-kix_cbmcswhr1ftl-3{list-style-type:none}ul.lst-kix_3jnvhqhv71nk-4{list-style-type:none}.lst-kix_pizqq4oicz3j-5>li:before{content:"\0025a0  "}ul.lst-kix_7kppodwmns3b-0{list-style-type:none}ul.lst-kix_7kppodwmns3b-1{list-style-type:none}ul.lst-kix_7kppodwmns3b-4{list-style-type:none}.lst-kix_e5f9ugvogyec-2>li:before{content:"\0025a0  "}.lst-kix_myo05blafbk-8>li:before{content:"\0025a0  "}ul.lst-kix_7kppodwmns3b-5{list-style-type:none}ul.lst-kix_7kppodwmns3b-2{list-style-type:none}ul.lst-kix_7kppodwmns3b-3{list-style-type:none}.lst-kix_cmqbbecr9wqr-4>li:before{content:"\0025cb  "}.lst-kix_olh7ea290trr-5>li:before{content:"\0025a0  "}ul.lst-kix_7kppodwmns3b-8{list-style-type:none}ul.lst-kix_7kppodwmns3b-6{list-style-type:none}ul.lst-kix_7kppodwmns3b-7{list-style-type:none}.lst-kix_9drbe3jtkkuh-5>li{counter-increment:lst-ctn-kix_9drbe3jtkkuh-5}.lst-kix_olh7ea290trr-2>li:before{content:"\0025a0  "}.lst-kix_l7t8lma3v5dd-0>li:before{content:"\0025cf  "}.lst-kix_whzqgipsg1bt-4>li:before{content:"\0025cb  "}.lst-kix_cmqbbecr9wqr-3>li:before{content:"\0025cf  "}.lst-kix_z0ttp4mliy4e-5>li:before{content:"\0025a0  "}.lst-kix_vqtjz6k7exn7-8>li:before{content:"\0025a0  "}ul.lst-kix_xc244bhq3ube-2{list-style-type:none}ul.lst-kix_xc244bhq3ube-3{list-style-type:none}ul.lst-kix_xc244bhq3ube-0{list-style-type:none}ul.lst-kix_xc244bhq3ube-1{list-style-type:none}ul.lst-kix_xc244bhq3ube-6{list-style-type:none}ul.lst-kix_xc244bhq3ube-7{list-style-type:none}ul.lst-kix_xc244bhq3ube-4{list-style-type:none}ul.lst-kix_xc244bhq3ube-5{list-style-type:none}.lst-kix_3kgcjgrfg4nh-4>li:before{content:"\0025cb  "}ul.lst-kix_xc244bhq3ube-8{list-style-type:none}.lst-kix_l7t8lma3v5dd-6>li:before{content:"\0025cf  "}.lst-kix_7rljtqi4tpf8-7>li:before{content:"\0025cb  "}.lst-kix_x7iifx7mthae-2>li:before{content:"\0025a0  "}.lst-kix_x7iifx7mthae-3>li:before{content:"\0025cf  "}.lst-kix_10avetia2xt4-5>li:before{content:"\0025a0  "}.lst-kix_pdbqztwjr65g-0>li:before{content:"\0025cf  "}ul.lst-kix_x7iifx7mthae-8{list-style-type:none}ul.lst-kix_x7iifx7mthae-6{list-style-type:none}ol.lst-kix_9drbe3jtkkuh-7.start{counter-reset:lst-ctn-kix_9drbe3jtkkuh-7 0}ul.lst-kix_x7iifx7mthae-7{list-style-type:none}ul.lst-kix_x7iifx7mthae-4{list-style-type:none}ul.lst-kix_x7iifx7mthae-5{list-style-type:none}ul.lst-kix_x7iifx7mthae-2{list-style-type:none}ul.lst-kix_x7iifx7mthae-3{list-style-type:none}.lst-kix_owpom8h3nkwo-5>li:before{content:"\0025a0  "}.lst-kix_g3lepnw8n0x6-8>li:before{content:"\0025a0  "}.lst-kix_owpom8h3nkwo-0>li:before{content:"\0025cf  "}ul.lst-kix_x7iifx7mthae-1{list-style-type:none}ul.lst-kix_x7iifx7mthae-0{list-style-type:none}.lst-kix_velyrizglpvf-2>li:before{content:"\0025a0  "}.lst-kix_vz631vhbp5pc-7>li:before{content:"\0025cb  "}.lst-kix_e5f9ugvogyec-5>li:before{content:"\0025a0  "}.lst-kix_olh7ea290trr-8>li:before{content:"\0025a0  "}.lst-kix_xq12dxu3f08q-4>li:before{content:"\0025cb  "}.lst-kix_olh7ea290trr-3>li:before{content:"\0025cf  "}.lst-kix_g3lepnw8n0x6-6>li:before{content:"\0025cf  "}.lst-kix_cmqbbecr9wqr-8>li:before{content:"\0025a0  "}.lst-kix_z0ttp4mliy4e-4>li:before{content:"\0025cb  "}.lst-kix_4tq1geabcxa-6>li:before{content:"\0025cf  "}ul.lst-kix_euwmeg8cbpt8-2{list-style-type:none}ul.lst-kix_euwmeg8cbpt8-3{list-style-type:none}ul.lst-kix_euwmeg8cbpt8-0{list-style-type:none}ul.lst-kix_euwmeg8cbpt8-1{list-style-type:none}.lst-kix_qzhxzy2sbaso-8>li:before{content:"\0025a0  "}ul.lst-kix_euwmeg8cbpt8-6{list-style-type:none}.lst-kix_xc244bhq3ube-6>li:before{content:"\0025cf  "}ul.lst-kix_euwmeg8cbpt8-7{list-style-type:none}ul.lst-kix_euwmeg8cbpt8-4{list-style-type:none}.lst-kix_hzkhz4hhlr8l-8>li:before{content:"\0025a0  "}ul.lst-kix_euwmeg8cbpt8-5{list-style-type:none}ul.lst-kix_euwmeg8cbpt8-8{list-style-type:none}ol{margin:0;padding:0}.c61{vertical-align:top;width:510.2pt;border-style:solid;background-color:#f3f3f3;border-color:#000000;border-width:0pt;padding:5pt 5pt 5pt 5pt}.c28{vertical-align:top;width:510.2pt;border-style:solid;background-color:#ffffff;border-color:#000000;border-width:0pt;padding:5pt 5pt 5pt 5pt}.c50{vertical-align:top;width:510.2pt;border-style:solid;background-color:#efefef;border-color:#000000;border-width:0pt;padding:5pt 5pt 5pt 5pt}.c41{vertical-align:top;width:58.7pt;border-style:solid;background-color:#f9cb9c;border-color:#b7b7b7;border-width:1pt}.c49{vertical-align:top;width:150.8pt;border-style:solid;border-color:#000000;border-width:1pt;padding:5pt 5pt 5pt 5pt}.c24{vertical-align:top;width:108pt;border-style:solid;border-color:#000000;border-width:1pt;padding:5pt 5pt 5pt 5pt}.c38{vertical-align:middle;width:24.8pt;border-style:solid;background-color:#cccccc;border-color:#b7b7b7;border-width:1pt}.c29{vertical-align:top;width:53.2pt;border-style:solid;background-color:#f9cb9c;border-color:#b7b7b7;border-width:1pt}.c74{vertical-align:top;width:58.7pt;border-style:solid;background-color:#ffffff;border-color:#b7b7b7;border-width:1pt}.c44{vertical-align:middle;width:21.8pt;border-style:solid;background-color:#cccccc;border-color:#b7b7b7;border-width:1pt}.c30{vertical-align:top;width:62.2pt;border-style:solid;border-color:#000000;border-width:1pt;padding:5pt 5pt 5pt 5pt}.c32{vertical-align:middle;width:26.2pt;border-style:solid;background-color:#cccccc;border-color:#b7b7b7;border-width:1pt}.c20{vertical-align:top;width:66pt;border-style:solid;border-color:#000000;border-width:1pt;padding:5pt 5pt 5pt 5pt}.c69{vertical-align:middle;width:35.2pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c52{vertical-align:middle;width:37.5pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c68{vertical-align:middle;width:29.2pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c60{vertical-align:top;width:23.2pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c67{vertical-align:middle;width:41.2pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c75{vertical-align:top;width:53.2pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c39{vertical-align:top;width:41.2pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c33{vertical-align:top;width:30.8pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c70{vertical-align:middle;width:23.2pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c62{vertical-align:middle;width:26.2pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c56{vertical-align:middle;width:30.8pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c73{vertical-align:middle;width:42pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c19{vertical-align:top;width:37.5pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c13{vertical-align:top;width:27.8pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c1{vertical-align:middle;width:27.8pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c51{vertical-align:top;width:21.8pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c45{vertical-align:top;width:26.2pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c8{vertical-align:top;width:29.2pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c72{vertical-align:middle;width:24.8pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c64{vertical-align:middle;width:21.8pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c59{vertical-align:top;width:35.2pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c47{vertical-align:top;width:24.8pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c34{vertical-align:top;width:42pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c48{max-width:510.2pt;background-color:#ffffff;padding:42.5pt 42.5pt 42.5pt 42.5pt}.c46{margin-right:auto;border-collapse:collapse}.c12{padding-left:0pt;margin-left:36pt}.c11{color:#1155cc;text-decoration:underline}.c37{padding-top:14pt;padding-bottom:4pt}.c31{padding-top:14pt;padding-bottom:14pt}.c7{font-size:10pt;font-family:"Courier New"}.c4{font-size:12pt;font-family:"Times New Roman"}.c3{font-family:"Times New Roman";font-weight:bold}.c26{padding-left:4.5pt;margin-left:30.8pt}.c2{text-align:justify;direction:ltr}.c9{text-indent:-18pt;margin-left:36pt}.c25{color:inherit;text-decoration:inherit}.c58{padding-left:0pt;margin-left:72pt}.c5{line-height:1.0;height:11pt}.c15{margin:0;padding:0}.c65{padding-top:24pt;padding-bottom:6pt}.c0{line-height:1.0;direction:ltr}.c6{direction:ltr}.c14{height:11pt}.c17{font-size:14pt}.c22{line-height:1.0}.c43{margin-left:36pt}.c23{margin-left:18pt}.c40{text-indent:36pt}.c10{height:0pt}.c35{height:13pt}.c66{font-size:12pt}.c18{font-weight:bold}.c55{background-color:#cccccc}.c36{color:#cc0000}.c21{text-align:center}.c42{text-decoration:underline}.c63{height:16pt}.c71{font-family:"Courier New"}.c27{font-style:italic}.c53{color:#ff0000}.c16{font-family:"Times New Roman"}.c54{font-size:10pt}.c57{margin-left:54pt}.title{padding-top:0pt;line-height:1.15;text-align:left;color:#000000;font-size:21pt;font-family:"Trebuchet MS";padding-bottom:0pt}.subtitle{padding-top:0pt;line-height:1.15;text-align:left;color:#666666;font-style:italic;font-size:13pt;font-family:"Trebuchet MS";padding-bottom:10pt}li{color:#000000;font-size:11pt;font-family:"Arial"}p{color:#000000;font-size:11pt;margin:0;font-family:"Arial"}h1{padding-top:10pt;line-height:1.15;text-align:left;color:#000000;font-size:16pt;font-family:"Trebuchet MS";padding-bottom:0pt}h2{padding-top:10pt;line-height:1.15;text-align:left;color:#000000;font-size:13pt;font-family:"Trebuchet MS";font-weight:bold;padding-bottom:0pt}h3{padding-top:8pt;line-height:1.15;text-align:left;color:#666666;font-size:12pt;font-family:"Trebuchet MS";font-weight:bold;padding-bottom:0pt}h4{padding-top:8pt;line-height:1.15;text-align:left;color:#666666;font-size:11pt;text-decoration:underline;font-family:"Trebuchet MS";padding-bottom:0pt}h5{padding-top:8pt;line-height:1.15;text-align:left;color:#666666;font-size:11pt;font-family:"Trebuchet MS";padding-bottom:0pt}h6{padding-top:8pt;line-height:1.15;text-align:left;color:#666666;font-style:italic;font-size:11pt;font-family:"Trebuchet MS";padding-bottom:0pt}</style></head><body class="c48"><div><p class="c6 c14"><span></span></p></div><p class="c65 c6 c21 title"><a name="h.5r0tnx612rs0"></a><span class="c3">GeneValidator</span></p><p class="c6 c21 c65 subtitle"><a name="h.jmdsip5u710q"></a><span class="c3">Identify problems with predicted genes</span></p><p class="c6 c14"><span class="c4 c18"></span></p><p class="c6"><span class="c3 c17">This is a </span><span class="c11 c3 c17"><a class="c25" href="">NESCent project</a></span><span class="c3 c17">, funded by Google during </span><span class="c11 c3 c17"><a class="c25" href="http://www.google-melange.com/gsoc/homepage/google/gsoc2013%20">Google Summer of Code 2013</a></span><span class="c3 c17">&nbsp;and mentored by Anurag Priyam and Yannick Wurm.</span></p><h3 class="c6 c37"><a name="h.yluxu51txcqj"></a><span class="c17 c16">Authors</span></h3><ul class="c15 lst-kix_pizqq4oicz3j-0 start"><li class="c12 c6"><span class="c3 c17">GSoC student:</span><span class="c3 c17 c36"><a class="c25" href="http://swarm.cs.pub.ro/%7Emdragan/gsoc2013/Monica_Dragan_CV.pdf">&nbsp;</a></span><span class="c11 c3 c17"><a class="c25" href="http://swarm.cs.pub.ro/%7Emdragan/gsoc2013/Monica_Dragan_CV.pdf">Monica Dragan</a></span><span class="c3 c17 c36">&nbsp;</span></li><li class="c12 c6"><span class="c3 c17">Mentors:</span><span class="c3 c17 c36"><a class="c25" href="https://plus.google.com/114122400102590087616/about">&nbsp;</a></span><span class="c11 c3 c17"><a class="c25" href="https://plus.google.com/114122400102590087616/about">Anurag Priyam</a></span><span class="c3 c17 c36">&nbsp;</span><span class="c3 c17">and</span><span class="c3 c17 c36"><a class="c25" href="http://yannick.poulet.org/">&nbsp;</a></span><span class="c11 c3 c17"><a class="c25" href="http://yannick.poulet.org/">Yannick Wurm</a></span></li></ul><p class="c6 c14"><span class="c17 c36 c16"></span></p><p class="c6"><span class="c3 c17">Resources</span></p><ul class="c15 lst-kix_6064le485ne7-0 start"><li class="c12 c6"><span class="c3 c17">Source code: </span><span class="c11 c3 c17"><a class="c25" href="https://github.com/monicadragan/GeneValidator">github</a></span></li><li class="c12 c6"><span class="c11 c3 c17"><a class="c25" href="http://gene-prediction.blogspot.ch/">Blog</a></span></li><li class="c12 c6"><span class="c11 c3 c17"><a class="c25" href="http://swarm.cs.pub.ro/~mdragan/gsoc2013/genevalidator">Output</a></span></li></ul><p class="c6 c14"><span class="c3 c17 c36 c42"></span></p><p class="c6 c14"><span class="c3 c17 c36 c42"></span></p><p class="c6"><span class="c3 c17">Summary</span></p><p class="c6 c23"><span class="c11 c4"><a class="c25" href="#h.aquxglil7al4">A. Project description</a></span></p><p class="c6 c23"><span class="c11 c4"><a class="c25" href="#h.48ubnp31f0km">B. Validations</a></span></p><p class="c6 c43"><span class="c11 c4"><a class="c25" href="#h.pay33gn4jmyv">1) Length Validation via clusterization</a></span></p><p class="c6 c43"><span class="c11 c4"><a class="c25" href="#h.8l1xsn8jugrl">2) Length Validation via ranking</a></span></p><p class="c6 c43"><span class="c11 c4"><a class="c25" href="#h.x5xi5uouhpqm">3) Duplication Check</a></span></p><p class="c6 c43"><span class="c11 c4"><a class="c25" href="#h.mci2ebh73ez2">4) Gene merge Validation</a></span></p><p class="c6 c43"><span class="c11 c4"><a class="c25" href="#h.lpsoaxpm11ri">5) Validation based on multiple alignment - for proteins only</a></span></p><p class="c6 c43"><span class="c11 c4"><a class="c25" href="#h.f63zo0greej2">6) Blast Reading Frame Validation - for nucleotides only</a></span></p><p class="c6 c43"><span class="c11 c4"><a class="c25" href="#h.3elev4f70kdl">7) Open reading frame - for nucleotides only</a></span></p><p class="c6 c43"><span class="c11 c4"><a class="c25" href="#h.whvjphfxn6rd">8) Codon usage - this is future work</a></span></p><p class="c6 c23"><span class="c11 c4"><a class="c25" href="#h.cw7l8waunvwl">C. Running time</a></span></p><p class="c6 c23"><span class="c11 c4"><a class="c25" href="#h.njhshrkdurp1">D. Code Structure</a></span></p><p class="c6 c23"><span class="c11 c4"><a class="c25" href="#h.8hgqxxlaoy94">E. How to run the tool</a></span></p><p class="c6 c23"><span class="c11 c4"><a class="c25" href="#h.qi34bbc42hqf">F. Add a new validation test</a></span></p><p class="c6 c57"><span class="c11 c4"><a class="c25" href="#h.l1h18sojuad4">Extend 2 parent classes</a></span></p><p class="c6 c57"><span class="c11 c4"><a class="c25" href="#h.52nozdazdy71">Add the new validation to the tool</a></span></p><p class="c6 c57"><span class="c11 c4"><a class="c25" href="#h.j12khye9z0zk">Add plots</a></span></p><p class="c6 c23"><span class="c11 c4"><a class="c25" href="#h.4rd7mxpo7dua">G. Output results</a></span></p><p class="c6 c14"><span class="c4 c18 c36 c42"></span></p><h1 class="c6 c21 c63"><a name="h.htgpgoygm0nz"></a></h1><h1 class="c6 c21 c63"><a name="h.rz0ao0fey7r8"></a></h1><h1 class="c6 c21"><a name="h.aquxglil7al4"></a><span class="c3">A. Project description</span></h1><p class="c6 c14"><span></span></p><p class="c6 c14"><span class="c16"></span></p><p class="c2"><span class="c17 c16">Genome sequencing is now possible at almost no cost. However, obtaining accurate gene predictions remains a target hard to achieve with the existing technology. </span></p><p class="c2 c14"><span class="c17 c16"></span></p><p class="c2"><span class="c17 c16">We designed a tool that identifies problems with gene predictions, based on similarities with data from public databases (like Swissprot and Uniprot). We apply a set of validation tests that provide useful information about the problems that appear in the predictions, in order to make evidence about how the gene curation can be made or whether a certain predicted gene may not be considered in other analysis. </span></p><p class="c2 c14"><span class="c17 c16"></span></p><p class="c2"><span class="c17 c16">Our main target users are the biologists, who have large amounts of genetic data that has to be manually curated, and </span><span class="c17 c16">our tool will prioritize the genes</span><span class="c17 c16">&nbsp;that need to be checked and suggest possible error causes.</span></p><p class="c2 c14"><span class="c3"></span></p><p class="c2 c14"><span class="c3"></span></p><h1 class="c6 c21"><a name="h.48ubnp31f0km"></a><span class="c3">B. Validations</span></h1><p class="c6 c14 c21"><span class="c4"></span></p><p class="c2"><span class="c17 c16">In order to highlight the problems that appear in the predictions and suggest possible error causes, we developed 7 validation test:</span></p><p class="c2 c14"><span class="c17 c16"></span></p><p class="c6"><span class="c3 c17">1) Length validation via clusterization</span></p><p class="c6"><span class="c3 c17">2) Length validation via ranking</span></p><p class="c6"><span class="c3 c17">3) Duplication</span></p><p class="c6"><span class="c3 c17">4) Gene merge</span></p><p class="c6"><span class="c3 c17">5) Multiple alignment</span></p><p class="c6"><span class="c3 c17">6) Blast Reading Frame - applicable only for nucleotide sequences</span></p><p class="c6"><span class="c3 c17">7) Open reading frame - applicable only for nucleotide sequences</span></p><p class="c6 c14"><span class="c3 c17 c27"></span></p><p class="c6 c14"><span class="c3 c17 c27"></span></p><p class="c6 c14"><span class="c3 c17 c27"></span></p><p class="c6 c14"><span class="c3 c17 c27"></span></p><p class="c6 c14"><span class="c3 c17 c27"></span></p><p class="c6 c14"><span class="c3 c17 c27"></span></p><hr style="page-break-before:always;display:none;"><p class="c6 c14"><span class="c3 c17 c27"></span></p><p class="c6"><span class="c17 c16">Data needed for each validation is retrieved from the BLAST output and used as follows (aliases and tags are those used in BLAST):</span></p><p class="c6 c14"><span class="c17 c16"></span></p><p class="c6"><span class="c4 c18">x &nbsp; = mandatory parameter</span></p><p class="c6 c14"><span class="c3"></span></p><a href="#" name="d147262e4d5479fb7862dd16956252730bedd62d"></a><a href="#" name="0"></a><table cellpadding="0" cellspacing="0" class="c46"><tbody><tr class="c10"><td class="c74"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c55 c75"><p class="c0 c21"><span class="c3">Blast param</span></p></td><td class="c56 c55"><p class="c0 c21"><span class="c3">sseqid</span></p></td><td class="c70 c55"><p class="c0 c21"><span class="c3">sacc</span></p></td><td class="c44"><p class="c0 c21"><span class="c3">slen</span></p></td><td class="c55 c56"><p class="c0 c21"><span class="c3">qstart</span></p></td><td class="c1 c55"><p class="c0 c21"><span class="c3">qend</span></p></td><td class="c68 c55"><p class="c0 c21"><span class="c3">sstart</span></p></td><td class="c1 c55"><p class="c0 c21"><span class="c3">send</span></p></td><td class="c55 c69"><p class="c0 c21"><span class="c3">length</span></p></td><td class="c32"><p class="c0 c21"><span class="c3">qseq</span></p></td><td class="c38"><p class="c0 c21"><span class="c3">sseq</span></p></td><td class="c52 c55"><p class="c0 c21"><span class="c3">qframe</span></p></td><td class="c73 c55"><p class="c0 c21"><span class="c3">Query raw seq</span></p></td><td class="c55 c67"><p class="c0 c21"><span class="c3">Hit &nbsp;raw seq</span></p></td></tr><tr><td class="c41"><p class="c0 c21"><span class="c3">Validation</span></p></td><td class="c29"><p class="c0 c21"><span class="c3">Alias</span></p></td><td class="c56"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c70"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c64"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c56"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c1"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c68"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c1"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c69"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c62"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c72"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c52"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c73"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c67"><p class="c5 c6 c21"><span class="c3"></span></p></td></tr><tr><td class="c41"><p class="c0 c21"><span class="c3">Length by clustering </span></p></td><td class="c29"><p class="c0 c21"><span class="c3">lenc</span></p></td><td class="c33"><p class="c0 c21"><span class="c3">x</span></p></td><td class="c60"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c51"><p class="c0 c21"><span class="c3">x</span></p></td><td class="c33"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c13"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c8"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c13"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c59"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c45"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c47"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c19"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c34"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c39"><p class="c5 c6 c21"><span class="c3"></span></p></td></tr><tr><td class="c41"><p class="c0 c21"><span class="c3">Length by rank</span></p></td><td class="c29"><p class="c0 c21"><span class="c3">lenr</span></p></td><td class="c33"><p class="c0 c21"><span class="c3">x</span></p></td><td class="c60"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c51"><p class="c0 c21"><span class="c3">x</span></p></td><td class="c33"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c13"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c8"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c13"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c59"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c45"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c47"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c19"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c34"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c39"><p class="c5 c6 c21"><span class="c3"></span></p></td></tr><tr><td class="c41"><p class="c0 c21"><span class="c3">Reading Frame </span></p></td><td class="c29"><p class="c0 c21"><span class="c3">frame</span></p></td><td class="c33"><p class="c0 c21"><span class="c3">x</span></p></td><td class="c60"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c51"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c33"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c13"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c8"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c13"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c59"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c45"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c47"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c19"><p class="c0 c21"><span class="c3">x</span></p></td><td class="c34"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c39"><p class="c5 c6 c21"><span class="c3"></span></p></td></tr><tr><td class="c41"><p class="c0 c21"><span class="c3">Gene merge</span></p></td><td class="c29"><p class="c0 c21"><span class="c3">merge</span></p></td><td class="c33"><p class="c0 c21"><span class="c3">x</span></p></td><td class="c60"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c51"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c33"><p class="c0 c21"><span class="c3">x</span></p></td><td class="c13"><p class="c0 c21"><span class="c3">x</span></p></td><td class="c8"><p class="c0 c21"><span class="c3">x</span></p></td><td class="c13"><p class="c0 c21"><span class="c3">x</span></p></td><td class="c59"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c45"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c47"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c19"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c34"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c39"><p class="c5 c6 c21"><span class="c3"></span></p></td></tr><tr><td class="c41"><p class="c0 c21"><span class="c3">Duplication</span></p></td><td class="c29"><p class="c0 c21"><span class="c3">dup</span></p></td><td class="c33"><p class="c0 c21"><span class="c3">x</span></p></td><td class="c60"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c51"><p class="c0 c21"><span class="c3">x</span></p></td><td class="c33"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c13"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c8"><p class="c0 c21"><span class="c3">x</span></p></td><td class="c13"><p class="c0 c21"><span class="c3">x</span></p></td><td class="c59"><p class="c0 c21"><span class="c3">x</span></p></td><td class="c45"><p class="c0 c21"><span class="c3">x</span></p></td><td class="c47"><p class="c0 c21"><span class="c3">x</span></p></td><td class="c19"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c34"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c39"><p class="c5 c6 c21"><span class="c3"></span></p></td></tr><tr><td class="c41"><p class="c0 c21"><span class="c3">Open Reading Frame</span></p></td><td class="c29"><p class="c0 c21"><span class="c3">orf</span></p></td><td class="c33"><p class="c0 c21"><span class="c3">x</span></p></td><td class="c60"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c51"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c33"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c13"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c8"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c13"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c59"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c45"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c47"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c19"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c34"><p class="c0 c21"><span class="c3">x</span></p></td><td class="c39"><p class="c5 c6 c21"><span class="c3"></span></p></td></tr><tr><td class="c41"><p class="c0 c21"><span class="c3">Multiple align. based</span></p></td><td class="c29"><p class="c0 c21"><span class="c3">align</span></p></td><td class="c33"><p class="c0 c21"><span class="c3">x</span></p></td><td class="c60"><p class="c0 c21"><span class="c3">x</span></p></td><td class="c51"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c33"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c13"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c8"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c13"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c59"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c45"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c47"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c19"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c34"><p class="c5 c6 c21"><span class="c3"></span></p></td><td class="c39"><p class="c0 c21"><span class="c3">x</span></p></td></tr></tbody></table><p class="c6 c14"><span class="c3"></span></p><p class="c6 c14"><span class="c4 c18"></span></p><p class="c6"><span class="c4 c18">Each validation is described further in this section.</span></p><p class="c6 c14"><span class="c4 c18"></span></p><h2 class="c6"><a name="h.pay33gn4jmyv"></a><span class="c16">1) Length Validation via clusterization</span></h2><a href="#" name="f6da51ae0dd1aac4fd05f3bd2cec16af728807f8"></a><a href="#" name="1"></a><table cellpadding="0" cellspacing="0" class="c46"><tbody><tr class="c10"><td class="c28"><h4 class="c0"><a name="h.1kmh3ff8b3xt"></a><span>Error causes</span></h4><p class="c5 c6"><span></span></p><ul class="c15 lst-kix_3jnvhqhv71nk-0 start"><li class="c12 c0"><span class="c4">sequencing error: some parts of the gene were lost/added on the way</span></li></ul><ul class="c15 lst-kix_mk8t5sy0k589-0 start"><li class="c12 c0"><span class="c4">gene prediction error: after &ldquo;reading&rdquo; a genome sequence with new generation sequencing techniques, the start/end of the gene was not well estimated</span></li><li class="c12 c0"><span class="c4">the gene had a low expression level</span></li><li class="c12 c0"><span class="c4">the sequenced mrna incorrectly contains an some introns </span></li></ul><p class="c5 c6"><span class="c4"></span></p></td></tr><tr><td class="c28"><h4 class="c0"><a name="h.5jizasmleana"></a><span>Input data</span></h4><p class="c5 c6"><span></span></p><ul class="c15 lst-kix_4ed4y1s6lj5w-0 start"><li class="c12 c6"><span class="c4">lengths of the hits: this data is retrieved after parsing the blast output file</span></li><li class="c12 c6"><span class="c4">length of the prediction: this number is the length of the current query from the fasta. In case of nucleotide sequences, we are interested in the length of the corresponding protein translation of the query (which is the length of the query divided by 3, plus or minus 1 or 2 residues, depending on the reading frame -- this detail is not taken into consideration here)</span></li></ul><p class="c5 c6"><span class="c16"></span></p></td></tr><tr class="c10"><td class="c28"><h4 class="c0"><a name="h.wlnymr8sa2ta"></a><span>Class information:</span></h4><p class="c5 c6"><span></span></p><ul class="c15 lst-kix_7rljtqi4tpf8-0 start"><li class="c12 c0"><span class="c4">header: Length Cluster</span></li><li class="c12 c0"><span class="c4">short description: Check whether the prediction length fits most of the BLAST hit lengths, by 1D hierarchical clusterization. Meaning of the output displayed: Prediction_len [Main Cluster Length Interval]</span></li><li class="c12 c0"><span class="c4">alias: lenc</span></li></ul></td></tr><tr><td class="c28"><h4 class="c0"><a name="h.pe5jadwr5x2b"></a><span>Workflow</span></h4><p class="c5 c6"><span></span></p><p class="c0"><span class="c4 c18">Aim :</span><span class="c4">&nbsp;we are interested to find out if the length of the predicted sequence belongs to the distribution of the hit lengths (in other words, how close is the length of the prediction to the majority of the lengths of the hits)</span></p><p class="c5 c6"><span class="c4"></span></p><ul class="c15 lst-kix_xq12dxu3f08q-0 start"><li class="c12 c2 c22"><span class="c4">&nbsp;By plotting the histogram of the length distribution of the hits we observe that the distribution does not fit a Bell Curve, so we cannot apply the classical T-test. </span></li></ul><p class="c2 c5"><span class="c4"></span></p><p class="c2 c5"><span class="c16"></span></p><p class="c0 c21"><img height="347" src="images/image14.jpg" width="347"></p><p class="c0 c21"><span class="c3">(1)</span></p><p class="c2 c5"><span class="c16"></span></p><p class="c2 c22"><span class="c4">Our approach to find the majority of lengths among the reference lengths &nbsp;is by a typical </span><span class="c4 c18">hierarchical clusterization</span><span class="c4">: </span></p><p class="c2 c5"><span class="c4"></span></p><ul class="c15 lst-kix_aps1ppntqbvm-0 start"><li class="c12 c2 c22"><span class="c4">Firstly we assume that each length belongs to a separate cluster. Each step we merge the closest two clusters, until a cluster that contains more than 50% of the reference sequences is obtained. Each clusters is represented with a different colors in Figure 2. The one colored in red is called the &ldquo;main cluster&rdquo;.</span></li></ul><p class="c2 c5"><span class="c16"></span></p><p class="c0 c21"><img height="400" src="images/image16.jpg" width="400"></p><p class="c0 c21"><span class="c3">(2)</span></p><p class="c5 c6 c21"><span class="c16"></span></p><ul class="c15 lst-kix_lf42ux7u6753-0 start"><li class="c12 c2 c22"><span class="c4">Finally we are interested to check whether the length of the</span><span class="c4">&nbsp;prediction belongs to the main cluster or not. The validation test will pass if the length of the prediction belongs to the main cluster of lengths (see Figure 3) and will fail otherwise (Figure 4).</span></li></ul><p class="c2 c5"><span class="c16"></span></p><p class="c2 c5"><span class="c16"></span></p><p class="c0 c21"><img height="269" src="images/image05.jpg" width="280"><img height="268" src="images/image01.png" width="284"></p><p class="c0 c21"><span class="c3">(3) &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; (4)</span></p><p class="c5 c6"><span class="c3"></span></p></td></tr><tr><td class="c28"><h4 class="c5 c6"><a name="h.m7z1grnnpo13"></a></h4><h4 class="c0"><a name="h.37r1qyy1rn29"></a><span class="c11"><a class="c25" href="https://github.com/monicadragan/GeneValidator/blob/rubygem/lib/genevalidator/validation_length_cluster.rb#L67%20">Code</a></span></h4><p class="c5 c6"><span class="c3"></span></p><p class="c0"><span class="c7">#clusterization by length</span></p><p class="c0"><span class="c7">contents = lst.map{ |x| x.length_protein.to_i }.sort{|a,b| a&lt;=&gt;b}</span></p><p class="c5 c6"><span class="c7"></span></p><p class="c0"><span class="c7">hc = HierarchicalClusterization.new(contents)</span></p><p class="c0"><span class="c7">clusters = hc.hierarchical_clusterization</span></p><p class="c5 c6"><span class="c7"></span></p><p class="c0"><span class="c7">max_density = 0;</span></p><p class="c0"><span class="c7">max_density_cluster_idx = 0;</span></p><p class="c0"><span class="c7">clusters.each_with_index do |item, i|</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;if item.density &gt; max_density</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;max_density = item.density</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;max_density_cluster_idx = i;</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;end</span></p><p class="c0"><span class="c7">end</span></p><p class="c5 c6"><span class="c7"></span></p><p class="c0"><span class="c7">clusterization = [clusters, max_density_cluster_idx]</span></p><p class="c5 c6"><span class="c7"></span></p><p class="c0"><span class="c7">@clusters = clusterization[0]</span></p><p class="c0"><span class="c7">@max_density_cluster = clusterization[1]</span></p><p class="c0"><span class="c7">limits = @clusters[@max_density_cluster].get_limits</span></p><p class="c0"><span class="c7">prediction_len = @prediction.length_protein</span></p><p class="c5 c6"><span class="c7"></span></p><p class="c0"><span class="c7">@validation_report = LengthClusterValidationOutput.new(prediction_len, limits)</span></p><p class="c0"><span class="c7">plot1 = plot_histo_clusters</span></p><p class="c0"><span class="c7">@validation_report.plot_files.push(plot1)</span></p><p class="c0"><span class="c7">plot2 = plot_len_clusters</span></p><p class="c0"><span class="c7">@validation_report.plot_files.push(plot2)</span></p><p class="c5 c6"><span class="c16 c54"></span></p><p class="c5 c6"><span class="c16 c54"></span></p></td></tr><tr><td class="c28"><h4 class="c0"><a name="h.t0qkl9cnekf7"></a><span>Plots</span></h4><p class="c5 c6"><span></span></p><ul class="c15 lst-kix_euwmeg8cbpt8-0 start"><li class="c12 c2 c22"><span class="c4">length distribution histogram </span><span class="c4">(prediction passes the validation test in Figure 3 and fails in Figure 4)</span></li></ul><p class="c2 c5"><span class="c16"></span></p><p class="c0 c21"><img height="266" src="images/image05.jpg" width="276"><img height="266" src="images/image01.png" width="282"></p><p class="c0 c21"><span class="c3">&nbsp; (3) &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; (4) </span></p><p class="c5 c6 c21"><span class="c3"></span></p><ul class="c15 lst-kix_x7iifx7mthae-0 start"><li class="c12 c0"><span class="c4">line plot representing the regions of the hits that are matched by the prediction query</span></li></ul><p class="c5 c6"><span class="c16"></span></p><p class="c0 c21"><img height="373" src="images/image10.png" width="452"></p><p class="c5 c6"><span class="c3"></span></p></td></tr></tbody></table><h2 class="c6"><a name="h.8l1xsn8jugrl"></a><span class="c16">2) Length Validation via ranking</span></h2><p class="c6 c14"><span class="c16"></span></p><a href="#" name="b8b6b60508fb87f593280c7164c3b361aa7e0ec4"></a><a href="#" name="2"></a><table cellpadding="0" cellspacing="0" class="c46"><tbody><tr class="c10"><td class="c50"><h4 class="c0"><a name="h.c49wi06d2y6n"></a><span class="c66">Error causes</span></h4><p class="c5 c6"><span></span></p><ul class="c15 lst-kix_mk8t5sy0k589-0"><li class="c12 c0"><span class="c4">sequencing error: some parts of the gene were lost/added on the way</span></li><li class="c12 c0"><span class="c4">gene prediction error: after &ldquo;reading&rdquo; a genome sequence with new generation sequencing techniques, the start/end of the gene was not well estimated</span></li><li class="c12 c0"><span class="c4">the gene had a low expression level</span></li><li class="c12 c0"><span class="c4">the introns were not well removed from the gene</span></li></ul></td></tr><tr class="c10"><td class="c50"><h4 class="c0"><a name="h.59mfpm8tx3g3"></a><span>Input data</span></h4><p class="c5 c6"><span></span></p><ul class="c15 lst-kix_4ed4y1s6lj5w-0"><li class="c12 c6"><span class="c4">lengths of the hits: this data is retrieved after parsing the blast output file</span></li><li class="c12 c6"><span class="c4">length of the prediction: this number is the length of the current query from the fasta. In case of nucleotide sequences, we are interested in the length of the corresponding protein translation of the query (which is the length of the query divided by 3, plus or minus 1 or 2 residues, depending on the reading frame -- this detail is not taken into consideration here)</span></li></ul></td></tr><tr class="c10"><td class="c50"><h4 class="c0"><a name="h.z3l9ocp2n17f"></a><span>Class information:</span></h4><p class="c5 c6"><span></span></p><ul class="c15 lst-kix_7kppodwmns3b-0 start"><li class="c12 c0"><span class="c4">header: Length Rank</span></li><li class="c12 c0"><span class="c4">description: Check whether the rank of the prediction length lies among 80% of all the BLAST hit lengths. Meaning of the output displayed: no of extreme length hits / total no of hits</span></li><li class="c12 c0"><span class="c4">alias: lenr</span></li></ul></td></tr><tr class="c10"><td class="c50"><h4 class="c0"><a name="h.rphxumlynikz"></a><span>Workflow</span></h4><p class="c5 c6"><span class="c4 c18"></span></p><p class="c0"><span class="c4 c18">Aim: </span><span class="c4">We want to see how extreme the length of the prediction is among all the other hits. </span></p><p class="c0"><span class="c4 c18">Approach:</span></p><ul class="c15 lst-kix_velyrizglpvf-0 start"><li class="c12 c0"><span class="c4">sort the hit lengths in ascending order.</span></li><li class="c12 c0"><span class="c4">find the median for the set of hit lengths</span></li><li class="c12 c0"><span class="c4">find which side of the median the length of the prediction belongs to</span></li><li class="c12 c0"><span class="c4">case 1: if the prediction length is on the right side of the median, count how many hit lengths are longer than the prediction length (Figure 1)</span></li><li class="c12 c0"><span class="c4">case 2: otherwise count how ma</span><span class="c4">ny hit lengths are</span><span class="c4">&nbsp;shorter than the prediction length (Figure 2)</span></li><li class="c12 c0"><span class="c4">hits that are longer than the prediction length (respectively shorter in case 2) are called extreme hits</span></li><li class="c12 c0"><span class="c4">return the percentage of the hits that are extreme hits. If this percentage is higher than 20%, the prediction will not pass this validation.</span></li></ul><p class="c5 c6"><span class="c4"></span></p><p class="c0 c21"><img height="263" src="images/image01.png" width="279"><span class="c4">&nbsp; &nbsp; &nbsp; </span><img height="262" src="images/image09.png" width="271"></p><p class="c0 c21"><span class="c4 c18">Figure 1 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; Figure 2</span></p></td></tr><tr class="c10"><td class="c50"><h4 class="c0"><a name="h.2tsg46gmlbs8"></a><span class="c11"><a class="c25" href="https://github.com/monicadragan/GeneValidator/blob/rubygem/lib/genevalidator/validation_length_rank.rb#L65%20">Code</a></span></h4><p class="c5 c6"><span class="c3"></span></p><p class="c0"><span class="c7">threshold = 0.2</span></p><p class="c0"><span class="c7">lengths = hits.map{ |x| x.length_protein.to_i }.sort{|a,b| a&lt;=&gt;b}</span></p><p class="c0"><span class="c7">len = lengths.length</span></p><p class="c0"><span class="c7">median = len % 2 == 1 ?</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp;lengths[len/2] :</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp;(lengths[len/2 - 1] + lengths[len/2]).to_f / 2</span></p><p class="c0"><span class="c7">predicted_len = prediction.length_protein</span></p><p class="c5 c6"><span class="c7"></span></p><p class="c0"><span class="c7">if hits.length == 1</span></p><p class="c0"><span class="c7">&nbsp; &nbsp;msg = &quot;&quot;</span></p><p class="c0"><span class="c7">&nbsp; &nbsp;percentage = 1</span></p><p class="c0"><span class="c7">else</span></p><p class="c0"><span class="c7">&nbsp; &nbsp;if predicted_len &lt; median</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;rank = lengths.find_all{|x| x &lt; predicted_len}.length</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;percentage = rank / (len + 0.0)</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;msg = &quot;TOO_SHORT&quot;</span></p><p class="c0"><span class="c7">&nbsp; &nbsp;else</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;rank = lengths.find_all{|x| x &gt; predicted_len}.length</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;percentage = rank / (len + 0.0)</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;msg = &quot;TOO_LONG&quot;</span></p><p class="c0"><span class="c7">&nbsp; &nbsp;end</span></p><p class="c0"><span class="c7">end</span></p><p class="c5 c6"><span class="c7"></span></p><p class="c0"><span class="c7">if percentage &gt;= threshold</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;msg = &quot;&quot;</span></p><p class="c0"><span class="c7">end</span></p><p class="c0"><span class="c7">&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;@validation_report = LengthRankValidationOutput.new(msg, percentage.round(2))</span></p><p class="c5 c6"><span class="c16 c54"></span></p></td></tr><tr class="c10"><td class="c50"><h4 class="c0"><a name="h.h26xfjvngbfo"></a><span>Plots</span></h4><p class="c0"><span class="c16">No plots are generated</span></p></td></tr></tbody></table><p class="c6 c14"><span class="c3"></span></p><h2 class="c6 c35"><a name="h.2taje05ufk8a"></a></h2><h2 class="c6"><a name="h.x5xi5uouhpqm"></a><span class="c16">3) Duplication Check</span></h2><p class="c6 c14"><span class="c16"></span></p><a href="#" name="e289c66d687a6eeaadf01c6065d46ede4bb64a23"></a><a href="#" name="3"></a><table cellpadding="0" cellspacing="0" class="c46"><tbody><tr class="c10"><td class="c28"><h4 class="c0"><a name="h.tmr9j2ocl9mh"></a><span>Error causes</span></h4><ul class="c15 lst-kix_9a6zvrglxdu8-0 start"><li class="c0 c12"><span class="c4">sequencing error</span></li><li class="c12 c0"><span class="c4">error in the replication mechanism of the cell</span></li></ul></td></tr><tr class="c10"><td class="c28"><h4 class="c0"><a name="h.gqw8t05qbpfv"></a><span>Input data</span></h4><ul class="c15 lst-kix_4ed4y1s6lj5w-0"><li class="c12 c6"><span class="c4">raw sequence for prediction</span></li><li class="c12 c6"><span class="c4">raw sequence for the best n hits</span></li><li class="c12 c6"><span class="c4">hit length</span></li><li class="c12 c6"><span class="c4">start/end index of the high-scoring segment pairs (hsp) for hits and query</span></li></ul><p class="c5 c6"><span class="c4"></span></p></td></tr><tr class="c10"><td class="c28"><h4 class="c0"><a name="h.lm93gxe78op4"></a><span>Class information:</span></h4><ul class="c15 lst-kix_e5f9ugvogyec-0 start"><li class="c12 c0"><span class="c4">header: Duplication</span></li><li class="c12 c0"><span class="c4">description: Check whether there is a duplicated subsequence in the predicted gene by counting the hsp residue coverage of the prediction, for each hit. Meaning of the output displayed: P-value of the Wilcoxon test which test the distribution of hit average coverage against 1. P-values higher than 5% pass the validation test.</span></li><li class="c12 c0"><span class="c4">alias: dup</span></li></ul><p class="c5 c6"><span class="c4"></span></p></td></tr><tr class="c10"><td class="c28"><h4 class="c0"><a name="h.ruzfferf17cu"></a><span>Workflow</span></h4><ul class="c15 lst-kix_sjinkw26u0wp-0 start"><li class="c12 c6"><span class="c4">get raw sequences for the first n hits (if they have not been already calculated)</span></li><li class="c12 c6"><span class="c4">in case of nucleotides -&gt; translate the </span><span class="c4">hsp in reading frame 0 or 1 (</span><span class="c4">hsp = </span><span class="c4">high-scoring segment</span><span class="c4">, i.e a pair of segments from the hit and the query that match with a high identity percent</span><span class="c4">) </span></li></ul><p class="c6 c14"><span class="c4"></span></p><p class="c6"><span class="c4">For each hsp:</span></p><ul class="c15 lst-kix_sjinkw26u0wp-0"><li class="c12 c6"><span class="c4">subtract the raw sequence of the prediction</span></li><li class="c12 c6"><span class="c4">subtract the raw sequence of the hsps</span></li><li class="c12 c6"><span class="c4">do local alignment for each hsp (in BLAST xml output the alignment is already provided)</span></li><li class="c12 c6"><span class="c4">initialize a coverage vector for every hit</span></li><li class="c12 c6"><span class="c4">pass through the alignment and increase by 1 the corresponding position in the coverage vector</span></li><li class="c12 c6"><span class="c4">calculate the average value of the values in the coverage vector. For a prediction without duplications, the average value should be very close to 1. For all the n hits we will end up with a vector of n average values </span></li><li class="c12 c6"><span class="c4">trick: average values are roun</span><span class="c4">ded to 2 d</span><span class="c4">ecimals to make the data more uniform </span></li><li class="c12 c6"><span class="c4">we test the null hypothesis (if the mean of the average values is different from one) using the one sample wilcoxon te</span><span class="c4">st and return the p-value. In case the p-value is higher than 5%, the null hypothesis is true and there is no duplication in the prediction. If the p-value is lower than 5%, then the </span><span class="c4 c18">alternative hypothesis </span><span class="c4">is true (and the mean is different from one) and </span><span class="c4">in this case there is a duplication in the sequence. </span></li></ul></td></tr><tr class="c10"><td class="c28"><h4 class="c0"><a name="h.90zrmq37aeyh"></a><span class="c11"><a class="c25" href="https://github.com/monicadragan/GeneValidator/blob/rubygem/lib/genevalidator/validation_duplication.rb#L65%20">Code</a></span><span>&nbsp;</span></h4><p class="c5 c6"><span class="c3"></span></p><p class="c0"><span class="c7"># get the first n hits</span></p><p class="c0"><span class="c7">less_hits = @hits[0..[n-1,@hits.length].min]</span></p><p class="c0"><span class="c7">useless_hits = []</span></p><p class="c0"><span class="c7">begin</span></p><p class="c0"><span class="c7">&nbsp; # get raw sequences for less_hits</span></p><p class="c0"><span class="c7">&nbsp; less_hits.map do |hit|</span></p><p class="c0"><span class="c7">&nbsp; #get gene by accession number</span></p><p class="c0"><span class="c7">&nbsp; if hit.raw_sequence == nil</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; if hit.type == :protein</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;hit.get_sequence_by_accession_no(hit.accession_no, &quot;protein&quot;)</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; else</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;hit.get_sequence_by_accession_no(hit.accession_no, &quot;nucleotide&quot;)</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; end</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; if hit.raw_sequence == &quot;&quot;</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;useless_hits.push(hit)</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; end</span></p><p class="c0"><span class="c7">&nbsp; end</span></p><p class="c0"><span class="c7">end</span></p><p class="c0"><span class="c7">rescue Exception =&gt; error</span></p><p class="c0"><span class="c7">&nbsp; &nbsp;raise NoInternetError</span></p><p class="c0"><span class="c7">end</span></p><p class="c5 c6"><span class="c7"></span></p><p class="c0"><span class="c7">useless_hits.each{|hit| less_hits.delete(hit)}</span></p><p class="c0"><span class="c7">averages = []</span></p><p class="c0"><span class="c7">less_hits.each do |hit|</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;coverage = Array.new(hit.length_protein,0)</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;hit.hsp_list.each do |hsp|</span></p><p class="c5 c6"><span class="c7"></span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;# align subsequences from the hit and prediction that match (if it&#39;s the case)</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;if hsp.hit_alignment != nil and hsp.query_alignment != nil</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; hit_alignment = hsp.hit_alignment</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; query_alignment = hsp.query_alignment</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;else</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; # indexing in blast starts from 1</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; hit_local = hit.raw_sequence[hsp.hit_from-1..hsp.hit_to-1]</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; query_local = prediction.raw_sequence[hsp.match_query_from-1..hsp.match_query_to-1]</span></p><p class="c5 c6"><span class="c7"></span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; # in case of nucleotide prediction sequence translate into protein</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; # use translate with reading frame 1 because</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; # to/from coordinates of the hsp already correspond to the</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; # reading frame in which the prediction was read to match this hsp</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; if @type == :nucleotide</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; s = Bio::Sequence::NA.new(query_local)</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;query_local = s.translate</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; end</span></p><p class="c5 c6"><span class="c7"></span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; # local alignment for hit and query</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; seqs = [hit_local, query_local]</span></p><p class="c5 c6"><span class="c7"></span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; begin</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;options = [&#39;--maxiterate&#39;, &#39;1000&#39;, &#39;--localpair&#39;, &#39;--quiet&#39;]</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;mafft = Bio::MAFFT.new(@mafft_path, options)</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;report = mafft.query_align(seqs)</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;raw_align = report.alignment</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;align = []</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;raw_align.each { |s| align.push(s.to_s) }</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;hit_alignment = align[0]</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;query_alignment = align[1]</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;rescue Exception =&gt; error</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;raise NoMafftInstallationError</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; end</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;end</span></p><p class="c5 c6"><span class="c7"></span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;# check multiple coverage</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;# for each hsp of the curent hit</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;# iterate through the alignment and count the matching residues</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[*(0 .. hit_alignment.length-1)].each do |i|</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;residue_hit = hit_alignment[i]</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;residue_query = query_alignment[i]</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;if residue_hit != &#39; &#39; and residue_hit != &#39;+&#39; and residue_hit != &#39;-&#39;</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; if residue_hit == residue_query</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; # indexing in blast starts from 1</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; &nbsp;idx = i + (hsp.hit_from-1) - hit_alignment[0..i].scan(/-/).length</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; coverage[idx] += 1</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; end</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;end</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;end</span></p><p class="c0"><span class="c7">&nbsp; &nbsp;end</span></p><p class="c0"><span class="c7">&nbsp; &nbsp;overlap = coverage.reject{|x| x==0}</span></p><p class="c0"><span class="c7">&nbsp; &nbsp;if overlap != []</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;averages.push(overlap.inject(:+)/(overlap.length + 0.0)).map{|x| x.round(2)}</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; end</span></p><p class="c0"><span class="c7">end</span></p><p class="c5 c6"><span class="c7"></span></p><p class="c0"><span class="c7"># if all hsps match only one time</span></p><p class="c0"><span class="c7">if averages.reject{|x| x==1} == []</span></p><p class="c0"><span class="c7">&nbsp; &nbsp;@validation_report = DuplicationValidationOutput.new(1)</span></p><p class="c0"><span class="c7">end</span></p><p class="c5 c6"><span class="c7"></span></p><p class="c0"><span class="c7">pval = wilcox_test(averages)</span></p><p class="c0"><span class="c7">@validation_report = DuplicationValidationOutput.new(pval)</span></p></td></tr><tr class="c10"><td class="c28"><h4 class="c0"><a name="h.jl85dzs94b1p"></a><span>Plots</span></h4><p class="c0"><span class="c4">No plots are generated</span></p></td></tr></tbody></table><p class="c6 c14"><span class="c16"></span></p><h2 class="c6"><a name="h.mci2ebh73ez2"></a><span class="c16">4) </span><span class="c16">Gene merge Validation</span></h2><p class="c6 c14"><span class="c16"></span></p><a href="#" name="604669cbf97079e10687e9c1b1b6608693bab5d0"></a><a href="#" name="4"></a><table cellpadding="0" cellspacing="0" class="c46"><tbody><tr class="c10"><td class="c61"><h4 class="c0"><a name="h.bg9kc4bj0vxm"></a><span>Error cause</span></h4><p class="c5 c6"><span></span></p><ul class="c15 lst-kix_gr4a4jnxityy-0 start"><li class="c12 c0"><span class="c4">gene prediction error (2 or more genes were merged together)</span></li></ul><p class="c5 c6"><span class="c4"></span></p></td></tr><tr class="c10"><td class="c61"><h4 class="c0"><a name="h.bdyywegj4lax"></a><span>Input data</span></h4><p class="c5 c6"><span></span></p><ul class="c15 lst-kix_4ed4y1s6lj5w-0"><li class="c12 c6"><span class="c4">start/end index of the</span><span class="c4">&nbsp;high-scoring segment pairs (hsp)</span><span class="c4">&nbsp;for hits and query</span></li></ul></td></tr><tr class="c10"><td class="c61"><h4 class="c0"><a name="h.acf7h4g6cpzt"></a><span>Class information:</span></h4><p class="c5 c6"><span></span></p><ul class="c15 lst-kix_cmqbbecr9wqr-0 start"><li class="c12 c0"><span class="c4">header: Gene Merge</span></li><li class="c12 c0"><span class="c4">description: Check whether BLAST hits make evidence about a merge of two genes that match the predicted gene. Meaning of the output displayed: slope of the linear regression of the relationship between the start and stop offsets of the hsps (see the plot). Invalid slopes are around 45 degrees.</span></li><li class="c12 c0"><span class="c4">alias: merge</span></li></ul><p class="c5 c6"><span class="c4"></span></p></td></tr><tr class="c10"><td class="c61"><h4 class="c0"><a name="h.m1smetvgx0h8"></a><span>Workflow</span></h4><p class="c5 c6"><span></span></p><ul class="c15 lst-kix_myo05blafbk-0 start"><li class="c12 c0"><span class="c4">plot a 2D graph using the start/end offsets of the high-scoring segments in the prediction that match the hits (due to blast local alignment)</span></li><li class="c12 c0"><span class="c4">draw a line obtained by linear regression </span></li><li class="c12 c0"><span class="c4">the prediction is a merge of two genes if the slope of the line is between 0.4 and 1.2 (these thresholds are obtained empirically, after observing the slope of a large amount of queries). An image depicting this is available just below:</span></li></ul><p class="c5 c6"><span class="c4"></span></p><p class="c0 c21"><img height="368" src="images/image06.png" width="578"></p><p class="c5 c6"><span class="c4"></span></p></td></tr><tr class="c10"><td class="c61"><h4 class="c0"><a name="h.lbam3as9ua3u"></a><span class="c11"><a class="c25" href="https://github.com/monicadragan/GeneValidator/blob/rubygem/lib/genevalidator/validation_gene_merge.rb#L82%20">Code</a></span></h4><p class="c5 c6"><span class="c3"></span></p><p class="c0"><span class="c7">lm_slope = slope[1]</span></p><p class="c0"><span class="c7">y_intercept = slope[0]</span></p><p class="c5 c6"><span class="c7"></span></p><p class="c0"><span class="c7">@validation_report = GeneMergeValidationOutput.new(lm_slope)</span></p><p class="c5 c6"><span class="c7"></span></p><p class="c0"><span class="c7">plot1 = plot_2d_start_from(lm_slope, y_intercept)</span></p><p class="c0"><span class="c7">@validation_report.plot_files.push(plot1)</span></p><p class="c0"><span class="c7">plot2 = plot_matched_regions</span></p><p class="c0"><span class="c7">@validation_report.plot_files.push(plot2)</span></p><p class="c0"><span class="c7">@running_time = Time.now - start</span></p><p class="c0"><span class="c7">return @validation_report</span></p></td></tr><tr class="c10"><td class="c61"><h4 class="c0"><a name="h.7u98e7kx9p5n"></a><span>Plots</span></h4><ul class="c15 lst-kix_whzqgipsg1bt-0 start"><li class="c12 c0"><span class="c4">scatterplot representing a 2D plot of the start/end offsets of the high-scoring segments in the prediction that match the hits (gene merge in Figure 1, no merge in Figure 2)</span></li></ul><p class="c0 c21"><img height="243" src="images/image11.png" width="269"><img height="242" src="images/image04.png" width="258"></p><p class="c0 c21"><span class="c3">(1) &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;(2)</span></p><ul class="c15 lst-kix_3kgcjgrfg4nh-0 start"><li class="c12 c0"><span class="c4">line plot representing the regions in the prediction that are matched by each </span><span class="c4">hit</span><span class="c4">. The plot highlights the case when there is a merge between multiple genes (Figure 3) or no merge (Figure 4)</span></li></ul><p class="c5 c6"><span class="c16"></span></p><p class="c0 c21"><img height="230" src="images/image03.png" width="293"><img height="231" src="images/image07.png" width="276"></p><p class="c0 c21"><span class="c3">(3) &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; (4)</span></p></td></tr></tbody></table><h2 class="c6 c35"><a name="h.hxel8tqfbg4y"></a></h2><h2 class="c6"><a name="h.lpsoaxpm11ri"></a><span class="c16">5) Validation based on multiple alignment - for proteins only</span></h2><a href="#" name="4e71e50f856d7c889a35c5c0981af14627a0ec28"></a><a href="#" name="5"></a><table cellpadding="0" cellspacing="0" class="c46"><tbody><tr class="c10"><td class="c28"><h4 class="c0"><a name="h.8l6fap8glzwt"></a><span>Error causes</span></h4><ul class="c15 lst-kix_hzkhz4hhlr8l-0 start"><li class="c12 c0"><span class="c4">sequencing error</span></li></ul></td></tr><tr class="c10"><td class="c28"><h4 class="c0"><a name="h.pasu6gpa26d"></a><span>Input data</span></h4><ul class="c15 lst-kix_4ed4y1s6lj5w-0"><li class="c12 c6"><span class="c4">raw sequence for prediction</span></li><li class="c12 c6"><span class="c4">raw sequence for the best n hits</span></li></ul><p class="c5 c6"><span class="c4"></span></p></td></tr><tr class="c10"><td class="c28"><h4 class="c0"><a name="h.34yhnxv3f21q"></a><span>Class information:</span></h4><ul class="c15 lst-kix_4tq1geabcxa-0 start"><li class="c12 c0"><span class="c4">header = &quot;Missing/Extra sequences&quot;</span></li><li class="c12 c0"><span class="c4">description = &quot;Finds missing and extra sequences in the prediction, based on the multiple alignment of the best hits. Meaning of the output displayed: the percentages of the missing/extra sequences with respect to the multiple alignment. Validation fails if one of these values is higher than 20%</span></li><li class="c12 c0"><span class="c4">alias = &quot;align&quot;</span></li></ul><p class="c5 c6"><span class="c4"></span></p></td></tr><tr class="c10"><td class="c28"><h4 class="c0"><a name="h.i4z2t4x1xxka"></a><span>Workflow</span></h4><ul class="c15 lst-kix_olh7ea290trr-0 start"><li class="c12 c6"><span class="c4">get raw sequences for the first n hits (if they have not been already calculated)</span></li><li class="c12 c6"><span class="c4">multiple align the hits with the prediction (mafft)</span></li><li class="c12 c6"><span class="c4">compute the pssm (Position Specific Scoring Matrix) profile for the hits</span></li><li class="c12 c6"><span class="c4">remove isolated residues from the hits and from the prediction</span></li><li class="c12 c6"><span class="c4">compute gap coverage: count gaps that are in the prediction but not in the statistical model (see Figure </span><span class="c4">1</span><span class="c4">) and return the percentage of the missing residues from the total length of the prediction</span></li><li class="c12 c6"><span class="c4">extra sequences: count residues that are in the prediction but not in the statistical model (see Figure 1) and return the percentage of the extra residues from the total length of the prediction</span></li><li class="c12 c6"><span class="c4">conserved regions: count the number of the residues conserved among the hits that appear in the prediction and return the percentage of conserved residues from the total length of the prediction</span></li><li class="c12 c6"><span class="c4">a gene which passes the test would have both the percentage of missing and extra sequences lower than 20%. Otherwise the query will fail on this validation test.</span></li></ul><p class="c6 c14"><span class="c4"></span></p><p class="c6 c21"><img height="108" src="images/image12.png" width="547"></p><p class="c6 c21"><span class="c4 c18">(1)</span></p></td></tr><tr class="c10"><td class="c28"><h4 class="c0"><a name="h.xm9ljjeu6om5"></a><span class="c11"><a class="c25" href="https://github.com/monicadragan/GeneValidator/blob/rubygem/lib/genevalidator/validation_alignment.rb#L65%20">Code</a></span></h4><p class="c5 c6"><span class="c16 c54"></span></p><p class="c0"><span class="c7"># get the first n hits</span></p><p class="c0"><span class="c7">less_hits = @hits[0..[n-1,@hits.length].min]</span></p><p class="c0"><span class="c7">useless_hits = []</span></p><p class="c0"><span class="c7">begin</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; # get raw sequences for less_hits</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; less_hits.map do |hit|</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;if hit.raw_sequence == nil</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; #get gene by accession number</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; if hit.type == :protein</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;hit.get_sequence_by_accession_no(hit.accession_no, &quot;protein&quot;)</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; else</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;hit.get_sequence_by_accession_no(hit.accession_no, &quot;nucleotide&quot;)</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; end</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; if hit.raw_sequence == &quot;&quot;</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;useless_hits.push(hit)</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; end</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;end</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; end</span></p><p class="c0"><span class="c7">end</span></p><p class="c0"><span class="c7">useless_hits.each{|hit| less_hits.delete(hit)}</span></p><p class="c5 c6"><span class="c7"></span></p><p class="c0"><span class="c7">begin</span></p><p class="c0"><span class="c7">&nbsp; # multiple align sequences from &nbsp;less_hits with the prediction</span></p><p class="c0"><span class="c7">&nbsp; multiple_align_mafft(prediction, less_hits)</span></p><p class="c0"><span class="c7">&nbsp; rescue Exception =&gt; error</span></p><p class="c0"><span class="c7">&nbsp; raise NoInternetError</span></p><p class="c0"><span class="c7">end</span></p><p class="c5 c6"><span class="c7"></span></p><p class="c0"><span class="c7">sm &nbsp;= get_sm_pssm(@multiple_alignment[0..@multiple_alignment.length-2])</span></p><p class="c5 c6"><span class="c7"></span></p><p class="c0"><span class="c7"># remove isolated residues from the predicted sequence</span></p><p class="c0"><span class="c7">prediction_raw = remove_isolated_residues(@multiple_alignment[@multiple_alignment.length-1])</span></p><p class="c0"><span class="c7"># remove isolated residues from the statistical model</span></p><p class="c0"><span class="c7">sm = remove_isolated_residues(sm)</span></p><p class="c5 c6"><span class="c7"></span></p><p class="c0"><span class="c7">plot1 = plot_alignment(sm)</span></p><p class="c0"><span class="c7">gaps = gap_validation(prediction_raw, sm)</span></p><p class="c0"><span class="c7">extra_seq = extra_sequence_validation(prediction_raw, sm)</span></p><p class="c0"><span class="c7">consensus = consensus_validation(prediction_raw, get_consensus(@multiple_alignment[0..@multiple_alignment.length-2]))</span></p><p class="c5 c6"><span class="c7"></span></p><p class="c0"><span class="c7">@validation_report = AlignmentValidationOutput.new(gaps, extra_seq, consensus)</span></p><p class="c0"><span class="c7">@validation_report.plot_files.push(plot1)</span></p><p class="c5 c6"><span class="c16 c54"></span></p></td></tr><tr class="c10"><td class="c28"><h4 class="c0"><a name="h.du366f6idqnj"></a><span>Plots</span></h4><ul class="c15 lst-kix_yghjnj8s9xuc-0 start"><li class="c12 c0"><span class="c4">line plot representing the multiple alignment, the prediction and the statistical model (colors highlight gaps and extra sequences and consensus regions among the hits). Figure 1 - good prediction, Figure 2- weak prediction.</span></li></ul><p class="c5 c6"><span class="c4"></span></p><p class="c0 c21"><img height="232" src="images/image13.png" width="293"><img height="229" src="images/image00.png" width="280"></p><p class="c0 c21"><span class="c3">(1) &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;(2)</span></p></td></tr></tbody></table><p class="c6 c14"><span class="c3"></span></p><h2 class="c6"><a name="h.f63zo0greej2"></a><span class="c16">6) Blast Reading Frame Validation - for nucleotides only</span></h2><a href="#" name="62e5f0a5efee187f3a460e06a2908cf912568aa3"></a><a href="#" name="6"></a><table cellpadding="0" cellspacing="0" class="c46"><tbody><tr class="c10"><td class="c61"><h4 class="c0"><a name="h.f7tme0bpawv4"></a><span>Error causes</span></h4><ul class="c15 lst-kix_vqtjz6k7exn7-0 start"><li class="c12 c0"><span class="c4">frame shift sequencing errors: various reading frames of the same sign (e.g single nucleotide was accidentally inserted, sequences are read in the same direction). </span></li><li class="c12 c2 c22"><span class="c4">various reading frames of different sign may still highlight a problem in the prediction, not a reading frame shift, but a possible merge between two genes (and this case is treated by the gene merge validation)</span><span class="c4 c18">&nbsp;</span></li></ul></td></tr><tr class="c10"><td class="c61"><h4 class="c0"><a name="h.ug7m82xgyjul"></a><span>Input data</span></h4><ul class="c15 lst-kix_4ed4y1s6lj5w-0"><li class="c12 c6"><span class="c4">reading frame provided by the blast output</span></li></ul></td></tr><tr class="c10"><td class="c61"><h4 class="c0"><a name="h.3zrcjhz1synl"></a><span>Class information:</span></h4><ul class="c15 lst-kix_48dltnwvsqfp-0 start"><li class="c58 c0"><span class="c4">header: Reading Frame</span></li><li class="c58 c0"><span class="c4">description: Check whether there is a single reading frame among BLAST hits. Otherwise there might be a reading frame shift in the query sequence. Meaning of the output displayed: (reading frame: no hsps)</span></li><li class="c58 c0"><span class="c4">alias: frame</span></li></ul><p class="c5 c6"><span class="c4"></span></p></td></tr><tr class="c10"><td class="c61"><h4 class="c0"><a name="h.j0jjou21atn5"></a><span>Workflow</span></h4><ul class="c15 lst-kix_kmsnp6mmp404-0 start"><li class="c12 c0"><span class="c4">blastx translates each query in each of the six reading frames, and all the resulting six-protein sequences are compared to data in the databases (there&#39;s no evidence where the codon starts from, so all the 3 possibilities of starting the gene translation are considered in BLASTX; also another 3 possibilities when reading the sequence in reverse). </span></li></ul><ul class="c15 lst-kix_508olx789l51-0 start"><li class="c12 c0"><span class="c4">for all the hsps of each hit we check the reading frame number</span></li><li class="c12 c0"><span class="c4">store in a hash table the number of occurrences for each reading frame</span></li><li class="c12 c0"><span class="c4">when we count more reading frames of the same sign, there must have occurred a frame shift during the sequencing </span><span class="c4">process</span><span class="c4">. </span></li><li class="c12 c0"><span class="c4">return a hash with the key each one of the 6 possible reading frames and the value - the number of hsps found by blast which are read in that reading frame</span></li></ul></td></tr><tr class="c10"><td class="c61"><h4 class="c0"><a name="h.i7vlof3ljop1"></a><span class="c11"><a class="c25" href="https://github.com/monicadragan/GeneValidator/blob/rubygem/lib/genevalidator/validation_blast_reading_frame.rb#L69%20">Code</a></span></h4><p class="c5 c6"><span class="c3"></span></p><p class="c0"><span class="c7">if type.to_s != &quot;nucleotide&quot;</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;@validation_report = ValidationReport.new(&quot;&quot;, :unapplicable)</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;return @validation_report</span></p><p class="c0"><span class="c7">end</span></p><p class="c5 c6"><span class="c7"></span></p><p class="c0"><span class="c7">rfs = &nbsp;lst.map{ |x| x.hsp_list.map{ |y| y.query_reading_frame}}.flatten</span></p><p class="c0"><span class="c7">&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;frames_histo = Hash[rfs.group_by { |x| x }.map { |k, vs| [k, vs.length] }]</span></p><p class="c5 c6"><span class="c7"></span></p><p class="c0"><span class="c7"># get the main reading frame</span></p><p class="c0"><span class="c7">main_rf = frames_histo.map{|k,v| v}.max</span></p><p class="c0"><span class="c7">@prediction.nucleotide_rf = frames_histo.select{|k,v| v==main_rf}.first.first</span></p><p class="c5 c6"><span class="c7"></span></p><p class="c0"><span class="c7">@validation_report = BlastRFValidationOutput.new(frames_histo)</span></p><p class="c5 c6"><span class="c7"></span></p></td></tr><tr class="c10"><td class="c61"><h4 class="c0"><a name="h.kzye1e8i7r8t"></a><span>Plots</span></h4><p class="c0"><span class="c16">No plots are generated</span></p></td></tr></tbody></table><p class="c6 c14"><span class="c3"></span></p><h2 class="c6"><a name="h.3elev4f70kdl"></a><span class="c16">7) Open reading frame - for nucleotides only</span></h2><p class="c6 c14"><span class="c16"></span></p><a href="#" name="44123c880f31c93b3f070d97baf8ef569b6f3e97"></a><a href="#" name="7"></a><table cellpadding="0" cellspacing="0" class="c46"><tbody><tr class="c10"><td class="c28"><h4 class="c0"><a name="h.frtucgcrc7ap"></a><span>Error causes</span></h4><ul class="c15 lst-kix_g96zcqu3kh41-0 start"><li class="c12 c0"><span class="c4">many small ORF</span></li><li class="c12 c0"><span class="c4">frame shift</span></li></ul></td></tr><tr class="c10"><td class="c28"><h4 class="c0"><a name="h.58tttrdk1gyx"></a><span>Input data</span></h4><ul class="c15 lst-kix_4ed4y1s6lj5w-0"><li class="c12 c6"><span class="c4">only query raw sequence is needed for this validation (got from the fasta file)</span></li></ul><p class="c5 c6"><span class="c4"></span></p></td></tr><tr class="c10"><td class="c28"><h4 class="c0"><a name="h.6slvdpgdbg5r"></a><span>Class information:</span></h4><ul class="c15 lst-kix_x7t5b7jbczs4-0 start"><li class="c12 c0"><span class="c4">header: Main ORF</span></li><li class="c12 c0"><span class="c4">description: Check whether there is a single main Open Reading Frame in the predicted gene. Applicable only for nucleotide queries. Meaning of the output displayed: %=MAIN ORF COVERAGE. Coverage higher than 80% passe the validation test.</span></li><li class="c12 c0"><span class="c4">alias: orf</span></li></ul><p class="c5 c6"><span class="c4"></span></p></td></tr><tr class="c10"><td class="c28"><h4 class="c0"><a name="h.1dg3y6evjaci"></a><span>Workflow</span></h4><ul class="c15 lst-kix_g3lepnw8n0x6-0 start"><li class="c12 c0"><span class="c4">for each reading frame get all the ORFs, with customizable stop and start codons (the subsequeces that start with a start codon and end with an end codon)</span></li><li class="c12 c0"><span class="c4">return the coverage percentage of the longest reading frame</span></li><li class="c12 c0"><span class="c4">If there is any continuous ORF read in a certain reading frame that is longer than 80% of the length of the prediction, that prediction is considered to be a good prediction. Otherwise it won&rsquo;t pass the test.</span></li></ul><p class="c5 c6"><span class="c4"></span></p></td></tr><tr class="c10"><td class="c28"><h4 class="c0"><a name="h.gwio1xu42jrm"></a><span class="c11"><a class="c25" href="https://github.com/monicadragan/GeneValidator/blob/rubygem/lib/genevalidator/validation_open_reading_frame.rb#L76%20">Code</a></span></h4><p class="c5 c6"><span></span></p><p class="c0"><span class="c16 c54">if type.to_s != &quot;nucleotide&quot;</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;@validation_report = ValidationReport.new(&quot;&quot;, :unapplicable)</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;return @validation_report</span></p><p class="c0"><span class="c7">&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;end</span></p><p class="c0"><span class="c7">&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;orfs = get_orfs</span></p><p class="c5 c6"><span class="c7"></span></p><p class="c0"><span class="c7">&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;# check if longest ORF / prediction &gt; 0.8 (ok)</span></p><p class="c0"><span class="c7">&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;prediction_len = prediction.raw_sequence.length</span></p><p class="c0"><span class="c7">&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;longest_orf = orfs.map{|elem| elem[1].map{|orf| orf[1]-orf[0]}}.flatten.max</span></p><p class="c0"><span class="c7">&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;ratio = &nbsp;longest_orf/(prediction_len + 0.0)</span></p><p class="c5 c6"><span class="c7"></span></p><p class="c0"><span class="c7">&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;len = @prediction.raw_sequence.length</span></p><p class="c5 c6"><span class="c7"></span></p><p class="c0"><span class="c7">&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;f = File.open(&quot;#{@filename}_orfs.json&quot; , &quot;w&quot;)</span></p><p class="c0"><span class="c7">&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;lst = @hits.sort{|a,b| a.length&lt;=&gt;b.length}</span></p><p class="c0"><span class="c7">&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;f.write((orfs.each_with_index.map{|elem, i| {&quot;y&quot;=&gt;elem[0], &quot;start&quot;=&gt;0, &quot;stop&quot;=&gt;len, &quot;color&quot;=&gt;&quot;black&quot;}} +</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;orfs.each_with_index.map{|elem, i| elem[1].map{|orf| {&quot;y&quot;=&gt;elem[0], &quot;start&quot;=&gt;orf[0], &quot;stop&quot;=&gt;orf[1], &quot;color&quot;=&gt;&quot;red&quot;}}}.flatten).to_json)</span></p><p class="c0"><span class="c7">&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;f.close</span></p><p class="c0"><span class="c7">&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;@plot_files.push(Plot.new(&quot;#{@filename}_orfs.json&quot;.scan(/\/([^\/]+)$/)[0][0],</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;:lines,</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&quot;Open reading frame with START codon&quot;,</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&quot;&quot;,</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&quot;length&quot;,</span></p><p class="c0"><span class="c7">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&quot;Reading Frame&quot;))</span></p><p class="c5 c6"><span class="c7"></span></p><p class="c0"><span class="c7">@validation_report = ORFValidationOutput.new(orfs, ratio)</span></p><p class="c5 c6"><span class="c16 c54"></span></p></td></tr><tr class="c10"><td class="c28"><h4 class="c0"><a name="h.8803ag5rytmq"></a><span>Plots</span></h4><p class="c5 c6"><span class="c16"></span></p><ul class="c15 lst-kix_l7t8lma3v5dd-0 start"><li class="c12 c0"><span class="c4">line plot depicting the ORFs for each reading frame (Figure 1 - plot for a good prediction)</span></li></ul><p class="c5 c6"><span class="c16"></span></p><p class="c0 c21"><img height="342" src="images/image15.png" width="328"></p><p class="c0 c21"><span class="c3">(1)</span></p></td></tr></tbody></table><p class="c6 c14"><span class="c16"></span></p><hr style="page-break-before:always;display:none;"><p class="c6 c14"><span class="c16"></span></p><p class="c6 c14"><span class="c16"></span></p><a href="#" name="fb9fb93011ad78ca1e36c4c1e56fdf04d009bd67"></a><a href="#" name="8"></a><table cellpadding="0" cellspacing="0" class="c46"><tbody></tbody></table><h1 class="c6 c21"><a name="h.cw7l8waunvwl"></a><span class="c3">C. Running time</span></h1><p class="c6 c14"><span class="c16"></span></p><a href="#" name="6b4ca8c605ff2a614a81234917ffce16338615ac"></a><a href="#" name="9"></a><table cellpadding="0" cellspacing="0" class="c46"><tbody><tr class="c10"><td class="c24"><p class="c0"><span class="c4">Validation</span></p></td><td class="c20"><p class="c0"><span class="c4">10</span><span class="c4">&nbsp;q</span><span class="c4">ueries</span></p></td><td class="c30"><p class="c0"><span class="c4">30 queries</span></p></td><td class="c49"><p class="c0"><span class="c4">Time consumers</span></p></td></tr><tr><td class="c24"><p class="c0"><span class="c4">Length Cluster </span></p></td><td class="c20"><p class="c0"><span class="c4">0.104 s/q</span></p></td><td class="c30"><p class="c0"><span class="c4">0.025 s/q</span></p></td><td class="c49"><p class="c5 c6"><span class="c4"></span></p></td></tr><tr class="c10"><td class="c24"><p class="c0"><span class="c4">Length Rank </span></p></td><td class="c20"><p class="c0"><span class="c4">0.00 s/q</span></p></td><td class="c30"><p class="c0"><span class="c4">0.00 s/q</span></p></td><td class="c49"><p class="c5 c6"><span class="c4"></span></p></td></tr><tr class="c10"><td class="c24"><p class="c0"><span class="c4">Gene Merge</span></p></td><td class="c20"><p class="c0"><span class="c4">0.003 s/q</span></p></td><td class="c30"><p class="c0"><span class="c4">0.01 s/q</span></p></td><td class="c49"><p class="c5 c6"><span class="c4"></span></p></td></tr><tr><td class="c24"><p class="c0"><span class="c4">Duplication </span></p></td><td class="c20"><p class="c0"><span class="c4">9.458 s/q</span></p></td><td class="c30"><p class="c0"><span class="c4">10.488 s/q</span></p></td><td class="c49"><ul class="c15 lst-kix_fscfg01f3awm-0 start"><li class="c12 c0"><span class="c4">potential use of the webservice to get raw sequences</span></li><li class="c12 c0"><span class="c4">MAFFT local alignment</span></li></ul></td></tr><tr class="c10"><td class="c24"><p class="c0"><span class="c4">Multiple Alignment </span></p></td><td class="c20"><p class="c0"><span class="c4">1.785 s/q</span></p><p class="c5 c6"><span class="c4"></span></p></td><td class="c30"><p class="c0"><span class="c4">0.645 s/q</span></p></td><td class="c49"><ul class="c15 lst-kix_dpynbghv8b9j-0 start"><li class="c12 c0"><span class="c4">potential use of the webservice to get raw sequences (if was not retrieved in a previous validation)</span></li><li class="c12 c0"><span class="c4">MAFFT multiple alignment</span></li></ul></td></tr><tr class="c10"><td class="c24"><p class="c0"><span class="c4">Frame Validation</span></p></td><td class="c20"><p class="c0"><span class="c4">0.0 s/q</span></p></td><td class="c30"><p class="c0"><span class="c4">0.0 s/q</span></p></td><td class="c49"><p class="c5 c6 c9"><span class="c4"></span></p></td></tr><tr class="c10"><td class="c24"><p class="c0"><span class="c4">ORF Validation</span></p></td><td class="c20"><p class="c0"><span class="c4">0.015 s/q</span></p></td><td class="c30"><p class="c0"><span class="c4">0.012 s/q</span></p></td><td class="c49"><p class="c9 c5 c6"><span class="c4"></span></p></td></tr></tbody></table><p class="c6 c14"><span class="c16"></span></p><p class="c6 c14 c21"><span class="c3 c17 c36"></span></p><p class="c6 c14 c21"><span class="c3 c17 c36"></span></p><hr style="page-break-before:always;display:none;"><p class="c6 c14 c21"><span class="c3 c17 c36"></span></p><p class="c6 c14 c21"><span class="c3 c17 c36"></span></p><h1 class="c6 c21"><a name="h.njhshrkdurp1"></a><span class="c3">D. Code Structure</span></h1><p class="c6 c14 c21"><span class="c3 c17"></span></p><p class="c6 c21"><img height="594" src="images/image08.jpg" width="893"></p><p class="c6 c14 c21"><span class="c3 c17"></span></p><p class="c6 c14 c21"><span class="c3 c17"></span></p><hr style="page-break-before:always;display:none;"><p class="c6 c14 c21"><span class="c3 c17"></span></p><h1 class="c6 c21"><a name="h.8hgqxxlaoy94"></a><span class="c3">E. How to run the tool</span></h1><p class="c6 c14"><span class="c16"></span></p><h2 class="c6"><a name="h.es2ozy57vw91"></a><span class="c66">1. Prerequisite:</span></h2><p class="c6 c14"><span class="c4"></span></p><ul class="c15 lst-kix_cbmcswhr1ftl-0 start"><li class="c12 c6"><span class="c4 c18">Ruby</span><span class="c4">&nbsp;(&gt;= 1.9.3)</span></li><li class="c12 c6"><span class="c4 c18">RubyGems</span><span class="c4">&nbsp;(&gt;= 1.3.6)</span></li><li class="c12 c6"><span class="c4 c18">NCBI BLAST+</span><span class="c4">&nbsp;(&gt;= 2.2.25+)</span></li><li class="c12 c6"><span class="c4 c18">MAFFT installation</span><span class="c4">&nbsp;(download it from :</span><span class="c4"><a class="c25" href="http://mafft.cbrc.jp/alignment/software/">&nbsp;</a></span><span class="c11 c4"><a class="c25" href="http://mafft.cbrc.jp/alignment/software/">http://mafft.cbrc.jp/alignment/software/</a></span><span class="c4">&nbsp;).</span></li></ul><p class="c6 c14"><span class="c4"></span></p><p class="c6 c21"><span class="c4 c18">Linux and MacOS are officially supported!</span></p><h2 class="c6"><a name="h.pszg00frp7dj"></a><span class="c66">2. Installation:</span></h2><p class="c6 c14"><span class="c4 c18"></span></p><ol class="c15 lst-kix_9drbe3jtkkuh-0 start" start="1"><li class="c12 c6"><span class="c4">Get the source code</span></li></ol><p class="c6 c40"><span class="c4">$ git clone git@github.com:monicadragan/GeneValidator.git</span></p><ol class="c15 lst-kix_9drbe3jtkkuh-0"><li class="c12 c6"><span class="c4">Be sudo and build the gem</span></li></ol><p class="c6 c40"><span class="c4">$ sudo rake</span></p><ol class="c15 lst-kix_9drbe3jtkkuh-0"><li class="c12 c6"><span class="c4">Run GeneValidation</span></li></ol><p class="c6 c40"><span class="c4">$ genevalidator [validations] [skip_blast] [start] [tabular] [mafft] [raw_seq] FILE</span></p><p class="c6 c14 c40"><span class="c4"></span></p><p class="c6"><span class="c4">Example that runs all validations on a set of ant gene predictions:</span></p><p class="c6"><span class="c4">$ genevalidator -x data/prot_Solenopsis.xml data/prot_Solenopsis.fa</span></p><p class="c6 c14"><span class="c4"></span></p><p class="c6"><span class="c4">To learn more:</span></p><p class="c6"><span class="c4">$ genevalidator -h</span></p><p class="c6 c14"><span class="c4"></span></p><h2 class="c6"><a name="h.m9jfyg83mj71"></a><span class="c66">3. Input</span></h2><p class="c6 c14"><span class="c4 c18"></span></p><ul class="c15 lst-kix_xc244bhq3ube-0 start"><li class="c12 c6"><span class="c4 c18">FASTA file</span><span class="c4">&nbsp;with the prediction queries (queries must be of the same type: </span><span class="c4 c18">mRNA or proteins</span><span class="c4">. Mixed type FASTA files are not allowed!).</span></li><li class="c12 c6"><span class="c4">[optional] </span><span class="c4 c18">Precalculated BLAST output file</span><span class="c4">&nbsp;in xml format (-outfmt 5) or tabular format (-outfmt 6 or -outfmt 7 or &ldquo;Hit Table (CSV)&rdquo; in the online version of BLAST). Please specify the --tabular argument if you use tabular format input with nonstandard columns. Also, </span><span class="c4 c18">we search only in protein database using a blastp for protein queries and blastx for nucleotide queries. </span></li></ul><p class="c6 c14"><span class="c4 c18"></span></p><p class="c6 c14"><span class="c4 c18"></span></p><h2 class="c6"><a name="h.plih6twef6rr"></a><span class="c66">4. Output</span></h2><p class="c6 c14"><span class="c4 c18"></span></p><p class="c6"><span class="c4">By running GeneValidator on your dataset you get numbers and plots. Some relevant files will be generated at the same path with the input file. The results are available in 3 formats:</span></p><ul class="c15 lst-kix_jamjx369pdr8-0 start"><li class="c12 c6"><span class="c4 c18">console table output</span></li><li class="c12 c6"><span class="c4">validation results in </span><span class="c4 c18">YAML format</span><span class="c4">&nbsp;(the YAML file has the same name with the input file + YAML extension)</span></li><li class="c12 c6"><span class="c4 c18">html output</span><span class="c4">&nbsp;with plot visualization (the useful files will be generated in the &#39;html&#39; directory, at the same path with the input file). Check the </span><span class="c4 c18">index.html</span><span class="c4">&nbsp;file to see the output.</span></li></ul><p class="c6 c14 c21"><span class="c4"></span></p><p class="c6 c21"><span class="c4 c18 c36">! Note: for the moment check the html output with Firefox browser only !</span></p><p class="c6 c21"><span class="c4">(because our d3js cannot be visualized from other browsers without a local webserver) </span></p><p class="c6 c14"><span class="c4 c18"></span></p><p class="c6 c14"><span class="c4"></span></p><h2 class="c6"><a name="h.26zdfyct6x5w"></a><span class="c66">5. Scenarios</span></h2><p class="c6 c14"><span class="c4 c18"></span></p><p class="c6"><span class="c4 c18">5.1</span><span class="c4">&nbsp;Quickly test the tool or validate a small set of predictions </span><span class="c4 c18">(internet connection required).</span></p><p class="c6 c21"><span class="c4">$ genevalidator predictions.fasta</span></p><p class="c6"><span class="c4 c27">What happens backstage: blast hits and raw sequences are retrieved from remote databases and analysed for each </span><span class="c4 c27">query</span><span class="c4 c27">.</span></p><p class="c6 c14"><span class="c4 c18"></span></p><p class="c2"><span class="c4 c18">5.2</span><span class="c4">&nbsp;</span><span class="c4 c18">Precalculate the input files </span></p><p class="c2 c14"><span class="c4 c18"></span></p><p class="c2"><span class="c4 c18">5.2.1 Precalculate BALST output</span></p><p class="c2"><span class="c4">If you have &gt;100 queries &amp; you access to a more powerful machine that pre-calculates overnight the blast output file. </span><span class="c4 c18">Use blastp for proteins and blastx for nucleotide queries.</span></p><p class="c2 c14"><span class="c4 c18"></span></p><p class="c6"><span class="c4 c18">a)</span><span class="c4">&nbsp;If</span><span class="c4 c18">&nbsp;</span><span class="c4">you have time and a really powerful machine you can </span><span class="c4 c18">take the blast xml output</span><span class="c4">. This way you will save time when running GeneValidator (because you precalculate local alignments with BLAST)</span></p><p class="c6"><span class="c4">&nbsp; </span></p><p class="c6"><span class="c4">Example: </span></p><p class="c6 c14"><span class="c4"></span></p><p class="c6"><span class="c7"># take blast output for </span><span class="c7 c18">protein queries </span><span class="c7">(for </span><span class="c7 c18">mrna queries </span><span class="c7">use </span><span class="c7 c18">blastx</span><span class="c7">)</span></p><p class="c6"><span class="c7">$ blastp -query predictions.fasta -db /path/to/db/nr/nr -outfmt 5 -max_target_seqs 200 -num_threads 48 -out predictions_blast.xml</span></p><p class="c6 c14"><span class="c7"></span></p><p class="c6"><span class="c7"># run validations</span></p><p class="c6"><span class="c7">$ genevalidator -x predictions_blast.xml predictions.fasta</span></p><p class="c6 c14"><span class="c4"></span></p><p class="c6"><span class="c4 c18">b) Take blast tabular output</span><span class="c4">&nbsp;(customized so that you have all the necessary columns -- see the table with data needed for validations in section B)</span></p><p class="c6 c14"><span class="c4"></span></p><p class="c6"><span class="c4">Example: </span></p><p class="c6 c14"><span class="c4"></span></p><p class="c6"><span class="c7"># take blast output for protein queries</span></p><p class="c6"><span class="c7">$ blastp -query predictions.fasta -db /path/to/db/nr/nr -outfmt &quot;6 qseqid sseqid sacc slen qstart qend sstart send length qframe pident evalue&quot; -max_target_seqs 200 -num_threads 48 -out predictions_blast.tab</span></p><p class="c6 c14"><span class="c7"></span></p><p class="c6"><span class="c7"># run validations</span></p><p class="c6"><span class="c7">$ genevalidator -x predictions_blast.tab -t &quot;qseqid sseqid sacc slen qstart qend sstart send length qframe pident evalue&quot; predictions.fasta </span></p><p class="c6 c14"><span class="c4"></span></p><p class="c6 c14"><span class="c4"></span></p><p class="c6"><span class="c4 c18">5.2.1 Precalculate the raw sequences</span><span class="c4">&nbsp;for the first n hits</span><span class="c4 c18">&nbsp;from a local BAST database</span><span class="c4">, using the identifiers from your BLAST output</span></p><p class="c6 c14"><span class="c4 c18"></span></p><p class="c6"><span class="c7">$ get</span><span class="c4">_raw_sequences BLAST_OUTPUT_FILE -o OUTPUT_FILE [-t TABULAR_FORMAT] [-d DATABASES] [-n NR]</span></p><p class="c6 c14"><span class="c4"></span></p><p class="c6"><span class="c4">Example:</span></p><p class="c6 c14"><span class="c4"></span></p><p class="c6"><span class="c7"># get raw sequences from tabular BLAST output </span></p><p class="c6"><span class="c7">$ get_raw_sequences predictions_blast.tab -o predictions.raw_seq -t &quot;qseqid sseqid sacc slen qstart qend sstart send pident length qframe evalue&quot; -d path/to/ncbi/blast/db/nr/nr</span></p><p class="c6 c14"><span class="c7"></span></p><p class="c6 c14"><span class="c7"></span></p><p class="c6"><span class="c7"># get raw sequences from xml BLAST output</span></p><p class="c6"><span class="c7">$ get_raw_sequences predictions_blast.xml -o predictions.raw_seq -d path/to/ncbi/blast/db/nr/nr</span></p><p class="c6 c14"><span class="c7"></span></p><p class="c6"><span class="c7"># run validations</span></p><p class="c6"><span class="c7">$ genevalidator -x data/all_validations_prot/all_validations_prot.xml data/all_validations_prot/all_validations_prot.fasta -t &quot;qseqid sseqid sacc slen qstart qend sstart send pident length qframe evalue&quot; -r data/all_validations_prot/all_validations_prot.xml.raw_seq</span></p><p class="c6 c14"><span class="c4"></span></p><p class="c6"><span class="c4 c18">5.3</span><span class="c4">&nbsp;You are interested in </span><span class="c4 c18">certain validations</span><span class="c4">&nbsp;only (let&rsquo;s take length validations and duplication check). Look up the CLI names for these validations in the table (Section B) and add them as argument so that GeneValidator will process those 3 only.</span></p><p class="c6 c14"><span class="c4"></span></p><p class="c6"><span class="c7">$ genevalidator -x predictions_blast.xml predictions.fasta -v &lsquo;lenc lenr dup&rsquo;</span></p><p class="c6 c14"><span class="c4"></span></p><p class="c6"><span class="c4 c18">5.4 Start validating from a certain query in the fasta file: </span><span class="c4">the following command skips the first 9 queries</span></p><p class="c6 c14"><span class="c4"></span></p><p class="c6"><span class="c7">$ genevalidator -x predictions_blast.xml predictions.fasta -s 10</span></p><p class="c6 c14"><span class="c16"></span></p><p class="c6 c14"><span class="c16"></span></p><hr style="page-break-before:always;display:none;"><p class="c6 c14"><span class="c16"></span></p><h1 class="c6 c21"><a name="h.qi34bbc42hqf"></a><span class="c3">F. Add a new </span><span class="c3">validation</span></h1><p class="c6 c14"><span></span></p><p class="c6"><span>The code architecture is designed so that it is very easy to add new validations. There is a </span><span class="c4">&lsquo;template&rsquo; that the code of the new validation must fit, which is discussed furt</span><span class="c4">her</span><span class="c4">:</span></p><a href="#" name="013ba5ea08b97c10ec4b2adf9081ba34a620cfb4"></a><a href="#" name="10"></a><table cellpadding="0" cellspacing="0" class="c46"><tbody><tr class="c10"><td class="c28"><h3 class="c0"><a name="h.l1h18sojuad4"></a><span class="c16">Extend 2 parent classes</span></h3><p class="c2 c22"><span class="c4"><br></span><span class="c4 c18">(1) extend </span><span class="c11 c4 c18"><a class="c25" href="https://github.com/monicadragan/GeneValidator/blob/rubygem/lib/genevalidator/validation_output.rb%20">ValidationReport</a></span><span class="c4">- stores the output of the validation test and some methods used for data representation, that may need to be overridden:</span></p><ul class="c15 lst-kix_owpom8h3nkwo-0 start"><li class="c12 c0"><span class="c4 c18">validation</span><span class="c4">: returns </span><span class="c4">one</span><span class="c4">&nbsp;of the 4 results: </span></li></ul><ul class="c15 lst-kix_owpom8h3nkwo-1 start"><li class="c58 c0"><span class="c4">:yes </span></li><li class="c58 c0"><span class="c4">:no </span></li><li class="c58 c0"><span class="c4">:unapplicable </span></li><li class="c0 c58"><span class="c4">:error</span></li></ul><ul class="c15 lst-kix_owpom8h3nkwo-0"><li class="c12 c2 c22"><span class="c4 c18">print</span><span class="c4">: returns the output that will be displayed in the outputs</span></li><li class="c12 c2 c22"><span class="c4 c18">color</span><span class="c4">: based on the validation result returns the background color for the corresponding cell in the html output (at the moment the possible values are &ldquo;success&rdquo;, &ldquo;warning&rdquo; and &ldquo;danger&rdquo; -- which are classes in the CSS file)</span></li></ul><p class="c5 c6"><span class="c4"></span></p><p class="c2 c22"><span class="c4">Initialize internal variables (which initially have some default values):</span></p><ul class="c15 lst-kix_pdbqztwjr65g-0 start"><li class="c12 c2 c22"><span class="c4 c18">expected</span><span class="c4">&nbsp;: the expected correct result for the validation. The test will pass if the expected result is equal to the validation result (:yes or :no)</span></li><li class="c12 c2 c22"><span class="c4">initialize </span><span class="c4 c18">plot_files</span><span class="c4">&nbsp;with the empty Array in case you will generate plots (by default this variable is nil)</span></li><li class="c12 c2 c22"><span class="c4">initialize </span><span class="c4 c18">errors</span><span class="c4">&nbsp;with the empty Array in case you will handle and display in the output the possible errors of this validation</span></li></ul><p class="c2 c22"><span class="c4"><br></span><span class="c4 c18">(2) extend </span><span class="c11 c4 c18"><a class="c25" href="https://github.com/monicadragan/GeneValidator/blob/rubygem/lib/genevalidator/validation_test.rb%20">ValidationTest</a></span><span class="c4">&nbsp;</span><span class="c4">(the &#39;run&#39; method must be overloaded, run must return ValidationReport class type, &nbsp;some fields have to be updated: the validation name used in the header, description, plot_files)</span></p><p class="c5 c6"><span class="c4 c18"></span></p><p class="c2 c22"><span class="c4">Initialize internal variables (which initially have some default values):</span></p><ul class="c15 lst-kix_48dltnwvsqfp-0"><li class="c0 c26"><span class="c4 c18">short_header</span><span class="c4">: will appear in the console output (may not contain spaces)</span></li><li class="c26 c0"><span class="c4 c18">header</span><span class="c4">: will appear in the header of the table - in html output </span></li><li class="c26 c0"><span class="c4 c18">description</span><span class="c4">: short description of the validation -- will appear in the html output</span></li><li class="c26 c0"><span class="c4 c18">cli_name</span><span class="c4">: validation can be referred by this alias when selecting the validations to be run (-v argument in the command line)</span></li><li class="c26 c0"><span class="c4 c18">running_time</span><span class="c4">: if you want to make overall running time statistics</span></li></ul><p class="c5 c6"><span class="c4"></span></p></td></tr><tr class="c10"><td class="c28"><h3 class="c0"><a name="h.52nozdazdy71"></a><span class="c16">Add the new validation to the tool</span></h3><p class="c6 c31"><span class="c4">Add the validation to the validations list, which is further processed for yaml/html/console visualization<br><br></span><span class="c11 c4 c18"><a class="c25" href="https://github.com/monicadragan/GeneValidator/blob/rubygem/lib/genevalidator/validation.rb#L284%20">Code</a></span><span class="c4 c18">:</span></p><p class="c2"><span class="c4"># context code #</span></p><p class="c2"><span class="c4">validations = []</span></p><p class="c2"><span class="c4">validations.push LengthClusterValidation.new(@type, prediction, hits, plot_path, plots)</span></p><p class="c2"><span class="c4">&nbsp;&hellip;</span></p><p class="c2"><span class="c4 c18 c53">## what you need to add is this line that adds your validation class to the list of validations. Your class must extend ValidationTest ##</span></p><p class="c2 c14"><span class="c4 c18 c53"></span></p><p class="c2"><span class="c4 c18 c53">validations.push YourValidationClass(type, prediction, hits, other_arguments)</span></p><p class="c2 c14"><span class="c4"></span></p><p class="c2"><span class="c4"># context code #</span></p><p class="c2"><span class="c4"># check the class type of the elements in the list</span></p><p class="c2"><span class="c4"># this will raise an error if YourValidationClass does not extend ValidationTest validations.map do |v|</span></p><p class="c2"><span class="c4">&nbsp; raise ValidationClassError unless v.is_a? ValidationTest</span></p><p class="c2"><span class="c4">end</span></p><p class="c2 c14"><span class="c4"></span></p><p class="c2"><span class="c4"># run validations</span></p><p class="c2"><span class="c4">validations.map{|v| v.run}</span></p><p class="c2 c14"><span class="c4"></span></p><p class="c2"><span class="c4"># check the class type of the validation reports</span></p><p class="c2"><span class="c4"># this will raise an error if the run method of YourValidationClass does not return ValidationReport</span></p><p class="c2"><span class="c4">validations.map do |v|</span></p><p class="c2"><span class="c4">&nbsp; raise ValidationClassError unless v.validation_report.is_a? ValidationReport</span></p><p class="c2"><span class="c4">end</span></p></td></tr><tr class="c10"><td class="c28"><h3 class="c0"><a name="h.j12khye9z0zk"></a><span class="c16">Add plots</span></h3><p class="c5 c6"><span class="c4 c18"></span></p><h4 class="c0"><a name="h.af1kbdhs4wdx"></a><span class="c4">A. Generate json file according to the type of plot</span></h4><p class="c5 c6"><span class="c4 c18"></span></p><p class="c0"><span class="c4 c18">1) (clustered) bars plot</span><span class="c4">, main cluster is in red: [ [{&quot;key&quot;:109,&quot;value&quot;:1,&quot;main&quot;:false}], [{&quot;key&quot;:122,&quot;value&quot;:32,&quot;main&quot;:true},{&quot;key&quot;:123,&quot;value&quot;:33,&quot;main&quot;:true}], [...]]</span></p><p class="c5 c6"><span class="c4 c18"></span></p><p class="c0"><span class="c4 c18">2) scatter plot</span><span class="c4">: </span></p><p class="c0"><span class="c4">[{&quot;x&quot;:1,&quot;y&quot;:626},{&quot;x&quot;:1,&quot;y&quot;:626}...]</span></p><p class="c5 c6"><span class="c4"></span></p><p class="c0"><span class="c4 c18">3) lines</span><span class="c4">: [{&quot;y&quot;:1,&quot;start&quot;:0,&quot;stop&quot;:356,&quot;color&quot;:&quot;black&quot;},{&quot;y&quot;:1,&quot;start&quot;:40,&quot;stop&quot;:200,&quot;color&quot;:&quot;red&quot;}...]</span></p><p class="c5 c6"><span class="c4"></span></p><p class="c5 c6"><span class="c4"></span></p><h4 class="c0"><a name="h.v27rr25j084l"></a><span class="c4">B. Create a plot object, according to the plot type </span></h4><p class="c0"><span class="c4">&nbsp; &nbsp;plot1 = Plot.new(filemane,</span></p><p class="c0"><span class="c4">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;plot_type # which can be </span><span class="c4 c18">:scatter, :line or :bar</span></p><p class="c0"><span class="c4">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;plot_title,</span></p><p class="c0"><span class="c4">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;legend**,</span></p><p class="c0"><span class="c4">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;xTitle,</span></p><p class="c0"><span class="c4">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;yTitle,</span></p><p class="c0"><span class="c4">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;aux_parameter1*,</span></p><p class="c0"><span class="c4">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;aux_parameter2*)</span></p><p class="c5 c6"><span class="c4"></span></p><p class="c0"><span class="c4 c18">** legend is a String that _must_ have the following format:</span></p><p class="c0"><span class="c4 c18">&quot;item, color1;item with spaces, color2&quot;</span></p><p class="c0"><span class="c4 c18">* auxiliary parameters</span><span class="c4">&nbsp;were used by now for passing the length of prediction (in length histogram) or the slope of the regression line (gene merge validation) or the number of lines (line plots)</span></p><p class="c5 c6"><span class="c4"></span></p><h4 class="c0"><a name="h.vbwt0lpnzk07"></a><span class="c4">C. Add the plot object to the plot_files array</span></h4><p class="c5 c6"><span class="c4"></span></p><p class="c0"><span class="c4">@validation_report.plot_files.push(plot1)</span></p></td></tr></tbody></table><p class="c6 c14"><span class="c3 c17"></span></p><p class="c6 c14 c21"><span class="c3 c17"></span></p><hr style="page-break-before:always;display:none;"><p class="c6 c14 c21"><span class="c3 c17"></span></p><h1 class="c6 c21"><a name="h.4rd7mxpo7dua"></a><span class="c3">G. Output results</span></h1><p class="c6 c14 c21"><span class="c3 c17 c36"></span></p><p class="c6"><span class="c3 c17 c36">Some validation outputs for protein and mRNA are available here:</span></p><p class="c6 c14"><span class="c3 c17 c36"></span></p><p class="c6"><span class="c11 c17 c16"><a class="c25" href="http://swarm.cs.pub.ro/~mdragan/gsoc2013/genevalidator">http://swarm.cs.pub.ro/~mdragan/gsoc2013/</a></span><span class="c11 c17 c16"><a class="c25" href="http://swarm.cs.pub.ro/~mdragan/gsoc2013/genevalidator">genevalidator</a></span><span class="c17 c16">&nbsp; </span></p><p class="c6 c14"><span class="c16"></span></p><p class="c6 c14"><span class="c16"></span></p><p class="c6"><img height="358" src="images/image02.png" width="792"></p></body></html>