<!DOCTYPE html>
<html>
<head>
  <title>GeneValidator: identify problems with gene predictions</title>
  <meta content="text/html; charset=utf-8" http-equiv="Content-Type">

  <script type="text/javascript" src="files/js/jquery-2.1.1.min.js"></script>
  <script type="text/javascript" src="files/js/bootstrap.min.js"></script>
  <script type="text/javascript" src="files/js/jquery.tablesorter.min.js"></script>
  <script type="text/javascript" src="files/js/d3.v3.min.js"></script>
  <script type="text/javascript" src="files/js/script.js"></script>
  <script type="text/javascript" src="files/js/plots.js"></script>

  <link href="files/css/bootstrap.min.css" rel="stylesheet" type="text/css">
  <link href="files/css/font-awesome.min.css" rel="stylesheet" type="text/css">
  <link href="files/css/style.css" rel="stylesheet" type="text/css">

</head>
<body>
  <div class="container">
    <span class="menu_icon pull-right"><a href="#" id="show_all_plots" onclick="show_all_plots(this);"><i class="fa fa-2x fa-bar-chart-o"></i><br>Show All Charts</a></span>
    <span class="menu_icon pull-right"><a href="https://github.com/monicadragan/GeneValidator" target="_blank"><i class="fa fa-2x fa-github"></i><br>Source Code</a></span>
    <span class="menu_icon pull-right"><a href="http://wurmlab.github.io/tools/genevalidator_documentation/v1/"><i class="fa fa-2x fa-info-circle"></i><br>About</a></span>
    <div class="clearfix"></div>
    <img class="logo" src="files/img/gene.png" alt="logo">
    <div class="page-title">
      <h1>Gene Validator <small>v1.3.6</small></h1>
      <h4 class="subheading">Identify Problems with Gene Predictions</h4>
    </div>
    <br/><br/>
    <div id="report">
      <div id="report_1"></div>
    </div>
    <br><br>
      <table id="sortable_table" class="table table-collapsed table-bordered table-condensed tablesorter">
        <thead>
          <tr id="header">
            <th>#</th>
            <th>Ranking</th>
            <th>Sequence Definition
              <span data-toggle="tooltip" title="Query definition as it apears in the input fasta file." data-placement="top"><i class="fa fa-question-circle"></i></span>
            </th>
            <th>No. Hits
              <span data-toggle="tooltip" title="Number of non-identical hits found by BLAST." data-placement="top"><i class="fa fa-question-circle"></i></span>
            </th>
                        <th class="sorter-false">
              <b>Length Cluster</b>
                                  <span data-toggle="tooltip" title="Charts available for this validation" data-placement="top"><i class="fa fa-bar-chart-o chartIcon"></i></span>&nbsp;<span data-toggle="tooltip" title="Check whether the prediction length fits most of the BLAST hit lengths, by 1D hierarchical clusterization. Meaning of the output displayed: Query_length [Main Cluster Length Interval]" data-placement="top"><i class="fa fa-question-circle"></i></span>
                            </th>
                        <th class="sorter-false">
              <b>Length Rank</b>
                                   <span data-toggle="tooltip" title="Check whether the rank of the prediction length lies among 80% of all the BLAST hit lengths." data-placement="top"><i class="fa fa-question-circle"></i></span>
                            </th>
                        <th class="sorter-false">
              <b>Gene Merge</b>
                                  <span data-toggle="tooltip" title="Charts available for this validation" data-placement="top"><i class="fa fa-bar-chart-o chartIcon"></i></span>&nbsp;<span data-toggle="tooltip" title="Check whether BLAST hits make evidence about a merge of two genes that match the predicted gene." data-placement="top"><i class="fa fa-question-circle"></i></span>
                            </th>
                        <th class="sorter-false">
              <b>Duplication</b>
                                   <span data-toggle="tooltip" title="Check whether there is a duplicated subsequence in the predicted gene by counting the hsp residue coverage of the prediction, for each hit." data-placement="top"><i class="fa fa-question-circle"></i></span>
                            </th>
                        <th class="sorter-false">
              <b>Reading Frame</b>
                                   <span data-toggle="tooltip" title="Check whether there is a single reading frame among BLAST hits. Otherwise there might be a reading frame shift in the query sequence." data-placement="top"><i class="fa fa-question-circle"></i></span>
                            </th>
                        <th class="sorter-false">
              <b>Main ORF</b>
                                  <span data-toggle="tooltip" title="Charts available for this validation" data-placement="top"><i class="fa fa-bar-chart-o chartIcon"></i></span>&nbsp;<span data-toggle="tooltip" title="Check whether there is a single main Open Reading Frame in the predicted gene. Applicable only for nucleotide queries." data-placement="top"><i class="fa fa-question-circle"></i></span>
                            </th>
                        <th class="sorter-false">
              <b>Missing/Extra sequences</b>
                                  <span data-toggle="tooltip" title="Charts available for this validation" data-placement="top"><i class="fa fa-bar-chart-o chartIcon"></i></span>&nbsp;<span data-toggle="tooltip" title="Finds missing and extra sequences in the prediction, based on the multiple alignment of the best hits. Also counts the percentage of the  conserved regions that appear in the prediction." data-placement="top"><i class="fa fa-question-circle"></i></span>
                            </th>
                        <th class="sorter-false chart-column"></th>
          </tr>
        </thead>
        <tbody><tr>
  <td>1</td>
  <td data-score="75"><div class="ratings"><div class="empty-stars"></div><div class="full-stars" style="width:63.75%;"></div></div></td>
  <td title="Definition" class="seq_definition">gi|514746961|ref|XM_005019748.1|  PREDICTED: Anas platyrhynchos insulin (INS), mRNA</td>
  <td title="No. Hits">199</td>
      <td title="Length Cluster" class="danger my-btn-danger" data-approach="If the query sequence is well conserved and similar sequences (BLAST hits) are correct, we can expect query and hit sequences to have similar lengths. Here, we cluster the lengths of hit sequences and compare the length of our query sequence to the most dense cluster of hit lengths. " data-explanation="The most dense length-cluster of BLAST hits includes sequences that are from 51 to 51 amino-acids long. The query sequence is 108 amino-acids long and is thus outside the most dense length-cluster of BLAST hits." data-conclusion="This suggests that the query sequence may be too long." onclick="showDiv(this, 'toggle1');
          addPlot('toggle1', 'files/json/mrna_data.fasta_1_len_clusters.json', 'bars', 'Length Cluster Validation: Distribution of BLAST hit lengths', 'query, black;most dense cluster,red;other hits, blue', 'Sequence Lenght', 'Number of Sequences', '108', '');
    AddExplanation(this, 'toggle1');">108&nbsp;[51,&nbsp;51]</td>
        <td title="Length Rank" class="success" data-approach="If the query sequence is well conserved and similar sequences (BLAST hits) are correct, we can expect query and hit sequences to have similar lengths. " data-explanation="The query sequence is  108 amino-acids long. BLAST identified 199 hit sequences with lengths from 51 to 248 amino-acids (median: 107; mean: 109). 84 of these hit sequences (i.e. 42%) are shorter than the query sequence." data-conclusion="There is no reason to believe there is any problem with the length of the query sequence." onclick="showDiv(this, 'toggle1'); AddExplanation(this,'toggle1');">42%</td>
        <td title="Gene Merge" class="success my-btn-success" data-approach="We expect the query sequence to encode a single protein-coding gene. Here, we analyse the High-scoring Segment Pairs (HSPs) identified by BLAST to determine whether the query includes sequence from two or more genes." data-explanation="The start coordinates and the end coordinates of HSPs are unimodally distributed." data-conclusion="This suggest that the query sequence represents a single gene." onclick="showDiv(this, 'toggle1');
          addPlot('toggle1', 'files/json/mrna_data.fasta_1_match_2d.json', 'scatter', 'Gene Merge Validation: Start/end of matching hit coord. on query (1 point/hit)', '', 'Start Offset (most left hsp)', 'End Offset (most right hsp)', '', '');
          addPlot('toggle1', 'files/json/mrna_data.fasta_1_match.json', 'lines', 'Gene Merge Validation: Query coord covered by blast hit (1 line/hit)', '', 'Offset in Prediction', 'Hit Number', '199', '');
    AddExplanation(this, 'toggle1');">0.0</td>
        <td title="Duplication" class="success" data-approach="We expect each BLAST hit to match each region of the query at most once. Here, we calculate the distribution of hit coverage against the query sequence and use the Wilcoxon test to determine if it is higher than 1." data-explanation="The Wilcoxon test produced a p-value of 1." data-conclusion="This suggests that the query sequence contains no erroneous duplications." onclick="showDiv(this, 'toggle1'); AddExplanation(this,'toggle1');">1.0</td>
        <td title="Reading Frame" class="success" data-approach="We expect the query sequence to encode a single gene, thus it should contain one main Open Reading Frame (ORF). All all BLAST hits are thus expected to align within this ORF." data-explanation="BLAST identified 199 High-scoring Segment Pairs (HSPs); all of these align in frame 1." data-conclusion="This is as expected." onclick="showDiv(this, 'toggle1'); AddExplanation(this,'toggle1');">199&nbsp;HSPs&nbsp;align&nbsp;in&nbsp;frame&nbsp;1</td>
        <td title="Main ORF" class="success my-btn-success" data-approach="We expect the query sequence to encode a single gene, thus it should contain one main Open Reading Frame (ORF) that occupies most of the query sequence." data-explanation=" The longest ORF is in frame 1, where it  occupies 100% of the query sequence." data-conclusion="There is no evidence to believe that there is any problem with the ORF of the query sequence." onclick="showDiv(this, 'toggle1');
          addPlot('toggle1', 'files/json/mrna_data.fasta_1_orfs.json', 'lines', 'Open reading frames in the 6 frames', '', 'Offset in the Prediction', 'Reading Frame', '14', '');
    AddExplanation(this, 'toggle1');">100%&nbsp;(frame&nbsp;1)</td>
        <td title="Missing/Extra sequences" class="success my-btn-success" data-approach="We expect the query sequence to be similar to the top ten BLAST hits. Here, we create a statistical consensus model of those top hits and compare the query to this  model." data-explanation="The query sequence includes 94% amino-acid residues present in the consensus model. 1% of residues in the query sequence are absent from the consensus profile.  0% of residues in the consensus profile are absent from the query sequence." data-conclusion="There is no evidence based on the top 10 BLAST hits to suggest any problems with the query sequence." onclick="showDiv(this, 'toggle1');
          addPlot('toggle1', 'files/json/mrna_data.fasta_1_ma.json', 'align', 'Missing/Extra sequences Validation: Multiple Align. & Statistical model of hits', 'Conserved Region, Yellow', 'Offset in the Alignment', '', '12', 'Prediction, hit&nbsp;1, hit&nbsp;2, hit&nbsp;3, hit&nbsp;4, hit&nbsp;5, hit&nbsp;6, hit&nbsp;7, hit&nbsp;8, hit&nbsp;9, hit&nbsp;10, Statistical Model');
    AddExplanation(this, 'toggle1');">94%&nbsp;conserved; 1%&nbsp;extra; 0%&nbsp;missing.</td>
  
  <td>
    <button title="Show plots" name="plot_btn" class="plot_btn btn btn-default" onclick="showDiv(this, 'toggle1');
                   addPlot('toggle1', 'files/json/mrna_data.fasta_1_len_clusters.json', 'bars', 'Length Cluster Validation: Distribution of BLAST hit lengths', 'query, black;most dense cluster,red;other hits, blue', 'Sequence Lenght', 'Number of Sequences', '108', '');                                   addPlot('toggle1', 'files/json/mrna_data.fasta_1_match_2d.json', 'scatter', 'Gene Merge Validation: Start/end of matching hit coord. on query (1 point/hit)', '', 'Start Offset (most left hsp)', 'End Offset (most right hsp)', '', ''); addPlot('toggle1', 'files/json/mrna_data.fasta_1_match.json', 'lines', 'Gene Merge Validation: Query coord covered by blast hit (1 line/hit)', '', 'Offset in Prediction', 'Hit Number', '199', '');                                             addPlot('toggle1', 'files/json/mrna_data.fasta_1_orfs.json', 'lines', 'Open reading frames in the 6 frames', '', 'Offset in the Prediction', 'Reading Frame', '14', '');                         addPlot('toggle1', 'files/json/mrna_data.fasta_1_ma.json', 'align', 'Missing/Extra sequences Validation: Multiple Align. & Statistical model of hits', 'Conserved Region, Yellow', 'Offset in the Alignment', '', '12', 'Prediction, hit&nbsp;1, hit&nbsp;2, hit&nbsp;3, hit&nbsp;4, hit&nbsp;5, hit&nbsp;6, hit&nbsp;7, hit&nbsp;8, hit&nbsp;9, hit&nbsp;10, Statistical Model');          "><i class="fa fa-bar-chart-o"></i></button>
  </td>
</tr>
<tr class="tablesorter-childRow" name="plot_row" style="display:none;">
  <td colspan="14" id="toggle1row">
    <div id="toggle1" class="expanded-child"></div>
  </td>
</tr>
<tr>
  <td>3</td>
  <td data-score="33"><div class="ratings"><div class="empty-stars"></div><div class="full-stars" style="width:28.05%;"></div></div></td>
  <td title="Definition" class="seq_definition">gi|251823817|ref|NM_010785.2|  Mus musculus transformed mouse 3T3 cell double minute 1 (Mdm1), transcript variant 1, mRNA</td>
  <td title="No. Hits">200</td>
      <td title="Length Cluster" class="danger my-btn-danger" data-approach="If the query sequence is well conserved and similar sequences (BLAST hits) are correct, we can expect query and hit sequences to have similar lengths. Here, we cluster the lengths of hit sequences and compare the length of our query sequence to the most dense cluster of hit lengths. " data-explanation="The most dense length-cluster of BLAST hits includes sequences that are from 454 to 454 amino-acids long. The query sequence is 1023 amino-acids long and is thus outside the most dense length-cluster of BLAST hits." data-conclusion="This suggests that the query sequence may be too long." onclick="showDiv(this, 'toggle3');
          addPlot('toggle3', 'files/json/mrna_data.fasta_3_len_clusters.json', 'bars', 'Length Cluster Validation: Distribution of BLAST hit lengths', 'query, black;most dense cluster,red;other hits, blue', 'Sequence Lenght', 'Number of Sequences', '1023', '');
    AddExplanation(this, 'toggle3');">1023&nbsp;[454,&nbsp;454]</td>
        <td title="Length Rank" class="danger" data-approach="If the query sequence is well conserved and similar sequences (BLAST hits) are correct, we can expect query and hit sequences to have similar lengths. " data-explanation="The query sequence is  1023 amino-acids long. BLAST identified 200 hit sequences with lengths from 454 to 969 amino-acids (median: 710; mean: 705). All hit sequences are shorter than the query sequence." data-conclusion="The sequence may be too long." onclick="showDiv(this, 'toggle3'); AddExplanation(this,'toggle3');">0%&nbsp;(too&nbsp;long)</td>
        <td title="Gene Merge" class="success my-btn-success" data-approach="We expect the query sequence to encode a single protein-coding gene. Here, we analyse the High-scoring Segment Pairs (HSPs) identified by BLAST to determine whether the query includes sequence from two or more genes." data-explanation="The start coordinates and the end coordinates of HSPs are unimodally distributed." data-conclusion="This suggest that the query sequence represents a single gene." onclick="showDiv(this, 'toggle3');
          addPlot('toggle3', 'files/json/mrna_data.fasta_3_match_2d.json', 'scatter', 'Gene Merge Validation: Start/end of matching hit coord. on query (1 point/hit)', '', 'Start Offset (most left hsp)', 'End Offset (most right hsp)', '', '');
          addPlot('toggle3', 'files/json/mrna_data.fasta_3_match.json', 'lines', 'Gene Merge Validation: Query coord covered by blast hit (1 line/hit)', '', 'Offset in Prediction', 'Hit Number', '200', '');
    AddExplanation(this, 'toggle3');">0.0</td>
        <td title="Duplication" class="success" data-approach="We expect each BLAST hit to match each region of the query at most once. Here, we calculate the distribution of hit coverage against the query sequence and use the Wilcoxon test to determine if it is higher than 1." data-explanation="The Wilcoxon test produced a p-value of 1." data-conclusion="This suggests that the query sequence contains no erroneous duplications." onclick="showDiv(this, 'toggle3'); AddExplanation(this,'toggle3');">1.0</td>
        <td title="Reading Frame" class="success" data-approach="We expect the query sequence to encode a single gene, thus it should contain one main Open Reading Frame (ORF). All all BLAST hits are thus expected to align within this ORF." data-explanation="BLAST identified 200 High-scoring Segment Pairs (HSPs); all of these align in frame 3." data-conclusion="This is as expected." onclick="showDiv(this, 'toggle3'); AddExplanation(this,'toggle3');">200&nbsp;HSPs&nbsp;align&nbsp;in&nbsp;frame&nbsp;3</td>
        <td title="Main ORF" class="danger my-btn-danger" data-approach="We expect the query sequence to encode a single gene, thus it should contain one main Open Reading Frame (ORF) that occupies most of the query sequence." data-explanation=" The longest ORF is in frame 3, where it  occupies 72% of the query sequence." data-conclusion="This only represents a portion of the query sequence. In some cases this indicates that a frame shift exists in the query sequence." onclick="showDiv(this, 'toggle3');
          addPlot('toggle3', 'files/json/mrna_data.fasta_3_orfs.json', 'lines', 'Open reading frames in the 6 frames', '', 'Offset in the Prediction', 'Reading Frame', '14', '');
    AddExplanation(this, 'toggle3');">72%&nbsp;(frame&nbsp;3)</td>
        <td title="Missing/Extra sequences" class="danger my-btn-danger" data-approach="We expect the query sequence to be similar to the top ten BLAST hits. Here, we create a statistical consensus model of those top hits and compare the query to this  model." data-explanation="The query sequence includes 22% amino-acid residues present in the consensus model. 34% of residues in the query sequence are absent from the consensus profile.  5% of residues in the consensus profile are absent from the query sequence." data-conclusion="These results suggest that there may be some problems with the query sequence. The query sequence has a high percentage (34%) of extra residues absent from the statistical profile (the cut-off is 20%)." onclick="showDiv(this, 'toggle3');
          addPlot('toggle3', 'files/json/mrna_data.fasta_3_ma.json', 'align', 'Missing/Extra sequences Validation: Multiple Align. & Statistical model of hits', 'Conserved Region, Yellow', 'Offset in the Alignment', '', '12', 'Prediction, hit&nbsp;1, hit&nbsp;2, hit&nbsp;3, hit&nbsp;4, hit&nbsp;5, hit&nbsp;6, hit&nbsp;7, hit&nbsp;8, hit&nbsp;9, hit&nbsp;10, Statistical Model');
    AddExplanation(this, 'toggle3');">22%&nbsp;conserved; 34%&nbsp;extra; 5%&nbsp;missing.</td>
  
  <td>
    <button title="Show plots" name="plot_btn" class="plot_btn btn btn-default" onclick="showDiv(this, 'toggle3');
                   addPlot('toggle3', 'files/json/mrna_data.fasta_3_len_clusters.json', 'bars', 'Length Cluster Validation: Distribution of BLAST hit lengths', 'query, black;most dense cluster,red;other hits, blue', 'Sequence Lenght', 'Number of Sequences', '1023', '');                                   addPlot('toggle3', 'files/json/mrna_data.fasta_3_match_2d.json', 'scatter', 'Gene Merge Validation: Start/end of matching hit coord. on query (1 point/hit)', '', 'Start Offset (most left hsp)', 'End Offset (most right hsp)', '', ''); addPlot('toggle3', 'files/json/mrna_data.fasta_3_match.json', 'lines', 'Gene Merge Validation: Query coord covered by blast hit (1 line/hit)', '', 'Offset in Prediction', 'Hit Number', '200', '');                                             addPlot('toggle3', 'files/json/mrna_data.fasta_3_orfs.json', 'lines', 'Open reading frames in the 6 frames', '', 'Offset in the Prediction', 'Reading Frame', '14', '');                         addPlot('toggle3', 'files/json/mrna_data.fasta_3_ma.json', 'align', 'Missing/Extra sequences Validation: Multiple Align. & Statistical model of hits', 'Conserved Region, Yellow', 'Offset in the Alignment', '', '12', 'Prediction, hit&nbsp;1, hit&nbsp;2, hit&nbsp;3, hit&nbsp;4, hit&nbsp;5, hit&nbsp;6, hit&nbsp;7, hit&nbsp;8, hit&nbsp;9, hit&nbsp;10, Statistical Model');          "><i class="fa fa-bar-chart-o"></i></button>
  </td>
</tr>
<tr class="tablesorter-childRow" name="plot_row" style="display:none;">
  <td colspan="14" id="toggle3row">
    <div id="toggle3" class="expanded-child"></div>
  </td>
</tr>
<tr>
  <td>7</td>
  <td data-score="17"><div class="ratings"><div class="empty-stars"></div><div class="full-stars" style="width:14.45%;"></div></div></td>
  <td title="Definition" class="seq_definition">Brasy-EspV1T00061 </td>
  <td title="No. Hits">200</td>
      <td title="Length Cluster" class="danger my-btn-danger" data-approach="If the query sequence is well conserved and similar sequences (BLAST hits) are correct, we can expect query and hit sequences to have similar lengths. Here, we cluster the lengths of hit sequences and compare the length of our query sequence to the most dense cluster of hit lengths. " data-explanation="The most dense length-cluster of BLAST hits includes sequences that are from 276 to 276 amino-acids long. The query sequence is 466 amino-acids long and is thus outside the most dense length-cluster of BLAST hits." data-conclusion="This suggests that the query sequence may be too long." onclick="showDiv(this, 'toggle7');
          addPlot('toggle7', 'files/json/mrna_data.fasta_7_len_clusters.json', 'bars', 'Length Cluster Validation: Distribution of BLAST hit lengths', 'query, black;most dense cluster,red;other hits, blue', 'Sequence Lenght', 'Number of Sequences', '466', '');
    AddExplanation(this, 'toggle7');">466&nbsp;[276,&nbsp;276]</td>
        <td title="Length Rank" class="danger" data-approach="If the query sequence is well conserved and similar sequences (BLAST hits) are correct, we can expect query and hit sequences to have similar lengths. " data-explanation="The query sequence is  466 amino-acids long. BLAST identified 200 hit sequences with lengths from 276 to 430 amino-acids (median: 377; mean: 367). All hit sequences are shorter than the query sequence." data-conclusion="The sequence may be too long." onclick="showDiv(this, 'toggle7'); AddExplanation(this,'toggle7');">0%&nbsp;(too&nbsp;long)</td>
        <td title="Gene Merge" class="success my-btn-success" data-approach="We expect the query sequence to encode a single protein-coding gene. Here, we analyse the High-scoring Segment Pairs (HSPs) identified by BLAST to determine whether the query includes sequence from two or more genes." data-explanation="The start coordinates and the end coordinates of HSPs are unimodally distributed." data-conclusion="This suggest that the query sequence represents a single gene." onclick="showDiv(this, 'toggle7');
          addPlot('toggle7', 'files/json/mrna_data.fasta_7_match_2d.json', 'scatter', 'Gene Merge Validation: Start/end of matching hit coord. on query (1 point/hit)', '', 'Start Offset (most left hsp)', 'End Offset (most right hsp)', '', '');
          addPlot('toggle7', 'files/json/mrna_data.fasta_7_match.json', 'lines', 'Gene Merge Validation: Query coord covered by blast hit (1 line/hit)', '', 'Offset in Prediction', 'Hit Number', '200', '');
    AddExplanation(this, 'toggle7');">0.0</td>
        <td title="Duplication" class="success" data-approach="We expect each BLAST hit to match each region of the query at most once. Here, we calculate the distribution of hit coverage against the query sequence and use the Wilcoxon test to determine if it is higher than 1." data-explanation="The Wilcoxon test produced a p-value of 1." data-conclusion="This suggests that the query sequence contains no erroneous duplications." onclick="showDiv(this, 'toggle7'); AddExplanation(this,'toggle7');">1.0</td>
        <td title="Reading Frame" class="danger" data-approach="We expect the query sequence to encode a single gene, thus it should contain one main Open Reading Frame (ORF). All all BLAST hits are thus expected to align within this ORF." data-explanation="BLAST identified 600 High-scoring Segment Pairs (HSPs): 200 HSPs align in frame 1; 400 HSPs align in frame 2; ." data-conclusion="The HSPs align in mulitple reading frames, this suggests there may be a frame shift in the query sequence." onclick="showDiv(this, 'toggle7'); AddExplanation(this,'toggle7');">200&nbsp;HSPs&nbsp;align&nbsp;in&nbsp;frame&nbsp;1; 400&nbsp;HSPs&nbsp;align&nbsp;in&nbsp;frame&nbsp;2</td>
        <td title="Main ORF" class="danger my-btn-danger" data-approach="We expect the query sequence to encode a single gene, thus it should contain one main Open Reading Frame (ORF) that occupies most of the query sequence." data-explanation=" The longest ORF is in frame 1, where it  occupies 49% of the query sequence." data-conclusion="This only represents a portion of the query sequence. In some cases this indicates that a frame shift exists in the query sequence." onclick="showDiv(this, 'toggle7');
          addPlot('toggle7', 'files/json/mrna_data.fasta_7_orfs.json', 'lines', 'Open reading frames in the 6 frames', '', 'Offset in the Prediction', 'Reading Frame', '14', '');
    AddExplanation(this, 'toggle7');">49%&nbsp;(frame&nbsp;1)</td>
        <td title="Missing/Extra sequences" class="danger my-btn-danger" data-approach="We expect the query sequence to be similar to the top ten BLAST hits. Here, we create a statistical consensus model of those top hits and compare the query to this  model." data-explanation="The query sequence includes 46% amino-acid residues present in the consensus model. 38% of residues in the query sequence are absent from the consensus profile.  23% of residues in the consensus profile are absent from the query sequence." data-conclusion="These results suggest that there may be some problems with the query sequence. The query sequence has a high percentage (38%) of extra residues absent from the statistical profile (the cut-off is 20%). The query sequence has a high percentage (23%) of missing residues when compared to the statistical profile (the cut-off is 20%)." onclick="showDiv(this, 'toggle7');
          addPlot('toggle7', 'files/json/mrna_data.fasta_7_ma.json', 'align', 'Missing/Extra sequences Validation: Multiple Align. & Statistical model of hits', 'Conserved Region, Yellow', 'Offset in the Alignment', '', '12', 'Prediction, hit&nbsp;1, hit&nbsp;2, hit&nbsp;3, hit&nbsp;4, hit&nbsp;5, hit&nbsp;6, hit&nbsp;7, hit&nbsp;8, hit&nbsp;9, hit&nbsp;10, Statistical Model');
    AddExplanation(this, 'toggle7');">46%&nbsp;conserved; 38%&nbsp;extra; 23%&nbsp;missing.</td>
  
  <td>
    <button title="Show plots" name="plot_btn" class="plot_btn btn btn-default" onclick="showDiv(this, 'toggle7');
                   addPlot('toggle7', 'files/json/mrna_data.fasta_7_len_clusters.json', 'bars', 'Length Cluster Validation: Distribution of BLAST hit lengths', 'query, black;most dense cluster,red;other hits, blue', 'Sequence Lenght', 'Number of Sequences', '466', '');                                   addPlot('toggle7', 'files/json/mrna_data.fasta_7_match_2d.json', 'scatter', 'Gene Merge Validation: Start/end of matching hit coord. on query (1 point/hit)', '', 'Start Offset (most left hsp)', 'End Offset (most right hsp)', '', ''); addPlot('toggle7', 'files/json/mrna_data.fasta_7_match.json', 'lines', 'Gene Merge Validation: Query coord covered by blast hit (1 line/hit)', '', 'Offset in Prediction', 'Hit Number', '200', '');                                             addPlot('toggle7', 'files/json/mrna_data.fasta_7_orfs.json', 'lines', 'Open reading frames in the 6 frames', '', 'Offset in the Prediction', 'Reading Frame', '14', '');                         addPlot('toggle7', 'files/json/mrna_data.fasta_7_ma.json', 'align', 'Missing/Extra sequences Validation: Multiple Align. & Statistical model of hits', 'Conserved Region, Yellow', 'Offset in the Alignment', '', '12', 'Prediction, hit&nbsp;1, hit&nbsp;2, hit&nbsp;3, hit&nbsp;4, hit&nbsp;5, hit&nbsp;6, hit&nbsp;7, hit&nbsp;8, hit&nbsp;9, hit&nbsp;10, Statistical Model');          "><i class="fa fa-bar-chart-o"></i></button>
  </td>
</tr>
<tr class="tablesorter-childRow" name="plot_row" style="display:none;">
  <td colspan="14" id="toggle7row">
    <div id="toggle7" class="expanded-child"></div>
  </td>
</tr>
<tr>
  <td>6</td>
  <td data-score="17"><div class="ratings"><div class="empty-stars"></div><div class="full-stars" style="width:14.45%;"></div></div></td>
  <td title="Definition" class="seq_definition">Brasy-EspV1T00022 </td>
  <td title="No. Hits">200</td>
      <td title="Length Cluster" class="danger my-btn-danger" data-approach="If the query sequence is well conserved and similar sequences (BLAST hits) are correct, we can expect query and hit sequences to have similar lengths. Here, we cluster the lengths of hit sequences and compare the length of our query sequence to the most dense cluster of hit lengths. " data-explanation="The most dense length-cluster of BLAST hits includes sequences that are from 96 to 144 amino-acids long. The query sequence is 1625 amino-acids long and is thus outside the most dense length-cluster of BLAST hits." data-conclusion="This suggests that the query sequence may be too long." onclick="showDiv(this, 'toggle6');
          addPlot('toggle6', 'files/json/mrna_data.fasta_6_len_clusters.json', 'bars', 'Length Cluster Validation: Distribution of BLAST hit lengths', 'query, black;most dense cluster,red;other hits, blue', 'Sequence Lenght', 'Number of Sequences', '1625', '');
    AddExplanation(this, 'toggle6');">1625&nbsp;[96,&nbsp;144]</td>
        <td title="Length Rank" class="danger" data-approach="If the query sequence is well conserved and similar sequences (BLAST hits) are correct, we can expect query and hit sequences to have similar lengths. " data-explanation="The query sequence is  1625 amino-acids long. BLAST identified 200 hit sequences with lengths from 96 to 1570 amino-acids (median: 661; mean: 665). All hit sequences are shorter than the query sequence." data-conclusion="The sequence may be too long." onclick="showDiv(this, 'toggle6'); AddExplanation(this,'toggle6');">0%&nbsp;(too&nbsp;long)</td>
        <td title="Gene Merge" class="success my-btn-success" data-approach="We expect the query sequence to encode a single protein-coding gene. Here, we analyse the High-scoring Segment Pairs (HSPs) identified by BLAST to determine whether the query includes sequence from two or more genes." data-explanation="The start coordinates and the end coordinates of HSPs are unimodally distributed." data-conclusion="This suggest that the query sequence represents a single gene." onclick="showDiv(this, 'toggle6');
          addPlot('toggle6', 'files/json/mrna_data.fasta_6_match_2d.json', 'scatter', 'Gene Merge Validation: Start/end of matching hit coord. on query (1 point/hit)', '', 'Start Offset (most left hsp)', 'End Offset (most right hsp)', '', '');
          addPlot('toggle6', 'files/json/mrna_data.fasta_6_match.json', 'lines', 'Gene Merge Validation: Query coord covered by blast hit (1 line/hit)', '', 'Offset in Prediction', 'Hit Number', '200', '');
    AddExplanation(this, 'toggle6');">0.0</td>
        <td title="Duplication" class="success" data-approach="We expect each BLAST hit to match each region of the query at most once. Here, we calculate the distribution of hit coverage against the query sequence and use the Wilcoxon test to determine if it is higher than 1." data-explanation="The Wilcoxon test produced a p-value of 1." data-conclusion="This suggests that the query sequence contains no erroneous duplications." onclick="showDiv(this, 'toggle6'); AddExplanation(this,'toggle6');">1.0</td>
        <td title="Reading Frame" class="danger" data-approach="We expect the query sequence to encode a single gene, thus it should contain one main Open Reading Frame (ORF). All all BLAST hits are thus expected to align within this ORF." data-explanation="BLAST identified 343 High-scoring Segment Pairs (HSPs): 163 HSPs align in frame 3; 180 HSPs align in frame 1; ." data-conclusion="The HSPs align in mulitple reading frames, this suggests there may be a frame shift in the query sequence." onclick="showDiv(this, 'toggle6'); AddExplanation(this,'toggle6');">163&nbsp;HSPs&nbsp;align&nbsp;in&nbsp;frame&nbsp;3; 180&nbsp;HSPs&nbsp;align&nbsp;in&nbsp;frame&nbsp;1</td>
        <td title="Main ORF" class="danger my-btn-danger" data-approach="We expect the query sequence to encode a single gene, thus it should contain one main Open Reading Frame (ORF) that occupies most of the query sequence." data-explanation=" The longest ORF is in frame 3, where it  occupies 71% of the query sequence." data-conclusion="This only represents a portion of the query sequence. In some cases this indicates that a frame shift exists in the query sequence." onclick="showDiv(this, 'toggle6');
          addPlot('toggle6', 'files/json/mrna_data.fasta_6_orfs.json', 'lines', 'Open reading frames in the 6 frames', '', 'Offset in the Prediction', 'Reading Frame', '14', '');
    AddExplanation(this, 'toggle6');">71%&nbsp;(frame&nbsp;3)</td>
        <td title="Missing/Extra sequences" class="danger my-btn-danger" data-approach="We expect the query sequence to be similar to the top ten BLAST hits. Here, we create a statistical consensus model of those top hits and compare the query to this  model." data-explanation="The query sequence includes 39% amino-acid residues present in the consensus model. 27% of residues in the query sequence are absent from the consensus profile.  6% of residues in the consensus profile are absent from the query sequence." data-conclusion="These results suggest that there may be some problems with the query sequence. The query sequence has a high percentage (27%) of extra residues absent from the statistical profile (the cut-off is 20%)." onclick="showDiv(this, 'toggle6');
          addPlot('toggle6', 'files/json/mrna_data.fasta_6_ma.json', 'align', 'Missing/Extra sequences Validation: Multiple Align. & Statistical model of hits', 'Conserved Region, Yellow', 'Offset in the Alignment', '', '12', 'Prediction, hit&nbsp;1, hit&nbsp;2, hit&nbsp;3, hit&nbsp;4, hit&nbsp;5, hit&nbsp;6, hit&nbsp;7, hit&nbsp;8, hit&nbsp;9, hit&nbsp;10, Statistical Model');
    AddExplanation(this, 'toggle6');">39%&nbsp;conserved; 27%&nbsp;extra; 6%&nbsp;missing.</td>
  
  <td>
    <button title="Show plots" name="plot_btn" class="plot_btn btn btn-default" onclick="showDiv(this, 'toggle6');
                   addPlot('toggle6', 'files/json/mrna_data.fasta_6_len_clusters.json', 'bars', 'Length Cluster Validation: Distribution of BLAST hit lengths', 'query, black;most dense cluster,red;other hits, blue', 'Sequence Lenght', 'Number of Sequences', '1625', '');                                   addPlot('toggle6', 'files/json/mrna_data.fasta_6_match_2d.json', 'scatter', 'Gene Merge Validation: Start/end of matching hit coord. on query (1 point/hit)', '', 'Start Offset (most left hsp)', 'End Offset (most right hsp)', '', ''); addPlot('toggle6', 'files/json/mrna_data.fasta_6_match.json', 'lines', 'Gene Merge Validation: Query coord covered by blast hit (1 line/hit)', '', 'Offset in Prediction', 'Hit Number', '200', '');                                             addPlot('toggle6', 'files/json/mrna_data.fasta_6_orfs.json', 'lines', 'Open reading frames in the 6 frames', '', 'Offset in the Prediction', 'Reading Frame', '14', '');                         addPlot('toggle6', 'files/json/mrna_data.fasta_6_ma.json', 'align', 'Missing/Extra sequences Validation: Multiple Align. & Statistical model of hits', 'Conserved Region, Yellow', 'Offset in the Alignment', '', '12', 'Prediction, hit&nbsp;1, hit&nbsp;2, hit&nbsp;3, hit&nbsp;4, hit&nbsp;5, hit&nbsp;6, hit&nbsp;7, hit&nbsp;8, hit&nbsp;9, hit&nbsp;10, Statistical Model');          "><i class="fa fa-bar-chart-o"></i></button>
  </td>
</tr>
<tr class="tablesorter-childRow" name="plot_row" style="display:none;">
  <td colspan="14" id="toggle6row">
    <div id="toggle6" class="expanded-child"></div>
  </td>
</tr>
<tr>
  <td>8</td>
  <td data-score="17"><div class="ratings"><div class="empty-stars"></div><div class="full-stars" style="width:14.45%;"></div></div></td>
  <td title="Definition" class="seq_definition">Brasy-EspV1T00037 </td>
  <td title="No. Hits">200</td>
      <td title="Length Cluster" class="danger my-btn-danger" data-approach="If the query sequence is well conserved and similar sequences (BLAST hits) are correct, we can expect query and hit sequences to have similar lengths. Here, we cluster the lengths of hit sequences and compare the length of our query sequence to the most dense cluster of hit lengths. " data-explanation="The most dense length-cluster of BLAST hits includes sequences that are from 91 to 176 amino-acids long. The query sequence is 1325 amino-acids long and is thus outside the most dense length-cluster of BLAST hits." data-conclusion="This suggests that the query sequence may be too long." onclick="showDiv(this, 'toggle8');
          addPlot('toggle8', 'files/json/mrna_data.fasta_8_len_clusters.json', 'bars', 'Length Cluster Validation: Distribution of BLAST hit lengths', 'query, black;most dense cluster,red;other hits, blue', 'Sequence Lenght', 'Number of Sequences', '1325', '');
    AddExplanation(this, 'toggle8');">1325&nbsp;[91,&nbsp;176]</td>
        <td title="Length Rank" class="danger" data-approach="If the query sequence is well conserved and similar sequences (BLAST hits) are correct, we can expect query and hit sequences to have similar lengths. " data-explanation="The query sequence is  1325 amino-acids long. BLAST identified 200 hit sequences with lengths from 91 to 1789 amino-acids (median: 826; mean: 766). 1 of these hit sequences (i.e. 1%) are shorter than the query sequence." data-conclusion="The sequence may be too long." onclick="showDiv(this, 'toggle8'); AddExplanation(this,'toggle8');">1%&nbsp;(too&nbsp;long)</td>
        <td title="Gene Merge" class="success my-btn-success" data-approach="We expect the query sequence to encode a single protein-coding gene. Here, we analyse the High-scoring Segment Pairs (HSPs) identified by BLAST to determine whether the query includes sequence from two or more genes." data-explanation="The distribution of start and/or end-coordinates of HSPs are multi-modal. To detect potential problems we performed a linear regression (with coordinates weighted inversely proportionally to  hit strength). The resulting slope is -0.05." data-conclusion="This slope is  outside our empirically calculated thresholds (0.4 and 1.2). There is no evidence that the query contains sequence from multiple genes." onclick="showDiv(this, 'toggle8');
          addPlot('toggle8', 'files/json/mrna_data.fasta_8_match_2d.json', 'scatter', 'Gene Merge Validation: Start/end of matching hit coord. on query (1 point/hit)', '', 'Start Offset (most left hsp)', 'End Offset (most right hsp)', '1165.8889725718573', '-0.051763551763702904');
          addPlot('toggle8', 'files/json/mrna_data.fasta_8_match.json', 'lines', 'Gene Merge Validation: Query coord covered by blast hit (1 line/hit)', '', 'Offset in Prediction', 'Hit Number', '200', '');
    AddExplanation(this, 'toggle8');">-0.05</td>
        <td title="Duplication" class="success" data-approach="We expect each BLAST hit to match each region of the query at most once. Here, we calculate the distribution of hit coverage against the query sequence and use the Wilcoxon test to determine if it is higher than 1." data-explanation="The Wilcoxon test produced a p-value of 1." data-conclusion="This suggests that the query sequence contains no erroneous duplications." onclick="showDiv(this, 'toggle8'); AddExplanation(this,'toggle8');">1.0</td>
        <td title="Reading Frame" class="danger" data-approach="We expect the query sequence to encode a single gene, thus it should contain one main Open Reading Frame (ORF). All all BLAST hits are thus expected to align within this ORF." data-explanation="BLAST identified 406 High-scoring Segment Pairs (HSPs): 404 HSPs align in frame 2; 1 HSPs align in frame 1; 1 HSPs align in frame -2; ." data-conclusion="The HSPs align in mulitple reading frames, this suggests there may be a frame shift in the query sequence." onclick="showDiv(this, 'toggle8'); AddExplanation(this,'toggle8');">404&nbsp;HSPs&nbsp;align&nbsp;in&nbsp;frame&nbsp;2; 1&nbsp;HSPs&nbsp;align&nbsp;in&nbsp;frame&nbsp;1; 1&nbsp;HSPs&nbsp;align&nbsp;in&nbsp;frame&nbsp;-2</td>
        <td title="Main ORF" class="danger my-btn-danger" data-approach="We expect the query sequence to encode a single gene, thus it should contain one main Open Reading Frame (ORF) that occupies most of the query sequence." data-explanation=" The longest ORF is in frame 2, where it  occupies 56% of the query sequence." data-conclusion="This only represents a portion of the query sequence. In some cases this indicates that a frame shift exists in the query sequence." onclick="showDiv(this, 'toggle8');
          addPlot('toggle8', 'files/json/mrna_data.fasta_8_orfs.json', 'lines', 'Open reading frames in the 6 frames', '', 'Offset in the Prediction', 'Reading Frame', '14', '');
    AddExplanation(this, 'toggle8');">56%&nbsp;(frame&nbsp;2)</td>
        <td title="Missing/Extra sequences" class="danger my-btn-danger" data-approach="We expect the query sequence to be similar to the top ten BLAST hits. Here, we create a statistical consensus model of those top hits and compare the query to this  model." data-explanation="The query sequence includes 26% amino-acid residues present in the consensus model. 35% of residues in the query sequence are absent from the consensus profile.  15% of residues in the consensus profile are absent from the query sequence." data-conclusion="These results suggest that there may be some problems with the query sequence. The query sequence has a high percentage (35%) of extra residues absent from the statistical profile (the cut-off is 20%)." onclick="showDiv(this, 'toggle8');
          addPlot('toggle8', 'files/json/mrna_data.fasta_8_ma.json', 'align', 'Missing/Extra sequences Validation: Multiple Align. & Statistical model of hits', 'Conserved Region, Yellow', 'Offset in the Alignment', '', '12', 'Prediction, hit&nbsp;1, hit&nbsp;2, hit&nbsp;3, hit&nbsp;4, hit&nbsp;5, hit&nbsp;6, hit&nbsp;7, hit&nbsp;8, hit&nbsp;9, hit&nbsp;10, Statistical Model');
    AddExplanation(this, 'toggle8');">26%&nbsp;conserved; 35%&nbsp;extra; 15%&nbsp;missing.</td>
  
  <td>
    <button title="Show plots" name="plot_btn" class="plot_btn btn btn-default" onclick="showDiv(this, 'toggle8');
                   addPlot('toggle8', 'files/json/mrna_data.fasta_8_len_clusters.json', 'bars', 'Length Cluster Validation: Distribution of BLAST hit lengths', 'query, black;most dense cluster,red;other hits, blue', 'Sequence Lenght', 'Number of Sequences', '1325', '');                                   addPlot('toggle8', 'files/json/mrna_data.fasta_8_match_2d.json', 'scatter', 'Gene Merge Validation: Start/end of matching hit coord. on query (1 point/hit)', '', 'Start Offset (most left hsp)', 'End Offset (most right hsp)', '1165.8889725718573', '-0.051763551763702904'); addPlot('toggle8', 'files/json/mrna_data.fasta_8_match.json', 'lines', 'Gene Merge Validation: Query coord covered by blast hit (1 line/hit)', '', 'Offset in Prediction', 'Hit Number', '200', '');                                             addPlot('toggle8', 'files/json/mrna_data.fasta_8_orfs.json', 'lines', 'Open reading frames in the 6 frames', '', 'Offset in the Prediction', 'Reading Frame', '14', '');                         addPlot('toggle8', 'files/json/mrna_data.fasta_8_ma.json', 'align', 'Missing/Extra sequences Validation: Multiple Align. & Statistical model of hits', 'Conserved Region, Yellow', 'Offset in the Alignment', '', '12', 'Prediction, hit&nbsp;1, hit&nbsp;2, hit&nbsp;3, hit&nbsp;4, hit&nbsp;5, hit&nbsp;6, hit&nbsp;7, hit&nbsp;8, hit&nbsp;9, hit&nbsp;10, Statistical Model');          "><i class="fa fa-bar-chart-o"></i></button>
  </td>
</tr>
<tr class="tablesorter-childRow" name="plot_row" style="display:none;">
  <td colspan="14" id="toggle8row">
    <div id="toggle8" class="expanded-child"></div>
  </td>
</tr>
<tr>
  <td>9</td>
  <td data-score="17"><div class="ratings"><div class="empty-stars"></div><div class="full-stars" style="width:14.45%;"></div></div></td>
  <td title="Definition" class="seq_definition">Brasy-EspV1T00038 </td>
  <td title="No. Hits">200</td>
      <td title="Length Cluster" class="danger my-btn-danger" data-approach="If the query sequence is well conserved and similar sequences (BLAST hits) are correct, we can expect query and hit sequences to have similar lengths. Here, we cluster the lengths of hit sequences and compare the length of our query sequence to the most dense cluster of hit lengths. " data-explanation="The most dense length-cluster of BLAST hits includes sequences that are from 91 to 176 amino-acids long. The query sequence is 1325 amino-acids long and is thus outside the most dense length-cluster of BLAST hits." data-conclusion="This suggests that the query sequence may be too long." onclick="showDiv(this, 'toggle9');
          addPlot('toggle9', 'files/json/mrna_data.fasta_9_len_clusters.json', 'bars', 'Length Cluster Validation: Distribution of BLAST hit lengths', 'query, black;most dense cluster,red;other hits, blue', 'Sequence Lenght', 'Number of Sequences', '1325', '');
    AddExplanation(this, 'toggle9');">1325&nbsp;[91,&nbsp;176]</td>
        <td title="Length Rank" class="danger" data-approach="If the query sequence is well conserved and similar sequences (BLAST hits) are correct, we can expect query and hit sequences to have similar lengths. " data-explanation="The query sequence is  1325 amino-acids long. BLAST identified 200 hit sequences with lengths from 91 to 1789 amino-acids (median: 826; mean: 766). 1 of these hit sequences (i.e. 1%) are shorter than the query sequence." data-conclusion="The sequence may be too long." onclick="showDiv(this, 'toggle9'); AddExplanation(this,'toggle9');">1%&nbsp;(too&nbsp;long)</td>
        <td title="Gene Merge" class="success my-btn-success" data-approach="We expect the query sequence to encode a single protein-coding gene. Here, we analyse the High-scoring Segment Pairs (HSPs) identified by BLAST to determine whether the query includes sequence from two or more genes." data-explanation="The distribution of start and/or end-coordinates of HSPs are multi-modal. To detect potential problems we performed a linear regression (with coordinates weighted inversely proportionally to  hit strength). The resulting slope is -0.04." data-conclusion="This slope is  outside our empirically calculated thresholds (0.4 and 1.2). There is no evidence that the query contains sequence from multiple genes." onclick="showDiv(this, 'toggle9');
          addPlot('toggle9', 'files/json/mrna_data.fasta_9_match_2d.json', 'scatter', 'Gene Merge Validation: Start/end of matching hit coord. on query (1 point/hit)', '', 'Start Offset (most left hsp)', 'End Offset (most right hsp)', '1164.732188147934', '-0.043641558262501184');
          addPlot('toggle9', 'files/json/mrna_data.fasta_9_match.json', 'lines', 'Gene Merge Validation: Query coord covered by blast hit (1 line/hit)', '', 'Offset in Prediction', 'Hit Number', '200', '');
    AddExplanation(this, 'toggle9');">-0.04</td>
        <td title="Duplication" class="success" data-approach="We expect each BLAST hit to match each region of the query at most once. Here, we calculate the distribution of hit coverage against the query sequence and use the Wilcoxon test to determine if it is higher than 1." data-explanation="The Wilcoxon test produced a p-value of 1." data-conclusion="This suggests that the query sequence contains no erroneous duplications." onclick="showDiv(this, 'toggle9'); AddExplanation(this,'toggle9');">1.0</td>
        <td title="Reading Frame" class="danger" data-approach="We expect the query sequence to encode a single gene, thus it should contain one main Open Reading Frame (ORF). All all BLAST hits are thus expected to align within this ORF." data-explanation="BLAST identified 409 High-scoring Segment Pairs (HSPs): 407 HSPs align in frame 2; 1 HSPs align in frame 1; 1 HSPs align in frame -2; ." data-conclusion="The HSPs align in mulitple reading frames, this suggests there may be a frame shift in the query sequence." onclick="showDiv(this, 'toggle9'); AddExplanation(this,'toggle9');">407&nbsp;HSPs&nbsp;align&nbsp;in&nbsp;frame&nbsp;2; 1&nbsp;HSPs&nbsp;align&nbsp;in&nbsp;frame&nbsp;1; 1&nbsp;HSPs&nbsp;align&nbsp;in&nbsp;frame&nbsp;-2</td>
        <td title="Main ORF" class="danger my-btn-danger" data-approach="We expect the query sequence to encode a single gene, thus it should contain one main Open Reading Frame (ORF) that occupies most of the query sequence." data-explanation=" The longest ORF is in frame 2, where it  occupies 56% of the query sequence." data-conclusion="This only represents a portion of the query sequence. In some cases this indicates that a frame shift exists in the query sequence." onclick="showDiv(this, 'toggle9');
          addPlot('toggle9', 'files/json/mrna_data.fasta_9_orfs.json', 'lines', 'Open reading frames in the 6 frames', '', 'Offset in the Prediction', 'Reading Frame', '14', '');
    AddExplanation(this, 'toggle9');">56%&nbsp;(frame&nbsp;2)</td>
        <td title="Missing/Extra sequences" class="danger my-btn-danger" data-approach="We expect the query sequence to be similar to the top ten BLAST hits. Here, we create a statistical consensus model of those top hits and compare the query to this  model." data-explanation="The query sequence includes 26% amino-acid residues present in the consensus model. 35% of residues in the query sequence are absent from the consensus profile.  16% of residues in the consensus profile are absent from the query sequence." data-conclusion="These results suggest that there may be some problems with the query sequence. The query sequence has a high percentage (35%) of extra residues absent from the statistical profile (the cut-off is 20%)." onclick="showDiv(this, 'toggle9');
          addPlot('toggle9', 'files/json/mrna_data.fasta_9_ma.json', 'align', 'Missing/Extra sequences Validation: Multiple Align. & Statistical model of hits', 'Conserved Region, Yellow', 'Offset in the Alignment', '', '12', 'Prediction, hit&nbsp;1, hit&nbsp;2, hit&nbsp;3, hit&nbsp;4, hit&nbsp;5, hit&nbsp;6, hit&nbsp;7, hit&nbsp;8, hit&nbsp;9, hit&nbsp;10, Statistical Model');
    AddExplanation(this, 'toggle9');">26%&nbsp;conserved; 35%&nbsp;extra; 16%&nbsp;missing.</td>
  
  <td>
    <button title="Show plots" name="plot_btn" class="plot_btn btn btn-default" onclick="showDiv(this, 'toggle9');
                   addPlot('toggle9', 'files/json/mrna_data.fasta_9_len_clusters.json', 'bars', 'Length Cluster Validation: Distribution of BLAST hit lengths', 'query, black;most dense cluster,red;other hits, blue', 'Sequence Lenght', 'Number of Sequences', '1325', '');                                   addPlot('toggle9', 'files/json/mrna_data.fasta_9_match_2d.json', 'scatter', 'Gene Merge Validation: Start/end of matching hit coord. on query (1 point/hit)', '', 'Start Offset (most left hsp)', 'End Offset (most right hsp)', '1164.732188147934', '-0.043641558262501184'); addPlot('toggle9', 'files/json/mrna_data.fasta_9_match.json', 'lines', 'Gene Merge Validation: Query coord covered by blast hit (1 line/hit)', '', 'Offset in Prediction', 'Hit Number', '200', '');                                             addPlot('toggle9', 'files/json/mrna_data.fasta_9_orfs.json', 'lines', 'Open reading frames in the 6 frames', '', 'Offset in the Prediction', 'Reading Frame', '14', '');                         addPlot('toggle9', 'files/json/mrna_data.fasta_9_ma.json', 'align', 'Missing/Extra sequences Validation: Multiple Align. & Statistical model of hits', 'Conserved Region, Yellow', 'Offset in the Alignment', '', '12', 'Prediction, hit&nbsp;1, hit&nbsp;2, hit&nbsp;3, hit&nbsp;4, hit&nbsp;5, hit&nbsp;6, hit&nbsp;7, hit&nbsp;8, hit&nbsp;9, hit&nbsp;10, Statistical Model');          "><i class="fa fa-bar-chart-o"></i></button>
  </td>
</tr>
<tr class="tablesorter-childRow" name="plot_row" style="display:none;">
  <td colspan="14" id="toggle9row">
    <div id="toggle9" class="expanded-child"></div>
  </td>
</tr>
<tr>
  <td>10</td>
  <td data-score="17"><div class="ratings"><div class="empty-stars"></div><div class="full-stars" style="width:14.45%;"></div></div></td>
  <td title="Definition" class="seq_definition">Brasy-EspV1T00039 </td>
  <td title="No. Hits">200</td>
      <td title="Length Cluster" class="danger my-btn-danger" data-approach="If the query sequence is well conserved and similar sequences (BLAST hits) are correct, we can expect query and hit sequences to have similar lengths. Here, we cluster the lengths of hit sequences and compare the length of our query sequence to the most dense cluster of hit lengths. " data-explanation="The most dense length-cluster of BLAST hits includes sequences that are from 246 to 279 amino-acids long. The query sequence is 1618 amino-acids long and is thus outside the most dense length-cluster of BLAST hits." data-conclusion="This suggests that the query sequence may be too long." onclick="showDiv(this, 'toggle10');
          addPlot('toggle10', 'files/json/mrna_data.fasta_10_len_clusters.json', 'bars', 'Length Cluster Validation: Distribution of BLAST hit lengths', 'query, black;most dense cluster,red;other hits, blue', 'Sequence Lenght', 'Number of Sequences', '1618', '');
    AddExplanation(this, 'toggle10');">1618&nbsp;[246,&nbsp;279]</td>
        <td title="Length Rank" class="danger" data-approach="If the query sequence is well conserved and similar sequences (BLAST hits) are correct, we can expect query and hit sequences to have similar lengths. " data-explanation="The query sequence is  1618 amino-acids long. BLAST identified 200 hit sequences with lengths from 246 to 1311 amino-acids (median: 807; mean: 807). All hit sequences are shorter than the query sequence." data-conclusion="The sequence may be too long." onclick="showDiv(this, 'toggle10'); AddExplanation(this,'toggle10');">0%&nbsp;(too&nbsp;long)</td>
        <td title="Gene Merge" class="success my-btn-success" data-approach="We expect the query sequence to encode a single protein-coding gene. Here, we analyse the High-scoring Segment Pairs (HSPs) identified by BLAST to determine whether the query includes sequence from two or more genes." data-explanation="The distribution of start and/or end-coordinates of HSPs are multi-modal. To detect potential problems we performed a linear regression (with coordinates weighted inversely proportionally to  hit strength). The resulting slope is 0.35." data-conclusion="This slope is  outside our empirically calculated thresholds (0.4 and 1.2). There is no evidence that the query contains sequence from multiple genes." onclick="showDiv(this, 'toggle10');
          addPlot('toggle10', 'files/json/mrna_data.fasta_10_match_2d.json', 'scatter', 'Gene Merge Validation: Start/end of matching hit coord. on query (1 point/hit)', '', 'Start Offset (most left hsp)', 'End Offset (most right hsp)', '932.8089445678366', '0.3528288374126873');
          addPlot('toggle10', 'files/json/mrna_data.fasta_10_match.json', 'lines', 'Gene Merge Validation: Query coord covered by blast hit (1 line/hit)', '', 'Offset in Prediction', 'Hit Number', '200', '');
    AddExplanation(this, 'toggle10');">0.35</td>
        <td title="Duplication" class="success" data-approach="We expect each BLAST hit to match each region of the query at most once. Here, we calculate the distribution of hit coverage against the query sequence and use the Wilcoxon test to determine if it is higher than 1." data-explanation="The Wilcoxon test produced a p-value of 1." data-conclusion="This suggests that the query sequence contains no erroneous duplications." onclick="showDiv(this, 'toggle10'); AddExplanation(this,'toggle10');">1.0</td>
        <td title="Reading Frame" class="danger" data-approach="We expect the query sequence to encode a single gene, thus it should contain one main Open Reading Frame (ORF). All all BLAST hits are thus expected to align within this ORF." data-explanation="BLAST identified 279 High-scoring Segment Pairs (HSPs): 278 HSPs align in frame 2; 1 HSPs align in frame 1; ." data-conclusion="The HSPs align in mulitple reading frames, this suggests there may be a frame shift in the query sequence." onclick="showDiv(this, 'toggle10'); AddExplanation(this,'toggle10');">278&nbsp;HSPs&nbsp;align&nbsp;in&nbsp;frame&nbsp;2; 1&nbsp;HSPs&nbsp;align&nbsp;in&nbsp;frame&nbsp;1</td>
        <td title="Main ORF" class="danger my-btn-danger" data-approach="We expect the query sequence to encode a single gene, thus it should contain one main Open Reading Frame (ORF) that occupies most of the query sequence." data-explanation=" The longest ORF is in frame 2, where it  occupies 49% of the query sequence." data-conclusion="This only represents a portion of the query sequence. In some cases this indicates that a frame shift exists in the query sequence." onclick="showDiv(this, 'toggle10');
          addPlot('toggle10', 'files/json/mrna_data.fasta_10_orfs.json', 'lines', 'Open reading frames in the 6 frames', '', 'Offset in the Prediction', 'Reading Frame', '14', '');
    AddExplanation(this, 'toggle10');">49%&nbsp;(frame&nbsp;2)</td>
        <td title="Missing/Extra sequences" class="danger my-btn-danger" data-approach="We expect the query sequence to be similar to the top ten BLAST hits. Here, we create a statistical consensus model of those top hits and compare the query to this  model." data-explanation="The query sequence includes 28% amino-acid residues present in the consensus model. 45% of residues in the query sequence are absent from the consensus profile.  12% of residues in the consensus profile are absent from the query sequence." data-conclusion="These results suggest that there may be some problems with the query sequence. The query sequence has a high percentage (45%) of extra residues absent from the statistical profile (the cut-off is 20%)." onclick="showDiv(this, 'toggle10');
          addPlot('toggle10', 'files/json/mrna_data.fasta_10_ma.json', 'align', 'Missing/Extra sequences Validation: Multiple Align. & Statistical model of hits', 'Conserved Region, Yellow', 'Offset in the Alignment', '', '12', 'Prediction, hit&nbsp;1, hit&nbsp;2, hit&nbsp;3, hit&nbsp;4, hit&nbsp;5, hit&nbsp;6, hit&nbsp;7, hit&nbsp;8, hit&nbsp;9, hit&nbsp;10, Statistical Model');
    AddExplanation(this, 'toggle10');">28%&nbsp;conserved; 45%&nbsp;extra; 12%&nbsp;missing.</td>
  
  <td>
    <button title="Show plots" name="plot_btn" class="plot_btn btn btn-default" onclick="showDiv(this, 'toggle10');
                   addPlot('toggle10', 'files/json/mrna_data.fasta_10_len_clusters.json', 'bars', 'Length Cluster Validation: Distribution of BLAST hit lengths', 'query, black;most dense cluster,red;other hits, blue', 'Sequence Lenght', 'Number of Sequences', '1618', '');                                   addPlot('toggle10', 'files/json/mrna_data.fasta_10_match_2d.json', 'scatter', 'Gene Merge Validation: Start/end of matching hit coord. on query (1 point/hit)', '', 'Start Offset (most left hsp)', 'End Offset (most right hsp)', '932.8089445678366', '0.3528288374126873'); addPlot('toggle10', 'files/json/mrna_data.fasta_10_match.json', 'lines', 'Gene Merge Validation: Query coord covered by blast hit (1 line/hit)', '', 'Offset in Prediction', 'Hit Number', '200', '');                                             addPlot('toggle10', 'files/json/mrna_data.fasta_10_orfs.json', 'lines', 'Open reading frames in the 6 frames', '', 'Offset in the Prediction', 'Reading Frame', '14', '');                         addPlot('toggle10', 'files/json/mrna_data.fasta_10_ma.json', 'align', 'Missing/Extra sequences Validation: Multiple Align. & Statistical model of hits', 'Conserved Region, Yellow', 'Offset in the Alignment', '', '12', 'Prediction, hit&nbsp;1, hit&nbsp;2, hit&nbsp;3, hit&nbsp;4, hit&nbsp;5, hit&nbsp;6, hit&nbsp;7, hit&nbsp;8, hit&nbsp;9, hit&nbsp;10, Statistical Model');          "><i class="fa fa-bar-chart-o"></i></button>
  </td>
</tr>
<tr class="tablesorter-childRow" name="plot_row" style="display:none;">
  <td colspan="14" id="toggle10row">
    <div id="toggle10" class="expanded-child"></div>
  </td>
</tr>
<tr>
  <td>13</td>
  <td data-score="50"><div class="ratings"><div class="empty-stars"></div><div class="full-stars" style="width:42.5%;"></div></div></td>
  <td title="Definition" class="seq_definition">Brasy-EspV1T00042 </td>
  <td title="No. Hits">73</td>
      <td title="Length Cluster" class="danger my-btn-danger" data-approach="If the query sequence is well conserved and similar sequences (BLAST hits) are correct, we can expect query and hit sequences to have similar lengths. Here, we cluster the lengths of hit sequences and compare the length of our query sequence to the most dense cluster of hit lengths. " data-explanation="The most dense length-cluster of BLAST hits includes sequences that are from 261 to 261 amino-acids long. The query sequence is 226 amino-acids long and is thus outside the most dense length-cluster of BLAST hits." data-conclusion="This suggests that the query sequence may be too short." onclick="showDiv(this, 'toggle13');
          addPlot('toggle13', 'files/json/mrna_data.fasta_13_len_clusters.json', 'bars', 'Length Cluster Validation: Distribution of BLAST hit lengths', 'query, black;most dense cluster,red;other hits, blue', 'Sequence Lenght', 'Number of Sequences', '226', '');
    AddExplanation(this, 'toggle13');">226&nbsp;[261,&nbsp;261]</td>
        <td title="Length Rank" class="danger" data-approach="If the query sequence is well conserved and similar sequences (BLAST hits) are correct, we can expect query and hit sequences to have similar lengths. " data-explanation="The query sequence is  226 amino-acids long. BLAST identified 73 hit sequences with lengths from 261 to 1213 amino-acids (median: 862; mean: 862). All hit sequences are longer than the query sequence." data-conclusion="The sequence may be too short." onclick="showDiv(this, 'toggle13'); AddExplanation(this,'toggle13');">0%&nbsp;(too&nbsp;short)</td>
        <td title="Gene Merge" class="success my-btn-success" data-approach="We expect the query sequence to encode a single protein-coding gene. Here, we analyse the High-scoring Segment Pairs (HSPs) identified by BLAST to determine whether the query includes sequence from two or more genes." data-explanation="The distribution of start and/or end-coordinates of HSPs are multi-modal. To detect potential problems we performed a linear regression (with coordinates weighted inversely proportionally to  hit strength). The resulting slope is -0.0." data-conclusion="This slope is  outside our empirically calculated thresholds (0.4 and 1.2). There is no evidence that the query contains sequence from multiple genes." onclick="showDiv(this, 'toggle13');
          addPlot('toggle13', 'files/json/mrna_data.fasta_13_match_2d.json', 'scatter', 'Gene Merge Validation: Start/end of matching hit coord. on query (1 point/hit)', '', 'Start Offset (most left hsp)', 'End Offset (most right hsp)', '226.97208531181383', '-0.000599534443489048');
          addPlot('toggle13', 'files/json/mrna_data.fasta_13_match.json', 'lines', 'Gene Merge Validation: Query coord covered by blast hit (1 line/hit)', '', 'Offset in Prediction', 'Hit Number', '73', '');
    AddExplanation(this, 'toggle13');">-0.0</td>
        <td title="Duplication" class="success" data-approach="We expect each BLAST hit to match each region of the query at most once. Here, we calculate the distribution of hit coverage against the query sequence and use the Wilcoxon test to determine if it is higher than 1." data-explanation="The Wilcoxon test produced a p-value of 1." data-conclusion="This suggests that the query sequence contains no erroneous duplications." onclick="showDiv(this, 'toggle13'); AddExplanation(this,'toggle13');">1.0</td>
        <td title="Reading Frame" class="success" data-approach="We expect the query sequence to encode a single gene, thus it should contain one main Open Reading Frame (ORF). All all BLAST hits are thus expected to align within this ORF." data-explanation="BLAST identified 73 High-scoring Segment Pairs (HSPs); all of these align in frame 1." data-conclusion="This is as expected." onclick="showDiv(this, 'toggle13'); AddExplanation(this,'toggle13');">73&nbsp;HSPs&nbsp;align&nbsp;in&nbsp;frame&nbsp;1</td>
        <td title="Main ORF" class="success my-btn-success" data-approach="We expect the query sequence to encode a single gene, thus it should contain one main Open Reading Frame (ORF) that occupies most of the query sequence." data-explanation=" The longest ORF is in frame 1, where it  occupies 100% of the query sequence." data-conclusion="There is no evidence to believe that there is any problem with the ORF of the query sequence." onclick="showDiv(this, 'toggle13');
          addPlot('toggle13', 'files/json/mrna_data.fasta_13_orfs.json', 'lines', 'Open reading frames in the 6 frames', '', 'Offset in the Prediction', 'Reading Frame', '14', '');
    AddExplanation(this, 'toggle13');">100%&nbsp;(frame&nbsp;1)</td>
        <td title="Missing/Extra sequences" class="danger my-btn-danger" data-approach="We expect the query sequence to be similar to the top ten BLAST hits. Here, we create a statistical consensus model of those top hits and compare the query to this  model." data-explanation="The query sequence includes 40% amino-acid residues present in the consensus model. 0% of residues in the query sequence are absent from the consensus profile.  61% of residues in the consensus profile are absent from the query sequence." data-conclusion="These results suggest that there may be some problems with the query sequence. The query sequence has a high percentage (61%) of missing residues when compared to the statistical profile (the cut-off is 20%)." onclick="showDiv(this, 'toggle13');
          addPlot('toggle13', 'files/json/mrna_data.fasta_13_ma.json', 'align', 'Missing/Extra sequences Validation: Multiple Align. & Statistical model of hits', 'Conserved Region, Yellow', 'Offset in the Alignment', '', '12', 'Prediction, hit&nbsp;1, hit&nbsp;2, hit&nbsp;3, hit&nbsp;4, hit&nbsp;5, hit&nbsp;6, hit&nbsp;7, hit&nbsp;8, hit&nbsp;9, hit&nbsp;10, Statistical Model');
    AddExplanation(this, 'toggle13');">40%&nbsp;conserved; 0%&nbsp;extra; 61%&nbsp;missing.</td>
  
  <td>
    <button title="Show plots" name="plot_btn" class="plot_btn btn btn-default" onclick="showDiv(this, 'toggle13');
                   addPlot('toggle13', 'files/json/mrna_data.fasta_13_len_clusters.json', 'bars', 'Length Cluster Validation: Distribution of BLAST hit lengths', 'query, black;most dense cluster,red;other hits, blue', 'Sequence Lenght', 'Number of Sequences', '226', '');                                   addPlot('toggle13', 'files/json/mrna_data.fasta_13_match_2d.json', 'scatter', 'Gene Merge Validation: Start/end of matching hit coord. on query (1 point/hit)', '', 'Start Offset (most left hsp)', 'End Offset (most right hsp)', '226.97208531181383', '-0.000599534443489048'); addPlot('toggle13', 'files/json/mrna_data.fasta_13_match.json', 'lines', 'Gene Merge Validation: Query coord covered by blast hit (1 line/hit)', '', 'Offset in Prediction', 'Hit Number', '73', '');                                             addPlot('toggle13', 'files/json/mrna_data.fasta_13_orfs.json', 'lines', 'Open reading frames in the 6 frames', '', 'Offset in the Prediction', 'Reading Frame', '14', '');                         addPlot('toggle13', 'files/json/mrna_data.fasta_13_ma.json', 'align', 'Missing/Extra sequences Validation: Multiple Align. & Statistical model of hits', 'Conserved Region, Yellow', 'Offset in the Alignment', '', '12', 'Prediction, hit&nbsp;1, hit&nbsp;2, hit&nbsp;3, hit&nbsp;4, hit&nbsp;5, hit&nbsp;6, hit&nbsp;7, hit&nbsp;8, hit&nbsp;9, hit&nbsp;10, Statistical Model');          "><i class="fa fa-bar-chart-o"></i></button>
  </td>
</tr>
<tr class="tablesorter-childRow" name="plot_row" style="display:none;">
  <td colspan="14" id="toggle13row">
    <div id="toggle13" class="expanded-child"></div>
  </td>
</tr>
<tr>
  <td>12</td>
  <td data-score="17"><div class="ratings"><div class="empty-stars"></div><div class="full-stars" style="width:14.45%;"></div></div></td>
  <td title="Definition" class="seq_definition">Brasy-EspV1T00041 </td>
  <td title="No. Hits">200</td>
      <td title="Length Cluster" class="danger my-btn-danger" data-approach="If the query sequence is well conserved and similar sequences (BLAST hits) are correct, we can expect query and hit sequences to have similar lengths. Here, we cluster the lengths of hit sequences and compare the length of our query sequence to the most dense cluster of hit lengths. " data-explanation="The most dense length-cluster of BLAST hits includes sequences that are from 246 to 279 amino-acids long. The query sequence is 1620 amino-acids long and is thus outside the most dense length-cluster of BLAST hits." data-conclusion="This suggests that the query sequence may be too long." onclick="showDiv(this, 'toggle12');
          addPlot('toggle12', 'files/json/mrna_data.fasta_12_len_clusters.json', 'bars', 'Length Cluster Validation: Distribution of BLAST hit lengths', 'query, black;most dense cluster,red;other hits, blue', 'Sequence Lenght', 'Number of Sequences', '1620', '');
    AddExplanation(this, 'toggle12');">1620&nbsp;[246,&nbsp;279]</td>
        <td title="Length Rank" class="danger" data-approach="If the query sequence is well conserved and similar sequences (BLAST hits) are correct, we can expect query and hit sequences to have similar lengths. " data-explanation="The query sequence is  1620 amino-acids long. BLAST identified 200 hit sequences with lengths from 246 to 1311 amino-acids (median: 807; mean: 807). All hit sequences are shorter than the query sequence." data-conclusion="The sequence may be too long." onclick="showDiv(this, 'toggle12'); AddExplanation(this,'toggle12');">0%&nbsp;(too&nbsp;long)</td>
        <td title="Gene Merge" class="success my-btn-success" data-approach="We expect the query sequence to encode a single protein-coding gene. Here, we analyse the High-scoring Segment Pairs (HSPs) identified by BLAST to determine whether the query includes sequence from two or more genes." data-explanation="The distribution of start and/or end-coordinates of HSPs are multi-modal. To detect potential problems we performed a linear regression (with coordinates weighted inversely proportionally to  hit strength). The resulting slope is 0.35." data-conclusion="This slope is  outside our empirically calculated thresholds (0.4 and 1.2). There is no evidence that the query contains sequence from multiple genes." onclick="showDiv(this, 'toggle12');
          addPlot('toggle12', 'files/json/mrna_data.fasta_12_match_2d.json', 'scatter', 'Gene Merge Validation: Start/end of matching hit coord. on query (1 point/hit)', '', 'Start Offset (most left hsp)', 'End Offset (most right hsp)', '934.3541424128539', '0.35172297092245264');
          addPlot('toggle12', 'files/json/mrna_data.fasta_12_match.json', 'lines', 'Gene Merge Validation: Query coord covered by blast hit (1 line/hit)', '', 'Offset in Prediction', 'Hit Number', '200', '');
    AddExplanation(this, 'toggle12');">0.35</td>
        <td title="Duplication" class="success" data-approach="We expect each BLAST hit to match each region of the query at most once. Here, we calculate the distribution of hit coverage against the query sequence and use the Wilcoxon test to determine if it is higher than 1." data-explanation="The Wilcoxon test produced a p-value of 1." data-conclusion="This suggests that the query sequence contains no erroneous duplications." onclick="showDiv(this, 'toggle12'); AddExplanation(this,'toggle12');">1.0</td>
        <td title="Reading Frame" class="danger" data-approach="We expect the query sequence to encode a single gene, thus it should contain one main Open Reading Frame (ORF). All all BLAST hits are thus expected to align within this ORF." data-explanation="BLAST identified 279 High-scoring Segment Pairs (HSPs): 278 HSPs align in frame 2; 1 HSPs align in frame 1; ." data-conclusion="The HSPs align in mulitple reading frames, this suggests there may be a frame shift in the query sequence." onclick="showDiv(this, 'toggle12'); AddExplanation(this,'toggle12');">278&nbsp;HSPs&nbsp;align&nbsp;in&nbsp;frame&nbsp;2; 1&nbsp;HSPs&nbsp;align&nbsp;in&nbsp;frame&nbsp;1</td>
        <td title="Main ORF" class="danger my-btn-danger" data-approach="We expect the query sequence to encode a single gene, thus it should contain one main Open Reading Frame (ORF) that occupies most of the query sequence." data-explanation=" The longest ORF is in frame 2, where it  occupies 49% of the query sequence." data-conclusion="This only represents a portion of the query sequence. In some cases this indicates that a frame shift exists in the query sequence." onclick="showDiv(this, 'toggle12');
          addPlot('toggle12', 'files/json/mrna_data.fasta_12_orfs.json', 'lines', 'Open reading frames in the 6 frames', '', 'Offset in the Prediction', 'Reading Frame', '14', '');
    AddExplanation(this, 'toggle12');">49%&nbsp;(frame&nbsp;2)</td>
        <td title="Missing/Extra sequences" class="danger my-btn-danger" data-approach="We expect the query sequence to be similar to the top ten BLAST hits. Here, we create a statistical consensus model of those top hits and compare the query to this  model." data-explanation="The query sequence includes 27% amino-acid residues present in the consensus model. 45% of residues in the query sequence are absent from the consensus profile.  13% of residues in the consensus profile are absent from the query sequence." data-conclusion="These results suggest that there may be some problems with the query sequence. The query sequence has a high percentage (45%) of extra residues absent from the statistical profile (the cut-off is 20%)." onclick="showDiv(this, 'toggle12');
          addPlot('toggle12', 'files/json/mrna_data.fasta_12_ma.json', 'align', 'Missing/Extra sequences Validation: Multiple Align. & Statistical model of hits', 'Conserved Region, Yellow', 'Offset in the Alignment', '', '12', 'Prediction, hit&nbsp;1, hit&nbsp;2, hit&nbsp;3, hit&nbsp;4, hit&nbsp;5, hit&nbsp;6, hit&nbsp;7, hit&nbsp;8, hit&nbsp;9, hit&nbsp;10, Statistical Model');
    AddExplanation(this, 'toggle12');">27%&nbsp;conserved; 45%&nbsp;extra; 13%&nbsp;missing.</td>
  
  <td>
    <button title="Show plots" name="plot_btn" class="plot_btn btn btn-default" onclick="showDiv(this, 'toggle12');
                   addPlot('toggle12', 'files/json/mrna_data.fasta_12_len_clusters.json', 'bars', 'Length Cluster Validation: Distribution of BLAST hit lengths', 'query, black;most dense cluster,red;other hits, blue', 'Sequence Lenght', 'Number of Sequences', '1620', '');                                   addPlot('toggle12', 'files/json/mrna_data.fasta_12_match_2d.json', 'scatter', 'Gene Merge Validation: Start/end of matching hit coord. on query (1 point/hit)', '', 'Start Offset (most left hsp)', 'End Offset (most right hsp)', '934.3541424128539', '0.35172297092245264'); addPlot('toggle12', 'files/json/mrna_data.fasta_12_match.json', 'lines', 'Gene Merge Validation: Query coord covered by blast hit (1 line/hit)', '', 'Offset in Prediction', 'Hit Number', '200', '');                                             addPlot('toggle12', 'files/json/mrna_data.fasta_12_orfs.json', 'lines', 'Open reading frames in the 6 frames', '', 'Offset in the Prediction', 'Reading Frame', '14', '');                         addPlot('toggle12', 'files/json/mrna_data.fasta_12_ma.json', 'align', 'Missing/Extra sequences Validation: Multiple Align. & Statistical model of hits', 'Conserved Region, Yellow', 'Offset in the Alignment', '', '12', 'Prediction, hit&nbsp;1, hit&nbsp;2, hit&nbsp;3, hit&nbsp;4, hit&nbsp;5, hit&nbsp;6, hit&nbsp;7, hit&nbsp;8, hit&nbsp;9, hit&nbsp;10, Statistical Model');          "><i class="fa fa-bar-chart-o"></i></button>
  </td>
</tr>
<tr class="tablesorter-childRow" name="plot_row" style="display:none;">
  <td colspan="14" id="toggle12row">
    <div id="toggle12" class="expanded-child"></div>
  </td>
</tr>
<tr>
  <td>11</td>
  <td data-score="17"><div class="ratings"><div class="empty-stars"></div><div class="full-stars" style="width:14.45%;"></div></div></td>
  <td title="Definition" class="seq_definition">Brasy-EspV1T00040 </td>
  <td title="No. Hits">200</td>
      <td title="Length Cluster" class="danger my-btn-danger" data-approach="If the query sequence is well conserved and similar sequences (BLAST hits) are correct, we can expect query and hit sequences to have similar lengths. Here, we cluster the lengths of hit sequences and compare the length of our query sequence to the most dense cluster of hit lengths. " data-explanation="The most dense length-cluster of BLAST hits includes sequences that are from 91 to 176 amino-acids long. The query sequence is 1253 amino-acids long and is thus outside the most dense length-cluster of BLAST hits." data-conclusion="This suggests that the query sequence may be too long." onclick="showDiv(this, 'toggle11');
          addPlot('toggle11', 'files/json/mrna_data.fasta_11_len_clusters.json', 'bars', 'Length Cluster Validation: Distribution of BLAST hit lengths', 'query, black;most dense cluster,red;other hits, blue', 'Sequence Lenght', 'Number of Sequences', '1253', '');
    AddExplanation(this, 'toggle11');">1253&nbsp;[91,&nbsp;176]</td>
        <td title="Length Rank" class="danger" data-approach="If the query sequence is well conserved and similar sequences (BLAST hits) are correct, we can expect query and hit sequences to have similar lengths. " data-explanation="The query sequence is  1253 amino-acids long. BLAST identified 200 hit sequences with lengths from 91 to 1789 amino-acids (median: 826; mean: 766). 1 of these hit sequences (i.e. 1%) are shorter than the query sequence." data-conclusion="The sequence may be too long." onclick="showDiv(this, 'toggle11'); AddExplanation(this,'toggle11');">1%&nbsp;(too&nbsp;long)</td>
        <td title="Gene Merge" class="success my-btn-success" data-approach="We expect the query sequence to encode a single protein-coding gene. Here, we analyse the High-scoring Segment Pairs (HSPs) identified by BLAST to determine whether the query includes sequence from two or more genes." data-explanation="The distribution of start and/or end-coordinates of HSPs are multi-modal. To detect potential problems we performed a linear regression (with coordinates weighted inversely proportionally to  hit strength). The resulting slope is -0.04." data-conclusion="This slope is  outside our empirically calculated thresholds (0.4 and 1.2). There is no evidence that the query contains sequence from multiple genes." onclick="showDiv(this, 'toggle11');
          addPlot('toggle11', 'files/json/mrna_data.fasta_11_match_2d.json', 'scatter', 'Gene Merge Validation: Start/end of matching hit coord. on query (1 point/hit)', '', 'Start Offset (most left hsp)', 'End Offset (most right hsp)', '1163.978621356794', '-0.04179047933740076');
          addPlot('toggle11', 'files/json/mrna_data.fasta_11_match.json', 'lines', 'Gene Merge Validation: Query coord covered by blast hit (1 line/hit)', '', 'Offset in Prediction', 'Hit Number', '200', '');
    AddExplanation(this, 'toggle11');">-0.04</td>
        <td title="Duplication" class="success" data-approach="We expect each BLAST hit to match each region of the query at most once. Here, we calculate the distribution of hit coverage against the query sequence and use the Wilcoxon test to determine if it is higher than 1." data-explanation="The Wilcoxon test produced a p-value of 1." data-conclusion="This suggests that the query sequence contains no erroneous duplications." onclick="showDiv(this, 'toggle11'); AddExplanation(this,'toggle11');">1.0</td>
        <td title="Reading Frame" class="danger" data-approach="We expect the query sequence to encode a single gene, thus it should contain one main Open Reading Frame (ORF). All all BLAST hits are thus expected to align within this ORF." data-explanation="BLAST identified 409 High-scoring Segment Pairs (HSPs): 407 HSPs align in frame 2; 1 HSPs align in frame 1; 1 HSPs align in frame -3; ." data-conclusion="The HSPs align in mulitple reading frames, this suggests there may be a frame shift in the query sequence." onclick="showDiv(this, 'toggle11'); AddExplanation(this,'toggle11');">407&nbsp;HSPs&nbsp;align&nbsp;in&nbsp;frame&nbsp;2; 1&nbsp;HSPs&nbsp;align&nbsp;in&nbsp;frame&nbsp;1; 1&nbsp;HSPs&nbsp;align&nbsp;in&nbsp;frame&nbsp;-3</td>
        <td title="Main ORF" class="danger my-btn-danger" data-approach="We expect the query sequence to encode a single gene, thus it should contain one main Open Reading Frame (ORF) that occupies most of the query sequence." data-explanation=" The longest ORF is in frame 2, where it  occupies 60% of the query sequence." data-conclusion="This only represents a portion of the query sequence. In some cases this indicates that a frame shift exists in the query sequence." onclick="showDiv(this, 'toggle11');
          addPlot('toggle11', 'files/json/mrna_data.fasta_11_orfs.json', 'lines', 'Open reading frames in the 6 frames', '', 'Offset in the Prediction', 'Reading Frame', '14', '');
    AddExplanation(this, 'toggle11');">60%&nbsp;(frame&nbsp;2)</td>
        <td title="Missing/Extra sequences" class="danger my-btn-danger" data-approach="We expect the query sequence to be similar to the top ten BLAST hits. Here, we create a statistical consensus model of those top hits and compare the query to this  model." data-explanation="The query sequence includes 26% amino-acid residues present in the consensus model. 33% of residues in the query sequence are absent from the consensus profile.  17% of residues in the consensus profile are absent from the query sequence." data-conclusion="These results suggest that there may be some problems with the query sequence. The query sequence has a high percentage (33%) of extra residues absent from the statistical profile (the cut-off is 20%)." onclick="showDiv(this, 'toggle11');
          addPlot('toggle11', 'files/json/mrna_data.fasta_11_ma.json', 'align', 'Missing/Extra sequences Validation: Multiple Align. & Statistical model of hits', 'Conserved Region, Yellow', 'Offset in the Alignment', '', '12', 'Prediction, hit&nbsp;1, hit&nbsp;2, hit&nbsp;3, hit&nbsp;4, hit&nbsp;5, hit&nbsp;6, hit&nbsp;7, hit&nbsp;8, hit&nbsp;9, hit&nbsp;10, Statistical Model');
    AddExplanation(this, 'toggle11');">26%&nbsp;conserved; 33%&nbsp;extra; 17%&nbsp;missing.</td>
  
  <td>
    <button title="Show plots" name="plot_btn" class="plot_btn btn btn-default" onclick="showDiv(this, 'toggle11');
                   addPlot('toggle11', 'files/json/mrna_data.fasta_11_len_clusters.json', 'bars', 'Length Cluster Validation: Distribution of BLAST hit lengths', 'query, black;most dense cluster,red;other hits, blue', 'Sequence Lenght', 'Number of Sequences', '1253', '');                                   addPlot('toggle11', 'files/json/mrna_data.fasta_11_match_2d.json', 'scatter', 'Gene Merge Validation: Start/end of matching hit coord. on query (1 point/hit)', '', 'Start Offset (most left hsp)', 'End Offset (most right hsp)', '1163.978621356794', '-0.04179047933740076'); addPlot('toggle11', 'files/json/mrna_data.fasta_11_match.json', 'lines', 'Gene Merge Validation: Query coord covered by blast hit (1 line/hit)', '', 'Offset in Prediction', 'Hit Number', '200', '');                                             addPlot('toggle11', 'files/json/mrna_data.fasta_11_orfs.json', 'lines', 'Open reading frames in the 6 frames', '', 'Offset in the Prediction', 'Reading Frame', '14', '');                         addPlot('toggle11', 'files/json/mrna_data.fasta_11_ma.json', 'align', 'Missing/Extra sequences Validation: Multiple Align. & Statistical model of hits', 'Conserved Region, Yellow', 'Offset in the Alignment', '', '12', 'Prediction, hit&nbsp;1, hit&nbsp;2, hit&nbsp;3, hit&nbsp;4, hit&nbsp;5, hit&nbsp;6, hit&nbsp;7, hit&nbsp;8, hit&nbsp;9, hit&nbsp;10, Statistical Model');          "><i class="fa fa-bar-chart-o"></i></button>
  </td>
</tr>
<tr class="tablesorter-childRow" name="plot_row" style="display:none;">
  <td colspan="14" id="toggle11row">
    <div id="toggle11" class="expanded-child"></div>
  </td>
</tr>
<tr>
  <td>2</td>
  <td data-score="33"><div class="ratings"><div class="empty-stars"></div><div class="full-stars" style="width:28.05%;"></div></div></td>
  <td title="Definition" class="seq_definition">gi|514746961|ref|XM_005019748.1|dup  PREDICTED: Anas platyrhynchos insulin (INS), mRNA</td>
  <td title="No. Hits">200</td>
      <td title="Length Cluster" class="danger my-btn-danger" data-approach="If the query sequence is well conserved and similar sequences (BLAST hits) are correct, we can expect query and hit sequences to have similar lengths. Here, we cluster the lengths of hit sequences and compare the length of our query sequence to the most dense cluster of hit lengths. " data-explanation="The most dense length-cluster of BLAST hits includes sequences that are from 55 to 55 amino-acids long. The query sequence is 163 amino-acids long and is thus outside the most dense length-cluster of BLAST hits." data-conclusion="This suggests that the query sequence may be too long." onclick="showDiv(this, 'toggle2');
          addPlot('toggle2', 'files/json/mrna_data.fasta_2_len_clusters.json', 'bars', 'Length Cluster Validation: Distribution of BLAST hit lengths', 'query, black;most dense cluster,red;other hits, blue', 'Sequence Lenght', 'Number of Sequences', '163', '');
    AddExplanation(this, 'toggle2');">163&nbsp;[55,&nbsp;55]</td>
        <td title="Length Rank" class="danger" data-approach="If the query sequence is well conserved and similar sequences (BLAST hits) are correct, we can expect query and hit sequences to have similar lengths. " data-explanation="The query sequence is  163 amino-acids long. BLAST identified 200 hit sequences with lengths from 55 to 248 amino-acids (median: 107; mean: 109). 4 of these hit sequences (i.e. 2%) are shorter than the query sequence." data-conclusion="The sequence may be too long." onclick="showDiv(this, 'toggle2'); AddExplanation(this,'toggle2');">2%&nbsp;(too&nbsp;long)</td>
        <td title="Gene Merge" class="success my-btn-success" data-approach="We expect the query sequence to encode a single protein-coding gene. Here, we analyse the High-scoring Segment Pairs (HSPs) identified by BLAST to determine whether the query includes sequence from two or more genes." data-explanation="The start coordinates and the end coordinates of HSPs are unimodally distributed." data-conclusion="This suggest that the query sequence represents a single gene." onclick="showDiv(this, 'toggle2');
          addPlot('toggle2', 'files/json/mrna_data.fasta_2_match_2d.json', 'scatter', 'Gene Merge Validation: Start/end of matching hit coord. on query (1 point/hit)', '', 'Start Offset (most left hsp)', 'End Offset (most right hsp)', '', '');
          addPlot('toggle2', 'files/json/mrna_data.fasta_2_match.json', 'lines', 'Gene Merge Validation: Query coord covered by blast hit (1 line/hit)', '', 'Offset in Prediction', 'Hit Number', '200', '');
    AddExplanation(this, 'toggle2');">0.0</td>
        <td title="Duplication" class="danger" data-approach="We expect each BLAST hit to match each region of the query at most once. Here, we calculate the distribution of hit coverage against the query sequence and use the Wilcoxon test to determine if it is higher than 1." data-explanation="The Wilcoxon test produced a p-value of 0.002 (average = 1.56)." data-conclusion="The null hypothesis is rejected - thus a region of the query sequence is likely repeated more than once." onclick="showDiv(this, 'toggle2'); AddExplanation(this,'toggle2');">0.0</td>
        <td title="Reading Frame" class="danger" data-approach="We expect the query sequence to encode a single gene, thus it should contain one main Open Reading Frame (ORF). All all BLAST hits are thus expected to align within this ORF." data-explanation="BLAST identified 401 High-scoring Segment Pairs (HSPs): 201 HSPs align in frame 2; 200 HSPs align in frame 1; ." data-conclusion="The HSPs align in mulitple reading frames, this suggests there may be a frame shift in the query sequence." onclick="showDiv(this, 'toggle2'); AddExplanation(this,'toggle2');">201&nbsp;HSPs&nbsp;align&nbsp;in&nbsp;frame&nbsp;2; 200&nbsp;HSPs&nbsp;align&nbsp;in&nbsp;frame&nbsp;1</td>
        <td title="Main ORF" class="danger my-btn-danger" data-approach="We expect the query sequence to encode a single gene, thus it should contain one main Open Reading Frame (ORF) that occupies most of the query sequence." data-explanation=" The longest ORF is in frame 2, where it  occupies 61% of the query sequence." data-conclusion="This only represents a portion of the query sequence. In some cases this indicates that a frame shift exists in the query sequence." onclick="showDiv(this, 'toggle2');
          addPlot('toggle2', 'files/json/mrna_data.fasta_2_orfs.json', 'lines', 'Open reading frames in the 6 frames', '', 'Offset in the Prediction', 'Reading Frame', '14', '');
    AddExplanation(this, 'toggle2');">61%&nbsp;(frame&nbsp;2)</td>
        <td title="Missing/Extra sequences" class="danger my-btn-danger" data-approach="We expect the query sequence to be similar to the top ten BLAST hits. Here, we create a statistical consensus model of those top hits and compare the query to this  model." data-explanation="The query sequence includes 81% amino-acid residues present in the consensus model. 36% of residues in the query sequence are absent from the consensus profile.  3% of residues in the consensus profile are absent from the query sequence." data-conclusion="These results suggest that there may be some problems with the query sequence. The query sequence has a high percentage (36%) of extra residues absent from the statistical profile (the cut-off is 20%)." onclick="showDiv(this, 'toggle2');
          addPlot('toggle2', 'files/json/mrna_data.fasta_2_ma.json', 'align', 'Missing/Extra sequences Validation: Multiple Align. & Statistical model of hits', 'Conserved Region, Yellow', 'Offset in the Alignment', '', '12', 'Prediction, hit&nbsp;1, hit&nbsp;2, hit&nbsp;3, hit&nbsp;4, hit&nbsp;5, hit&nbsp;6, hit&nbsp;7, hit&nbsp;8, hit&nbsp;9, hit&nbsp;10, Statistical Model');
    AddExplanation(this, 'toggle2');">81%&nbsp;conserved; 36%&nbsp;extra; 3%&nbsp;missing.</td>
  
  <td>
    <button title="Show plots" name="plot_btn" class="plot_btn btn btn-default" onclick="showDiv(this, 'toggle2');
                   addPlot('toggle2', 'files/json/mrna_data.fasta_2_len_clusters.json', 'bars', 'Length Cluster Validation: Distribution of BLAST hit lengths', 'query, black;most dense cluster,red;other hits, blue', 'Sequence Lenght', 'Number of Sequences', '163', '');                                   addPlot('toggle2', 'files/json/mrna_data.fasta_2_match_2d.json', 'scatter', 'Gene Merge Validation: Start/end of matching hit coord. on query (1 point/hit)', '', 'Start Offset (most left hsp)', 'End Offset (most right hsp)', '', ''); addPlot('toggle2', 'files/json/mrna_data.fasta_2_match.json', 'lines', 'Gene Merge Validation: Query coord covered by blast hit (1 line/hit)', '', 'Offset in Prediction', 'Hit Number', '200', '');                                             addPlot('toggle2', 'files/json/mrna_data.fasta_2_orfs.json', 'lines', 'Open reading frames in the 6 frames', '', 'Offset in the Prediction', 'Reading Frame', '14', '');                         addPlot('toggle2', 'files/json/mrna_data.fasta_2_ma.json', 'align', 'Missing/Extra sequences Validation: Multiple Align. & Statistical model of hits', 'Conserved Region, Yellow', 'Offset in the Alignment', '', '12', 'Prediction, hit&nbsp;1, hit&nbsp;2, hit&nbsp;3, hit&nbsp;4, hit&nbsp;5, hit&nbsp;6, hit&nbsp;7, hit&nbsp;8, hit&nbsp;9, hit&nbsp;10, Statistical Model');          "><i class="fa fa-bar-chart-o"></i></button>
  </td>
</tr>
<tr class="tablesorter-childRow" name="plot_row" style="display:none;">
  <td colspan="14" id="toggle2row">
    <div id="toggle2" class="expanded-child"></div>
  </td>
</tr>
<tr>
  <td>5</td>
  <td data-score="33"><div class="ratings"><div class="empty-stars"></div><div class="full-stars" style="width:28.05%;"></div></div></td>
  <td title="Definition" class="seq_definition">gi|251823817|ref|NM_010785.2|dup2  Mus musculus transformed mouse 3T3 cell double minute 1 (Mdm1), transcript variant 1, mRNA</td>
  <td title="No. Hits">200</td>
      <td title="Length Cluster" class="danger my-btn-danger" data-approach="If the query sequence is well conserved and similar sequences (BLAST hits) are correct, we can expect query and hit sequences to have similar lengths. Here, we cluster the lengths of hit sequences and compare the length of our query sequence to the most dense cluster of hit lengths. " data-explanation="The most dense length-cluster of BLAST hits includes sequences that are from 434 to 434 amino-acids long. The query sequence is 1303 amino-acids long and is thus outside the most dense length-cluster of BLAST hits." data-conclusion="This suggests that the query sequence may be too long." onclick="showDiv(this, 'toggle5');
          addPlot('toggle5', 'files/json/mrna_data.fasta_5_len_clusters.json', 'bars', 'Length Cluster Validation: Distribution of BLAST hit lengths', 'query, black;most dense cluster,red;other hits, blue', 'Sequence Lenght', 'Number of Sequences', '1303', '');
    AddExplanation(this, 'toggle5');">1303&nbsp;[434,&nbsp;434]</td>
        <td title="Length Rank" class="danger" data-approach="If the query sequence is well conserved and similar sequences (BLAST hits) are correct, we can expect query and hit sequences to have similar lengths. " data-explanation="The query sequence is  1303 amino-acids long. BLAST identified 200 hit sequences with lengths from 434 to 969 amino-acids (median: 710; mean: 704). All hit sequences are shorter than the query sequence." data-conclusion="The sequence may be too long." onclick="showDiv(this, 'toggle5'); AddExplanation(this,'toggle5');">0%&nbsp;(too&nbsp;long)</td>
        <td title="Gene Merge" class="success my-btn-success" data-approach="We expect the query sequence to encode a single protein-coding gene. Here, we analyse the High-scoring Segment Pairs (HSPs) identified by BLAST to determine whether the query includes sequence from two or more genes." data-explanation="The start coordinates and the end coordinates of HSPs are unimodally distributed." data-conclusion="This suggest that the query sequence represents a single gene." onclick="showDiv(this, 'toggle5');
          addPlot('toggle5', 'files/json/mrna_data.fasta_5_match_2d.json', 'scatter', 'Gene Merge Validation: Start/end of matching hit coord. on query (1 point/hit)', '', 'Start Offset (most left hsp)', 'End Offset (most right hsp)', '', '');
          addPlot('toggle5', 'files/json/mrna_data.fasta_5_match.json', 'lines', 'Gene Merge Validation: Query coord covered by blast hit (1 line/hit)', '', 'Offset in Prediction', 'Hit Number', '200', '');
    AddExplanation(this, 'toggle5');">0.0</td>
        <td title="Duplication" class="danger" data-approach="We expect each BLAST hit to match each region of the query at most once. Here, we calculate the distribution of hit coverage against the query sequence and use the Wilcoxon test to determine if it is higher than 1." data-explanation="The Wilcoxon test produced a p-value of 0.002 (average = 1.36)." data-conclusion="The null hypothesis is rejected - thus a region of the query sequence is likely repeated more than once." onclick="showDiv(this, 'toggle5'); AddExplanation(this,'toggle5');">0.0</td>
        <td title="Reading Frame" class="danger" data-approach="We expect the query sequence to encode a single gene, thus it should contain one main Open Reading Frame (ORF). All all BLAST hits are thus expected to align within this ORF." data-explanation="BLAST identified 600 High-scoring Segment Pairs (HSPs): 200 HSPs align in frame 1; 200 HSPs align in frame 3; 200 HSPs align in frame 2; ." data-conclusion="The HSPs align in mulitple reading frames, this suggests there may be a frame shift in the query sequence." onclick="showDiv(this, 'toggle5'); AddExplanation(this,'toggle5');">200&nbsp;HSPs&nbsp;align&nbsp;in&nbsp;frame&nbsp;1; 200&nbsp;HSPs&nbsp;align&nbsp;in&nbsp;frame&nbsp;3; 200&nbsp;HSPs&nbsp;align&nbsp;in&nbsp;frame&nbsp;2</td>
        <td title="Main ORF" class="danger my-btn-danger" data-approach="We expect the query sequence to encode a single gene, thus it should contain one main Open Reading Frame (ORF) that occupies most of the query sequence." data-explanation=" The longest ORF is in frame 1, where it  occupies 37% of the query sequence." data-conclusion="This only represents a portion of the query sequence. In some cases this indicates that a frame shift exists in the query sequence." onclick="showDiv(this, 'toggle5');
          addPlot('toggle5', 'files/json/mrna_data.fasta_5_orfs.json', 'lines', 'Open reading frames in the 6 frames', '', 'Offset in the Prediction', 'Reading Frame', '14', '');
    AddExplanation(this, 'toggle5');">37%&nbsp;(frame&nbsp;1)</td>
        <td title="Missing/Extra sequences" class="danger my-btn-danger" data-approach="We expect the query sequence to be similar to the top ten BLAST hits. Here, we create a statistical consensus model of those top hits and compare the query to this  model." data-explanation="The query sequence includes 76% amino-acid residues present in the consensus model. 46% of residues in the query sequence are absent from the consensus profile.  2% of residues in the consensus profile are absent from the query sequence." data-conclusion="These results suggest that there may be some problems with the query sequence. The query sequence has a high percentage (46%) of extra residues absent from the statistical profile (the cut-off is 20%)." onclick="showDiv(this, 'toggle5');
          addPlot('toggle5', 'files/json/mrna_data.fasta_5_ma.json', 'align', 'Missing/Extra sequences Validation: Multiple Align. & Statistical model of hits', 'Conserved Region, Yellow', 'Offset in the Alignment', '', '12', 'Prediction, hit&nbsp;1, hit&nbsp;2, hit&nbsp;3, hit&nbsp;4, hit&nbsp;5, hit&nbsp;6, hit&nbsp;7, hit&nbsp;8, hit&nbsp;9, hit&nbsp;10, Statistical Model');
    AddExplanation(this, 'toggle5');">76%&nbsp;conserved; 46%&nbsp;extra; 2%&nbsp;missing.</td>
  
  <td>
    <button title="Show plots" name="plot_btn" class="plot_btn btn btn-default" onclick="showDiv(this, 'toggle5');
                   addPlot('toggle5', 'files/json/mrna_data.fasta_5_len_clusters.json', 'bars', 'Length Cluster Validation: Distribution of BLAST hit lengths', 'query, black;most dense cluster,red;other hits, blue', 'Sequence Lenght', 'Number of Sequences', '1303', '');                                   addPlot('toggle5', 'files/json/mrna_data.fasta_5_match_2d.json', 'scatter', 'Gene Merge Validation: Start/end of matching hit coord. on query (1 point/hit)', '', 'Start Offset (most left hsp)', 'End Offset (most right hsp)', '', ''); addPlot('toggle5', 'files/json/mrna_data.fasta_5_match.json', 'lines', 'Gene Merge Validation: Query coord covered by blast hit (1 line/hit)', '', 'Offset in Prediction', 'Hit Number', '200', '');                                             addPlot('toggle5', 'files/json/mrna_data.fasta_5_orfs.json', 'lines', 'Open reading frames in the 6 frames', '', 'Offset in the Prediction', 'Reading Frame', '14', '');                         addPlot('toggle5', 'files/json/mrna_data.fasta_5_ma.json', 'align', 'Missing/Extra sequences Validation: Multiple Align. & Statistical model of hits', 'Conserved Region, Yellow', 'Offset in the Alignment', '', '12', 'Prediction, hit&nbsp;1, hit&nbsp;2, hit&nbsp;3, hit&nbsp;4, hit&nbsp;5, hit&nbsp;6, hit&nbsp;7, hit&nbsp;8, hit&nbsp;9, hit&nbsp;10, Statistical Model');          "><i class="fa fa-bar-chart-o"></i></button>
  </td>
</tr>
<tr class="tablesorter-childRow" name="plot_row" style="display:none;">
  <td colspan="14" id="toggle5row">
    <div id="toggle5" class="expanded-child"></div>
  </td>
</tr>
<tr>
  <td>4</td>
  <td data-score="33"><div class="ratings"><div class="empty-stars"></div><div class="full-stars" style="width:28.05%;"></div></div></td>
  <td title="Definition" class="seq_definition">gi|251823817|ref|NM_010785.2|dup  Mus musculus transformed mouse 3T3 cell double minute 1 (Mdm1), transcript variant 1, mRNA</td>
  <td title="No. Hits">200</td>
      <td title="Length Cluster" class="danger my-btn-danger" data-approach="If the query sequence is well conserved and similar sequences (BLAST hits) are correct, we can expect query and hit sequences to have similar lengths. Here, we cluster the lengths of hit sequences and compare the length of our query sequence to the most dense cluster of hit lengths. " data-explanation="The most dense length-cluster of BLAST hits includes sequences that are from 454 to 454 amino-acids long. The query sequence is 1327 amino-acids long and is thus outside the most dense length-cluster of BLAST hits." data-conclusion="This suggests that the query sequence may be too long." onclick="showDiv(this, 'toggle4');
          addPlot('toggle4', 'files/json/mrna_data.fasta_4_len_clusters.json', 'bars', 'Length Cluster Validation: Distribution of BLAST hit lengths', 'query, black;most dense cluster,red;other hits, blue', 'Sequence Lenght', 'Number of Sequences', '1327', '');
    AddExplanation(this, 'toggle4');">1327&nbsp;[454,&nbsp;454]</td>
        <td title="Length Rank" class="danger" data-approach="If the query sequence is well conserved and similar sequences (BLAST hits) are correct, we can expect query and hit sequences to have similar lengths. " data-explanation="The query sequence is  1327 amino-acids long. BLAST identified 200 hit sequences with lengths from 454 to 969 amino-acids (median: 710; mean: 705). All hit sequences are shorter than the query sequence." data-conclusion="The sequence may be too long." onclick="showDiv(this, 'toggle4'); AddExplanation(this,'toggle4');">0%&nbsp;(too&nbsp;long)</td>
        <td title="Gene Merge" class="success my-btn-success" data-approach="We expect the query sequence to encode a single protein-coding gene. Here, we analyse the High-scoring Segment Pairs (HSPs) identified by BLAST to determine whether the query includes sequence from two or more genes." data-explanation="The start coordinates and the end coordinates of HSPs are unimodally distributed." data-conclusion="This suggest that the query sequence represents a single gene." onclick="showDiv(this, 'toggle4');
          addPlot('toggle4', 'files/json/mrna_data.fasta_4_match_2d.json', 'scatter', 'Gene Merge Validation: Start/end of matching hit coord. on query (1 point/hit)', '', 'Start Offset (most left hsp)', 'End Offset (most right hsp)', '', '');
          addPlot('toggle4', 'files/json/mrna_data.fasta_4_match.json', 'lines', 'Gene Merge Validation: Query coord covered by blast hit (1 line/hit)', '', 'Offset in Prediction', 'Hit Number', '200', '');
    AddExplanation(this, 'toggle4');">0.0</td>
        <td title="Duplication" class="danger" data-approach="We expect each BLAST hit to match each region of the query at most once. Here, we calculate the distribution of hit coverage against the query sequence and use the Wilcoxon test to determine if it is higher than 1." data-explanation="The Wilcoxon test produced a p-value of 0.002 (average = 1.46)." data-conclusion="The null hypothesis is rejected - thus a region of the query sequence is likely repeated more than once." onclick="showDiv(this, 'toggle4'); AddExplanation(this,'toggle4');">0.0</td>
        <td title="Reading Frame" class="danger" data-approach="We expect the query sequence to encode a single gene, thus it should contain one main Open Reading Frame (ORF). All all BLAST hits are thus expected to align within this ORF." data-explanation="BLAST identified 400 High-scoring Segment Pairs (HSPs): 200 HSPs align in frame 1; 200 HSPs align in frame 3; ." data-conclusion="The HSPs align in mulitple reading frames, this suggests there may be a frame shift in the query sequence." onclick="showDiv(this, 'toggle4'); AddExplanation(this,'toggle4');">200&nbsp;HSPs&nbsp;align&nbsp;in&nbsp;frame&nbsp;1; 200&nbsp;HSPs&nbsp;align&nbsp;in&nbsp;frame&nbsp;3</td>
        <td title="Main ORF" class="danger my-btn-danger" data-approach="We expect the query sequence to encode a single gene, thus it should contain one main Open Reading Frame (ORF) that occupies most of the query sequence." data-explanation=" The longest ORF is in frame 1, where it  occupies 48% of the query sequence." data-conclusion="This only represents a portion of the query sequence. In some cases this indicates that a frame shift exists in the query sequence." onclick="showDiv(this, 'toggle4');
          addPlot('toggle4', 'files/json/mrna_data.fasta_4_orfs.json', 'lines', 'Open reading frames in the 6 frames', '', 'Offset in the Prediction', 'Reading Frame', '14', '');
    AddExplanation(this, 'toggle4');">48%&nbsp;(frame&nbsp;1)</td>
        <td title="Missing/Extra sequences" class="danger my-btn-danger" data-approach="We expect the query sequence to be similar to the top ten BLAST hits. Here, we create a statistical consensus model of those top hits and compare the query to this  model." data-explanation="The query sequence includes 93% amino-acid residues present in the consensus model. 47% of residues in the query sequence are absent from the consensus profile.  1% of residues in the consensus profile are absent from the query sequence." data-conclusion="These results suggest that there may be some problems with the query sequence. The query sequence has a high percentage (47%) of extra residues absent from the statistical profile (the cut-off is 20%)." onclick="showDiv(this, 'toggle4');
          addPlot('toggle4', 'files/json/mrna_data.fasta_4_ma.json', 'align', 'Missing/Extra sequences Validation: Multiple Align. & Statistical model of hits', 'Conserved Region, Yellow', 'Offset in the Alignment', '', '12', 'Prediction, hit&nbsp;1, hit&nbsp;2, hit&nbsp;3, hit&nbsp;4, hit&nbsp;5, hit&nbsp;6, hit&nbsp;7, hit&nbsp;8, hit&nbsp;9, hit&nbsp;10, Statistical Model');
    AddExplanation(this, 'toggle4');">93%&nbsp;conserved; 47%&nbsp;extra; 1%&nbsp;missing.</td>
  
  <td>
    <button title="Show plots" name="plot_btn" class="plot_btn btn btn-default" onclick="showDiv(this, 'toggle4');
                   addPlot('toggle4', 'files/json/mrna_data.fasta_4_len_clusters.json', 'bars', 'Length Cluster Validation: Distribution of BLAST hit lengths', 'query, black;most dense cluster,red;other hits, blue', 'Sequence Lenght', 'Number of Sequences', '1327', '');                                   addPlot('toggle4', 'files/json/mrna_data.fasta_4_match_2d.json', 'scatter', 'Gene Merge Validation: Start/end of matching hit coord. on query (1 point/hit)', '', 'Start Offset (most left hsp)', 'End Offset (most right hsp)', '', ''); addPlot('toggle4', 'files/json/mrna_data.fasta_4_match.json', 'lines', 'Gene Merge Validation: Query coord covered by blast hit (1 line/hit)', '', 'Offset in Prediction', 'Hit Number', '200', '');                                             addPlot('toggle4', 'files/json/mrna_data.fasta_4_orfs.json', 'lines', 'Open reading frames in the 6 frames', '', 'Offset in the Prediction', 'Reading Frame', '14', '');                         addPlot('toggle4', 'files/json/mrna_data.fasta_4_ma.json', 'align', 'Missing/Extra sequences Validation: Multiple Align. & Statistical model of hits', 'Conserved Region, Yellow', 'Offset in the Alignment', '', '12', 'Prediction, hit&nbsp;1, hit&nbsp;2, hit&nbsp;3, hit&nbsp;4, hit&nbsp;5, hit&nbsp;6, hit&nbsp;7, hit&nbsp;8, hit&nbsp;9, hit&nbsp;10, Statistical Model');          "><i class="fa fa-bar-chart-o"></i></button>
  </td>
</tr>
<tr class="tablesorter-childRow" name="plot_row" style="display:none;">
  <td colspan="14" id="toggle4row">
    <div id="toggle4" class="expanded-child"></div>
  </td>
</tr>
</tbody>
</table>
<script>
  var evaluation_div = document.getElementById('report_1');
  var content_less = "<br>Query score evaluation for mrna_data.fasta:<br>There were validated 13 predictions from which:<br>One good prediction<br>12 are possibly weak predictions<div class=\"clearfix\"></div><br><button type=\"button\" onClick='add_content()' class=\"btn btn-primary btn-sm\"> Show More</button>";
  var content = "<br>Query score evaluation for mrna_data.fasta:<br>There were validated 13 predictions from which:<br>One good prediction<br>12 are possibly weak predictions<br><br>Average running time for LengthCluster Validation: 0.255s per validation<br>Average running time for LengthRank Validation: 0.001s per validation<br>Average running time for Gene_Merge Validation: 0.669s per validation<br>Average running time for Duplication Validation: 0.781s per validation<br>Average running time for Frame Validation: 0.0s per validation<br>Average running time for ORF Validation: 0.271s per validation<br>Average running time for MA Validation: 2.453s per validation <div class=\"clearfix\"></div><br><button type=\"button\" onClick='evaluation_div.innerHTML=content_less' class=\"btn btn-primary btn-sm\"> Show Less</button>";
  evaluation_div.innerHTML = content_less;
  function add_content(){
    evaluation_div = document.getElementById('report_1')
    evaluation_div.innerHTML=content;
    addPlot('report_1', 'files/json/mrna_data.fasta_statistics.json', 'simplebars', 'Overall evaluation', '', 'validation score', 'number of queries');
  }
</script>
    <div aria-hidden="true" aria-labelledby="myModalLabel3" class="modal" id="spinner1" role="dialog" tabindex="-1">
      <div class="modal-dialog">
        <div class="modal-content">
          <div class="modal-body text-center">
            <h2>Loading ...</h2>
            <i class="fa fa-spinner fa-5x fa-spin"></i>
          </div>
        </div>
      </div>
    </div>

    <div class="modal fade" id="alert" aria-hidden="true" aria-labelledby="myModalLabel5" role="dialog" tabindex="-1">
      <div class="modal-dialog">
        <div class="modal-content">
          <div class="modal-header">
            <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
            <h4 class="modal-title">Oops, Something went wrong!</h4>
          </div>
          <div class="modal-body">
            <p>This operation is not posible. There seems to be too many queries...</p>
          </div>
          <div class="modal-footer">
            <button type="button" class="btn btn-default" data-dismiss="modal">Close</button>
          </div>
        </div>
      </div>
    </div>

    <div class="modal fade" id="browseralert" aria-hidden="true" aria-labelledby="browser" role="dialog" tabindex="-1">
      <div class="modal-dialog">
        <div class="modal-content browser-alert">
          <div class="modal-header">
            <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
            <h4 class="modal-title">Incompatible Browser - Please use Mozilla Firefox!</h4>
          </div>
          <div class="modal-body" id="browseralertbody">
            <p id="browseralertText">The plugin that we use to display the validation plots means that your browser is currently unsupported.</p>
            <p>Please use <strong>Mozilla FireFox</strong> to view this file.</p>
          </div>
        </div>
      </div>
    </div>

</div>

<footer >
  <div class="container center-block">
  <p class="text-muted text-center">Please cite: "Dragan M, Moghul MI, Priyam A & Wurm Y <em>(in prep)</em> GeneValidator: identify problematic gene predictions"<br/>
  Developed at <a href="http://yannick.poulet.org" target="_blank">Wurm Lab</a>, <a href="http://www.sbcs.qmul.ac.uk" target="_blank">QMUL</a> with funding by <a href="http://www.bbsrc.ac.uk/home/home.aspx" target="_blank">BBSRC</a> and <a href="https://www.google-melange.com/gsoc/homepage/google/gsoc2013" target="_blank">Google Summer of Code 2013</a><br/>
  This page was created by <a href="https://github.com/monicadragan/GeneValidator" target="_blank" >GeneValidator</a> v1.3.6</p>
  </div>
</footer>

</body>
</html>