<html><head><title>GeneValidator - validation explanations</title><meta content="text/html; charset=UTF-8" http-equiv="content-type"><style type="text/css">.lst-kix_euwmeg8cbpt8-4>li:before{content:"\0025cb  "}.lst-kix_g3lepnw8n0x6-4>li:before{content:"\0025cb  "}.lst-kix_9drbe3jtkkuh-4>li:before{content:"" counter(lst-ctn-kix_9drbe3jtkkuh-4,lower-latin) ". "}ul.lst-kix_hzkhz4hhlr8l-6{list-style-type:none}ul.lst-kix_hzkhz4hhlr8l-7{list-style-type:none}ul.lst-kix_hzkhz4hhlr8l-8{list-style-type:none}.lst-kix_z0ttp4mliy4e-6>li:before{content:"\0025cf  "}.lst-kix_tjc8wwk9cseo-2>li:before{content:"\0025a0  "}ol.lst-kix_9drbe3jtkkuh-5.start{counter-reset:lst-ctn-kix_9drbe3jtkkuh-5 0}ul.lst-kix_pizqq4oicz3j-5{list-style-type:none}ul.lst-kix_pizqq4oicz3j-4{list-style-type:none}ul.lst-kix_hzkhz4hhlr8l-0{list-style-type:none}ul.lst-kix_pizqq4oicz3j-3{list-style-type:none}ul.lst-kix_hzkhz4hhlr8l-1{list-style-type:none}ul.lst-kix_pizqq4oicz3j-2{list-style-type:none}ul.lst-kix_hzkhz4hhlr8l-2{list-style-type:none}ul.lst-kix_pizqq4oicz3j-1{list-style-type:none}.lst-kix_tjc8wwk9cseo-3>li:before{content:"\0025cf  "}ul.lst-kix_hzkhz4hhlr8l-3{list-style-type:none}ul.lst-kix_pizqq4oicz3j-0{list-style-type:none}ul.lst-kix_hzkhz4hhlr8l-4{list-style-type:none}.lst-kix_dpynbghv8b9j-3>li:before{content:"\0025cf  "}ul.lst-kix_hzkhz4hhlr8l-5{list-style-type:none}ul.lst-kix_pizqq4oicz3j-6{list-style-type:none}ul.lst-kix_pizqq4oicz3j-7{list-style-type:none}ul.lst-kix_pizqq4oicz3j-8{list-style-type:none}.lst-kix_hzkhz4hhlr8l-6>li:before{content:"\0025cf  "}.lst-kix_7rljtqi4tpf8-2>li:before{content:"\0025a0  "}.lst-kix_yghjnj8s9xuc-2>li:before{content:"\0025a0  "}.lst-kix_cbmcswhr1ftl-6>li:before{content:"\0025cf  "}.lst-kix_dpynbghv8b9j-0>li:before{content:"\0025cf  "}.lst-kix_lf42ux7u6753-5>li:before{content:"\0025a0  "}.lst-kix_sjinkw26u0wp-3>li:before{content:"\0025cf  "}.lst-kix_pdbqztwjr65g-8>li:before{content:"\0025a0  "}.lst-kix_qzhxzy2sbaso-7>li:before{content:"\0025cb  "}.lst-kix_3jnvhqhv71nk-4>li:before{content:"\0025cb  "}.lst-kix_fscfg01f3awm-4>li:before{content:"\0025cb  "}.lst-kix_olh7ea290trr-4>li:before{content:"\0025cb  "}.lst-kix_xq12dxu3f08q-6>li:before{content:"\0025cf  "}.lst-kix_hzkhz4hhlr8l-4>li:before{content:"\0025cb  "}.lst-kix_3jnvhqhv71nk-5>li:before{content:"\0025a0  "}.lst-kix_jamjx369pdr8-2>li:before{content:"\0025a0  "}.lst-kix_aps1ppntqbvm-5>li:before{content:"\0025a0  "}.lst-kix_x7iifx7mthae-8>li:before{content:"\0025a0  "}.lst-kix_vqtjz6k7exn7-1>li:before{content:"\0025cb  "}.lst-kix_g96zcqu3kh41-8>li:before{content:"\0025a0  "}.lst-kix_gr4a4jnxityy-3>li:before{content:"\0025cf  "}ul.lst-kix_lf42ux7u6753-0{list-style-type:none}.lst-kix_whzqgipsg1bt-2>li:before{content:"\0025a0  "}.lst-kix_g96zcqu3kh41-0>li:before{content:"\0025cf  "}ul.lst-kix_lf42ux7u6753-2{list-style-type:none}ul.lst-kix_lf42ux7u6753-1{list-style-type:none}.lst-kix_puzs83wo6j0o-3>li:before{content:"\0025cf  "}ul.lst-kix_lf42ux7u6753-4{list-style-type:none}ul.lst-kix_lf42ux7u6753-3{list-style-type:none}ul.lst-kix_lf42ux7u6753-6{list-style-type:none}.lst-kix_e5f9ugvogyec-7>li:before{content:"\0025cb  "}ul.lst-kix_lf42ux7u6753-5{list-style-type:none}ul.lst-kix_lf42ux7u6753-8{list-style-type:none}ul.lst-kix_lf42ux7u6753-7{list-style-type:none}.lst-kix_48dltnwvsqfp-1>li:before{content:"\0025cb  "}.lst-kix_4ed4y1s6lj5w-0>li:before{content:"\0025cf  "}.lst-kix_z0ttp4mliy4e-3>li:before{content:"\0025cf  "}.lst-kix_fscfg01f3awm-8>li:before{content:"\0025a0  "}.lst-kix_yghjnj8s9xuc-7>li:before{content:"\0025cb  "}.lst-kix_10avetia2xt4-3>li:before{content:"\0025cf  "}.lst-kix_508olx789l51-3>li:before{content:"\0025cf  "}.lst-kix_10avetia2xt4-8>li:before{content:"\0025a0  "}.lst-kix_9a6zvrglxdu8-1>li:before{content:"\0025cb  "}.lst-kix_z0ttp4mliy4e-7>li:before{content:"\0025cb  "}.lst-kix_vz631vhbp5pc-0>li:before{content:"\0025cf  "}.lst-kix_kmsnp6mmp404-3>li:before{content:"\0025cf  "}.lst-kix_q40ki98g1i63-6>li:before{content:"\0025cf  "}.lst-kix_aps1ppntqbvm-1>li:before{content:"\0025cb  "}ol.lst-kix_9drbe3jtkkuh-2.start{counter-reset:lst-ctn-kix_9drbe3jtkkuh-2 0}.lst-kix_9drbe3jtkkuh-0>li{counter-increment:lst-ctn-kix_9drbe3jtkkuh-0}.lst-kix_7kppodwmns3b-2>li:before{content:"\0025a0  "}.lst-kix_vqtjz6k7exn7-3>li:before{content:"\0025cf  "}ol.lst-kix_9drbe3jtkkuh-0.start{counter-reset:lst-ctn-kix_9drbe3jtkkuh-0 0}.lst-kix_ao4z72tcnso6-3>li:before{content:"\0025cf  "}.lst-kix_pizqq4oicz3j-4>li:before{content:"\0025cb  "}.lst-kix_owpom8h3nkwo-7>li:before{content:"\0025cb  "}.lst-kix_7kppodwmns3b-5>li:before{content:"\0025a0  "}.lst-kix_e5f9ugvogyec-0>li:before{content:"\0025cf  "}.lst-kix_puzs83wo6j0o-2>li:before{content:"\0025a0  "}.lst-kix_gr4a4jnxityy-0>li:before{content:"\0025cf  "}ul.lst-kix_dpynbghv8b9j-8{list-style-type:none}.lst-kix_euwmeg8cbpt8-6>li:before{content:"\0025cf  "}ul.lst-kix_dpynbghv8b9j-6{list-style-type:none}ul.lst-kix_dpynbghv8b9j-7{list-style-type:none}ul.lst-kix_dpynbghv8b9j-4{list-style-type:none}ul.lst-kix_dpynbghv8b9j-5{list-style-type:none}ul.lst-kix_dpynbghv8b9j-2{list-style-type:none}ul.lst-kix_dpynbghv8b9j-3{list-style-type:none}ul.lst-kix_dpynbghv8b9j-1{list-style-type:none}ul.lst-kix_puzs83wo6j0o-6{list-style-type:none}ul.lst-kix_dpynbghv8b9j-0{list-style-type:none}ul.lst-kix_puzs83wo6j0o-5{list-style-type:none}ul.lst-kix_puzs83wo6j0o-8{list-style-type:none}ul.lst-kix_puzs83wo6j0o-7{list-style-type:none}.lst-kix_pdbqztwjr65g-5>li:before{content:"\0025a0  "}ul.lst-kix_puzs83wo6j0o-0{list-style-type:none}.lst-kix_myo05blafbk-5>li:before{content:"\0025a0  "}ul.lst-kix_puzs83wo6j0o-2{list-style-type:none}ul.lst-kix_puzs83wo6j0o-1{list-style-type:none}ul.lst-kix_puzs83wo6j0o-4{list-style-type:none}ul.lst-kix_puzs83wo6j0o-3{list-style-type:none}.lst-kix_48dltnwvsqfp-5>li:before{content:"\0025a0  "}.lst-kix_e5f9ugvogyec-8>li:before{content:"\0025a0  "}.lst-kix_fscfg01f3awm-0>li:before{content:"\0025cf  "}.lst-kix_riib1nssx348-5>li:before{content:"\0025a0  "}.lst-kix_aps1ppntqbvm-2>li:before{content:"\0025a0  "}.lst-kix_508olx789l51-4>li:before{content:"\0025cb  "}.lst-kix_48dltnwvsqfp-7>li:before{content:"\0025cb  "}.lst-kix_10avetia2xt4-4>li:before{content:"\0025cb  "}.lst-kix_7kppodwmns3b-6>li:before{content:"\0025cf  "}.lst-kix_48dltnwvsqfp-3>li:before{content:"\0025cf  "}.lst-kix_4tq1geabcxa-2>li:before{content:"\0025a0  "}ol.lst-kix_9drbe3jtkkuh-0{list-style-type:none}.lst-kix_qzhxzy2sbaso-1>li:before{content:"\0025cb  "}.lst-kix_4tq1geabcxa-7>li:before{content:"\0025cb  "}.lst-kix_7rljtqi4tpf8-3>li:before{content:"\0025cf  "}.lst-kix_puzs83wo6j0o-5>li:before{content:"\0025a0  "}.lst-kix_b24v03uyl4k9-4>li:before{content:"\0025cb  "}.lst-kix_velyrizglpvf-6>li:before{content:"\0025cf  "}.lst-kix_sjinkw26u0wp-5>li:before{content:"\0025a0  "}.lst-kix_xc244bhq3ube-8>li:before{content:"\0025a0  "}.lst-kix_6064le485ne7-6>li:before{content:"\0025cf  "}.lst-kix_hzkhz4hhlr8l-1>li:before{content:"\0025cb  "}.lst-kix_l7t8lma3v5dd-7>li:before{content:"\0025cb  "}.lst-kix_jamjx369pdr8-5>li:before{content:"\0025a0  "}.lst-kix_vqtjz6k7exn7-0>li:before{content:"\0025cf  "}.lst-kix_6064le485ne7-0>li:before{content:"\0025cf  "}.lst-kix_e5f9ugvogyec-1>li:before{content:"\0025cb  "}.lst-kix_4ed4y1s6lj5w-7>li:before{content:"\0025cb  "}.lst-kix_olh7ea290trr-7>li:before{content:"\0025cb  "}.lst-kix_48dltnwvsqfp-6>li:before{content:"\0025cf  "}.lst-kix_velyrizglpvf-0>li:before{content:"\0025cf  "}.lst-kix_yghjnj8s9xuc-6>li:before{content:"\0025cf  "}.lst-kix_mk8t5sy0k589-2>li:before{content:"\0025a0  "}.lst-kix_b24v03uyl4k9-0>li:before{content:"\0025cf  "}.lst-kix_mk8t5sy0k589-4>li:before{content:"\0025cb  "}.lst-kix_g96zcqu3kh41-6>li:before{content:"\0025cf  "}.lst-kix_whzqgipsg1bt-3>li:before{content:"\0025cf  "}.lst-kix_euwmeg8cbpt8-0>li:before{content:"\0025cf  "}.lst-kix_pizqq4oicz3j-8>li:before{content:"\0025a0  "}.lst-kix_olh7ea290trr-6>li:before{content:"\0025cf  "}ol.lst-kix_9drbe3jtkkuh-3{list-style-type:none}ol.lst-kix_9drbe3jtkkuh-4{list-style-type:none}ol.lst-kix_9drbe3jtkkuh-1{list-style-type:none}.lst-kix_g3lepnw8n0x6-5>li:before{content:"\0025a0  "}ol.lst-kix_9drbe3jtkkuh-2{list-style-type:none}ol.lst-kix_9drbe3jtkkuh-7{list-style-type:none}ol.lst-kix_9drbe3jtkkuh-8{list-style-type:none}ol.lst-kix_9drbe3jtkkuh-5{list-style-type:none}ol.lst-kix_9drbe3jtkkuh-6{list-style-type:none}.lst-kix_owpom8h3nkwo-3>li:before{content:"\0025cf  "}ul.lst-kix_g96zcqu3kh41-1{list-style-type:none}.lst-kix_gr4a4jnxityy-8>li:before{content:"\0025a0  "}.lst-kix_dpynbghv8b9j-5>li:before{content:"\0025a0  "}ul.lst-kix_g96zcqu3kh41-2{list-style-type:none}ul.lst-kix_g96zcqu3kh41-3{list-style-type:none}ul.lst-kix_g96zcqu3kh41-4{list-style-type:none}ul.lst-kix_jamjx369pdr8-8{list-style-type:none}ul.lst-kix_jamjx369pdr8-7{list-style-type:none}.lst-kix_7rljtqi4tpf8-8>li:before{content:"\0025a0  "}ul.lst-kix_jamjx369pdr8-6{list-style-type:none}ul.lst-kix_g96zcqu3kh41-0{list-style-type:none}ul.lst-kix_jamjx369pdr8-5{list-style-type:none}.lst-kix_ao4z72tcnso6-1>li:before{content:"\0025cb  "}.lst-kix_g3lepnw8n0x6-0>li:before{content:"\0025cf  "}.lst-kix_x7iifx7mthae-0>li:before{content:"\0025cf  "}.lst-kix_mk8t5sy0k589-5>li:before{content:"\0025a0  "}.lst-kix_q40ki98g1i63-7>li:before{content:"\0025cb  "}.lst-kix_10avetia2xt4-6>li:before{content:"\0025cf  "}.lst-kix_6064le485ne7-1>li:before{content:"\0025cb  "}.lst-kix_9drbe3jtkkuh-6>li:before{content:"" counter(lst-ctn-kix_9drbe3jtkkuh-6,decimal) ". "}.lst-kix_cmqbbecr9wqr-2>li:before{content:"\0025a0  "}.lst-kix_6064le485ne7-2>li:before{content:"\0025a0  "}ul.lst-kix_jamjx369pdr8-3{list-style-type:none}.lst-kix_l7t8lma3v5dd-3>li:before{content:"\0025cf  "}ul.lst-kix_jamjx369pdr8-4{list-style-type:none}ul.lst-kix_jamjx369pdr8-1{list-style-type:none}ul.lst-kix_jamjx369pdr8-2{list-style-type:none}.lst-kix_9drbe3jtkkuh-3>li:before{content:"" counter(lst-ctn-kix_9drbe3jtkkuh-3,decimal) ". "}ul.lst-kix_jamjx369pdr8-0{list-style-type:none}.lst-kix_7kppodwmns3b-3>li:before{content:"\0025cf  "}.lst-kix_sjinkw26u0wp-1>li:before{content:"\0025cb  "}.lst-kix_tjc8wwk9cseo-4>li:before{content:"\0025cb  "}ul.lst-kix_sjinkw26u0wp-0{list-style-type:none}.lst-kix_7rljtqi4tpf8-6>li:before{content:"\0025cf  "}.lst-kix_l7t8lma3v5dd-5>li:before{content:"\0025a0  "}.lst-kix_4ed4y1s6lj5w-8>li:before{content:"\0025a0  "}ul.lst-kix_g3lepnw8n0x6-0{list-style-type:none}ul.lst-kix_g3lepnw8n0x6-1{list-style-type:none}ul.lst-kix_g3lepnw8n0x6-2{list-style-type:none}ul.lst-kix_g3lepnw8n0x6-3{list-style-type:none}ul.lst-kix_g3lepnw8n0x6-4{list-style-type:none}ul.lst-kix_g3lepnw8n0x6-5{list-style-type:none}ul.lst-kix_g3lepnw8n0x6-6{list-style-type:none}ul.lst-kix_g3lepnw8n0x6-7{list-style-type:none}ul.lst-kix_g3lepnw8n0x6-8{list-style-type:none}.lst-kix_mk8t5sy0k589-3>li:before{content:"\0025cf  "}.lst-kix_jamjx369pdr8-4>li:before{content:"\0025cb  "}.lst-kix_xc244bhq3ube-7>li:before{content:"\0025cb  "}.lst-kix_48dltnwvsqfp-8>li:before{content:"\0025a0  "}.lst-kix_puzs83wo6j0o-4>li:before{content:"\0025cb  "}.lst-kix_9drbe3jtkkuh-5>li:before{content:"" counter(lst-ctn-kix_9drbe3jtkkuh-5,lower-roman) ". "}.lst-kix_vz631vhbp5pc-8>li:before{content:"\0025a0  "}.lst-kix_lf42ux7u6753-6>li:before{content:"\0025cf  "}.lst-kix_xq12dxu3f08q-0>li:before{content:"\0025cf  "}.lst-kix_mk8t5sy0k589-6>li:before{content:"\0025cf  "}.lst-kix_508olx789l51-1>li:before{content:"\0025cb  "}ul.lst-kix_sjinkw26u0wp-8{list-style-type:none}.lst-kix_sjinkw26u0wp-6>li:before{content:"\0025cf  "}ul.lst-kix_sjinkw26u0wp-7{list-style-type:none}ul.lst-kix_sjinkw26u0wp-6{list-style-type:none}.lst-kix_owpom8h3nkwo-4>li:before{content:"\0025cb  "}ul.lst-kix_sjinkw26u0wp-5{list-style-type:none}.lst-kix_ao4z72tcnso6-6>li:before{content:"\0025cf  "}ul.lst-kix_sjinkw26u0wp-4{list-style-type:none}ul.lst-kix_sjinkw26u0wp-3{list-style-type:none}ul.lst-kix_sjinkw26u0wp-2{list-style-type:none}.lst-kix_gr4a4jnxityy-1>li:before{content:"\0025cb  "}ul.lst-kix_sjinkw26u0wp-1{list-style-type:none}.lst-kix_hzkhz4hhlr8l-3>li:before{content:"\0025cf  "}.lst-kix_cmqbbecr9wqr-6>li:before{content:"\0025cf  "}.lst-kix_whzqgipsg1bt-6>li:before{content:"\0025cf  "}.lst-kix_x7t5b7jbczs4-7>li:before{content:"\0025cb  "}.lst-kix_pizqq4oicz3j-6>li:before{content:"\0025cf  "}.lst-kix_x7t5b7jbczs4-2>li:before{content:"\0025a0  "}.lst-kix_pdbqztwjr65g-7>li:before{content:"\0025cb  "}.lst-kix_7kppodwmns3b-1>li:before{content:"\0025cb  "}.lst-kix_velyrizglpvf-1>li:before{content:"\0025cb  "}.lst-kix_x7iifx7mthae-6>li:before{content:"\0025cf  "}ul.lst-kix_l7t8lma3v5dd-7{list-style-type:none}ul.lst-kix_l7t8lma3v5dd-6{list-style-type:none}ul.lst-kix_l7t8lma3v5dd-5{list-style-type:none}.lst-kix_dpynbghv8b9j-7>li:before{content:"\0025cb  "}ul.lst-kix_l7t8lma3v5dd-4{list-style-type:none}ul.lst-kix_l7t8lma3v5dd-3{list-style-type:none}.lst-kix_pizqq4oicz3j-2>li:before{content:"\0025a0  "}ul.lst-kix_l7t8lma3v5dd-2{list-style-type:none}ul.lst-kix_l7t8lma3v5dd-1{list-style-type:none}ul.lst-kix_l7t8lma3v5dd-0{list-style-type:none}.lst-kix_whzqgipsg1bt-5>li:before{content:"\0025a0  "}.lst-kix_l7t8lma3v5dd-2>li:before{content:"\0025a0  "}ul.lst-kix_l7t8lma3v5dd-8{list-style-type:none}.lst-kix_riib1nssx348-8>li:before{content:"\0025a0  "}.lst-kix_myo05blafbk-7>li:before{content:"\0025cb  "}.lst-kix_vz631vhbp5pc-2>li:before{content:"\0025a0  "}.lst-kix_cbmcswhr1ftl-0>li:before{content:"\0025cf  "}.lst-kix_qzhxzy2sbaso-4>li:before{content:"\0025cb  "}.lst-kix_3jnvhqhv71nk-7>li:before{content:"\0025cb  "}.lst-kix_ao4z72tcnso6-0>li:before{content:"\0025cf  "}.lst-kix_3jnvhqhv71nk-6>li:before{content:"\0025cf  "}.lst-kix_x7t5b7jbczs4-5>li:before{content:"\0025a0  "}.lst-kix_mk8t5sy0k589-8>li:before{content:"\0025a0  "}.lst-kix_lf42ux7u6753-0>li:before{content:"\0025cf  "}.lst-kix_owpom8h3nkwo-8>li:before{content:"\0025a0  "}.lst-kix_mk8t5sy0k589-1>li:before{content:"\0025cb  "}.lst-kix_3kgcjgrfg4nh-2>li:before{content:"\0025a0  "}.lst-kix_7kppodwmns3b-4>li:before{content:"\0025cb  "}.lst-kix_b24v03uyl4k9-8>li:before{content:"\0025a0  "}ul.lst-kix_aps1ppntqbvm-2{list-style-type:none}ul.lst-kix_aps1ppntqbvm-3{list-style-type:none}.lst-kix_whzqgipsg1bt-8>li:before{content:"\0025a0  "}ul.lst-kix_aps1ppntqbvm-4{list-style-type:none}ul.lst-kix_aps1ppntqbvm-5{list-style-type:none}ul.lst-kix_aps1ppntqbvm-0{list-style-type:none}ul.lst-kix_aps1ppntqbvm-1{list-style-type:none}ul.lst-kix_4tq1geabcxa-8{list-style-type:none}ul.lst-kix_4tq1geabcxa-6{list-style-type:none}ul.lst-kix_4tq1geabcxa-7{list-style-type:none}ul.lst-kix_4tq1geabcxa-4{list-style-type:none}ul.lst-kix_4tq1geabcxa-5{list-style-type:none}ul.lst-kix_4tq1geabcxa-2{list-style-type:none}ul.lst-kix_4tq1geabcxa-3{list-style-type:none}ul.lst-kix_4tq1geabcxa-0{list-style-type:none}.lst-kix_myo05blafbk-3>li:before{content:"\0025cf  "}ul.lst-kix_4tq1geabcxa-1{list-style-type:none}ul.lst-kix_mk8t5sy0k589-3{list-style-type:none}ul.lst-kix_mk8t5sy0k589-4{list-style-type:none}ul.lst-kix_z0ttp4mliy4e-8{list-style-type:none}ul.lst-kix_mk8t5sy0k589-5{list-style-type:none}.lst-kix_myo05blafbk-0>li:before{content:"\0025cf  "}ul.lst-kix_z0ttp4mliy4e-7{list-style-type:none}ul.lst-kix_mk8t5sy0k589-6{list-style-type:none}.lst-kix_myo05blafbk-6>li:before{content:"\0025cf  "}.lst-kix_48dltnwvsqfp-2>li:before{content:"\0025a0  "}ul.lst-kix_mk8t5sy0k589-0{list-style-type:none}ul.lst-kix_mk8t5sy0k589-1{list-style-type:none}ul.lst-kix_mk8t5sy0k589-2{list-style-type:none}.lst-kix_3kgcjgrfg4nh-7>li:before{content:"\0025cb  "}.lst-kix_xq12dxu3f08q-2>li:before{content:"\0025a0  "}ul.lst-kix_yghjnj8s9xuc-7{list-style-type:none}ul.lst-kix_fscfg01f3awm-2{list-style-type:none}ul.lst-kix_yghjnj8s9xuc-8{list-style-type:none}ul.lst-kix_fscfg01f3awm-1{list-style-type:none}ul.lst-kix_yghjnj8s9xuc-5{list-style-type:none}ul.lst-kix_fscfg01f3awm-0{list-style-type:none}ul.lst-kix_yghjnj8s9xuc-6{list-style-type:none}ul.lst-kix_fscfg01f3awm-6{list-style-type:none}.lst-kix_q40ki98g1i63-3>li:before{content:"\0025cf  "}ul.lst-kix_fscfg01f3awm-5{list-style-type:none}ul.lst-kix_fscfg01f3awm-4{list-style-type:none}.lst-kix_vqtjz6k7exn7-5>li:before{content:"\0025a0  "}.lst-kix_9a6zvrglxdu8-2>li:before{content:"\0025a0  "}ul.lst-kix_fscfg01f3awm-3{list-style-type:none}.lst-kix_b24v03uyl4k9-1>li:before{content:"\0025cb  "}ul.lst-kix_z0ttp4mliy4e-0{list-style-type:none}ul.lst-kix_yghjnj8s9xuc-0{list-style-type:none}.lst-kix_508olx789l51-0>li:before{content:"\0025cf  "}ul.lst-kix_z0ttp4mliy4e-1{list-style-type:none}ul.lst-kix_fscfg01f3awm-8{list-style-type:none}ul.lst-kix_z0ttp4mliy4e-2{list-style-type:none}ul.lst-kix_fscfg01f3awm-7{list-style-type:none}ul.lst-kix_z0ttp4mliy4e-3{list-style-type:none}ul.lst-kix_yghjnj8s9xuc-3{list-style-type:none}ul.lst-kix_z0ttp4mliy4e-4{list-style-type:none}ul.lst-kix_yghjnj8s9xuc-4{list-style-type:none}ul.lst-kix_z0ttp4mliy4e-5{list-style-type:none}ul.lst-kix_yghjnj8s9xuc-1{list-style-type:none}ul.lst-kix_mk8t5sy0k589-8{list-style-type:none}ul.lst-kix_z0ttp4mliy4e-6{list-style-type:none}ul.lst-kix_yghjnj8s9xuc-2{list-style-type:none}.lst-kix_lf42ux7u6753-7>li:before{content:"\0025cb  "}ul.lst-kix_mk8t5sy0k589-7{list-style-type:none}.lst-kix_x7iifx7mthae-5>li:before{content:"\0025a0  "}.lst-kix_9a6zvrglxdu8-3>li:before{content:"\0025cf  "}.lst-kix_3jnvhqhv71nk-3>li:before{content:"\0025cf  "}.lst-kix_yghjnj8s9xuc-1>li:before{content:"\0025cb  "}ul.lst-kix_7rljtqi4tpf8-0{list-style-type:none}ul.lst-kix_7rljtqi4tpf8-1{list-style-type:none}ul.lst-kix_7rljtqi4tpf8-3{list-style-type:none}ul.lst-kix_q40ki98g1i63-6{list-style-type:none}.lst-kix_sjinkw26u0wp-4>li:before{content:"\0025cb  "}ul.lst-kix_7rljtqi4tpf8-2{list-style-type:none}ul.lst-kix_q40ki98g1i63-5{list-style-type:none}ul.lst-kix_7rljtqi4tpf8-5{list-style-type:none}ul.lst-kix_q40ki98g1i63-4{list-style-type:none}ul.lst-kix_7rljtqi4tpf8-4{list-style-type:none}ul.lst-kix_q40ki98g1i63-3{list-style-type:none}ul.lst-kix_7rljtqi4tpf8-7{list-style-type:none}ul.lst-kix_q40ki98g1i63-2{list-style-type:none}ul.lst-kix_7rljtqi4tpf8-6{list-style-type:none}ul.lst-kix_q40ki98g1i63-1{list-style-type:none}ul.lst-kix_q40ki98g1i63-0{list-style-type:none}ul.lst-kix_7rljtqi4tpf8-8{list-style-type:none}.lst-kix_pizqq4oicz3j-3>li:before{content:"\0025cf  "}.lst-kix_qzhxzy2sbaso-0>li:before{content:"\0025cf  "}.lst-kix_g96zcqu3kh41-7>li:before{content:"\0025cb  "}ul.lst-kix_q40ki98g1i63-8{list-style-type:none}ul.lst-kix_q40ki98g1i63-7{list-style-type:none}ul.lst-kix_e5f9ugvogyec-8{list-style-type:none}ul.lst-kix_cmqbbecr9wqr-3{list-style-type:none}ul.lst-kix_e5f9ugvogyec-7{list-style-type:none}ul.lst-kix_cmqbbecr9wqr-4{list-style-type:none}ul.lst-kix_cmqbbecr9wqr-1{list-style-type:none}.lst-kix_mk8t5sy0k589-7>li:before{content:"\0025cb  "}ul.lst-kix_cmqbbecr9wqr-2{list-style-type:none}ul.lst-kix_e5f9ugvogyec-4{list-style-type:none}.lst-kix_10avetia2xt4-2>li:before{content:"\0025a0  "}ul.lst-kix_e5f9ugvogyec-3{list-style-type:none}ul.lst-kix_cmqbbecr9wqr-0{list-style-type:none}ul.lst-kix_e5f9ugvogyec-6{list-style-type:none}ul.lst-kix_e5f9ugvogyec-5{list-style-type:none}.lst-kix_vz631vhbp5pc-6>li:before{content:"\0025cf  "}.lst-kix_xc244bhq3ube-4>li:before{content:"\0025cb  "}.lst-kix_kmsnp6mmp404-4>li:before{content:"\0025cb  "}.lst-kix_b24v03uyl4k9-7>li:before{content:"\0025cb  "}.lst-kix_b24v03uyl4k9-6>li:before{content:"\0025cf  "}ol.lst-kix_9drbe3jtkkuh-4.start{counter-reset:lst-ctn-kix_9drbe3jtkkuh-4 0}.lst-kix_b24v03uyl4k9-5>li:before{content:"\0025a0  "}.lst-kix_4ed4y1s6lj5w-6>li:before{content:"\0025cf  "}.lst-kix_e5f9ugvogyec-6>li:before{content:"\0025cf  "}ul.lst-kix_cmqbbecr9wqr-8{list-style-type:none}ul.lst-kix_cmqbbecr9wqr-7{list-style-type:none}.lst-kix_508olx789l51-6>li:before{content:"\0025cf  "}ul.lst-kix_cmqbbecr9wqr-6{list-style-type:none}ul.lst-kix_cmqbbecr9wqr-5{list-style-type:none}.lst-kix_4tq1geabcxa-5>li:before{content:"\0025a0  "}ol.lst-kix_9drbe3jtkkuh-6.start{counter-reset:lst-ctn-kix_9drbe3jtkkuh-6 0}.lst-kix_3jnvhqhv71nk-2>li:before{content:"\0025a0  "}.lst-kix_velyrizglpvf-7>li:before{content:"\0025cb  "}ul.lst-kix_qzhxzy2sbaso-7{list-style-type:none}ul.lst-kix_qzhxzy2sbaso-8{list-style-type:none}.lst-kix_9drbe3jtkkuh-8>li:before{content:"" counter(lst-ctn-kix_9drbe3jtkkuh-8,lower-roman) ". "}ul.lst-kix_qzhxzy2sbaso-5{list-style-type:none}ul.lst-kix_qzhxzy2sbaso-6{list-style-type:none}.lst-kix_b24v03uyl4k9-2>li:before{content:"\0025a0  "}ul.lst-kix_aps1ppntqbvm-6{list-style-type:none}ul.lst-kix_aps1ppntqbvm-7{list-style-type:none}ul.lst-kix_qzhxzy2sbaso-0{list-style-type:none}ul.lst-kix_aps1ppntqbvm-8{list-style-type:none}.lst-kix_4tq1geabcxa-1>li:before{content:"\0025cb  "}ul.lst-kix_e5f9ugvogyec-1{list-style-type:none}ul.lst-kix_qzhxzy2sbaso-3{list-style-type:none}ul.lst-kix_e5f9ugvogyec-2{list-style-type:none}ul.lst-kix_qzhxzy2sbaso-4{list-style-type:none}ul.lst-kix_qzhxzy2sbaso-1{list-style-type:none}ul.lst-kix_e5f9ugvogyec-0{list-style-type:none}ul.lst-kix_qzhxzy2sbaso-2{list-style-type:none}.lst-kix_5w9dslcj8r5z-6>li:before{content:"\0025cf  "}.lst-kix_cbmcswhr1ftl-5>li:before{content:"\0025a0  "}.lst-kix_q40ki98g1i63-5>li:before{content:"\0025a0  "}.lst-kix_kmsnp6mmp404-1>li:before{content:"\0025cb  "}.lst-kix_z0ttp4mliy4e-2>li:before{content:"\0025a0  "}.lst-kix_olh7ea290trr-0>li:before{content:"\0025cf  "}.lst-kix_x7t5b7jbczs4-4>li:before{content:"\0025cb  "}.lst-kix_ao4z72tcnso6-8>li:before{content:"\0025a0  "}.lst-kix_9drbe3jtkkuh-1>li:before{content:"" counter(lst-ctn-kix_9drbe3jtkkuh-1,lower-latin) ". "}.lst-kix_gr4a4jnxityy-7>li:before{content:"\0025cb  "}.lst-kix_x7t5b7jbczs4-1>li:before{content:"\0025cb  "}.lst-kix_whzqgipsg1bt-7>li:before{content:"\0025cb  "}.lst-kix_9drbe3jtkkuh-1>li{counter-increment:lst-ctn-kix_9drbe3jtkkuh-1}.lst-kix_jamjx369pdr8-7>li:before{content:"\0025cb  "}.lst-kix_mk8t5sy0k589-0>li:before{content:"\0025cf  "}.lst-kix_ao4z72tcnso6-2>li:before{content:"\0025a0  "}.lst-kix_48dltnwvsqfp-4>li:before{content:"\0025cb  "}.lst-kix_g96zcqu3kh41-3>li:before{content:"\0025cf  "}ul.lst-kix_xq12dxu3f08q-8{list-style-type:none}ul.lst-kix_xq12dxu3f08q-7{list-style-type:none}ul.lst-kix_xq12dxu3f08q-6{list-style-type:none}ul.lst-kix_xq12dxu3f08q-5{list-style-type:none}.lst-kix_9drbe3jtkkuh-4>li{counter-increment:lst-ctn-kix_9drbe3jtkkuh-4}ul.lst-kix_xq12dxu3f08q-4{list-style-type:none}ul.lst-kix_xq12dxu3f08q-3{list-style-type:none}ul.lst-kix_xq12dxu3f08q-2{list-style-type:none}.lst-kix_fscfg01f3awm-6>li:before{content:"\0025cf  "}.lst-kix_4ed4y1s6lj5w-2>li:before{content:"\0025a0  "}.lst-kix_kmsnp6mmp404-5>li:before{content:"\0025a0  "}.lst-kix_vz631vhbp5pc-5>li:before{content:"\0025a0  "}ul.lst-kix_xq12dxu3f08q-0{list-style-type:none}.lst-kix_qzhxzy2sbaso-2>li:before{content:"\0025a0  "}.lst-kix_xq12dxu3f08q-8>li:before{content:"\0025a0  "}ul.lst-kix_xq12dxu3f08q-1{list-style-type:none}.lst-kix_x7t5b7jbczs4-6>li:before{content:"\0025cf  "}.lst-kix_9drbe3jtkkuh-6>li{counter-increment:lst-ctn-kix_9drbe3jtkkuh-6}ul.lst-kix_g96zcqu3kh41-5{list-style-type:none}ul.lst-kix_g96zcqu3kh41-6{list-style-type:none}ul.lst-kix_g96zcqu3kh41-7{list-style-type:none}.lst-kix_euwmeg8cbpt8-2>li:before{content:"\0025a0  "}ul.lst-kix_g96zcqu3kh41-8{list-style-type:none}.lst-kix_10avetia2xt4-1>li:before{content:"\0025cb  "}.lst-kix_6064le485ne7-4>li:before{content:"\0025cb  "}.lst-kix_myo05blafbk-4>li:before{content:"\0025cb  "}.lst-kix_tjc8wwk9cseo-8>li:before{content:"\0025a0  "}.lst-kix_9a6zvrglxdu8-8>li:before{content:"\0025a0  "}.lst-kix_9a6zvrglxdu8-4>li:before{content:"\0025cb  "}.lst-kix_yghjnj8s9xuc-0>li:before{content:"\0025cf  "}.lst-kix_x7iifx7mthae-7>li:before{content:"\0025cb  "}.lst-kix_vz631vhbp5pc-3>li:before{content:"\0025cf  "}.lst-kix_g96zcqu3kh41-1>li:before{content:"\0025cb  "}.lst-kix_5w9dslcj8r5z-8>li:before{content:"\0025a0  "}.lst-kix_7rljtqi4tpf8-5>li:before{content:"\0025a0  "}.lst-kix_tjc8wwk9cseo-5>li:before{content:"\0025a0  "}.lst-kix_sjinkw26u0wp-2>li:before{content:"\0025a0  "}.lst-kix_4tq1geabcxa-4>li:before{content:"\0025cb  "}.lst-kix_5w9dslcj8r5z-7>li:before{content:"\0025cb  "}.lst-kix_q40ki98g1i63-8>li:before{content:"\0025a0  "}ul.lst-kix_gr4a4jnxityy-2{list-style-type:none}ul.lst-kix_gr4a4jnxityy-3{list-style-type:none}ul.lst-kix_gr4a4jnxityy-4{list-style-type:none}ul.lst-kix_gr4a4jnxityy-5{list-style-type:none}ul.lst-kix_gr4a4jnxityy-0{list-style-type:none}ul.lst-kix_gr4a4jnxityy-1{list-style-type:none}.lst-kix_owpom8h3nkwo-1>li:before{content:"\0025cb  "}ul.lst-kix_gr4a4jnxityy-6{list-style-type:none}ul.lst-kix_gr4a4jnxityy-7{list-style-type:none}.lst-kix_9drbe3jtkkuh-7>li{counter-increment:lst-ctn-kix_9drbe3jtkkuh-7}ul.lst-kix_gr4a4jnxityy-8{list-style-type:none}.lst-kix_48dltnwvsqfp-0>li:before{content:"\0025cf  "}.lst-kix_kmsnp6mmp404-7>li:before{content:"\0025cb  "}ul.lst-kix_vz631vhbp5pc-2{list-style-type:none}ul.lst-kix_vz631vhbp5pc-1{list-style-type:none}ul.lst-kix_vz631vhbp5pc-4{list-style-type:none}ul.lst-kix_vz631vhbp5pc-3{list-style-type:none}ul.lst-kix_vz631vhbp5pc-6{list-style-type:none}ul.lst-kix_vz631vhbp5pc-5{list-style-type:none}ul.lst-kix_vz631vhbp5pc-8{list-style-type:none}ul.lst-kix_vz631vhbp5pc-7{list-style-type:none}ul.lst-kix_vz631vhbp5pc-0{list-style-type:none}.lst-kix_jamjx369pdr8-3>li:before{content:"\0025cf  "}.lst-kix_xq12dxu3f08q-5>li:before{content:"\0025a0  "}.lst-kix_xc244bhq3ube-1>li:before{content:"\0025cb  "}ul.lst-kix_ao4z72tcnso6-5{list-style-type:none}ul.lst-kix_ao4z72tcnso6-6{list-style-type:none}ul.lst-kix_ao4z72tcnso6-3{list-style-type:none}ul.lst-kix_ao4z72tcnso6-4{list-style-type:none}.lst-kix_owpom8h3nkwo-6>li:before{content:"\0025cf  "}ul.lst-kix_ao4z72tcnso6-1{list-style-type:none}.lst-kix_aps1ppntqbvm-3>li:before{content:"\0025cf  "}ul.lst-kix_ao4z72tcnso6-2{list-style-type:none}ul.lst-kix_ao4z72tcnso6-0{list-style-type:none}.lst-kix_yghjnj8s9xuc-4>li:before{content:"\0025cb  "}.lst-kix_fscfg01f3awm-3>li:before{content:"\0025cf  "}ul.lst-kix_ao4z72tcnso6-8{list-style-type:none}ul.lst-kix_ao4z72tcnso6-7{list-style-type:none}.lst-kix_6064le485ne7-5>li:before{content:"\0025a0  "}.lst-kix_fscfg01f3awm-1>li:before{content:"\0025cb  "}.lst-kix_cbmcswhr1ftl-1>li:before{content:"\0025cb  "}.lst-kix_vz631vhbp5pc-1>li:before{content:"\0025cb  "}.lst-kix_owpom8h3nkwo-2>li:before{content:"\0025a0  "}.lst-kix_vqtjz6k7exn7-7>li:before{content:"\0025cb  "}ul.lst-kix_myo05blafbk-1{list-style-type:none}ul.lst-kix_myo05blafbk-2{list-style-type:none}ul.lst-kix_myo05blafbk-3{list-style-type:none}ul.lst-kix_myo05blafbk-4{list-style-type:none}ul.lst-kix_myo05blafbk-5{list-style-type:none}.lst-kix_dpynbghv8b9j-6>li:before{content:"\0025cf  "}ul.lst-kix_myo05blafbk-6{list-style-type:none}ul.lst-kix_myo05blafbk-7{list-style-type:none}ul.lst-kix_myo05blafbk-8{list-style-type:none}.lst-kix_5w9dslcj8r5z-2>li:before{content:"\0025a0  "}.lst-kix_gr4a4jnxityy-2>li:before{content:"\0025a0  "}ul.lst-kix_myo05blafbk-0{list-style-type:none}.lst-kix_aps1ppntqbvm-8>li:before{content:"\0025a0  "}.lst-kix_tjc8wwk9cseo-0>li:before{content:"\0025cf  "}.lst-kix_xq12dxu3f08q-3>li:before{content:"\0025cf  "}.lst-kix_puzs83wo6j0o-7>li:before{content:"\0025cb  "}.lst-kix_fscfg01f3awm-2>li:before{content:"\0025a0  "}.lst-kix_z0ttp4mliy4e-8>li:before{content:"\0025a0  "}.lst-kix_euwmeg8cbpt8-3>li:before{content:"\0025cf  "}.lst-kix_sjinkw26u0wp-7>li:before{content:"\0025cb  "}.lst-kix_gr4a4jnxityy-6>li:before{content:"\0025cf  "}.lst-kix_l7t8lma3v5dd-1>li:before{content:"\0025cb  "}.lst-kix_xc244bhq3ube-3>li:before{content:"\0025cf  "}.lst-kix_xc244bhq3ube-2>li:before{content:"\0025a0  "}.lst-kix_x7t5b7jbczs4-8>li:before{content:"\0025a0  "}.lst-kix_qzhxzy2sbaso-6>li:before{content:"\0025cf  "}ul.lst-kix_4ed4y1s6lj5w-8{list-style-type:none}.lst-kix_hzkhz4hhlr8l-0>li:before{content:"\0025cf  "}ul.lst-kix_4ed4y1s6lj5w-7{list-style-type:none}.lst-kix_z0ttp4mliy4e-0>li:before{content:"\0025cf  "}ul.lst-kix_4ed4y1s6lj5w-6{list-style-type:none}ul.lst-kix_4ed4y1s6lj5w-5{list-style-type:none}.lst-kix_q40ki98g1i63-4>li:before{content:"\0025cb  "}ul.lst-kix_whzqgipsg1bt-3{list-style-type:none}.lst-kix_e5f9ugvogyec-3>li:before{content:"\0025cf  "}ul.lst-kix_whzqgipsg1bt-2{list-style-type:none}ul.lst-kix_whzqgipsg1bt-1{list-style-type:none}ul.lst-kix_whzqgipsg1bt-0{list-style-type:none}ul.lst-kix_whzqgipsg1bt-7{list-style-type:none}ul.lst-kix_whzqgipsg1bt-6{list-style-type:none}ul.lst-kix_whzqgipsg1bt-5{list-style-type:none}ul.lst-kix_whzqgipsg1bt-4{list-style-type:none}.lst-kix_puzs83wo6j0o-0>li:before{content:"\0025cf  "}.lst-kix_x7iifx7mthae-1>li:before{content:"\0025cb  "}ul.lst-kix_whzqgipsg1bt-8{list-style-type:none}.lst-kix_9a6zvrglxdu8-6>li:before{content:"\0025cf  "}.lst-kix_5w9dslcj8r5z-5>li:before{content:"\0025a0  "}.lst-kix_x7t5b7jbczs4-0>li:before{content:"\0025cf  "}.lst-kix_pdbqztwjr65g-4>li:before{content:"\0025cb  "}.lst-kix_7kppodwmns3b-0>li:before{content:"\0025cf  "}.lst-kix_508olx789l51-5>li:before{content:"\0025a0  "}.lst-kix_vz631vhbp5pc-4>li:before{content:"\0025cb  "}.lst-kix_9a6zvrglxdu8-0>li:before{content:"\0025cf  "}.lst-kix_10avetia2xt4-7>li:before{content:"\0025cb  "}.lst-kix_q40ki98g1i63-0>li:before{content:"\0025cf  "}.lst-kix_3kgcjgrfg4nh-3>li:before{content:"\0025cf  "}ul.lst-kix_riib1nssx348-3{list-style-type:none}.lst-kix_6064le485ne7-7>li:before{content:"\0025cb  "}ul.lst-kix_riib1nssx348-4{list-style-type:none}ul.lst-kix_riib1nssx348-1{list-style-type:none}ul.lst-kix_riib1nssx348-2{list-style-type:none}ul.lst-kix_riib1nssx348-0{list-style-type:none}.lst-kix_tjc8wwk9cseo-1>li:before{content:"\0025cb  "}.lst-kix_ao4z72tcnso6-4>li:before{content:"\0025cb  "}.lst-kix_velyrizglpvf-5>li:before{content:"\0025a0  "}ul.lst-kix_olh7ea290trr-8{list-style-type:none}ul.lst-kix_olh7ea290trr-7{list-style-type:none}ul.lst-kix_olh7ea290trr-6{list-style-type:none}ul.lst-kix_olh7ea290trr-5{list-style-type:none}ul.lst-kix_olh7ea290trr-4{list-style-type:none}.lst-kix_9drbe3jtkkuh-8>li{counter-increment:lst-ctn-kix_9drbe3jtkkuh-8}ul.lst-kix_olh7ea290trr-3{list-style-type:none}ul.lst-kix_olh7ea290trr-2{list-style-type:none}ul.lst-kix_olh7ea290trr-1{list-style-type:none}ul.lst-kix_riib1nssx348-8{list-style-type:none}ul.lst-kix_olh7ea290trr-0{list-style-type:none}ul.lst-kix_riib1nssx348-7{list-style-type:none}ul.lst-kix_riib1nssx348-6{list-style-type:none}ul.lst-kix_riib1nssx348-5{list-style-type:none}.lst-kix_4ed4y1s6lj5w-1>li:before{content:"\0025cb  "}.lst-kix_puzs83wo6j0o-8>li:before{content:"\0025a0  "}.lst-kix_q40ki98g1i63-1>li:before{content:"\0025cb  "}.lst-kix_lf42ux7u6753-1>li:before{content:"\0025cb  "}.lst-kix_pdbqztwjr65g-6>li:before{content:"\0025cf  "}.lst-kix_cbmcswhr1ftl-2>li:before{content:"\0025a0  "}.lst-kix_euwmeg8cbpt8-8>li:before{content:"\0025a0  "}.lst-kix_aps1ppntqbvm-4>li:before{content:"\0025cb  "}.lst-kix_7kppodwmns3b-8>li:before{content:"\0025a0  "}.lst-kix_euwmeg8cbpt8-7>li:before{content:"\0025cb  "}ol.lst-kix_9drbe3jtkkuh-3.start{counter-reset:lst-ctn-kix_9drbe3jtkkuh-3 0}ul.lst-kix_x7t5b7jbczs4-8{list-style-type:none}.lst-kix_3kgcjgrfg4nh-1>li:before{content:"\0025cb  "}.lst-kix_cmqbbecr9wqr-0>li:before{content:"\0025cf  "}ul.lst-kix_x7t5b7jbczs4-4{list-style-type:none}ul.lst-kix_b24v03uyl4k9-1{list-style-type:none}ul.lst-kix_3kgcjgrfg4nh-6{list-style-type:none}ul.lst-kix_x7t5b7jbczs4-5{list-style-type:none}ul.lst-kix_b24v03uyl4k9-0{list-style-type:none}ul.lst-kix_3kgcjgrfg4nh-5{list-style-type:none}ul.lst-kix_x7t5b7jbczs4-6{list-style-type:none}ul.lst-kix_3kgcjgrfg4nh-8{list-style-type:none}ul.lst-kix_x7t5b7jbczs4-7{list-style-type:none}.lst-kix_ao4z72tcnso6-7>li:before{content:"\0025cb  "}ul.lst-kix_3kgcjgrfg4nh-7{list-style-type:none}ul.lst-kix_x7t5b7jbczs4-0{list-style-type:none}ul.lst-kix_9a6zvrglxdu8-0{list-style-type:none}ul.lst-kix_x7t5b7jbczs4-1{list-style-type:none}ul.lst-kix_x7t5b7jbczs4-2{list-style-type:none}ul.lst-kix_x7t5b7jbczs4-3{list-style-type:none}ul.lst-kix_9a6zvrglxdu8-4{list-style-type:none}ul.lst-kix_48dltnwvsqfp-2{list-style-type:none}ul.lst-kix_9a6zvrglxdu8-3{list-style-type:none}ul.lst-kix_48dltnwvsqfp-3{list-style-type:none}ul.lst-kix_9a6zvrglxdu8-2{list-style-type:none}ul.lst-kix_48dltnwvsqfp-0{list-style-type:none}.lst-kix_lf42ux7u6753-8>li:before{content:"\0025a0  "}ul.lst-kix_9a6zvrglxdu8-1{list-style-type:none}ul.lst-kix_48dltnwvsqfp-1{list-style-type:none}ul.lst-kix_48dltnwvsqfp-7{list-style-type:none}.lst-kix_euwmeg8cbpt8-1>li:before{content:"\0025cb  "}ul.lst-kix_48dltnwvsqfp-6{list-style-type:none}ul.lst-kix_48dltnwvsqfp-5{list-style-type:none}ul.lst-kix_48dltnwvsqfp-4{list-style-type:none}ul.lst-kix_48dltnwvsqfp-8{list-style-type:none}ul.lst-kix_b24v03uyl4k9-4{list-style-type:none}ul.lst-kix_3kgcjgrfg4nh-1{list-style-type:none}ul.lst-kix_b24v03uyl4k9-5{list-style-type:none}ul.lst-kix_3kgcjgrfg4nh-2{list-style-type:none}.lst-kix_velyrizglpvf-4>li:before{content:"\0025cb  "}ul.lst-kix_b24v03uyl4k9-2{list-style-type:none}ul.lst-kix_3kgcjgrfg4nh-3{list-style-type:none}.lst-kix_kmsnp6mmp404-6>li:before{content:"\0025cf  "}ul.lst-kix_b24v03uyl4k9-3{list-style-type:none}ul.lst-kix_3kgcjgrfg4nh-4{list-style-type:none}.lst-kix_riib1nssx348-3>li:before{content:"\0025cf  "}ul.lst-kix_b24v03uyl4k9-8{list-style-type:none}ul.lst-kix_b24v03uyl4k9-6{list-style-type:none}ul.lst-kix_b24v03uyl4k9-7{list-style-type:none}.lst-kix_9drbe3jtkkuh-2>li{counter-increment:lst-ctn-kix_9drbe3jtkkuh-2}ul.lst-kix_3kgcjgrfg4nh-0{list-style-type:none}.lst-kix_kmsnp6mmp404-8>li:before{content:"\0025a0  "}.lst-kix_l7t8lma3v5dd-4>li:before{content:"\0025cb  "}.lst-kix_4ed4y1s6lj5w-5>li:before{content:"\0025a0  "}ol.lst-kix_9drbe3jtkkuh-1.start{counter-reset:lst-ctn-kix_9drbe3jtkkuh-1 0}.lst-kix_xc244bhq3ube-5>li:before{content:"\0025a0  "}.lst-kix_tjc8wwk9cseo-6>li:before{content:"\0025cf  "}.lst-kix_dpynbghv8b9j-4>li:before{content:"\0025cb  "}.lst-kix_riib1nssx348-0>li:before{content:"\0025cf  "}.lst-kix_sjinkw26u0wp-0>li:before{content:"\0025cf  "}.lst-kix_pdbqztwjr65g-1>li:before{content:"\0025cb  "}.lst-kix_jamjx369pdr8-1>li:before{content:"\0025cb  "}.lst-kix_xc244bhq3ube-0>li:before{content:"\0025cf  "}ul.lst-kix_9a6zvrglxdu8-5{list-style-type:none}.lst-kix_3kgcjgrfg4nh-5>li:before{content:"\0025a0  "}ul.lst-kix_9a6zvrglxdu8-6{list-style-type:none}ul.lst-kix_9a6zvrglxdu8-7{list-style-type:none}ul.lst-kix_9a6zvrglxdu8-8{list-style-type:none}.lst-kix_jamjx369pdr8-6>li:before{content:"\0025cf  "}.lst-kix_whzqgipsg1bt-0>li:before{content:"\0025cf  "}.lst-kix_hzkhz4hhlr8l-7>li:before{content:"\0025cb  "}ol.lst-kix_9drbe3jtkkuh-8.start{counter-reset:lst-ctn-kix_9drbe3jtkkuh-8 0}.lst-kix_cbmcswhr1ftl-7>li:before{content:"\0025cb  "}.lst-kix_q40ki98g1i63-2>li:before{content:"\0025a0  "}.lst-kix_myo05blafbk-1>li:before{content:"\0025cb  "}.lst-kix_puzs83wo6j0o-6>li:before{content:"\0025cf  "}.lst-kix_jamjx369pdr8-0>li:before{content:"\0025cf  "}.lst-kix_lf42ux7u6753-2>li:before{content:"\0025a0  "}.lst-kix_yghjnj8s9xuc-3>li:before{content:"\0025cf  "}.lst-kix_riib1nssx348-6>li:before{content:"\0025cf  "}.lst-kix_yghjnj8s9xuc-5>li:before{content:"\0025a0  "}.lst-kix_7rljtqi4tpf8-4>li:before{content:"\0025cb  "}.lst-kix_cmqbbecr9wqr-1>li:before{content:"\0025cb  "}.lst-kix_3jnvhqhv71nk-0>li:before{content:"\0025cf  "}.lst-kix_puzs83wo6j0o-1>li:before{content:"\0025cb  "}ul.lst-kix_kmsnp6mmp404-0{list-style-type:none}ul.lst-kix_kmsnp6mmp404-1{list-style-type:none}.lst-kix_3kgcjgrfg4nh-6>li:before{content:"\0025cf  "}.lst-kix_7rljtqi4tpf8-0>li:before{content:"\0025cf  "}.lst-kix_whzqgipsg1bt-1>li:before{content:"\0025cb  "}.lst-kix_vqtjz6k7exn7-6>li:before{content:"\0025cf  "}ul.lst-kix_kmsnp6mmp404-4{list-style-type:none}ul.lst-kix_kmsnp6mmp404-5{list-style-type:none}.lst-kix_pizqq4oicz3j-7>li:before{content:"\0025cb  "}ul.lst-kix_kmsnp6mmp404-2{list-style-type:none}ul.lst-kix_kmsnp6mmp404-3{list-style-type:none}.lst-kix_5w9dslcj8r5z-0>li:before{content:"\0025cf  "}.lst-kix_g3lepnw8n0x6-2>li:before{content:"\0025a0  "}.lst-kix_9a6zvrglxdu8-5>li:before{content:"\0025a0  "}ul.lst-kix_kmsnp6mmp404-8{list-style-type:none}.lst-kix_4ed4y1s6lj5w-3>li:before{content:"\0025cf  "}.lst-kix_6064le485ne7-3>li:before{content:"\0025cf  "}ul.lst-kix_kmsnp6mmp404-7{list-style-type:none}ul.lst-kix_kmsnp6mmp404-6{list-style-type:none}.lst-kix_10avetia2xt4-0>li:before{content:"\0025cf  "}.lst-kix_hzkhz4hhlr8l-2>li:before{content:"\0025a0  "}.lst-kix_g3lepnw8n0x6-1>li:before{content:"\0025cb  "}.lst-kix_kmsnp6mmp404-0>li:before{content:"\0025cf  "}.lst-kix_7rljtqi4tpf8-1>li:before{content:"\0025cb  "}.lst-kix_lf42ux7u6753-3>li:before{content:"\0025cf  "}ul.lst-kix_vqtjz6k7exn7-2{list-style-type:none}ul.lst-kix_10avetia2xt4-6{list-style-type:none}ul.lst-kix_vqtjz6k7exn7-3{list-style-type:none}ul.lst-kix_10avetia2xt4-7{list-style-type:none}ul.lst-kix_vqtjz6k7exn7-0{list-style-type:none}.lst-kix_cmqbbecr9wqr-7>li:before{content:"\0025cb  "}ul.lst-kix_10avetia2xt4-4{list-style-type:none}ul.lst-kix_vqtjz6k7exn7-1{list-style-type:none}ul.lst-kix_10avetia2xt4-5{list-style-type:none}ul.lst-kix_10avetia2xt4-2{list-style-type:none}ul.lst-kix_10avetia2xt4-3{list-style-type:none}.lst-kix_fscfg01f3awm-5>li:before{content:"\0025a0  "}.lst-kix_vqtjz6k7exn7-2>li:before{content:"\0025a0  "}ul.lst-kix_10avetia2xt4-0{list-style-type:none}ul.lst-kix_10avetia2xt4-1{list-style-type:none}.lst-kix_pdbqztwjr65g-2>li:before{content:"\0025a0  "}ul.lst-kix_vqtjz6k7exn7-8{list-style-type:none}ul.lst-kix_vqtjz6k7exn7-6{list-style-type:none}ul.lst-kix_vqtjz6k7exn7-7{list-style-type:none}.lst-kix_jamjx369pdr8-8>li:before{content:"\0025a0  "}ul.lst-kix_vqtjz6k7exn7-4{list-style-type:none}ul.lst-kix_10avetia2xt4-8{list-style-type:none}ul.lst-kix_vqtjz6k7exn7-5{list-style-type:none}.lst-kix_4ed4y1s6lj5w-4>li:before{content:"\0025cb  "}.lst-kix_pdbqztwjr65g-3>li:before{content:"\0025cf  "}.lst-kix_508olx789l51-7>li:before{content:"\0025cb  "}.lst-kix_xq12dxu3f08q-1>li:before{content:"\0025cb  "}.lst-kix_gr4a4jnxityy-4>li:before{content:"\0025cb  "}.lst-kix_qzhxzy2sbaso-3>li:before{content:"\0025cf  "}.lst-kix_tjc8wwk9cseo-7>li:before{content:"\0025cb  "}ul.lst-kix_velyrizglpvf-0{list-style-type:none}.lst-kix_9a6zvrglxdu8-7>li:before{content:"\0025cb  "}ul.lst-kix_velyrizglpvf-2{list-style-type:none}ul.lst-kix_velyrizglpvf-1{list-style-type:none}ul.lst-kix_velyrizglpvf-4{list-style-type:none}ul.lst-kix_velyrizglpvf-3{list-style-type:none}ul.lst-kix_velyrizglpvf-6{list-style-type:none}ul.lst-kix_velyrizglpvf-5{list-style-type:none}ul.lst-kix_velyrizglpvf-8{list-style-type:none}ul.lst-kix_velyrizglpvf-7{list-style-type:none}.lst-kix_fscfg01f3awm-7>li:before{content:"\0025cb  "}.lst-kix_pizqq4oicz3j-0>li:before{content:"\0025cf  "}.lst-kix_z0ttp4mliy4e-1>li:before{content:"\0025cb  "}.lst-kix_e5f9ugvogyec-4>li:before{content:"\0025cb  "}.lst-kix_vqtjz6k7exn7-4>li:before{content:"\0025cb  "}.lst-kix_cmqbbecr9wqr-5>li:before{content:"\0025a0  "}.lst-kix_7kppodwmns3b-7>li:before{content:"\0025cb  "}.lst-kix_velyrizglpvf-8>li:before{content:"\0025a0  "}.lst-kix_sjinkw26u0wp-8>li:before{content:"\0025a0  "}.lst-kix_x7t5b7jbczs4-3>li:before{content:"\0025cf  "}ul.lst-kix_6064le485ne7-0{list-style-type:none}.lst-kix_cbmcswhr1ftl-8>li:before{content:"\0025a0  "}.lst-kix_riib1nssx348-1>li:before{content:"\0025cb  "}.lst-kix_myo05blafbk-2>li:before{content:"\0025a0  "}.lst-kix_4tq1geabcxa-0>li:before{content:"\0025cf  "}ul.lst-kix_pdbqztwjr65g-2{list-style-type:none}ul.lst-kix_pdbqztwjr65g-1{list-style-type:none}ul.lst-kix_pdbqztwjr65g-0{list-style-type:none}.lst-kix_aps1ppntqbvm-6>li:before{content:"\0025cf  "}.lst-kix_g96zcqu3kh41-2>li:before{content:"\0025a0  "}ul.lst-kix_pdbqztwjr65g-6{list-style-type:none}ul.lst-kix_pdbqztwjr65g-5{list-style-type:none}ul.lst-kix_pdbqztwjr65g-4{list-style-type:none}ul.lst-kix_pdbqztwjr65g-3{list-style-type:none}ul.lst-kix_pdbqztwjr65g-8{list-style-type:none}ul.lst-kix_pdbqztwjr65g-7{list-style-type:none}ul.lst-kix_6064le485ne7-2{list-style-type:none}.lst-kix_dpynbghv8b9j-1>li:before{content:"\0025cb  "}ul.lst-kix_6064le485ne7-1{list-style-type:none}ul.lst-kix_6064le485ne7-4{list-style-type:none}ul.lst-kix_6064le485ne7-3{list-style-type:none}.lst-kix_3jnvhqhv71nk-1>li:before{content:"\0025cb  "}ul.lst-kix_6064le485ne7-6{list-style-type:none}.lst-kix_3kgcjgrfg4nh-0>li:before{content:"\0025cf  "}ul.lst-kix_6064le485ne7-5{list-style-type:none}.lst-kix_cbmcswhr1ftl-3>li:before{content:"\0025cf  "}ul.lst-kix_6064le485ne7-8{list-style-type:none}ul.lst-kix_6064le485ne7-7{list-style-type:none}.lst-kix_ao4z72tcnso6-5>li:before{content:"\0025a0  "}.lst-kix_cbmcswhr1ftl-4>li:before{content:"\0025cb  "}.lst-kix_pizqq4oicz3j-1>li:before{content:"\0025cb  "}.lst-kix_hzkhz4hhlr8l-5>li:before{content:"\0025a0  "}.lst-kix_9drbe3jtkkuh-0>li:before{content:"" counter(lst-ctn-kix_9drbe3jtkkuh-0,decimal) ". "}.lst-kix_5w9dslcj8r5z-4>li:before{content:"\0025cb  "}ul.lst-kix_508olx789l51-2{list-style-type:none}.lst-kix_5w9dslcj8r5z-3>li:before{content:"\0025cf  "}ul.lst-kix_508olx789l51-1{list-style-type:none}ul.lst-kix_508olx789l51-0{list-style-type:none}.lst-kix_4tq1geabcxa-3>li:before{content:"\0025cf  "}.lst-kix_kmsnp6mmp404-2>li:before{content:"\0025a0  "}.lst-kix_508olx789l51-2>li:before{content:"\0025a0  "}ul.lst-kix_508olx789l51-7{list-style-type:none}ul.lst-kix_508olx789l51-8{list-style-type:none}.lst-kix_aps1ppntqbvm-0>li:before{content:"\0025cf  "}ul.lst-kix_508olx789l51-5{list-style-type:none}.lst-kix_xq12dxu3f08q-7>li:before{content:"\0025cb  "}ul.lst-kix_508olx789l51-6{list-style-type:none}ul.lst-kix_508olx789l51-3{list-style-type:none}ul.lst-kix_508olx789l51-4{list-style-type:none}.lst-kix_g3lepnw8n0x6-7>li:before{content:"\0025cb  "}.lst-kix_euwmeg8cbpt8-5>li:before{content:"\0025a0  "}.lst-kix_508olx789l51-8>li:before{content:"\0025a0  "}.lst-kix_g96zcqu3kh41-4>li:before{content:"\0025cb  "}.lst-kix_9drbe3jtkkuh-7>li:before{content:"" counter(lst-ctn-kix_9drbe3jtkkuh-7,lower-latin) ". "}ul.lst-kix_tjc8wwk9cseo-1{list-style-type:none}ul.lst-kix_tjc8wwk9cseo-0{list-style-type:none}ul.lst-kix_owpom8h3nkwo-8{list-style-type:none}ul.lst-kix_tjc8wwk9cseo-3{list-style-type:none}ul.lst-kix_owpom8h3nkwo-7{list-style-type:none}ul.lst-kix_tjc8wwk9cseo-2{list-style-type:none}ul.lst-kix_tjc8wwk9cseo-5{list-style-type:none}ul.lst-kix_tjc8wwk9cseo-4{list-style-type:none}ul.lst-kix_tjc8wwk9cseo-7{list-style-type:none}ul.lst-kix_owpom8h3nkwo-4{list-style-type:none}ul.lst-kix_tjc8wwk9cseo-6{list-style-type:none}ul.lst-kix_owpom8h3nkwo-3{list-style-type:none}ul.lst-kix_owpom8h3nkwo-6{list-style-type:none}ul.lst-kix_tjc8wwk9cseo-8{list-style-type:none}ul.lst-kix_5w9dslcj8r5z-0{list-style-type:none}ul.lst-kix_owpom8h3nkwo-5{list-style-type:none}ul.lst-kix_5w9dslcj8r5z-1{list-style-type:none}ul.lst-kix_owpom8h3nkwo-0{list-style-type:none}ul.lst-kix_5w9dslcj8r5z-2{list-style-type:none}ul.lst-kix_5w9dslcj8r5z-3{list-style-type:none}ul.lst-kix_owpom8h3nkwo-2{list-style-type:none}ul.lst-kix_5w9dslcj8r5z-4{list-style-type:none}ul.lst-kix_owpom8h3nkwo-1{list-style-type:none}ul.lst-kix_5w9dslcj8r5z-6{list-style-type:none}ul.lst-kix_5w9dslcj8r5z-5{list-style-type:none}ul.lst-kix_5w9dslcj8r5z-8{list-style-type:none}ul.lst-kix_5w9dslcj8r5z-7{list-style-type:none}.lst-kix_riib1nssx348-4>li:before{content:"\0025cb  "}.lst-kix_yghjnj8s9xuc-8>li:before{content:"\0025a0  "}.lst-kix_l7t8lma3v5dd-8>li:before{content:"\0025a0  "}.lst-kix_dpynbghv8b9j-8>li:before{content:"\0025a0  "}.lst-kix_riib1nssx348-7>li:before{content:"\0025cb  "}.lst-kix_3kgcjgrfg4nh-8>li:before{content:"\0025a0  "}.lst-kix_g96zcqu3kh41-5>li:before{content:"\0025a0  "}.lst-kix_qzhxzy2sbaso-5>li:before{content:"\0025a0  "}.lst-kix_4tq1geabcxa-8>li:before{content:"\0025a0  "}.lst-kix_3jnvhqhv71nk-8>li:before{content:"\0025a0  "}.lst-kix_6064le485ne7-8>li:before{content:"\0025a0  "}.lst-kix_9drbe3jtkkuh-3>li{counter-increment:lst-ctn-kix_9drbe3jtkkuh-3}.lst-kix_velyrizglpvf-3>li:before{content:"\0025cf  "}.lst-kix_b24v03uyl4k9-3>li:before{content:"\0025cf  "}.lst-kix_riib1nssx348-2>li:before{content:"\0025a0  "}.lst-kix_dpynbghv8b9j-2>li:before{content:"\0025a0  "}.lst-kix_g3lepnw8n0x6-3>li:before{content:"\0025cf  "}ul.lst-kix_4ed4y1s6lj5w-0{list-style-type:none}ul.lst-kix_4ed4y1s6lj5w-4{list-style-type:none}ul.lst-kix_4ed4y1s6lj5w-3{list-style-type:none}ul.lst-kix_4ed4y1s6lj5w-2{list-style-type:none}ul.lst-kix_4ed4y1s6lj5w-1{list-style-type:none}.lst-kix_aps1ppntqbvm-7>li:before{content:"\0025cb  "}.lst-kix_gr4a4jnxityy-5>li:before{content:"\0025a0  "}ul.lst-kix_3jnvhqhv71nk-2{list-style-type:none}.lst-kix_olh7ea290trr-1>li:before{content:"\0025cb  "}ul.lst-kix_3jnvhqhv71nk-3{list-style-type:none}ul.lst-kix_3jnvhqhv71nk-0{list-style-type:none}ul.lst-kix_cbmcswhr1ftl-0{list-style-type:none}ul.lst-kix_3jnvhqhv71nk-1{list-style-type:none}.lst-kix_x7iifx7mthae-4>li:before{content:"\0025cb  "}ul.lst-kix_cbmcswhr1ftl-6{list-style-type:none}ul.lst-kix_cbmcswhr1ftl-5{list-style-type:none}.lst-kix_lf42ux7u6753-4>li:before{content:"\0025cb  "}ul.lst-kix_cbmcswhr1ftl-8{list-style-type:none}ul.lst-kix_cbmcswhr1ftl-7{list-style-type:none}ul.lst-kix_3jnvhqhv71nk-8{list-style-type:none}ul.lst-kix_cbmcswhr1ftl-2{list-style-type:none}ul.lst-kix_3jnvhqhv71nk-7{list-style-type:none}ul.lst-kix_cbmcswhr1ftl-1{list-style-type:none}ul.lst-kix_3jnvhqhv71nk-6{list-style-type:none}ul.lst-kix_cbmcswhr1ftl-4{list-style-type:none}ul.lst-kix_3jnvhqhv71nk-5{list-style-type:none}ul.lst-kix_cbmcswhr1ftl-3{list-style-type:none}.lst-kix_9drbe3jtkkuh-2>li:before{content:"" counter(lst-ctn-kix_9drbe3jtkkuh-2,lower-roman) ". "}ul.lst-kix_3jnvhqhv71nk-4{list-style-type:none}.lst-kix_pizqq4oicz3j-5>li:before{content:"\0025a0  "}ul.lst-kix_7kppodwmns3b-0{list-style-type:none}ul.lst-kix_7kppodwmns3b-1{list-style-type:none}ul.lst-kix_7kppodwmns3b-4{list-style-type:none}.lst-kix_e5f9ugvogyec-2>li:before{content:"\0025a0  "}ul.lst-kix_7kppodwmns3b-5{list-style-type:none}.lst-kix_myo05blafbk-8>li:before{content:"\0025a0  "}ul.lst-kix_7kppodwmns3b-2{list-style-type:none}.lst-kix_cmqbbecr9wqr-4>li:before{content:"\0025cb  "}ul.lst-kix_7kppodwmns3b-3{list-style-type:none}ul.lst-kix_7kppodwmns3b-8{list-style-type:none}.lst-kix_olh7ea290trr-5>li:before{content:"\0025a0  "}ul.lst-kix_7kppodwmns3b-6{list-style-type:none}ul.lst-kix_7kppodwmns3b-7{list-style-type:none}.lst-kix_9drbe3jtkkuh-5>li{counter-increment:lst-ctn-kix_9drbe3jtkkuh-5}.lst-kix_olh7ea290trr-2>li:before{content:"\0025a0  "}.lst-kix_l7t8lma3v5dd-0>li:before{content:"\0025cf  "}.lst-kix_whzqgipsg1bt-4>li:before{content:"\0025cb  "}.lst-kix_cmqbbecr9wqr-3>li:before{content:"\0025cf  "}.lst-kix_z0ttp4mliy4e-5>li:before{content:"\0025a0  "}.lst-kix_vqtjz6k7exn7-8>li:before{content:"\0025a0  "}ul.lst-kix_xc244bhq3ube-2{list-style-type:none}ul.lst-kix_xc244bhq3ube-3{list-style-type:none}ul.lst-kix_xc244bhq3ube-0{list-style-type:none}ul.lst-kix_xc244bhq3ube-1{list-style-type:none}ul.lst-kix_xc244bhq3ube-6{list-style-type:none}ul.lst-kix_xc244bhq3ube-7{list-style-type:none}ul.lst-kix_xc244bhq3ube-4{list-style-type:none}ul.lst-kix_xc244bhq3ube-5{list-style-type:none}.lst-kix_3kgcjgrfg4nh-4>li:before{content:"\0025cb  "}ul.lst-kix_xc244bhq3ube-8{list-style-type:none}.lst-kix_l7t8lma3v5dd-6>li:before{content:"\0025cf  "}.lst-kix_7rljtqi4tpf8-7>li:before{content:"\0025cb  "}.lst-kix_x7iifx7mthae-2>li:before{content:"\0025a0  "}.lst-kix_x7iifx7mthae-3>li:before{content:"\0025cf  "}.lst-kix_10avetia2xt4-5>li:before{content:"\0025a0  "}ul.lst-kix_x7iifx7mthae-8{list-style-type:none}.lst-kix_pdbqztwjr65g-0>li:before{content:"\0025cf  "}ul.lst-kix_x7iifx7mthae-6{list-style-type:none}ul.lst-kix_x7iifx7mthae-7{list-style-type:none}ol.lst-kix_9drbe3jtkkuh-7.start{counter-reset:lst-ctn-kix_9drbe3jtkkuh-7 0}ul.lst-kix_x7iifx7mthae-4{list-style-type:none}ul.lst-kix_x7iifx7mthae-5{list-style-type:none}ul.lst-kix_x7iifx7mthae-2{list-style-type:none}ul.lst-kix_x7iifx7mthae-3{list-style-type:none}.lst-kix_owpom8h3nkwo-5>li:before{content:"\0025a0  "}.lst-kix_g3lepnw8n0x6-8>li:before{content:"\0025a0  "}.lst-kix_5w9dslcj8r5z-1>li:before{content:"\0025cb  "}ul.lst-kix_x7iifx7mthae-1{list-style-type:none}.lst-kix_owpom8h3nkwo-0>li:before{content:"\0025cf  "}ul.lst-kix_x7iifx7mthae-0{list-style-type:none}.lst-kix_velyrizglpvf-2>li:before{content:"\0025a0  "}.lst-kix_vz631vhbp5pc-7>li:before{content:"\0025cb  "}.lst-kix_e5f9ugvogyec-5>li:before{content:"\0025a0  "}.lst-kix_olh7ea290trr-8>li:before{content:"\0025a0  "}.lst-kix_xq12dxu3f08q-4>li:before{content:"\0025cb  "}.lst-kix_olh7ea290trr-3>li:before{content:"\0025cf  "}.lst-kix_g3lepnw8n0x6-6>li:before{content:"\0025cf  "}.lst-kix_cmqbbecr9wqr-8>li:before{content:"\0025a0  "}.lst-kix_z0ttp4mliy4e-4>li:before{content:"\0025cb  "}.lst-kix_4tq1geabcxa-6>li:before{content:"\0025cf  "}ul.lst-kix_euwmeg8cbpt8-2{list-style-type:none}ul.lst-kix_euwmeg8cbpt8-3{list-style-type:none}ul.lst-kix_euwmeg8cbpt8-0{list-style-type:none}ul.lst-kix_euwmeg8cbpt8-1{list-style-type:none}.lst-kix_qzhxzy2sbaso-8>li:before{content:"\0025a0  "}ul.lst-kix_euwmeg8cbpt8-6{list-style-type:none}.lst-kix_xc244bhq3ube-6>li:before{content:"\0025cf  "}ul.lst-kix_euwmeg8cbpt8-7{list-style-type:none}ul.lst-kix_euwmeg8cbpt8-4{list-style-type:none}ul.lst-kix_euwmeg8cbpt8-5{list-style-type:none}.lst-kix_hzkhz4hhlr8l-8>li:before{content:"\0025a0  "}ul.lst-kix_euwmeg8cbpt8-8{list-style-type:none}ol{margin:0;padding:0}.c30{vertical-align:top;width:510.2pt;border-style:solid;background-color:#efefef;border-color:#000000;border-width:0pt;padding:5pt 5pt 5pt 5pt}.c50{vertical-align:top;width:507.8pt;border-style:solid;background-color:#ffffff;border-color:#000000;border-width:0pt;padding:5pt 5pt 5pt 5pt}.c1{padding-left:0pt;line-height:1.0;padding-top:0pt;text-align:left;direction:ltr;margin-left:36pt;padding-bottom:0pt}.c7{vertical-align:top;width:510.2pt;border-style:solid;background-color:#f3f3f3;border-color:#000000;border-width:0pt;padding:5pt 5pt 5pt 5pt}.c11{vertical-align:top;width:510.2pt;border-style:solid;background-color:#ffffff;border-color:#000000;border-width:0pt;padding:5pt 5pt 5pt 5pt}.c46{vertical-align:middle;width:26.2pt;border-style:solid;background-color:#cccccc;border-color:#b7b7b7;border-width:1pt}.c52{vertical-align:top;width:53.2pt;border-style:solid;background-color:#f9cb9c;border-color:#b7b7b7;border-width:1pt}.c49{vertical-align:middle;width:37.5pt;border-style:solid;background-color:#cccccc;border-color:#b7b7b7;border-width:1pt}.c65{vertical-align:top;width:53.2pt;border-style:solid;background-color:#cccccc;border-color:#b7b7b7;border-width:1pt}.c19{vertical-align:top;width:58.7pt;border-style:solid;background-color:#f9cb9c;border-color:#b7b7b7;border-width:1pt}.c47{vertical-align:top;width:58.7pt;border-style:solid;background-color:#cccccc;border-color:#b7b7b7;border-width:1pt}.c66{vertical-align:middle;width:27.8pt;border-style:solid;background-color:#cccccc;border-color:#b7b7b7;border-width:1pt}.c64{vertical-align:middle;width:23.2pt;border-style:solid;background-color:#cccccc;border-color:#b7b7b7;border-width:1pt}.c79{vertical-align:top;width:58.7pt;border-style:solid;background-color:#ffffff;border-color:#b7b7b7;border-width:1pt}.c68{vertical-align:middle;width:29.2pt;border-style:solid;background-color:#cccccc;border-color:#b7b7b7;border-width:1pt}.c36{vertical-align:top;width:23.2pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c35{vertical-align:middle;width:41.2pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c75{vertical-align:middle;width:26.2pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c25{vertical-align:top;width:26.2pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c16{vertical-align:top;width:42pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c71{vertical-align:middle;width:27.8pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c33{vertical-align:top;width:41.2pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c80{vertical-align:middle;width:23.2pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c78{vertical-align:middle;width:24.8pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c37{vertical-align:top;width:24.8pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c15{vertical-align:top;width:30.8pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c72{vertical-align:middle;width:29.2pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c70{vertical-align:middle;width:21.8pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c44{vertical-align:top;width:21.8pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c60{vertical-align:top;width:29.2pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c61{vertical-align:middle;width:42pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c34{vertical-align:middle;width:30.8pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c67{vertical-align:middle;width:37.5pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c31{vertical-align:top;width:27.8pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c69{vertical-align:baseline;color:#000000;font-size:11pt;font-style:normal;text-decoration:none}.c63{vertical-align:middle;width:35.2pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c53{vertical-align:top;width:37.5pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c48{vertical-align:top;width:35.2pt;border-style:solid;border-color:#b7b7b7;border-width:1pt}.c8{line-height:1.0;padding-top:0pt;text-align:center;padding-bottom:0pt}.c57{max-width:510.2pt;background-color:#ffffff;padding:42.5pt 42.5pt 42.5pt 42.5pt}.c6{margin-right:auto;border-collapse:collapse}.c42{widows:2;orphans:2;margin-left:36pt}.c45{widows:2;orphans:2;margin-left:18pt}.c27{widows:2;orphans:2;margin-left:54pt}.c23{margin:0;padding:0}.c81{padding-top:14pt;padding-bottom:4pt}.c3{text-indent:36pt;direction:ltr}.c43{font-size:10pt;font-family:"Verdana"}.c22{color:#cc0000;text-decoration:underline}.c5{font-size:12pt;font-family:"Times New Roman"}.c0{height:11pt;direction:ltr}.c51{widows:2;orphans:2}.c29{padding-top:0pt;padding-bottom:0pt}.c26{padding-left:4.5pt;margin-left:30.8pt}.c28{font-size:10pt;font-family:"Courier New"}.c41{padding-top:14pt;padding-bottom:14pt}.c24{color:inherit;text-decoration:inherit}.c14{padding-left:0pt;margin-left:36pt}.c54{padding-left:0pt;margin-left:72pt}.c21{color:#1155cc;text-decoration:underline}.c59{padding-top:24pt;padding-bottom:6pt}.c4{line-height:1.0;direction:ltr}.c9{text-align:center}.c76{color:#000000}.c77{height:11pt}.c20{line-height:1.0}.c58{font-size:10pt}.c12{font-weight:bold}.c18{font-size:14pt}.c40{text-align:left}.c13{text-align:justify}.c62{height:13pt}.c39{color:#cc0000}.c55{height:16pt}.c10{direction:ltr}.c17{font-family:"Times New Roman"}.c73{color:#ff0000}.c56{padding-top:0pt}.c32{font-size:12pt}.c2{height:0pt}.c38{font-style:italic}.c74{background-color:#cccccc}.title{widows:2;padding-top:0pt;line-height:1.15;orphans:2;text-align:left;color:#000000;font-size:21pt;font-family:"Trebuchet MS";padding-bottom:0pt;page-break-after:avoid}.subtitle{widows:2;padding-top:0pt;line-height:1.15;orphans:2;text-align:left;color:#666666;font-style:italic;font-size:13pt;font-family:"Trebuchet MS";padding-bottom:10pt;page-break-after:avoid}li{color:#000000;font-size:11pt;font-family:"Arial"}p{color:#000000;font-size:11pt;margin:0;font-family:"Arial"}h1{widows:2;padding-top:10pt;line-height:1.15;orphans:2;text-align:left;color:#000000;font-size:16pt;font-family:"Trebuchet MS";padding-bottom:0pt;page-break-after:avoid}h2{widows:2;padding-top:10pt;line-height:1.15;orphans:2;text-align:left;color:#000000;font-size:13pt;font-family:"Trebuchet MS";font-weight:bold;padding-bottom:0pt;page-break-after:avoid}h3{widows:2;padding-top:8pt;line-height:1.15;orphans:2;text-align:left;color:#666666;font-size:12pt;font-family:"Trebuchet MS";font-weight:bold;padding-bottom:0pt;page-break-after:avoid}h4{widows:2;padding-top:8pt;line-height:1.15;orphans:2;text-align:left;color:#666666;font-size:11pt;text-decoration:underline;font-family:"Trebuchet MS";padding-bottom:0pt;page-break-after:avoid}h5{widows:2;padding-top:8pt;line-height:1.15;orphans:2;text-align:left;color:#666666;font-size:11pt;font-family:"Trebuchet MS";padding-bottom:0pt;page-break-after:avoid}h6{widows:2;padding-top:8pt;line-height:1.15;orphans:2;text-align:left;color:#666666;font-style:italic;font-size:11pt;font-family:"Trebuchet MS";padding-bottom:0pt;page-break-after:avoid}</style></head><body class="c57"><div><p class="c0"><span></span></p></div><p class="c9 c10 c59 title"><a name="h.5r0tnx612rs0"></a><span class="c12 c17">GeneValidator</span></p><p class="c59 c9 c10 subtitle"><a name="h.jmdsip5u710q"></a><span class="c12 c17">Identify problems with predicted genes</span></p><p class="c0"><span class="c5 c12"></span></p><p class="c10"><span class="c12 c18 c17">This is a </span><span class="c21 c12 c18 c17"><a class="c24" href="">NESCent project</a></span><span class="c12 c18 c17">, funded by Google during </span><span class="c21 c12 c18 c17"><a class="c24" href="http://www.google-melange.com/gsoc/homepage/google/gsoc2013%20">Google Summer of Code 2013</a></span><span class="c12 c18 c17">&nbsp;and mentored by Anurag Priyam and Yannick Wurm.</span></p><h3 class="c10 c81"><a name="h.yluxu51txcqj"></a><span class="c18 c17 c76">Authors</span></h3><ul class="c23 lst-kix_pizqq4oicz3j-0 start"><li class="c14 c10"><span class="c12 c17 c18">GSoC student:</span><span class="c12 c18 c39 c17"><a class="c24" href="http://www.google.com/url?q=http%3A%2F%2Fswarm.cs.pub.ro%2F%257Emdragan%2Fgsoc2013%2FMonica_Dragan_CV.pdf&amp;sa=D&amp;sntz=1&amp;usg=AFQjCNE2sgYnN3NmFtzN0bDxNZtrLxykrA">&nbsp;</a></span><span class="c21 c12 c18 c17"><a class="c24" href="http://www.google.com/url?q=http%3A%2F%2Fswarm.cs.pub.ro%2F%257Emdragan%2Fgsoc2013%2FMonica_Dragan_CV.pdf&amp;sa=D&amp;sntz=1&amp;usg=AFQjCNE2sgYnN3NmFtzN0bDxNZtrLxykrA">Monica Dragan</a></span><span class="c12 c18 c39 c17">&nbsp;</span></li><li class="c14 c10"><span class="c12 c18 c17">Mentors:</span><span class="c12 c18 c39 c17"><a class="c24" href="https://plus.google.com/114122400102590087616/about">&nbsp;</a></span><span class="c21 c12 c18 c17"><a class="c24" href="https://plus.google.com/114122400102590087616/about">Anurag Priyam</a></span><span class="c12 c18 c39 c17">&nbsp;</span><span class="c12 c18 c17">and</span><span class="c12 c18 c39 c17"><a class="c24" href="http://www.google.com/url?q=http%3A%2F%2Fyannick.poulet.org%2F&amp;sa=D&amp;sntz=1&amp;usg=AFQjCNF0jaZdIo_nKWAP-go-xlg_W6zflA">&nbsp;</a></span><span class="c21 c12 c18 c17"><a class="c24" href="http://www.google.com/url?q=http%3A%2F%2Fyannick.poulet.org%2F&amp;sa=D&amp;sntz=1&amp;usg=AFQjCNF0jaZdIo_nKWAP-go-xlg_W6zflA">Yannick Wurm</a></span></li></ul><p class="c0"><span class="c18 c39 c17"></span></p><p class="c10"><span class="c12 c18 c17">Resources</span></p><ul class="c23 lst-kix_6064le485ne7-0 start"><li class="c14 c10"><span class="c12 c18 c17">Source code: </span><span class="c21 c12 c18 c17"><a class="c24" href="https://www.google.com/url?q=https%3A%2F%2Fgithub.com%2Fmonicadragan%2FGeneValidator&amp;sa=D&amp;sntz=1&amp;usg=AFQjCNEpEZUQSSyCqVUpkmmEyrqzf9X90A">github</a></span></li><li class="c14 c10"><span class="c21 c12 c18 c17"><a class="c24" href="http://gene-prediction.blogspot.ch/">Blog</a></span></li><li class="c14 c10"><span class="c21 c12 c18 c17"><a class="c24" href="http://www.google.com/url?q=http%3A%2F%2Fswarm.cs.pub.ro%2F~mdragan%2Fgsoc2013%2Fgenevalidator&amp;sa=D&amp;sntz=1&amp;usg=AFQjCNHevw0tB3SOBnPKIHmAemjta0Ke4w">Output</a></span></li></ul><p class="c0"><span class="c22 c12 c18 c17"></span></p><p class="c0"><span class="c22 c12 c18 c17"></span></p><p class="c10"><span class="c12 c18 c17">Summary</span></p><p class="c10 c27"><span class="c5 c21"><a class="c24" href="#h.yluxu51txcqj">Authors</a></span></p><p class="c45 c10"><span class="c5 c21"><a class="c24" href="#h.aquxglil7al4">A. Project description</a></span></p><p class="c45 c10"><span class="c5 c21"><a class="c24" href="#h.48ubnp31f0km">B. Validations</a></span></p><p class="c42 c10"><span class="c5 c21"><a class="c24" href="#h.pay33gn4jmyv">1) Length Validation via clusterization</a></span></p><p class="c42 c10"><span class="c5 c21"><a class="c24" href="#h.8l1xsn8jugrl">2) Length Validation by ranking</a></span></p><p class="c42 c10"><span class="c5 c21"><a class="c24" href="#h.x5xi5uouhpqm">3) Duplication Check</a></span></p><p class="c42 c10"><span class="c5 c21"><a class="c24" href="#h.mci2ebh73ez2">4) Gene Merge Validation</a></span></p><p class="c42 c10"><span class="c5 c21"><a class="c24" href="#h.lpsoaxpm11ri">5) Validation based on multiple alignment - for proteins only</a></span></p><p class="c10 c42"><span class="c5 c21"><a class="c24" href="#h.f63zo0greej2">6) Blast Reading Frame Validation - for nucleotides only</a></span></p><p class="c42 c10"><span class="c5 c21"><a class="c24" href="#h.3elev4f70kdl">7) Open reading frame - for nucleotides only</a></span></p><p class="c10 c45"><span class="c5 c21"><a class="c24" href="#h.njhshrkdurp1">C. Code Structure</a></span></p><p class="c45 c10"><span class="c5 c21"><a class="c24" href="#h.8hgqxxlaoy94">D. How to run the tool</a></span></p><p class="c45 c10"><span class="c5 c21"><a class="c24" href="#h.qi34bbc42hqf">E. Add a new validation</a></span></p><p class="c45 c10"><span class="c5 c21"><a class="c24" href="#h.o25pqypedjmx">F. Results</a></span></p><p class="c45 c10"><span class="c5 c21"><a class="c24" href="#h.4rd7mxpo7dua">G. GUI Output</a></span></p><p class="c0"><span class="c22 c5 c12"></span></p><h1 class="c9 c55 c10"><a name="h.htgpgoygm0nz"></a></h1><h1 class="c9 c55 c10"><a name="h.rz0ao0fey7r8"></a></h1><h1 class="c9 c10 c55"><a name="h.dr4jv2s5i6rv"></a></h1><h1 class="c9 c10"><a name="h.aquxglil7al4"></a><span class="c12 c17">A. Project description</span></h1><p class="c0"><span></span></p><p class="c0"><span class="c17"></span></p><p class="c13 c10"><span class="c18 c17">Genome sequencing is now possible at almost no cost. However, obtaining accurate gene predictions remains a target hard to achieve with the existing technology. </span></p><p class="c0 c13"><span class="c18 c17"></span></p><p class="c13 c10"><span class="c18 c17">We designed a tool that identifies problems with gene predictions, based on similarities with data from public databases (e.g Swissprot and Uniprot). We apply a set of validation tests that provide useful information about the problems which appear in the predictions, in order to make evidence about how the gene curation should be made or whether a certain predicted gene may not be considered in other analysis. </span></p><p class="c0 c13"><span class="c18 c17"></span></p><p class="c13 c10"><span class="c18 c17">Our main target users are the biologists, who have large amounts of genetic data that has to be manually curated and use </span><span class="c18 c17">our tool to prioritize the genes</span><span class="c18 c17">&nbsp;that need to be checked and be aware about possible error causes.</span></p><p class="c0 c13"><span class="c12 c17"></span></p><p class="c0 c13"><span class="c12 c17"></span></p><h1 class="c9 c10"><a name="h.48ubnp31f0km"></a><span class="c12 c17">B. Validations</span></h1><p class="c0 c9"><span class="c5"></span></p><p class="c13 c10"><span class="c18 c17">In order to highlight the problems that appear in the predictions and suggest possible error causes, we developed 7 validation test:</span></p><p class="c0 c13"><span class="c18 c17"></span></p><p class="c10"><span class="c12 c18 c17">1) Length validation via clusterization</span></p><p class="c10"><span class="c12 c18 c17">2) Length validation via ranking</span></p><p class="c10"><span class="c12 c18 c17">3) Duplication</span></p><p class="c10"><span class="c12 c18 c17">4) Gene merge</span></p><p class="c10"><span class="c12 c18 c17">5) Multiple alignment</span></p><p class="c10"><span class="c12 c18 c17">6) Blast Reading Frame - applicable only for nucleotide sequences</span></p><p class="c10"><span class="c12 c18 c17">7) Open reading frame - applicable only for nucleotide sequences</span></p><p class="c0"><span class="c12 c18 c17 c38"></span></p><p class="c0"><span class="c12 c18 c17 c38"></span></p><p class="c0"><span class="c12 c18 c17 c38"></span></p><p class="c0"><span class="c12 c18 c17 c38"></span></p><p class="c0"><span class="c12 c18 c17 c38"></span></p><p class="c0"><span class="c12 c18 c17 c38"></span></p><hr style="page-break-before:always;display:none;"><p class="c0"><span class="c12 c18 c17 c38"></span></p><p class="c10"><span class="c18 c17">Data needed for each validation is retrieved from the BLAST output and used as follows (aliases and tags are those used in BLAST &lsquo;outfmt&rsquo; argument):</span></p><p class="c0"><span class="c18 c17"></span></p><p class="c10"><span class="c5 c12">x &nbsp; = mandatory parameter</span></p><p class="c0"><span class="c12 c17"></span></p><a href="#" name="a925117327d0f6cda70f85570f7a83ad25da68f8"></a><a href="#" name="0"></a><table cellpadding="0" cellspacing="0" class="c6"><tbody><tr class="c2"><td class="c79"><p class="c8 c0"><span class="c69 c12 c17"></span></p></td><td class="c65"><p class="c8 c10"><span class="c12 c17">Blast param</span></p></td><td class="c34 c74"><p class="c4 c9"><span class="c12 c17">sseqid</span></p></td><td class="c64"><p class="c4 c9"><span class="c12 c17">sacc</span></p></td><td class="c70 c74"><p class="c4 c9"><span class="c12 c17">slen</span></p></td><td class="c34 c74"><p class="c4 c9"><span class="c12 c17">qstart</span></p></td><td class="c66"><p class="c4 c9"><span class="c12 c17">qend</span></p></td><td class="c68"><p class="c4 c9"><span class="c12 c17">sstart</span></p></td><td class="c66"><p class="c4 c9"><span class="c12 c17">send</span></p></td><td class="c63 c74"><p class="c4 c9"><span class="c12 c17">length</span></p></td><td class="c46"><p class="c4 c9"><span class="c12 c17">qseq</span></p></td><td class="c78 c74"><p class="c4 c9"><span class="c12 c17">sseq</span></p></td><td class="c49"><p class="c4 c9"><span class="c12 c17">qframe</span></p></td><td class="c61 c74"><p class="c4 c9"><span class="c12 c17">Query raw seq</span></p></td><td class="c35 c74"><p class="c4 c9"><span class="c12 c17">Hit &nbsp;raw seq</span></p></td></tr><tr class="c2"><td class="c47"><p class="c8 c10"><span class="c12 c17">Validation</span></p></td><td class="c65"><p class="c8 c10"><span class="c12 c17">Alias</span></p></td><td class="c34"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c80"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c70"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c34"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c71"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c72"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c71"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c63"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c75"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c78"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c67"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c61"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c35"><p class="c8 c0"><span class="c12 c17"></span></p></td></tr><tr class="c2"><td class="c19"><p class="c8 c10"><span class="c12 c17">Length by clustering </span></p></td><td class="c52"><p class="c8 c10"><span class="c12 c17">lenc</span></p></td><td class="c15"><p class="c4 c9"><span class="c12 c17">x</span></p></td><td class="c36"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c44"><p class="c8 c10"><span class="c12 c17">x</span></p></td><td class="c15"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c31"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c60"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c31"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c48"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c25"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c37"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c53"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c16"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c33"><p class="c8 c0"><span class="c12 c17"></span></p></td></tr><tr class="c2"><td class="c19"><p class="c8 c10"><span class="c12 c17">Length by rank</span></p></td><td class="c52"><p class="c8 c10"><span class="c12 c17">lenr</span></p></td><td class="c15"><p class="c4 c9"><span class="c12 c17">x</span></p></td><td class="c36"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c44"><p class="c8 c10"><span class="c12 c17">x</span></p></td><td class="c15"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c31"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c60"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c31"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c48"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c25"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c37"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c53"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c16"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c33"><p class="c8 c0"><span class="c12 c17"></span></p></td></tr><tr class="c2"><td class="c19"><p class="c8 c10"><span class="c12 c17">Reading Frame </span></p></td><td class="c52"><p class="c8 c10"><span class="c12 c17">frame</span></p></td><td class="c15"><p class="c4 c9"><span class="c12 c17">x</span></p></td><td class="c36"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c44"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c15"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c31"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c60"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c31"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c48"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c25"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c37"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c53"><p class="c8 c10"><span class="c12 c17">x</span></p></td><td class="c16"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c33"><p class="c8 c0"><span class="c12 c17"></span></p></td></tr><tr class="c2"><td class="c19"><p class="c8 c10"><span class="c12 c17">Gene merge</span></p></td><td class="c52"><p class="c8 c10"><span class="c12 c17">merge</span></p></td><td class="c15"><p class="c4 c9"><span class="c12 c17">x</span></p></td><td class="c36"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c44"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c15"><p class="c8 c10"><span class="c12 c17">x</span></p></td><td class="c31"><p class="c8 c10"><span class="c12 c17">x</span></p></td><td class="c60"><p class="c8 c10"><span class="c12 c17">x</span></p></td><td class="c31"><p class="c8 c10"><span class="c12 c17">x</span></p></td><td class="c48"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c25"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c37"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c53"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c16"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c33"><p class="c8 c0"><span class="c12 c17"></span></p></td></tr><tr class="c2"><td class="c19"><p class="c8 c10"><span class="c12 c17">Duplication</span></p></td><td class="c52"><p class="c8 c10"><span class="c12 c17">dup</span></p></td><td class="c15"><p class="c4 c9"><span class="c12 c17">x</span></p></td><td class="c36"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c44"><p class="c8 c10"><span class="c12 c17">x</span></p></td><td class="c15"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c31"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c60"><p class="c8 c10"><span class="c12 c17">x</span></p></td><td class="c31"><p class="c8 c10"><span class="c12 c17">x</span></p></td><td class="c48"><p class="c8 c10"><span class="c12 c17">x</span></p></td><td class="c25"><p class="c8 c10"><span class="c12 c17">x</span></p></td><td class="c37"><p class="c8 c10"><span class="c12 c17">x</span></p></td><td class="c53"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c16"><p class="c8 c10"><span class="c12 c17">x</span></p></td><td class="c33"><p class="c8 c10"><span class="c12 c17">x</span></p></td></tr><tr class="c2"><td class="c19"><p class="c8 c10"><span class="c12 c17">Open Reading Frame</span></p></td><td class="c52"><p class="c8 c10"><span class="c12 c17">orf</span></p></td><td class="c15"><p class="c4 c9"><span class="c12 c17">x</span></p></td><td class="c36"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c44"><p class="c0 c8"><span class="c12 c17"></span></p></td><td class="c15"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c31"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c60"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c31"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c48"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c25"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c37"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c53"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c16"><p class="c8 c10"><span class="c12 c17">x</span></p></td><td class="c33"><p class="c8 c0"><span class="c12 c17"></span></p></td></tr><tr class="c2"><td class="c19"><p class="c8 c10"><span class="c12 c17">Multiple align. based</span></p></td><td class="c52"><p class="c8 c10"><span class="c12 c17">align</span></p></td><td class="c15"><p class="c4 c9"><span class="c12 c17">x</span></p></td><td class="c36"><p class="c8 c10"><span class="c12 c17">x</span></p></td><td class="c44"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c15"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c31"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c60"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c31"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c48"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c25"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c37"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c53"><p class="c8 c0"><span class="c12 c17"></span></p></td><td class="c16"><p class="c8 c10"><span class="c12 c17">x</span></p></td><td class="c33"><p class="c8 c10"><span class="c12 c17">x</span></p></td></tr></tbody></table><p class="c0"><span class="c12 c17"></span></p><p class="c0"><span class="c5 c12"></span></p><p class="c10"><span class="c5 c12">Each validation is described further in this section.</span></p><p class="c0"><span class="c5 c12"></span></p><h2 class="c10"><a name="h.pay33gn4jmyv"></a><span class="c17">1) Length Validation via clusterization</span></h2><a href="#" name="61c369d908aba043975fd75a3cee4d8b3bd887b2"></a><a href="#" name="1"></a><table cellpadding="0" cellspacing="0" class="c6"><tbody><tr class="c2"><td class="c50"><h4 class="c4"><a name="h.1kmh3ff8b3xt"></a><span>Error causes</span></h4><p class="c0 c29 c20"><span></span></p><ul class="c23 lst-kix_3jnvhqhv71nk-0 start"><li class="c14 c4"><span class="c5">sequencing error: some parts of the gene were lost/added on the way or gene bounds were not well estimated</span></li></ul><ul class="c23 lst-kix_mk8t5sy0k589-0 start"><li class="c1"><span class="c5">the gene was low expressed</span></li><li class="c1"><span class="c5">the sequenced mRNA incorrectly contains an some introns </span></li></ul><p class="c0 c29 c20 c40"><span class="c5"></span></p></td></tr><tr class="c2"><td class="c50"><h4 class="c4"><a name="h.5jizasmleana"></a><span>Input data</span></h4><p class="c0 c29 c20"><span></span></p><ul class="c23 lst-kix_4ed4y1s6lj5w-0 start"><li class="c14 c10"><span class="c5">lengths of the hits: this data is retrieved after parsing the blast output file</span></li><li class="c14 c10"><span class="c5">length of the prediction: this number is the length of the current query from the fasta. In case of nucleotide sequences, we are interested in the length of the corresponding query translated into protein (which is the length of the query divided by 3)</span></li></ul></td></tr><tr class="c2"><td class="c50"><h4 class="c4"><a name="h.wlnymr8sa2ta"></a><span>Class information:</span></h4><p class="c0 c29 c20"><span></span></p><ul class="c23 lst-kix_7rljtqi4tpf8-0 start"><li class="c1"><span class="c5">header: Length Cluster</span></li><li class="c1"><span class="c5">short description: Check whether the prediction length fits most of the BLAST hit lengths, by 1D hierarchical clusterization. Meaning of the output displayed: Prediction_len [Main Cluster Length Interval]</span></li><li class="c1"><span class="c5">alias: lenc</span></li></ul></td></tr><tr class="c2"><td class="c50"><h4 class="c4"><a name="h.pe5jadwr5x2b"></a><span>Workflow</span></h4><p class="c0 c20 c29"><span></span></p><p class="c29 c4"><span class="c5 c12">Aim :</span><span class="c5">&nbsp;we are interested to find out if the length of the predicted sequence belongs to the distribution of the hit lengths (in other words, how close is the length of the prediction to the majority of the lengths of the hits)</span></p><p class="c0 c29 c20"><span class="c5"></span></p><ul class="c23 lst-kix_xq12dxu3f08q-0 start"><li class="c14 c4 c13"><span class="c5">&nbsp;By plotting the histogram of the length distribution of the hits we observe that the distribution does not fit a Bell Curve (see Figure 1), therefore we cannot apply the classical T-test. </span></li></ul><p class="c0 c13 c20"><span class="c5"></span></p><p class="c0 c20 c13"><span class="c17"></span></p><p class="c4 c9"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 307.00px; height: 220.24px;"><img alt="" src="images/image17.png" style="width: 307.00px; height: 220.24px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c4 c9"><span class="c12 c17">(1)</span></p><p class="c0 c20 c13"><span class="c17"></span></p><p class="c4 c13"><span class="c5">Our approach to find the majority of lengths among the reference lengths uses a typical </span><span class="c5 c12">hierarchical clusterization</span><span class="c5">: </span></p><p class="c0 c20 c13"><span class="c5"></span></p><ul class="c23 lst-kix_aps1ppntqbvm-0 start"><li class="c14 c4 c13"><span class="c5">Firstly we assume that each length belongs to a separate cluster. Each step we merge the closest two clusters, until a cluster that contains more than 50% of the reference sequences is obtained. Each clusters is represented with a different colors in Figure 2. The one colored in red is called the &ldquo;main cluster&rdquo;.</span></li></ul><p class="c0 c20 c13"><span class="c17"></span></p><p class="c4 c9"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 331.00px; height: 230.55px;"><img alt="" src="images/image13.png" style="width: 331.00px; height: 230.55px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c4 c9"><span class="c12 c17">(2)</span></p><p class="c0 c9 c20"><span class="c17"></span></p><ul class="c23 lst-kix_lf42ux7u6753-0 start"><li class="c14 c4 c13"><span class="c5">Finally we are interested to check whether the length of the</span><span class="c5">&nbsp;prediction belongs to the main cluster or not. The validation test will pass if the length of the prediction belongs to the main cluster of lengths (see Figure 3) and will fail otherwise (Figure 4).</span></li></ul><p class="c0 c20 c13"><span class="c17"></span></p><p class="c0 c20 c13"><span class="c17"></span></p><p class="c4 c9"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 311.12px; height: 246.50px;"><img alt="" src="images/image01.png" style="width: 311.12px; height: 246.50px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 304.29px; height: 242.50px;"><img alt="" src="images/image19.png" style="width: 304.29px; height: 242.50px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c4 c9"><span class="c12 c17">(3) &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; (4)</span></p><p class="c0 c29 c20 c40"><span class="c12 c17"></span></p></td></tr><tr class="c2"><td class="c50"><p class="c0 c29 c20"><span class="c58 c17"></span></p></td></tr><tr class="c2"><td class="c50"><h4 class="c4"><a name="h.t0qkl9cnekf7"></a><span>Plots</span></h4><p class="c0 c29 c20"><span></span></p><ul class="c23 lst-kix_euwmeg8cbpt8-0 start"><li class="c14 c4 c13"><span class="c5">length distribution histogram </span><span class="c5">(prediction passes the validation test in Figure 3 and fails in Figure 4)</span></li></ul><p class="c4 c40"><span class="c12 c17">&nbsp; </span></p></td></tr></tbody></table><h2 class="c10"><a name="h.8l1xsn8jugrl"></a><span class="c17">2</span><span class="c17">) </span><span class="c17">Length Validation by ranking</span></h2><p class="c0"><span class="c17"></span></p><a href="#" name="517d303ad854cb1f8ec6129a05eb146bc0814d94"></a><a href="#" name="2"></a><table cellpadding="0" cellspacing="0" class="c6"><tbody><tr class="c2"><td class="c30"><h4 class="c4"><a name="h.71j0exm2y7ro"></a><span class="c32">Error causes</span></h4><p class="c0 c29 c20 c51"><span></span></p><ul class="c23 lst-kix_3jnvhqhv71nk-0"><li class="c14 c4"><span class="c5">sequencing error: some parts of the gene were lost/added on the way or gene bounds were not well estimated</span></li></ul><ul class="c23 lst-kix_mk8t5sy0k589-0"><li class="c14 c4"><span class="c5">the gene was low expressed</span></li><li class="c14 c4"><span class="c5">the sequenced mRNA incorrectly contains an some introns </span></li></ul></td></tr><tr class="c2"><td class="c30"><h4 class="c4"><a name="h.59mfpm8tx3g3"></a><span>Input data</span></h4><p class="c0 c29 c20"><span></span></p><ul class="c23 lst-kix_4ed4y1s6lj5w-0"><li class="c14 c10"><span class="c5">lengths of the hits: this data is retrieved after parsing the blast output file</span></li><li class="c14 c10"><span class="c5">length of the prediction: this number is the length of the current query from the fasta. In case of nucleotide sequences, we are interested in the length of the corresponding query translated into protein (which is the length of the query divided by 3)</span></li></ul></td></tr><tr class="c2"><td class="c30"><h4 class="c4"><a name="h.z3l9ocp2n17f"></a><span>Class information:</span></h4><p class="c0 c29 c20"><span></span></p><ul class="c23 lst-kix_7kppodwmns3b-0 start"><li class="c14 c4"><span class="c5">header: Length Rank</span></li><li class="c14 c4"><span class="c5">description: Check whether the rank of the prediction length lies among 80% of all the BLAST hit lengths. Meaning of the output displayed: no of extreme length hits / total no of hits</span></li><li class="c14 c4"><span class="c5">alias: lenr</span></li></ul><p class="c0 c20"><span class="c5"></span></p></td></tr><tr class="c2"><td class="c30"><h4 class="c4"><a name="h.rphxumlynikz"></a><span>Workflow</span></h4><p class="c0 c20"><span class="c5 c12"></span></p><p class="c4"><span class="c5 c12">Aim: </span><span class="c5">How extreme the length of the prediction is among all the other hits. </span></p><p class="c4"><span class="c5 c12">Approach:</span></p><ul class="c23 lst-kix_velyrizglpvf-0 start"><li class="c14 c4"><span class="c5">sort the hit lengths in ascending order.</span></li><li class="c14 c4"><span class="c5">find the median for the set of hit lengths</span></li><li class="c14 c4"><span class="c5">find which side of the median the length of the prediction belongs to</span></li><li class="c14 c4"><span class="c5">case 1: if the prediction length is on the right side of the median, then count how many hit lengths are longer than the prediction length (Figure 1)</span></li><li class="c14 c4"><span class="c5">case 2: otherwise count how ma</span><span class="c5">ny hit lengths are</span><span class="c5">&nbsp;shorter than the prediction length (Figure 2)</span></li><li class="c14 c4"><span class="c5">hits that are longer than the prediction length (respectively shorter in case 2) are called extreme hits</span></li><li class="c14 c4"><span class="c5">return the percentage of the hits that are extreme hits. If this percentage is higher than 20%, the prediction will not pass this validation.</span></li></ul><p class="c0 c20"><span class="c5"></span></p><p class="c4 c9"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 279.00px; height: 263.00px;"><img alt="" src="images/image06.png" style="width: 279.00px; height: 263.00px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span><span class="c5">&nbsp; &nbsp; &nbsp; </span><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 271.00px; height: 262.00px;"><img alt="" src="images/image11.png" style="width: 271.00px; height: 262.00px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c4 c9"><span class="c5 c12">Figure 1 &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; Figure 2</span></p></td></tr><tr class="c2"><td class="c30"><p class="c0 c20"><span class="c58 c17"></span></p></td></tr><tr class="c2"><td class="c30"><h4 class="c4"><a name="h.h26xfjvngbfo"></a><span>Plots</span></h4><p class="c4"><span class="c17">No plots are generated</span></p></td></tr></tbody></table><p class="c0"><span class="c12 c17"></span></p><h2 class="c10 c62"><a name="h.2taje05ufk8a"></a></h2><h2 class="c10"><a name="h.x5xi5uouhpqm"></a><span class="c17">3) Duplication Check</span></h2><p class="c0"><span class="c17"></span></p><a href="#" name="09642a67e26aad76d3db374e6a0e2e556ec152ae"></a><a href="#" name="3"></a><table cellpadding="0" cellspacing="0" class="c6"><tbody><tr class="c2"><td class="c11"><h4 class="c4"><a name="h.tmr9j2ocl9mh"></a><span>Error causes</span></h4><ul class="c23 lst-kix_9a6zvrglxdu8-0 start"><li class="c1"><span class="c5">sequencing error</span></li><li class="c1"><span class="c5">error in the replication mechanism of the cell</span></li></ul></td></tr><tr class="c2"><td class="c11"><h4 class="c4"><a name="h.gqw8t05qbpfv"></a><span>Input data</span></h4><ul class="c23 lst-kix_4ed4y1s6lj5w-0"><li class="c14 c10"><span class="c5">raw sequence for prediction</span></li><li class="c14 c10"><span class="c5">raw sequence for the best n hits</span></li><li class="c14 c10"><span class="c5">hit length</span></li><li class="c14 c10"><span class="c5">start/end index of the high-scoring segment pairs (hsp) for hits and query</span></li></ul></td></tr><tr class="c2"><td class="c11"><h4 class="c4"><a name="h.lm93gxe78op4"></a><span>Class information:</span></h4><ul class="c23 lst-kix_e5f9ugvogyec-0 start"><li class="c14 c4"><span class="c5">header: Duplication</span></li><li class="c14 c4"><span class="c5">description: Check whether there is a duplicated subsequence in the predicted gene by counting the hsp residue coverage of the prediction, for each hit. Meaning of the output displayed: P-value of the Wilcoxon test which test the distribution of hit average coverage against 1. P-values higher than 5% pass the validation test.</span></li><li class="c14 c4"><span class="c5">alias: dup</span></li></ul><p class="c0 c20"><span class="c5"></span></p></td></tr><tr class="c2"><td class="c11"><h4 class="c4"><a name="h.ruzfferf17cu"></a><span>Workflow</span></h4><ul class="c23 lst-kix_sjinkw26u0wp-0 start"><li class="c14 c10"><span class="c5">get raw sequences for the first n hits (if they have not been already retrieved)</span></li><li class="c14 c10"><span class="c5">in case of nucleotide sequences -&gt; translate the </span><span class="c5">hsp in reading frame 0 or 1 (</span><span class="c5">hsp = </span><span class="c5">high-scoring segment</span><span class="c5">, i.e a pair of segments from the hit and the query that match with a high identity percent</span><span class="c5">) </span></li></ul><p class="c0"><span class="c5"></span></p><p class="c10"><span class="c5">For each hsp:</span></p><ul class="c23 lst-kix_sjinkw26u0wp-0"><li class="c14 c10"><span class="c5">subtract the raw sequence of the prediction</span></li><li class="c14 c10"><span class="c5">subtract the raw sequence of the hsps</span></li><li class="c14 c10"><span class="c5">compute local alignment for each hsp (in BLAST xml output the alignment is already provided)</span></li><li class="c14 c10"><span class="c5">initialize a coverage vector for every hit</span></li><li class="c14 c10"><span class="c5">pass through the alignment and increase by 1 the corresponding position in the coverage vector</span></li><li class="c14 c10"><span class="c5">calculate the average value of the values in the coverage vector (0 values are excluded). For a prediction without duplications, the average value should be very close to 1 (every residue from the hit should be covered by the prediction at most once). For all the n hits we will end up with a vector of n average values </span></li><li class="c14 c10"><span class="c5">trick: average values are roun</span><span class="c5">ded to 2 d</span><span class="c5">ecimals to make the data more uniform </span></li><li class="c14 c10"><span class="c5">we test the null hypothesis (if the mean of the average values is different from one) using the one sample wilcoxon te</span><span class="c5">st and return the p-value. In case the p-value is higher than 5%, the null hypothesis is true and there is no duplication in the prediction. If the p-value is lower than 5%, then the </span><span class="c5 c12">alternative hypothesis </span><span class="c5">is true (and the mean is different from one) and </span><span class="c5">in this case there is a duplication in the sequence. </span></li></ul></td></tr><tr class="c2"><td class="c11"><p class="c0 c20"><span class="c28"></span></p></td></tr><tr class="c2"><td class="c11"><h4 class="c4"><a name="h.jl85dzs94b1p"></a><span>Plots</span></h4><p class="c4"><span class="c5">No plots are generated</span></p></td></tr></tbody></table><p class="c0"><span class="c17"></span></p><h2 class="c10"><a name="h.mci2ebh73ez2"></a><span class="c17">4) </span><span class="c17">Gene Merge Validation</span></h2><p class="c0"><span class="c17"></span></p><a href="#" name="538b1b92648da3f1d5e628ef068aba7ad1263489"></a><a href="#" name="4"></a><table cellpadding="0" cellspacing="0" class="c6"><tbody><tr class="c2"><td class="c7"><h4 class="c4"><a name="h.bg9kc4bj0vxm"></a><span>Error cause</span></h4><ul class="c23 lst-kix_gr4a4jnxityy-0 start"><li class="c1"><span class="c5">gene prediction error (2 or more genes were merged together)</span></li></ul></td></tr><tr class="c2"><td class="c7"><h4 class="c4"><a name="h.bdyywegj4lax"></a><span>Input data</span></h4><ul class="c23 lst-kix_4ed4y1s6lj5w-0"><li class="c14 c10"><span class="c5">start/end index of the</span><span class="c5">&nbsp;high-scoring segment pairs (hsp)</span><span class="c5">&nbsp;for hits and query</span></li></ul></td></tr><tr class="c2"><td class="c7"><h4 class="c4"><a name="h.acf7h4g6cpzt"></a><span>Class information:</span></h4><p class="c0 c29 c20"><span></span></p><ul class="c23 lst-kix_cmqbbecr9wqr-0 start"><li class="c14 c4"><span class="c5">header: Gene Merge</span></li><li class="c14 c4"><span class="c5">description: Check whether BLAST hits make evidence about a merge of two genes that match the predicted gene. Meaning of the output displayed: slope of the linear regression of the relationship between the start and stop offsets of the hsps (see the plot). Invalid slopes are around 45 degrees.</span></li><li class="c14 c4"><span class="c5">alias: merge</span></li></ul><p class="c0 c20"><span class="c5"></span></p></td></tr><tr class="c2"><td class="c7"><h4 class="c4"><a name="h.m1smetvgx0h8"></a><span>Workflow</span></h4><p class="c0 c29 c20"><span></span></p><ul class="c23 lst-kix_myo05blafbk-0 start"><li class="c14 c4"><span class="c5">plot a 2D graph using the start/end offsets of the high-scoring segments in the prediction that match the hits (due to blast local alignment) -- take the coordinates of the most left and the most right hsp in the query sequence (qstart and qend)</span></li><li class="c14 c4"><span class="c5">check for the modality of the data in 2D space, using the &lsquo;Mode - median - mean&rsquo; modality test [1] on each of the two dimensions (if data is unimodal distributed on both axis that means there&rsquo;s only one mode)</span></li></ul><p class="c4"><span class="c5">[1] </span><span class="c21 c43"><a class="c24" href="http://www.google.com/url?q=http%3A%2F%2Fen.wikipedia.org%2Fwiki%2FUnimodal_distribution%23Unimodal_probability_distribution&amp;sa=D&amp;sntz=1&amp;usg=AFQjCNG3S4MIgcfIYSd8CqTe8t-qLetLKg">http://en.wikipedia.org/wiki/Unimodal_distribution#Unimodal_probability_distribution</a></span></p><ul class="c23 lst-kix_5w9dslcj8r5z-0 start"><li class="c14 c4"><span class="c5">if there is only one mode then there is no merge</span></li></ul><ul class="c23 lst-kix_myo05blafbk-0"><li class="c14 c4"><span class="c5">otherwise draw the line obtained by weighted linear regression among the start/end offset points. The weights of the points are inversely proportional to the rank (strength) of the hit.</span></li><li class="c14 c4"><span class="c5">the prediction is a merge of two genes if the slope of the line is between 0.4 and 1.2 (these thresholds are obtained empirically, after observing the slope of a large amount of queries). An image depicting this is available just below:</span></li></ul><p class="c0 c20"><span class="c5"></span></p><p class="c4 c9"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 578.00px; height: 368.00px;"><img alt="" src="images/image02.png" style="width: 578.00px; height: 368.00px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c0 c20"><span class="c5"></span></p></td></tr><tr class="c2"><td class="c7"><p class="c0 c20"><span class="c28"></span></p></td></tr><tr class="c2"><td class="c7"><h4 class="c0 c20"><a name="h.uzh0tehj82vi"></a></h4><h4 class="c4"><a name="h.7u98e7kx9p5n"></a><span>Plots</span></h4><ul class="c23 lst-kix_whzqgipsg1bt-0 start"><li class="c14 c4"><span class="c5">scatterplot representing a 2D plot of the start/end offsets of the high-scoring segments in the prediction that match the hits (gene merge in Figure 1, no merge in Figure 2)</span></li></ul><p class="c0 c20"><span class="c5"></span></p><p class="c4 c9"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 260.00px; height: 238.86px;"><img alt="" src="images/image15.png" style="width: 260.00px; height: 238.86px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 264.00px; height: 238.29px;"><img alt="" src="images/image16.png" style="width: 264.00px; height: 238.29px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c4 c9"><span class="c12 c17">(1) &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;(2)</span></p><ul class="c23 lst-kix_3kgcjgrfg4nh-0 start"><li class="c14 c4"><span class="c5">line plot representing the regions in the prediction that are matched by each </span><span class="c5">hit</span><span class="c5">. The plot highlights the case when there is a merge between multiple genes (Figure 3) or no merge (Figure 4)</span></li></ul><p class="c0 c20"><span class="c17"></span></p><p class="c4 c9"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 268.00px; height: 226.00px;"><img alt="" src="images/image04.png" style="width: 268.00px; height: 226.00px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 275.34px; height: 228.50px;"><img alt="" src="images/image09.png" style="width: 275.34px; height: 228.50px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c4 c9"><span class="c12 c17">(3) &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; (4)</span></p></td></tr></tbody></table><h2 class="c62 c10"><a name="h.hxel8tqfbg4y"></a></h2><h2 class="c10"><a name="h.lpsoaxpm11ri"></a><span class="c17">5) Validation based on multiple alignment - for proteins only</span></h2><a href="#" name="cf99eeef7d38d96bb4863d435eeea9e016ec3cb3"></a><a href="#" name="5"></a><table cellpadding="0" cellspacing="0" class="c6"><tbody><tr class="c2"><td class="c11"><h4 class="c4"><a name="h.8l6fap8glzwt"></a><span>Error causes</span></h4><ul class="c23 lst-kix_hzkhz4hhlr8l-0 start"><li class="c1"><span class="c5">sequencing error</span></li></ul></td></tr><tr class="c2"><td class="c11"><h4 class="c4"><a name="h.pasu6gpa26d"></a><span>Input data</span></h4><ul class="c23 lst-kix_4ed4y1s6lj5w-0"><li class="c14 c10"><span class="c5">raw sequence for prediction</span></li><li class="c14 c10"><span class="c5">raw sequence for the best n hits</span></li></ul><p class="c0 c20"><span class="c5"></span></p></td></tr><tr class="c2"><td class="c11"><h4 class="c4"><a name="h.34yhnxv3f21q"></a><span>Class information:</span></h4><ul class="c23 lst-kix_4tq1geabcxa-0 start"><li class="c14 c4"><span class="c5">header = &quot;Missing/Extra sequences&quot;</span></li><li class="c14 c4"><span class="c5">description = &quot;Finds missing and extra sequences in the prediction, based on the multiple alignment of the best hits. Meaning of the output displayed: the percentages of the missing/extra sequences with respect to the multiple alignment. Validation fails if one of these values is higher than 20%</span></li><li class="c14 c4"><span class="c5">alias = &quot;align&quot;</span></li></ul><p class="c0 c20"><span class="c5"></span></p></td></tr><tr class="c2"><td class="c11"><h4 class="c4"><a name="h.i4z2t4x1xxka"></a><span>Workflow</span></h4><ul class="c23 lst-kix_olh7ea290trr-0 start"><li class="c14 c10"><span class="c5">get raw sequences for the first n hits (if they have not been already calculated)</span></li><li class="c14 c10"><span class="c5">multiple align the hits with the prediction (mafft)</span></li><li class="c14 c10"><span class="c5">compute the pssm (Position Specific Scoring Matrix) profile for the hits</span></li><li class="c14 c10"><span class="c5">remove isolated residues from the hits and from the prediction</span></li><li class="c14 c10"><span class="c5">compute gap coverage: count gaps that are in the prediction but not in the statistical model (see Figure </span><span class="c5">1</span><span class="c5">) and return the percentage of the missing residues from the total length of the prediction</span></li><li class="c14 c10"><span class="c5">extra sequences: count residues that are in the prediction but not in the statistical model (see Figure 1) and return the percentage of the extra residues from the total length of the prediction</span></li><li class="c14 c10"><span class="c5">conserved regions: count the number of the residues conserved among the hits that appear in the prediction and return the percentage of conserved residues from the total length of the prediction</span></li><li class="c14 c10"><span class="c5">a gene which passes the test would have both the percentage of missing and extra sequences lower than 20%. Otherwise the query will fail on this validation test.</span></li></ul><p class="c0"><span class="c5"></span></p><p class="c9 c10"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 547.00px; height: 108.00px;"><img alt="" src="images/image18.png" style="width: 547.00px; height: 108.00px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c9 c10"><span class="c5 c12">(1)</span></p></td></tr><tr class="c2"><td class="c11"><p class="c0 c20"><span class="c58 c17"></span></p></td></tr><tr class="c2"><td class="c11"><h4 class="c4"><a name="h.du366f6idqnj"></a><span>Plots</span></h4><ul class="c23 lst-kix_yghjnj8s9xuc-0 start"><li class="c14 c4"><span class="c5">line plot representing the multiple alignment, the prediction and the statistical model (colors highlight gaps and extra sequences and consensus regions among the hits). Figure 1 - good prediction, Figure 2- weak prediction.</span></li></ul><p class="c0 c20"><span class="c5"></span></p><p class="c4 c9"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 308.00px; height: 146.75px;"><img alt="" src="images/image07.png" style="width: 308.00px; height: 146.75px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 317.59px; height: 146.50px;"><img alt="" src="images/image12.png" style="width: 317.59px; height: 146.50px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c4 c9"><span class="c12 c17">(1) &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;(2)</span></p></td></tr></tbody></table><p class="c0"><span class="c12 c17"></span></p><h2 class="c62 c10"><a name="h.w5051fey70a7"></a></h2><h2 class="c62 c10"><a name="h.wr8mkupp9mjz"></a></h2><h2 class="c10"><a name="h.f63zo0greej2"></a><span class="c17">6) Blast Reading Frame Validation - for nucleotides only</span></h2><a href="#" name="da7f0d61eed496f7fd2cd82dbaa19c8c29d6133c"></a><a href="#" name="6"></a><table cellpadding="0" cellspacing="0" class="c6"><tbody><tr class="c2"><td class="c7"><h4 class="c4"><a name="h.f7tme0bpawv4"></a><span>Error causes</span></h4><ul class="c23 lst-kix_vqtjz6k7exn7-0 start"><li class="c14 c4"><span class="c5">frame shift sequencing errors: various reading frames of the same sign (e.g single nucleotide was accidentally inserted, sequences are read in the same direction). </span></li><li class="c14 c4 c13"><span class="c5">various reading frames of different sign may still highlight a problem in the prediction, not a reading frame shift, but a possible merge between two genes (and this case is treated by the gene merge validation)</span><span class="c5 c12">&nbsp;</span></li></ul></td></tr><tr class="c2"><td class="c7"><h4 class="c4"><a name="h.ug7m82xgyjul"></a><span>Input data</span></h4><ul class="c23 lst-kix_4ed4y1s6lj5w-0"><li class="c14 c10"><span class="c5">reading frame provided by the blast output</span></li></ul></td></tr><tr class="c2"><td class="c7"><h4 class="c4"><a name="h.3zrcjhz1synl"></a><span>Class information:</span></h4><ul class="c23 lst-kix_48dltnwvsqfp-0 start"><li class="c54 c4"><span class="c5">header: Reading Frame</span></li><li class="c54 c4"><span class="c5">description: Check whether there is a single reading frame among BLAST hits. Otherwise there might be a reading frame shift in the query sequence. Meaning of the output displayed: (reading frame: no hsps)</span></li><li class="c54 c4"><span class="c5">alias: frame</span></li></ul><p class="c0 c20"><span class="c5"></span></p></td></tr><tr class="c2"><td class="c7"><h4 class="c4"><a name="h.j0jjou21atn5"></a><span>Workflow</span></h4><ul class="c23 lst-kix_kmsnp6mmp404-0 start"><li class="c14 c4"><span class="c5">blastx translates each query in each of the six reading frames, and all the resulting six-protein sequences are compared to data in the databases (there&#39;s no evidence where the codon starts from, so all the 3 possibilities of starting the gene translation are considered in BLASTX; also another 3 possibilities when reading the sequence in reverse). </span></li></ul><ul class="c23 lst-kix_508olx789l51-0 start"><li class="c14 c4"><span class="c5">for all the hsps of each hit we check the reading frame number</span></li><li class="c14 c4"><span class="c5">store in a hash table the number of occurrences for each reading frame</span></li><li class="c14 c4"><span class="c5">when we count more reading frames of the same sign, there must have occurred a frame shift during the sequencing </span><span class="c5">process</span><span class="c5">. </span></li><li class="c14 c4"><span class="c5">return a hash with the key each one of the 6 possible reading frames and the value - the number of hsps found by blast which are read in that reading frame</span></li></ul></td></tr><tr class="c2"><td class="c7"><p class="c0 c20"><span class="c28"></span></p></td></tr><tr class="c2"><td class="c7"><h4 class="c4"><a name="h.kzye1e8i7r8t"></a><span>Plots</span></h4><p class="c4"><span class="c17">No plots are generated</span></p></td></tr></tbody></table><p class="c0"><span class="c12 c17"></span></p><h2 class="c10"><a name="h.3elev4f70kdl"></a><span class="c17">7) Open reading frame - for nucleotides only</span></h2><p class="c0"><span class="c17"></span></p><a href="#" name="732325caf63d1bdde379f7a96ea6721e3039f864"></a><a href="#" name="7"></a><table cellpadding="0" cellspacing="0" class="c6"><tbody><tr class="c2"><td class="c11"><h4 class="c4"><a name="h.frtucgcrc7ap"></a><span>Error causes</span></h4><ul class="c23 lst-kix_g96zcqu3kh41-0 start"><li class="c1"><span class="c5">many small ORF</span></li><li class="c1"><span class="c5">frame shift</span></li></ul></td></tr><tr class="c2"><td class="c11"><h4 class="c4"><a name="h.58tttrdk1gyx"></a><span>Input data</span></h4><ul class="c23 lst-kix_4ed4y1s6lj5w-0"><li class="c14 c10"><span class="c5">only query raw sequence is needed for this validation (got from the fasta file)</span></li></ul><p class="c0 c20"><span class="c5"></span></p></td></tr><tr class="c2"><td class="c11"><h4 class="c4"><a name="h.6slvdpgdbg5r"></a><span>Class information:</span></h4><ul class="c23 lst-kix_x7t5b7jbczs4-0 start"><li class="c14 c4"><span class="c5">header: Main ORF</span></li><li class="c14 c4"><span class="c5">description: Check whether there is a single main Open Reading Frame in the predicted gene. Applicable only for nucleotide queries. Meaning of the output displayed: %=MAIN ORF COVERAGE. Coverage higher than 80% passe the validation test.</span></li><li class="c14 c4"><span class="c5">alias: orf</span></li></ul><p class="c0 c20"><span class="c5"></span></p></td></tr><tr class="c2"><td class="c11"><h4 class="c4"><a name="h.1dg3y6evjaci"></a><span>Workflow</span></h4><ul class="c23 lst-kix_g3lepnw8n0x6-0 start"><li class="c14 c4"><span class="c5">for each reading frame get all the ORFs, with customizable stop and start codons (the subsequeces that start with a start codon and end with an end codon)</span></li><li class="c14 c4"><span class="c5">return the coverage percentage of the longest reading frame</span></li><li class="c14 c4"><span class="c5">If there is any continuous ORF read in a certain reading frame that is longer than 80% of the length of the prediction, that prediction is considered to be a good prediction. Otherwise it won&rsquo;t pass the test.</span></li></ul><p class="c0 c20"><span class="c5"></span></p></td></tr><tr class="c2"><td class="c11"><p class="c0 c20"><span class="c58 c17"></span></p></td></tr><tr class="c2"><td class="c11"><h4 class="c4"><a name="h.8803ag5rytmq"></a><span>Plots</span></h4><p class="c0 c20"><span class="c17"></span></p><ul class="c23 lst-kix_l7t8lma3v5dd-0 start"><li class="c14 c4"><span class="c5">line plot depicting the ORFs for each reading frame (Figure 1 - plot for a good prediction)</span></li></ul><p class="c0 c20"><span class="c17"></span></p><p class="c4 c9"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 328.00px; height: 342.00px;"><img alt="" src="images/image08.png" style="width: 328.00px; height: 342.00px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c4 c9"><span class="c12 c17">(1)</span></p></td></tr></tbody></table><p class="c0"><span class="c17"></span></p><hr style="page-break-before:always;display:none;"><p class="c0"><span class="c17"></span></p><p class="c0"><span class="c17"></span></p><a href="#" name="fb9fb93011ad78ca1e36c4c1e56fdf04d009bd67"></a><a href="#" name="8"></a><table cellpadding="0" cellspacing="0" class="c6"><tbody></tbody></table><p class="c0 c40"><span class="c12 c18 c39 c17"></span></p><p class="c0 c9"><span class="c12 c18 c39 c17"></span></p><h1 class="c9 c10"><a name="h.njhshrkdurp1"></a><span class="c12 c17">C</span><span class="c12 c17">. Code Structure</span></h1><p class="c9 c10"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 818.43px; height: 425.50px;"><img alt="GeneValidator_block_diagram_2.png" src="images/image14.png" style="width: 818.43px; height: 425.50px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c0 c9"><span class="c12 c18 c17"></span></p><p class="c0 c9"><span class="c12 c18 c17"></span></p><p class="c0 c9"><span class="c12 c18 c17"></span></p><hr style="page-break-before:always;display:none;"><p class="c0 c9"><span class="c12 c18 c17"></span></p><h1 class="c9 c10"><a name="h.8hgqxxlaoy94"></a><span class="c12 c17">D</span><span class="c12 c17">. How to run the tool</span></h1><p class="c0"><span class="c17"></span></p><h2 class="c10"><a name="h.es2ozy57vw91"></a><span class="c32">1. Prerequisite:</span></h2><p class="c0"><span class="c5"></span></p><ul class="c23 lst-kix_cbmcswhr1ftl-0 start"><li class="c10 c14"><span class="c5 c12">Ruby</span><span class="c5">&nbsp;(&gt;= 1.9.3)</span></li><li class="c14 c10"><span class="c5 c12">RubyGems</span><span class="c5">&nbsp;(&gt;= 1.3.6)</span></li><li class="c14 c10"><span class="c5 c12">NCBI BLAST+</span><span class="c5">&nbsp;(&gt;= 2.2.25+)</span></li><li class="c14 c10"><span class="c5 c12">MAFFT installation</span><span class="c5">&nbsp;(download it from :</span><span class="c5"><a class="c24" href="http://www.google.com/url?q=http%3A%2F%2Fmafft.cbrc.jp%2Falignment%2Fsoftware%2F&amp;sa=D&amp;sntz=1&amp;usg=AFQjCNHEp_zxdXbofBWsJM1LNT1xEHndaQ">&nbsp;</a></span><span class="c5 c21"><a class="c24" href="http://www.google.com/url?q=http%3A%2F%2Fmafft.cbrc.jp%2Falignment%2Fsoftware%2F&amp;sa=D&amp;sntz=1&amp;usg=AFQjCNHEp_zxdXbofBWsJM1LNT1xEHndaQ">http://mafft.cbrc.jp/alignment/software/</a></span><span class="c5">&nbsp;).</span></li></ul><p class="c0"><span class="c5"></span></p><p class="c9 c10"><span class="c5 c12">Linux and MacOS are officially supported!</span></p><h2 class="c10"><a name="h.pszg00frp7dj"></a><span class="c32">2. Installation:</span></h2><p class="c0"><span class="c5 c12"></span></p><ol class="c23 lst-kix_9drbe3jtkkuh-0 start" start="1"><li class="c14 c10"><span class="c5">Get the source code</span></li></ol><p class="c3"><span class="c5">$ git clone git@github.com:monicadragan/GeneValidator.git</span></p><ol class="c23 lst-kix_9drbe3jtkkuh-0" start="2"><li class="c14 c10"><span class="c5">Be sudo and build the gem</span></li></ol><p class="c3"><span class="c5">$ sudo rake</span></p><ol class="c23 lst-kix_9drbe3jtkkuh-0" start="3"><li class="c14 c10"><span class="c5">Run GeneValidation</span></li></ol><p class="c3"><span class="c5">$ genevalidator [validations] [skip_blast] [start] [tabular] [mafft] [raw_seq] FILE</span></p><p class="c3 c77"><span class="c5"></span></p><p class="c10"><span class="c5">Example that runs all validations on a set of ant gene predictions:</span></p><p class="c10"><span class="c5">$ genevalidator -x data/prot_Solenopsis.xml data/prot_Solenopsis.fa</span></p><p class="c0"><span class="c5"></span></p><p class="c10"><span class="c5">To learn more:</span></p><p class="c10"><span class="c5">$ genevalidator -h</span></p><p class="c0"><span class="c5"></span></p><h2 class="c10"><a name="h.m9jfyg83mj71"></a><span class="c32">3. Input</span></h2><p class="c0"><span class="c5 c12"></span></p><ul class="c23 lst-kix_xc244bhq3ube-0 start"><li class="c14 c10"><span class="c5 c12">FASTA file</span><span class="c5">&nbsp;with the prediction queries (queries must be of the same type: </span><span class="c5 c12">mRNA or proteins</span><span class="c5">. Mixed type FASTA files are not allowed!).</span></li><li class="c14 c10"><span class="c5 c12">Precalculated BLAST output file</span><span class="c5">&nbsp;in xml format (-outfmt 5) or tabular format (-outfmt 6 or -outfmt 7 or &ldquo;Hit Table (CSV)&rdquo; in the online version of BLAST). Please specify the --tabular argument if you use tabular format input with nonstandard columns. Also, </span><span class="c5 c12">we search only in protein database using a blastp for protein queries and blastx for nucleotide queries. </span></li><li class="c14 c10"><span class="c5">[optional] </span><span class="c5 c12">Precalculated FASTA file </span><span class="c5">containing raw sequences of the best hits (will be looked up by the identifier)</span></li></ul><p class="c0"><span class="c5 c12"></span></p><h2 class="c10"><a name="h.plih6twef6rr"></a><span class="c32">4. Output</span></h2><p class="c0"><span class="c5 c12"></span></p><p class="c10"><span class="c5">By running GeneValidator on your dataset you get numbers and plots. Some relevant files will be generated at the same path with the input file. The results are available in 3 formats:</span></p><ul class="c23 lst-kix_jamjx369pdr8-0 start"><li class="c14 c10"><span class="c5 c12">console table output</span></li><li class="c14 c10"><span class="c5">validation results in </span><span class="c5 c12">YAML format</span><span class="c5">&nbsp;(the YAML file has the same name with the input file + YAML extension)</span></li><li class="c14 c10"><span class="c5 c12">html output</span><span class="c5">&nbsp;with plot visualization (the useful files will be generated in the &#39;html&#39; directory, at the same path with the input file). Check the </span><span class="c5 c12">index.html</span><span class="c5">&nbsp;file to see the output.</span></li></ul><p class="c0 c9"><span class="c5"></span></p><p class="c9 c10"><span class="c5 c12 c39">! Note: for the moment check the html output with Firefox browser only !</span></p><p class="c9 c10"><span class="c5">(because our d3js cannot be visualized from other browsers without a local webserver) </span></p><p class="c0"><span class="c5 c12"></span></p><p class="c0"><span class="c5"></span></p><h2 class="c10"><a name="h.26zdfyct6x5w"></a><span class="c32">5. Scenarios</span></h2><p class="c0"><span class="c5 c12"></span></p><p class="c10"><span class="c5 c12">5.1</span><span class="c5">&nbsp;Quickly test the tool or validate a small set of predictions </span><span class="c5 c12">(internet connection required).</span></p><p class="c9 c10"><span class="c5">$ genevalidator predictions.fasta</span></p><p class="c10"><span class="c5 c38">What happens backstage: blast hits and raw sequences are retrieved from remote databases and analysed for each </span><span class="c5 c38">query</span><span class="c5 c38">.</span></p><p class="c0"><span class="c5 c12"></span></p><p class="c13 c10"><span class="c5 c12">5.2</span><span class="c5">&nbsp;</span><span class="c5 c12">Precalculate the input files </span></p><p class="c0 c13"><span class="c5 c12"></span></p><p class="c13 c10"><span class="c5 c12">5.2.1 Precalculate BALST output</span></p><p class="c13 c10"><span class="c5">If you have &gt;100 queries &amp; you access to a more powerful machine that pre-calculates overnight the blast output file. </span><span class="c5 c12">Use blastp for proteins and blastx for nucleotide queries.</span></p><p class="c0 c13"><span class="c5 c12"></span></p><p class="c10"><span class="c5 c12">a)</span><span class="c5">&nbsp;If</span><span class="c5 c12">&nbsp;</span><span class="c5">you have time and a really powerful machine you can </span><span class="c5 c12">take the blast xml output</span><span class="c5">. This way you will save time when running GeneValidator (because you precalculate local alignments with BLAST)</span></p><p class="c10"><span class="c5">&nbsp; </span></p><p class="c10"><span class="c5">Example: </span></p><p class="c0"><span class="c5"></span></p><p class="c10"><span class="c28"># take blast output for </span><span class="c28 c12">protein queries </span><span class="c28">(for </span><span class="c28 c12">mrna queries </span><span class="c28">use </span><span class="c28 c12">blastx</span><span class="c28">)</span></p><p class="c10"><span class="c28">$ blastp -query predictions.fasta -db /path/to/db/nr/nr -outfmt 5 -max_target_seqs 200 -num_threads 48 -out predictions_blast.xml</span></p><p class="c0"><span class="c28"></span></p><p class="c10"><span class="c28"># run validations</span></p><p class="c10"><span class="c28">$ genevalidator -x predictions_blast.xml predictions.fasta</span></p><p class="c0"><span class="c5"></span></p><p class="c10"><span class="c5 c12">b) Take blast tabular output</span><span class="c5">&nbsp;(customized so that you have all the necessary columns -- see the table with data needed for validations in section B)</span></p><p class="c0"><span class="c5"></span></p><p class="c10"><span class="c5">Example: </span></p><p class="c0"><span class="c5"></span></p><p class="c10"><span class="c28"># take blast output for protein queries</span></p><p class="c10"><span class="c28">$ blastp -query predictions.fasta -db /path/to/db/nr/nr -outfmt &quot;6 qseqid sseqid sacc slen qstart qend sstart send length qframe pident evalue&quot; -max_target_seqs 200 -num_threads 48 -out predictions_blast.tab</span></p><p class="c0"><span class="c28"></span></p><p class="c10"><span class="c28"># run validations</span></p><p class="c10"><span class="c28">$ genevalidator -x predictions_blast.tab -t &quot;qseqid sseqid sacc slen qstart qend sstart send length qframe pident evalue&quot; predictions.fasta </span></p><p class="c0"><span class="c5"></span></p><p class="c0"><span class="c5"></span></p><p class="c10"><span class="c5 c12">5.2.1 Precalculate the raw sequences</span><span class="c5">&nbsp;for the first n hits</span><span class="c5 c12">&nbsp;from a local BAST database</span><span class="c5">, using the identifiers from your BLAST output</span></p><p class="c0"><span class="c5 c12"></span></p><p class="c10"><span class="c28">$ get</span><span class="c5">_raw_sequences BLAST_OUTPUT_FILE -o OUTPUT_FILE [-t TABULAR_FORMAT] [-d DATABASES] [-n NR]</span></p><p class="c0"><span class="c5"></span></p><p class="c10"><span class="c5">Example:</span></p><p class="c0"><span class="c5"></span></p><p class="c10"><span class="c28"># get raw sequences from tabular BLAST output </span></p><p class="c10"><span class="c28">$ get_raw_sequences predictions_blast.tab -o predictions.raw_seq -t &quot;qseqid sseqid sacc slen qstart qend sstart send pident length qframe evalue&quot; -d path/to/ncbi/blast/db/nr/nr</span></p><p class="c0"><span class="c28"></span></p><p class="c0"><span class="c28"></span></p><p class="c10"><span class="c28"># get raw sequences from xml BLAST output</span></p><p class="c10"><span class="c28">$ get_raw_sequences predictions_blast.xml -o predictions.raw_seq -d path/to/ncbi/blast/db/nr/nr</span></p><p class="c0"><span class="c28"></span></p><p class="c10"><span class="c28"># run validations</span></p><p class="c10"><span class="c28">$ genevalidator -x data/all_validations_prot/all_validations_prot.tab data/all_validations_prot/all_validations_prot.fasta -t &quot;qseqid sseqid sacc slen qstart qend sstart send pident length qframe evalue&quot; -r data/all_validations_prot/all_validations_prot.xml.raw_seq</span></p><p class="c0"><span class="c5"></span></p><p class="c10"><span class="c5 c12">5.3</span><span class="c5">&nbsp;You are interested in </span><span class="c5 c12">certain validations</span><span class="c5">&nbsp;only (let&rsquo;s take length validations and duplication check). Look up the CLI names for these validations in the table (Section B) and add them as argument so that GeneValidator will process those 3 only.</span></p><p class="c0"><span class="c5"></span></p><p class="c10"><span class="c28">$ genevalidator -x predictions_blast.xml predictions.fasta -v &lsquo;lenc lenr dup&rsquo;</span></p><p class="c0"><span class="c5"></span></p><p class="c0"><span class="c28"></span></p><p class="c0"><span class="c17"></span></p><p class="c0"><span class="c17"></span></p><hr style="page-break-before:always;display:none;"><p class="c0"><span class="c17"></span></p><h1 class="c9 c10"><a name="h.qi34bbc42hqf"></a><span class="c12 c17">E</span><span class="c12 c17">. Add a new </span><span class="c12 c17">validation</span></h1><p class="c0"><span></span></p><p class="c10"><span>The code architecture is designed so that it is very easy to add new validations. There is a </span><span class="c5">&lsquo;template&rsquo; that the code of the new validation must fit, which is discussed furt</span><span class="c5">her</span><span class="c5">:</span></p><a href="#" name="c2291361a0df17754ffa109d886614f1e3d01d6b"></a><a href="#" name="9"></a><table cellpadding="0" cellspacing="0" class="c6"><tbody><tr class="c2"><td class="c11"><h3 class="c4"><a name="h.l1h18sojuad4"></a><span class="c17">Extend 2 parent classes</span></h3><p class="c4 c13"><span class="c5"><br></span><span class="c5 c12">(1) extend </span><span class="c5 c21 c12"><a class="c24" href="https://www.google.com/url?q=https%3A%2F%2Fgithub.com%2Fmonicadragan%2FGeneValidator%2Fblob%2Frubygem%2Flib%2Fgenevalidator%2Fvalidation_output.rb%2520&amp;sa=D&amp;sntz=1&amp;usg=AFQjCNH4Nk5OeERS3x2HJHlkNRxtXjGMqA">ValidationReport</a></span><span class="c5">- stores the output of the validation test and some methods used for data representation, that may need to be overridden:</span></p><ul class="c23 lst-kix_owpom8h3nkwo-0 start"><li class="c14 c4"><span class="c5 c12">validation</span><span class="c5">: returns </span><span class="c5">one</span><span class="c5">&nbsp;of the 4 results: </span></li></ul><ul class="c23 lst-kix_owpom8h3nkwo-1 start"><li class="c54 c4"><span class="c5">:yes </span></li><li class="c54 c4"><span class="c5">:no </span></li><li class="c54 c4"><span class="c5">:unapplicable </span></li><li class="c4 c54"><span class="c5">:error</span></li></ul><ul class="c23 lst-kix_owpom8h3nkwo-0"><li class="c14 c4 c13"><span class="c5 c12">print</span><span class="c5">: returns the output that will be displayed in the outputs</span></li><li class="c14 c4 c13"><span class="c5 c12">color</span><span class="c5">: based on the validation result returns the background color for the corresponding cell in the html output (at the moment the possible values are &ldquo;success&rdquo;, &ldquo;warning&rdquo; and &ldquo;danger&rdquo; -- which are classes in the CSS file)</span></li></ul><p class="c0 c20"><span class="c5"></span></p><p class="c4 c13"><span class="c5">Initialize internal variables (which initially have some default values):</span></p><ul class="c23 lst-kix_pdbqztwjr65g-0 start"><li class="c14 c4 c13"><span class="c5 c12">expected</span><span class="c5">&nbsp;: the expected correct result for the validation. The test will pass if the expected result is equal to the validation result (:yes or :no)</span></li><li class="c14 c4 c13"><span class="c5">initialize </span><span class="c5 c12">plot_files</span><span class="c5">&nbsp;with the empty Array in case you will generate plots (by default this variable is nil)</span></li><li class="c14 c4 c13"><span class="c5">initialize </span><span class="c5 c12">errors</span><span class="c5">&nbsp;with the empty Array in case you will handle and display in the output the possible errors of this validation</span></li></ul><p class="c4 c13"><span class="c5"><br></span><span class="c5 c12">(2) extend </span><span class="c5 c21 c12"><a class="c24" href="https://www.google.com/url?q=https%3A%2F%2Fgithub.com%2Fmonicadragan%2FGeneValidator%2Fblob%2Frubygem%2Flib%2Fgenevalidator%2Fvalidation_test.rb%2520&amp;sa=D&amp;sntz=1&amp;usg=AFQjCNFiJkfBooOVaq3golwzSjneJyqenA">ValidationTest</a></span><span class="c5">&nbsp;</span><span class="c5">(the &#39;run&#39; method must be overloaded, run must return ValidationReport class type, &nbsp;some fields have to be updated: the validation name used in the header, description, plot_files)</span></p><p class="c0 c20"><span class="c5 c12"></span></p><p class="c4 c13"><span class="c5">Initialize internal variables (which initially have some default values):</span></p><ul class="c23 lst-kix_48dltnwvsqfp-0"><li class="c26 c4"><span class="c5 c12">short_header</span><span class="c5">: will appear in the console output (may not contain spaces)</span></li><li class="c4 c26"><span class="c5 c12">header</span><span class="c5">: will appear in the header of the table - in html output </span></li><li class="c26 c4"><span class="c5 c12">description</span><span class="c5">: short description of the validation -- will appear in the html output</span></li><li class="c26 c4"><span class="c5 c12">cli_name</span><span class="c5">: validation can be referred by this alias when selecting the validations to be run (-v argument in the command line)</span></li><li class="c26 c4"><span class="c5 c12">running_time</span><span class="c5">: if you want to make overall running time statistics</span></li></ul><p class="c0 c20"><span class="c5"></span></p></td></tr><tr class="c2"><td class="c11"><h3 class="c4"><a name="h.52nozdazdy71"></a><span class="c17">Add the new validation to the tool</span></h3><p class="c40 c10 c41"><span class="c5">Add the validation to the validations list, which is further processed for yaml/html/console visualization<br><br></span><span class="c5 c21 c12"><a class="c24" href="https://www.google.com/url?q=https%3A%2F%2Fgithub.com%2Fmonicadragan%2FGeneValidator%2Fblob%2Frubygem%2Flib%2Fgenevalidator%2Fvalidation.rb%23L284%2520&amp;sa=D&amp;sntz=1&amp;usg=AFQjCNG43N8PvUyklv8AKv9D_7KgzlQFlA">Code</a></span><span class="c5 c12">:</span></p><p class="c13 c10"><span class="c5"># context code #</span></p><p class="c13 c10"><span class="c5">validations = []</span></p><p class="c13 c10"><span class="c5">validations.push LengthClusterValidation.new(@type, prediction, hits, plot_path, plots)</span></p><p class="c13 c10"><span class="c5">&nbsp;&hellip;</span></p><p class="c13 c10"><span class="c5 c12 c73">## what you need to add is this line that adds your validation class to the list of validations. Your class must extend ValidationTest ##</span></p><p class="c0 c13"><span class="c5 c12 c73"></span></p><p class="c13 c10"><span class="c5 c12 c73">validations.push YourValidationClass(type, prediction, hits, other_arguments)</span></p><p class="c0 c13"><span class="c5"></span></p><p class="c13 c10"><span class="c5"># context code #</span></p><p class="c13 c10"><span class="c5"># check the class type of the elements in the list</span></p><p class="c13 c10"><span class="c5"># this will raise an error if YourValidationClass does not extend ValidationTest validations.map do |v|</span></p><p class="c13 c10"><span class="c5">&nbsp; raise ValidationClassError unless v.is_a? ValidationTest</span></p><p class="c13 c10"><span class="c5">end</span></p><p class="c0 c13"><span class="c5"></span></p><p class="c13 c10"><span class="c5"># run validations</span></p><p class="c13 c10"><span class="c5">validations.map{|v| v.run}</span></p><p class="c0 c13"><span class="c5"></span></p><p class="c13 c10"><span class="c5"># check the class type of the validation reports</span></p><p class="c13 c10"><span class="c5"># this will raise an error if the run method of YourValidationClass does not return ValidationReport</span></p><p class="c13 c10"><span class="c5">validations.map do |v|</span></p><p class="c13 c10"><span class="c5">&nbsp; raise ValidationClassError unless v.validation_report.is_a? ValidationReport</span></p><p class="c13 c10"><span class="c5">end</span></p></td></tr><tr class="c2"><td class="c11"><h3 class="c4"><a name="h.j12khye9z0zk"></a><span class="c17">Add plots</span></h3><p class="c0 c20"><span class="c5 c12"></span></p><h4 class="c4"><a name="h.af1kbdhs4wdx"></a><span class="c5">A. Generate json file according to the type of plot</span></h4><p class="c4"><span class="c5 c12">1) (clustered) bars plot</span><span class="c5">, main cluster is in red: [ [{&quot;key&quot;:109,&quot;value&quot;:1,&quot;main&quot;:false}], [{&quot;key&quot;:122,&quot;value&quot;:32,&quot;main&quot;:true},{&quot;key&quot;:123,&quot;value&quot;:33,&quot;main&quot;:true}], [...]]</span></p><p class="c4"><span class="c5 c12">2) scatter plot</span><span class="c5">: </span></p><p class="c4"><span class="c5">[{&quot;x&quot;:1,&quot;y&quot;:626},{&quot;x&quot;:1,&quot;y&quot;:626}...]</span></p><p class="c4"><span class="c5 c12">3) lines</span><span class="c5">: [{&quot;y&quot;:1,&quot;start&quot;:0,&quot;stop&quot;:356,&quot;color&quot;:&quot;black&quot;},{&quot;y&quot;:1,&quot;start&quot;:40,&quot;stop&quot;:200,&quot;color&quot;:&quot;red&quot;}...]</span></p><p class="c0 c20"><span class="c5"></span></p><h4 class="c4"><a name="h.v27rr25j084l"></a><span class="c5">B. Create a plot object, according to the plot type </span></h4><p class="c4"><span class="c5">&nbsp; &nbsp;plot1 = Plot.new(filemane,</span></p><p class="c4"><span class="c5">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;plot_type # which can be </span><span class="c5 c12">:scatter, :line or :bar</span></p><p class="c4"><span class="c5">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;plot_title,</span></p><p class="c4"><span class="c5">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;legend**,</span></p><p class="c4"><span class="c5">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;xTitle,</span></p><p class="c4"><span class="c5">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;yTitle,</span></p><p class="c4"><span class="c5">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;aux_parameter1*,</span></p><p class="c4"><span class="c5">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;aux_parameter2*)</span></p><p class="c0 c20"><span class="c5"></span></p><p class="c4"><span class="c5 c12">** legend is a String that _must_ have the following format:</span></p><p class="c4"><span class="c5 c12">&quot;item, color1;item with spaces, color2&quot;</span></p><p class="c4"><span class="c5 c12">* auxiliary parameters</span><span class="c5">&nbsp;were used by now for passing the length of prediction (in length histogram) or the slope of the regression line (gene merge validation) or the number of lines (line plots)</span></p><p class="c0 c20"><span class="c5"></span></p><h4 class="c4"><a name="h.vbwt0lpnzk07"></a><span class="c5">C. Add the plot object to the plot_files array</span></h4><p class="c0 c20"><span class="c5"></span></p><p class="c4"><span class="c5">@validation_report.plot_files.push(plot1)</span></p></td></tr></tbody></table><h1 class="c9 c10"><a name="h.o25pqypedjmx"></a><span class="c12 c17">F. Results</span></h1><p class="c4 c13 c56"><span>We prove the correctness of our approach by comparing the validation results of two releases of a genome project and by comparing validations of genes from weak and strong gene databases. The results show a quality improvement of the gene predictions from more recent releases (which have passed through intensive curation process) and for genes from stronger databases. </span></p><p class="c0 c51"><span></span></p><p class="c4 c13 c56"><span class="c12">Comparison between two versions of Honey Bee [5]</span></p><p class="c0 c9"><span></span></p><p class="c13 c10 c56"><span>The score difference between the corresponding sequences from two versions of &nbsp;Apis mellifera (honey bee) genome (versions 1.0 and 3.2) are presented in Figure 6. About 70% of the predictions in the most recent version improved after curation / genome reassembly (have better validation score than in the initial release).</span></p><p class="c9 c10"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 238.50px; height: 238.50px;"><img alt="amel_OGSv1.0_pep.fa.yaml.nr.png" src="images/image10.png" style="width: 238.50px; height: 238.50px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span><span>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;</span></p><p class="c9 c10 c56"><span>Figure 6: Score differences Honey Bee 1.0 and 3.2</span></p><p class="c0 c9"><span></span></p><p class="c4 c13 c56"><span class="c12">Comparison between a weak and a strong database</span></p><p class="c13 c10 c56"><span>The plot in Figure 7 shows the score quality (maximum score is 100) for 1,000 random genes from Swiss-Prot database - in red (strong database, with manually annotated and reviewed sequences) and 1,000 random genes from TrEMBL - in blue (weak database - genes are automatically annotated and not reviewed). In Figure 8 is presented a histogram with the number of passed validations for each single validation (the colors are associated in the same manner).</span></p><p class="c9 c10"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 306.50px; height: 306.50px;"><img alt="" src="images/image03.png" style="width: 306.50px; height: 306.50px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 301.00px; height: 301.00px;"><img alt="" src="images/image00.png" style="width: 301.00px; height: 301.00px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c40 c10"><span class="c58">Figure 7: Comparative outputs - overall score &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;Figure 8: Comparative outputs - score by validation</span></p><h1 class="c9 c55 c10"><a name="h.v23rg48n3u9t"></a></h1><h1 class="c9 c10"><a name="h.4rd7mxpo7dua"></a><span class="c12 c17">G</span><span class="c12 c17">. GUI Output </span></h1><p class="c10"><span class="c12 c18 c17 c39">Some validation outputs for protein and mRNA are available here:</span></p><p class="c0"><span class="c12 c18 c39 c17"></span></p><p class="c10"><span class="c21 c18 c17"><a class="c24" href="http://www.google.com/url?q=http%3A%2F%2Fswarm.cs.pub.ro%2F~mdragan%2Fgsoc2013%2Fgenevalidator&amp;sa=D&amp;sntz=1&amp;usg=AFQjCNHevw0tB3SOBnPKIHmAemjta0Ke4w">http://swarm.cs.pub.ro/~mdragan/gsoc2013/</a></span><span class="c21 c18 c17"><a class="c24" href="http://www.google.com/url?q=http%3A%2F%2Fswarm.cs.pub.ro%2F~mdragan%2Fgsoc2013%2Fgenevalidator&amp;sa=D&amp;sntz=1&amp;usg=AFQjCNHevw0tB3SOBnPKIHmAemjta0Ke4w">genevalidator</a></span><span class="c18 c17">&nbsp; </span></p><p class="c0"><span class="c17"></span></p><p class="c10"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 739.50px; height: 349.63px;"><img alt="" src="images/image05.png" style="width: 739.50px; height: 349.63px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c0"><span class="c17"></span></p></body></html>